// cjlib/nbp/nbp-loading/nbp-loading.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { NbpLoadingService } from './nbp-loading.service';
import { NbpPendingLoadingStatusService } from './nbp-pending-loading-status.service';
import { NbpLoadingStatus } from './../nbp-commons/nbp-commons.enums';
import { coerceBooleanProperty } from './../../utility/lib-utility';
/** intero autogenerato usato per id unici per nbpInput component */
var nextId = 0;
var NbpLoadingComponent = (function () {
    function NbpLoadingComponent(loadingService, pendingTaskService) {
        this.loadingService = loadingService;
        this.pendingTaskService = pendingTaskService;
        this._animationModeAuto = false;
        /**
         * Evento lanciato allo stop del loading.
         */
        this.nbpOnStopAnimation = new EventEmitter();
        /**
         * Evento lanciato allo start del laoding.
         */
        this.nbpOnStartAnimation = new EventEmitter();
    }
    Object.defineProperty(NbpLoadingComponent.prototype, "nbpShowAnimation", {
        /**
         * Flag che definisce la visibilità dell’animazione di caricamento
         */
        get: function () { return this._showAnimation; },
        set: function (value) { this._showAnimation = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NbpLoadingComponent.prototype, "nbpAnimationModeAuto", {
        /**
         * Flag che definisce la modalità automatica di visualizzazione dell'animazione
         * quando presente il componente ignora il parametro nbpShowAnimation e l'invocazione
         * dei metodi pubblici, ma si restistra al servizio NbpLoadingService
         */
        get: function () { return this._animationModeAuto; },
        set: function (value) {
            this._animationModeAuto = coerceBooleanProperty(value);
            if (this._showAnimation) {
                console.log('NbpLoadingComponent - Emitting event: nbpOnStartAnimation');
                this.nbpOnStartAnimation.emit();
            }
            else {
                console.log('NbpLoadingComponent - Emitting event: nbpOnStopAnimation');
                this.nbpOnStopAnimation.emit();
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NbpLoadingComponent.prototype, "nbpMessage", {
        /**
         * (Opzionale) Messaggio da visulizzare assieme all'immagine
         *
         */
        get: function () { return this._nbpMessage; },
        set: function (value) { if (value) {
            this._nbpMessage = value;
        } },
        enumerable: true,
        configurable: true
    });
    ;
    /**
     * 	metodo per far visualizzare il loading
     */
    NbpLoadingComponent.prototype.startAnimation = function () {
        this.nbpShowAnimation = true;
        this.nbpOnStartAnimation.emit();
    };
    /**
     * metodo per spegnere il loading
     */
    NbpLoadingComponent.prototype.stopAnimation = function () {
        this.nbpShowAnimation = false;
        this.nbpOnStopAnimation.emit();
    };
    NbpLoadingComponent.prototype.ngOnInit = function () {
        var _this = this;
        if (!this.id) {
            this.id = "nbp-loading-" + ++nextId;
        }
        if (!this.nbpDelayTime) {
            this.nbpDelayTime = 300;
        }
        this.pendingTaskService.initStreams(this.nbpDelayTime);
        if (this._animationModeAuto) {
            // registra al servizio
            this._loadingStatusStream = this.loadingService.getLoadingSpinnerStatusStream(this.id);
            this._loadingStatusStream.subscribe(function (request) {
                if (request.status === NbpLoadingStatus.START) {
                    _this.pendingTaskService.startLoading();
                }
                else {
                    _this.pendingTaskService.stopLoading();
                }
            });
            this._loadingVisibleStream = this.pendingTaskService.getPendingStatusStream();
        }
    };
    NbpLoadingComponent.prototype.ngOnChanges = function (changes) {
        if (!this._animationModeAuto) {
            if (changes.nbpShowAnimation) {
                if (changes.nbpShowAnimation.currentValue) {
                    this.nbpOnStartAnimation.emit();
                }
                else {
                    this.nbpOnStopAnimation.emit();
                }
            }
        }
    };
    return NbpLoadingComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpLoadingComponent.prototype, "name", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpLoadingComponent.prototype, "id", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpLoadingComponent.prototype, "nbpShowAnimation", null);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpLoadingComponent.prototype, "nbpAnimationModeAuto", null);
__decorate([
    Input(),
    __metadata("design:type", String),
    __metadata("design:paramtypes", [Object])
], NbpLoadingComponent.prototype, "nbpMessage", null);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpLoadingComponent.prototype, "nbpDelayTime", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpLoadingComponent.prototype, "nbpOnStopAnimation", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpLoadingComponent.prototype, "nbpOnStartAnimation", void 0);
NbpLoadingComponent = __decorate([
    Component({
        selector: 'nbp-loading',template: "<div *ngIf=\"(!nbpAnimationModeAuto && nbpShowAnimation) || (nbpAnimationModeAuto &&  (_loadingVisibleStream | async))\" id=\"id\"><div class=\"nbp-spinner-overlay\"></div><div class=\"nbp-spinner-wrapper\"><div class=\"nbp-message\" *ngIf=\"nbpMessage\" [innerHTML]=\"nbpMessage\"></div><div class=\"nbp-spinner\"></div></div></div>",
        styles: [".nbp-spinner-wrapper{position:fixed;width:100%;height:100%;display:flex;flex-direction:column;justify-content:center;align-items:center;background-color:transparent;top:0;left:0;z-index:99999}.nbp-spinner-overlay{position:fixed;width:100%;height:100%;display:flex;align-items:center;top:0;left:0;background-color:rgba(255,255,255,.8);z-index:99998}.nbp-spinner{display:flex;width:90px;height:90px;background-image:url(/cjlib/assets/images/spinner.svg);margin:0 auto}.nbp-message{display:flex;margin:0 auto;font-size:1.5em;font-weight:500;white-space:pre-line;text-align:center;margin-bottom:40px}@media all and (-ms-high-contrast:none),(-ms-high-contrast:active){.nbp-spinner{background-image:url(/cjlib/assets/images/ISP-loading-spinner.gif)!important}} /*# sourceMappingURL=nbp-loading.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [NbpLoadingService, NbpPendingLoadingStatusService])
], NbpLoadingComponent);
export { NbpLoadingComponent };
