// cjlib/nbp/nbp-index-alphabetic/nbp-index-alphabetic.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
/**
 * Componente che espone un elenco di lettere cliccabili. Utilizzo tipico: filtro o indice.
 */
var NbpIndexAlphabeticComponent = (function () {
    function NbpIndexAlphabeticComponent() {
        /**
         * Evento di notifica per passaggio ad una pagina qualsiasi
         */
        this.nbpSelectChar = new EventEmitter();
    }
    NbpIndexAlphabeticComponent.prototype.ngOnInit = function () {
    };
    NbpIndexAlphabeticComponent.prototype.onNbpSelectChar = function (char) {
        this.nbpChars.forEach(function (item) { item.selected = (item.label === char); });
        this.nbpSelectChar.emit(char);
    };
    return NbpIndexAlphabeticComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpIndexAlphabeticComponent.prototype, "nbpChars", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpIndexAlphabeticComponent.prototype, "nbpSelectChar", void 0);
NbpIndexAlphabeticComponent = __decorate([
    Component({
        selector: 'nbp-index-alphabetic',template: "<div class=\"contenitor\"><ng-container *ngFor=\"let char of nbpChars; index as i;\"><div class=\"letter\" (click)=\"onNbpSelectChar(char.label)\" [ngClass]=\"{ 'enabled': !char.disabled, 'disabled': char.disabled, 'searched': char.searched, 'selected': char.selected }\">{{char.label}}</div></ng-container></div>",
        styles: [".contenitor{display:table;border-collapse:separate;border-spacing:.875em 0;margin:0 auto}.letter{text-align:center}.enabled{background-color:transparent;font-size:1.125em;font-family:Arial;font-weight:400;margin:1px;color:#329600;display:table-cell;vertical-align:middle;cursor:pointer;padding-top:20px;padding-bottom:20px}.searched{font-size:1em;font-family:Arial;font-weight:700;margin:1px;color:#329600;background-color:transparent;display:table-cell;vertical-align:middle;cursor:pointer;padding-top:20px;padding-bottom:20px}.selected{font-size:1.125em;font-family:Arial;font-weight:700;margin:1px;color:#329600;background-color:transparent;display:table-cell;vertical-align:middle;cursor:pointer;padding-top:20px;padding-bottom:20px}.disabled{background-color:transparent;font-size:1.125em;font-family:Arial;font-weight:400;margin:1px;display:table-cell;vertical-align:middle;pointer-events:none;opacity:.5;color:#ccc;cursor:none;padding-top:20px;padding-bottom:20px} /*# sourceMappingURL=nbp-index-alphabetic.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpIndexAlphabeticComponent);
export { NbpIndexAlphabeticComponent };
