// cjlib/cj/cj-embedding/cj-embedding.utility.js
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
import { Injectable, Inject } from '@angular/core';
import { APP_CONFIG } from '../../module.config';
var CjEmbeddingUtility = (function () {
    function CjEmbeddingUtility(config) {
        var _this = this;
        this.config = config;
        if (this.config && this.config.getApplicationInfo) {
            this.config.getApplicationInfo().subscribe(function (appInfo) {
                _this.callerId = 'CJ' + (appInfo.application).toUpperCase();
            });
        }
        else {
            this.callerId = 'CjAPP';
        }
    }
    CjEmbeddingUtility.prototype.createReadyMessage = function () {
        var message = {
            eventId: 'CJ_READY',
            callerId: this.callerId,
            category: 'VIEW',
            body: {}
        };
        return message;
    };
    CjEmbeddingUtility.prototype.createModalWindowMessage = function (cmd) {
        var message = {
            eventId: 'MODAL_WINDOW',
            callerId: this.callerId,
            category: 'VIEW',
            body: {
                cmd: cmd
            }
        };
        return message;
    };
    CjEmbeddingUtility.prototype.createLoadingWindowMessage = function (cmd) {
        var message = {
            eventId: 'LOADING_WINDOW',
            callerId: this.callerId,
            category: 'VIEW',
            body: {
                cmd: cmd
            }
        };
        return message;
    };
    CjEmbeddingUtility.prototype.createPageSizeMessage = function (height) {
        var message = {
            eventId: 'PAGE_SIZE',
            callerId: this.callerId,
            category: 'VIEW',
            body: {
                height: height
            }
        };
        return message;
    };
    CjEmbeddingUtility.prototype.createGotoMessage = function (dest) {
        var message = {
            eventId: 'GOTO',
            callerId: this.callerId,
            category: 'VIEW',
            body: {
                dest: dest
            }
        };
        return message;
    };
    CjEmbeddingUtility.prototype.createCjStepMessage = function (body) {
        var message = {
            eventId: 'CJ_STEP',
            callerId: this.callerId,
            category: 'VIEW',
            body: {
                step: body
            }
        };
        return message;
    };
    CjEmbeddingUtility.prototype.createFontSizeMessage = function (body) {
        var message = {
            eventId: 'SET_FONT_SIZE',
            callerId: this.callerId,
            category: 'VIEW',
            body: {
                fontSize: body
            }
        };
        return message;
    };
    CjEmbeddingUtility.prototype.createContrastMessage = function (body) {
        var message = {
            eventId: 'SET_CONTRAST',
            callerId: this.callerId,
            category: 'VIEW',
            body: {
                contrast: body
            }
        };
        return message;
    };
    CjEmbeddingUtility.prototype.createPageScrollMessage = function (scrollY, headerHeight) {
        var message = {
            eventId: 'PAGE_SCROLL',
            callerId: this.callerId,
            category: 'VIEW',
            body: {
                deltaScrollY: scrollY,
                headerHeight: headerHeight
            }
        };
        return message;
    };
    CjEmbeddingUtility.prototype.createFloatingMenuMessage = function (body) {
        var message = {
            eventId: 'FLOATING_MENU',
            callerId: this.callerId,
            category: 'VIEW',
            body: {
                cmd: body
            }
        };
        return message;
    };
    CjEmbeddingUtility.prototype.createErrorMessage = function (error) {
        var message = {
            eventId: 'ERROR',
            callerId: this.callerId,
            category: 'VIEW',
            body: __assign({}, error)
        };
        return message;
    };
    return CjEmbeddingUtility;
}());
CjEmbeddingUtility = __decorate([
    Injectable(),
    __param(0, Inject(APP_CONFIG)),
    __metadata("design:paramtypes", [Object])
], CjEmbeddingUtility);
export { CjEmbeddingUtility };
