// cjlib/cj/cj-embedding/cj-embedding-test.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, ViewChild, ViewChildren, QueryList, ChangeDetectorRef, HostListener } from '@angular/core';
import { CjIFrameEmbeddingService } from './cj-iframe-embedding.service';
import { NbpEmbeddingMessageType } from './../../nbp/nbp-commons/nbp-commons.enums';
import { CjWizardComponent } from './../cj-wizard/cj-wizard.component';
import { NbpLoadingComponent } from './../../nbp/nbp-loading/nbp-loading.component';
import { NbpLoadingService } from './../../nbp/nbp-loading/nbp-loading.service';
import { NbpStyle, NbpSize } from './../../nbp/nbp-commons/nbp-commons.enums';
import { ActivatedRoute } from '@angular/router';
import 'rxjs/add/operator/scan';
import { NbpStepDimension } from './../../nbp/nbp-commons/nbp-commons.enums';
import { NbpEmbeddingService } from './../../nbp/nbp-embedding/nbp-embedding.service';
var WizardContainer = (function () {
    function WizardContainer() {
        this.data = {};
    }
    WizardContainer.prototype.getWizardData = function () {
        return this.data;
    };
    ;
    WizardContainer.prototype.setWizardData = function (data) {
        this.data = data;
    };
    return WizardContainer;
}());
export { WizardContainer };
;
var CjEmbeddingTestComponent = (function () {
    function CjEmbeddingTestComponent(cdr, embeddingService, iframeEmbeddingService, loadingService, route) {
        this.cdr = cdr;
        this.embeddingService = embeddingService;
        this.iframeEmbeddingService = iframeEmbeddingService;
        this.loadingService = loadingService;
        this.nbpStyle = NbpStyle;
        this.multiLoadingId = 'embeddingTestLoading';
        this.loadingId = 'demoLoading';
        // Modal tests
        this.isModalOpened = false;
        this.modalButtons = [
            { id: 'chiudiModale', label: 'Chiudi', disabled: false, visible: true, style: NbpStyle.PRIMARY },
        ];
        this.modalSize = NbpSize.SM;
        // Goto tests
        this.gotoDestination = 'home';
        // Wizard tests
        this.dataContainer = new WizardContainer();
        this.steps = [
            { active: false, fillPercent: 100, label: 'step 1', stepDimension: NbpStepDimension.MEDIO },
            { active: true, fillPercent: 50, label: 'step 2', stepDimension: NbpStepDimension.PICCOLO },
            { active: false, fillPercent: 0, label: 'step 3', stepDimension: NbpStepDimension.GRANDE },
            { active: true, fillPercent: 50, label: 'step 4' }
        ];
        this.embeddingUrlParams = route.snapshot.data['urlParams'];
    }
    CjEmbeddingTestComponent.prototype.ngOnInit = function () {
        this.incomingEvents = this.embeddingService
            .getIncomingEventStream()
            .scan(function (results, evt) { return results.concat(evt).slice(-10); }, [])
            .map(function (events) { return events.map(function (evt) { return JSON.stringify(evt); }).join('\n'); });
        this.outgoingMessages = this.iframeEmbeddingService
            .getOutgoingMessageStream()
            .scan(function (results, msg) { return results.concat(msg).slice(-10); }, [])
            .map(function (messages) { return messages.map(function (msg) { return JSON.stringify(msg); }).join('\n'); });
    };
    CjEmbeddingTestComponent.prototype.ngAfterViewInit = function () {
        this.sendCjReadyMessage();
        this.sendPageSizeMessage();
        this.initializeProgress();
        this.cdr.detectChanges();
    };
    CjEmbeddingTestComponent.prototype.initializeProgress = function () {
    };
    CjEmbeddingTestComponent.prototype.closeModal = function () {
        this.isModalOpened = false;
    };
    CjEmbeddingTestComponent.prototype.openModal = function () {
        this.isModalOpened = true;
    };
    CjEmbeddingTestComponent.prototype.startAnimationByMethod = function () {
        var _this = this;
        this.nbpLoadingComponent.startAnimation();
        setTimeout(function () {
            _this.nbpLoadingComponent.stopAnimation();
        }, 5000);
    };
    CjEmbeddingTestComponent.prototype.sendCjReadyMessage = function () {
        this.embeddingService.notifyOutgoingEvent({
            type: NbpEmbeddingMessageType.CJ_READY
        });
    };
    CjEmbeddingTestComponent.prototype.startAnimationByValue = function () {
        var _this = this;
        this.nbpshowAnimation = true;
        setTimeout(function () {
            _this.nbpshowAnimation = false;
        }, 5000);
    };
    CjEmbeddingTestComponent.prototype.startAnimationWithMessage = function () {
        var _this = this;
        this.nbpLoadingComponents.toArray()[1].startAnimation();
        setTimeout(function () {
            _this.nbpLoadingComponents.toArray()[1].stopAnimation();
        }, 5000);
    };
    CjEmbeddingTestComponent.prototype.startAnimationByService = function () {
        var _this = this;
        this.loadingService.notifyLoadingSpinnerRequest(true, this.loadingId);
        setTimeout(function () {
            _this.loadingService.notifyLoadingSpinnerRequest(false, _this.loadingId);
        }, 5000);
    };
    CjEmbeddingTestComponent.prototype.sendGotoMessage = function () {
        this.embeddingService.notifyOutgoingEvent({
            type: NbpEmbeddingMessageType.GOTO,
            body: this.gotoDestination
        });
    };
    CjEmbeddingTestComponent.prototype.sendPageSizeMessage = function () {
        this.embeddingService.notifyOutgoingEvent({
            type: NbpEmbeddingMessageType.PAGE_SIZE,
            body: this.getDocumentHeight()
        });
    };
    CjEmbeddingTestComponent.prototype.wizardNextStep = function () {
        this.wizard.navigateToNext();
    };
    CjEmbeddingTestComponent.prototype.wizardPrevStep = function () {
        this.wizard.navigateToPrev();
    };
    CjEmbeddingTestComponent.prototype.sendErrorMessage = function () {
        this.embeddingService.notifyOutgoingEvent({
            type: NbpEmbeddingMessageType.ERROR,
            body: {
                errorType: 'USER_TRIGGERED',
                errorCode: '000',
                errorMessage: "Errore creato dall'utente"
            }
        });
    };
    CjEmbeddingTestComponent.prototype.throwError = function () {
        throw new Error('Messaggio di errore');
    };
    CjEmbeddingTestComponent.prototype.getDocumentHeight = function () {
        /*return Math.max(
          document.body.scrollHeight, document.documentElement.scrollHeight,
          document.body.offsetHeight, document.documentElement.offsetHeight,
          document.body.clientHeight, document.documentElement.clientHeight
        );*/
        var resize;
        var element = document.getElementById('nbp-wrapper-container') ? document.getElementById('nbp-wrapper-container') : document.getElementById('body-container');
        if (element) {
            resize = element.clientHeight + 100 + 'px';
        }
        return resize;
    };
    return CjEmbeddingTestComponent;
}());
__decorate([
    ViewChild(CjWizardComponent),
    __metadata("design:type", CjWizardComponent)
], CjEmbeddingTestComponent.prototype, "wizard", void 0);
__decorate([
    ViewChild(NbpLoadingComponent),
    __metadata("design:type", NbpLoadingComponent)
], CjEmbeddingTestComponent.prototype, "nbpLoadingComponent", void 0);
__decorate([
    ViewChildren(NbpLoadingComponent),
    __metadata("design:type", QueryList)
], CjEmbeddingTestComponent.prototype, "nbpLoadingComponents", void 0);
__decorate([
    HostListener('window:resize'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], CjEmbeddingTestComponent.prototype, "sendPageSizeMessage", null);
CjEmbeddingTestComponent = __decorate([
    Component({
        selector: 'cj-embedding-test',template: "<nbp-wrapper><nbp-progress [nbpConfiguration]=\"steps\" nbpDisableSticky id=\"progressFlottante\"></nbp-progress><div class=\"row\"><div class=\"col-md-24\"><nbp-dividing-line nbpTitle=\"Gestione evento CJ-READY\"></nbp-dividing-line></div></div><div class=\"row\"><div class=\"col-md-4\"><nbp-button [nbpLabel]=\"'Send CJ_READY'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"sendCjReadyMessage()\"></nbp-button></div></div><h2>Loading Component with manual animation</h2><p>Start Animation by call startAnimation(), a Public nbpLoadingcomponent method (call stopAnimation() after 5s and is stops the Animation): <button class=\"btn btn-default\" (click)=\"startAnimationByMethod()\">Start Animation by Method Call</button></p><p>Start Animation By Change Boolean Value from False to True (set To false after 5s and this stops the Animation): <button class=\"btn btn-default\" (click)=\"startAnimationByValue()\">Start Animation by Value Change</button></p><p>Start Animation with message: <button class=\"btn btn-default\" (click)=\"startAnimationWithMessage()\">Start Animation with custom message</button></p><nbp-loading [nbpShowAnimation]=\"nbpshowAnimation\"></nbp-loading><nbp-loading [nbpMessage]=\"'12321321'\"></nbp-loading><h2>Loading component with auto animation</h2><p>Handle Animation By invoking:</p><ul><li>NbpLoadingService.setLoadingSpinnerStatus(true):</li><li>TIMEOUT 5 secondi - NbpLoadingService.setLoadingSpinnerStatus(false):</li></ul><button class=\"btn btn-default\" (click)=\"startAnimationByService()\">Start Animation by Service</button><nbp-loading [id]=\"loadingId\" nbpAnimationModeAuto></nbp-loading><nbp-loading [id]=\"multiLoadingId\"></nbp-loading><div class=\"row\"><div class=\"col-md-24\"><nbp-dividing-line nbpTitle=\"Parametri inizializzazione\"></nbp-dividing-line></div></div><div class=\"row\"><div class=\"col-md-24\">Parametri di inizializzazione recuperati dall'URL<pre> {{embeddingUrlParams | json}} </pre></div></div><div class=\"row\"><div class=\"col-md-24\"><nbp-dividing-line nbpTitle=\"Gestione evento MODAL_WINDOW\"></nbp-dividing-line></div></div><div class=\"row\"><div class=\"col-md-6\"><nbp-button [nbpLabel]=\"'Open modal'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"openModal()\"></nbp-button></div></div><nbp-modal #componente1 nbpTitle=\"Modale di esempio\" nbpSubTitle=\"Sottotitolo Esempio\" [(nbpOpened)]=\"isModalOpened\" [nbpButtons]=\"modalButtons\" (nbpOnButtonClicked)=\"closeModal()\" [nbpSize]=\"modalSize\">Modale</nbp-modal><div class=\"row\"><div class=\"col-md-24\"><nbp-dividing-line [nbpLineStyle]=\"nbpStyle.PRIMARY\" nbpTitle=\"Gestione Evento GOTO\"></nbp-dividing-line></div></div><div class=\"row\"><div class=\"col-md-6\"><nbp-input-container [nbpLabel]=\"'Destination'\"><nbp-input-text name=\"gotoDestination\" [(ngModel)]=\"gotoDestination\"></nbp-input-text></nbp-input-container></div><div class=\"col-md-6\" style=\"margin-top:35px;\"><nbp-button [nbpLabel]=\"'Send GOTO'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"sendGotoMessage()\"></nbp-button></div></div><div class=\"row\"><div class=\"col-md-24\"><nbp-dividing-line nbpTitle=\"Gestione Evento CJ_STEP\"></nbp-dividing-line></div></div><cj-wizard #wizard [cjWizardDataContainer]=\"dataContainer\"><cj-wizard-step id=\"Step_1\" name=\"Step_1\" cjLabel=\"Primo step\"><cj-wizard-page id=\"Step_1_Page_1\" name=\"Step_1_Page_1\" cjLabel=\"Pagina 1.1\"><h3>1.1</h3><nbp-button [nbpLabel]=\"'Next'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"wizardNextStep()\"></nbp-button></cj-wizard-page><cj-wizard-page id=\"Step_1_Page_2\" name=\"Step_1_Page_2\" cjLabel=\"Pagina 1.2\"><h3>1.2</h3><nbp-button [nbpLabel]=\"'Previous'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"wizardPrevStep()\"></nbp-button><nbp-button [nbpLabel]=\"'Next'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"wizardNextStep()\"></nbp-button></cj-wizard-page></cj-wizard-step><cj-wizard-step id=\"Step_2\" name=\"Step_2\" cjLabel=\"Secondo step\"><cj-wizard-page id=\"Step_2_Page_1\" name=\"Step_2_Page_1\" cjLabel=\"Pagina 2.1\"><h3>2.1</h3><nbp-button [nbpLabel]=\"'Previous'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"wizardPrevStep()\"></nbp-button><nbp-button [nbpLabel]=\"'Next'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"wizardNextStep()\"></nbp-button></cj-wizard-page><cj-wizard-page id=\"Step_2_Page_2\" name=\"Step_2_Page_2\" cjLabel=\"Pagina 2.2\"><h3>2.2</h3><nbp-button [nbpLabel]=\"'Previous'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"wizardPrevStep()\"></nbp-button></cj-wizard-page></cj-wizard-step></cj-wizard><div class=\"row\"><div class=\"col-md-24\"><nbp-dividing-line nbpTitle=\"Gestione evento PAGE_SIZE\"></nbp-dividing-line></div></div><div class=\"row\"><div class=\"col-md-6\"><nbp-button [nbpLabel]=\"'Send PAGE_SIZE message'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"sendPageSizeMessage()\"></nbp-button></div></div><div class=\"row\"><div class=\"col-md-24\"><nbp-dividing-line nbpTitle=\"Gestione Evento ERROR\"></nbp-dividing-line></div></div><div class=\"row\"><div class=\"col-md-6\"><nbp-button [nbpLabel]=\"'Send ERROR message'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"sendErrorMessage()\"></nbp-button></div><div class=\"col-md-6\"><nbp-button [nbpLabel]=\"'Throw Error'\" [nbpStyle]=\"nbpStyle.PRIMARY\" (nbpClick)=\"throwError()\"></nbp-button></div></div><div class=\"row\"><div class=\"col-md-24\"><nbp-dividing-line nbpTitle=\"Riepilogo eventi Inviati & Ricevuti\"></nbp-dividing-line></div></div><div class=\"row\"><div class=\"col-md-12\"><h2>Ultimi 10 messaggi inviati.</h2><pre>{{ outgoingMessages | async }}</pre></div><div class=\"col-md-12\"><h2>Ultimi 10 eventi ricevuti.</h2><pre>{{ incomingEvents | async }}</pre></div></div></nbp-wrapper>",
    }),
    __metadata("design:paramtypes", [ChangeDetectorRef,
        NbpEmbeddingService,
        CjIFrameEmbeddingService,
        NbpLoadingService,
        ActivatedRoute])
], CjEmbeddingTestComponent);
export { CjEmbeddingTestComponent };
