// cjlib/utility/lib-utility.js
/** Coerces a data-bound value (typically a string) to a boolean. */
/** Coerces a data-bound value (typically a string) to a boolean. */ export function coerceBooleanProperty(value) {
    return value != null && "" + value !== 'false';
}
function valueFormStringPathReducer() {
    return function (objectPartialData, property) {
        return objectPartialData[property];
    };
}
function stringPathToValueReducer(lastIndexOfArrayToReduce) {
    return function (objectPartialData, property, idx) {
        objectPartialData[property] = (idx === lastIndexOfArrayToReduce) ? null : {};
        return objectPartialData[property];
    };
}
export function getPropertyValueByPath(obj, stringPath) {
    return stringPath ? stringPath.split('.').reduce(valueFormStringPathReducer(), obj) : null;
}
export function generateObjectFromStringPath(obj, stringPath) {
    var arrayToReduceInObject = stringPath.split('.');
    arrayToReduceInObject.reduce(stringPathToValueReducer(arrayToReduceInObject.length - 1), obj);
    return obj;
}
export function scrollToTopPage() {
    if (document.body.scrollTop !== 0 || document.documentElement.scrollTop !== 0) {
        document.body.scrollTop = 0; // For Chrome, Safari and Opera 
        document.documentElement.scrollTop = 0; // For IE and Firefox
        console.log('Scrolling to top...');
    }
}
