// cjlib/cj/cj.module.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { NgModule, ErrorHandler, APP_INITIALIZER } from '@angular/core';
import { HttpClientModule } from '@angular/common/http';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
export * from './cj-signature/cj-signature.interfaces';
import { NbpModule } from './../nbp/nbp.module';
import { CjBodyComponent } from './cj-body/cj-body.component';
export { CjBodyComponent } from './cj-body/cj-body.component';
import { CjWizardComponent } from './cj-wizard/cj-wizard.component';
export { CjWizardComponent } from './cj-wizard/cj-wizard.component';
import { CjWizardPageComponent } from './cj-wizard-page/cj-wizard-page.component';
export { CjWizardPageComponent } from './cj-wizard-page/cj-wizard-page.component';
import { CjWizardStepComponent } from './cj-wizard-step/cj-wizard-step.component';
export { CjWizardStepComponent } from './cj-wizard-step/cj-wizard-step.component';
import { CjShoppingCartComponent } from './cj-shopping-cart/cj-shopping-cart.component';
export { CjShoppingCartComponent } from './cj-shopping-cart/cj-shopping-cart.component';
import { CjContextService } from './cj-context/cj-context.service';
export { CjContextService } from './cj-context/cj-context.service';
import { CjCustomContextService } from './cj-context/cj-custom-context.service';
export { ICjCustomContextService, CjCustomContextService } from './cj-context/cj-custom-context.service';
import { CjContextClientService } from './cj-context/cj-context-client.service';
export { CjContextClientService } from './cj-context/cj-context-client.service';
import { CjCustomContextClientService } from './cj-context/cj-custom-context-client.service';
export { CjCustomContextClientService } from './cj-context/cj-custom-context-client.service';
export { TestCjWizardContextSource } from './cj-wizard-context-source/cj-wizard-context-source.service';
import { CjWizardHelper } from './cj-wizard/cj-wizard-helper.service';
import { CjPrivacyFormComponent } from './cj-privacy-form/cj-privacy-form.component';
import { CjPrivacyComponent } from './cj-privacy/cj-privacy.component';
export { CjPrivacyComponent } from './cj-privacy/cj-privacy.component';
import { CjPrivacyClientService } from './cj-privacy/cj-privacy-client.service';
export { CjPrivacyClientService } from './cj-privacy/cj-privacy-client.service';
import { CjPrivacyService } from './cj-privacy/cj-privacy.service';
export { CjPrivacyService } from './cj-privacy/cj-privacy.service';
import { CjShoppingCartElementComponent } from './cj-shopping-cart-section/cj-shopping-cart-element.component';
import { CjShoppingCartProductComponent } from './cj-shopping-cart-product/cj-shopping-cart-product.component';
import { ShoppingCartUtility } from './cj-shopping-cart/cj-shopping-cart.utility';
export { ShoppingCartUtility } from './cj-shopping-cart/cj-shopping-cart.utility';
import { GlobalErrorHandlerService } from './../utility/global-error-handler/global-error-handler.service';
import { GlobalErrorHandlerHelperService } from './../utility/global-error-handler/global-error-handler-helper.service';
// ERROR_HANDLER - start
import { ApplicationStatusService } from './../utility/application-status/application-status.service';
export { ApplicationStatusService } from './../utility/application-status/application-status.service';
import { EventLoggerService, ApplicationEventLoggerFactory } from './../utility/event-logger/event-logger.service';
import { ServerEventLoggerService } from './../utility/event-logger/server-event-logger.service';
export { ServerEventLoggerService } from './../utility/event-logger/server-event-logger.service';
import { ConsoleEventLoggerService } from './../utility/event-logger/console-event-logger.service';
export { ConsoleEventLoggerService } from './../utility/event-logger/console-event-logger.service';
import { CjErrorModalComponent } from './cj-error-modal/cj-error-modal.component';
export { CjErrorModalComponent } from './cj-error-modal/cj-error-modal.component';
import { CjErrorViewComponent } from './cj-error-view/cj-error-view.component';
// ERROR_HANDLER: end
import { CjProgressConfigurationService } from './cj-progress-configuration/cj-progress-configuration.service';
import { CjAutocompleteComponent } from './cj-autocomplete/cj-autocomplete.component';
export { CjAutocompleteComponent } from './cj-autocomplete/cj-autocomplete.component';
import { CjAutocompleteElementComponent } from './cj-autocomplete-element/cj-autocomplete-element.component';
import { CjSignatureComponent } from './cj-signature/cj-signature.component';
export { CjSignatureComponent } from './cj-signature/cj-signature.component';
export { MockSignatureDatasource } from './cj-signature-datasource/mock-signature-datasource.service';
import { CjApplicationAlertComponent } from './cj-application-warning/cj-application-alert.component';
export { CjApplicationAlertComponent } from './cj-application-warning/cj-application-alert.component';
import { CjEmbeddingTestComponent } from './cj-embedding/cj-embedding-test.component';
export { CjEmbeddingTestComponent } from './cj-embedding/cj-embedding-test.component';
import { CjIFrameEmbeddingService } from './cj-embedding/cj-iframe-embedding.service';
import { CjEmbeddingUtility } from './cj-embedding/cj-embedding.utility';
import { CjEmbeddingParametersResolver } from './cj-embedding/cj-embedding-parameters.resolver';
import { NbpEmbeddingService } from './../nbp/nbp-embedding/nbp-embedding.service';
import { NbpViewSettingsService } from './../nbp/nbp-view-settings/nbp-view-settings.service';
export { CjEmbeddingParametersResolver } from './cj-embedding/cj-embedding-parameters.resolver';
import { CjEmbeddingEnableGuard } from './cj-embedding/cj-embedding-enable.guard';
export { CjEmbeddingEnableGuard } from './cj-embedding/cj-embedding-enable.guard';
var CjModule = (function () {
    function CjModule() {
    }
    return CjModule;
}());
CjModule = __decorate([
    NgModule({
        imports: [
            CommonModule,
            FormsModule,
            NbpModule,
            RouterModule,
            HttpClientModule
        ],
        declarations: [
            CjBodyComponent,
            CjWizardComponent,
            CjWizardPageComponent,
            CjWizardStepComponent,
            CjPrivacyComponent,
            CjPrivacyFormComponent,
            CjShoppingCartComponent,
            CjShoppingCartElementComponent,
            CjShoppingCartProductComponent,
            CjErrorModalComponent,
            CjErrorViewComponent,
            CjAutocompleteComponent,
            CjAutocompleteElementComponent,
            CjSignatureComponent,
            CjApplicationAlertComponent,
            CjEmbeddingTestComponent
        ],
        entryComponents: [
            CjWizardPageComponent
        ],
        providers: [
            CjContextService,
            CjContextClientService,
            CjCustomContextClientService,
            CjCustomContextService,
            CjWizardHelper,
            CjPrivacyService,
            CjPrivacyClientService,
            ShoppingCartUtility,
            CjProgressConfigurationService,
            // ERROR_HANDLER: start
            {
                provide: ErrorHandler,
                useClass: GlobalErrorHandlerService
            },
            GlobalErrorHandlerHelperService,
            ApplicationStatusService,
            EventLoggerService,
            {
                provide: APP_INITIALIZER,
                useFactory: ApplicationEventLoggerFactory,
                deps: [EventLoggerService],
                multi: true
            },
            ServerEventLoggerService,
            ConsoleEventLoggerService,
            // ERROR_HANDLER: stop
            CjIFrameEmbeddingService,
            CjEmbeddingUtility,
            CjEmbeddingParametersResolver,
            NbpEmbeddingService,
            NbpViewSettingsService,
            CjEmbeddingEnableGuard
        ],
        exports: [
            CjBodyComponent,
            CjWizardComponent,
            CjWizardPageComponent,
            CjWizardStepComponent,
            CjPrivacyComponent,
            CjShoppingCartComponent,
            CjErrorModalComponent,
            CjErrorViewComponent,
            CjAutocompleteComponent,
            CjSignatureComponent,
            CjApplicationAlertComponent,
            CjEmbeddingTestComponent
        ]
    })
], CjModule);
export { CjModule };
