// cjlib/nbp/nbp-accordion-button/nbp-accordion-button.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { NbpAccordionEvent } from '../nbp-accordion/nbp-accordion.interfaces';
var nextId = 0;
var NbpAccordionButtonComponent = (function () {
    function NbpAccordionButtonComponent() {
        // TODO NIS questo componente sarebbe da unire al nbp-accordion-showmore aggiungendo uno style magari!!
        this.id = "nbp-accordion-button-" + ++nextId;
        /**
         * (Opzionale) Booleano che pilota la visualizzazione iniziale dell’accordion collassato o espanso.
         *Default accordion collassato.
         */
        this.nbpCollapsed = true;
        /**
         * (Opzionale) Booleano che indica se allineare a destra l'heading del componente.
         * Default false.
         */
        this.nbpPullRight = false;
        /**
         * Evento emesso sul toggle del componente.
         * @param {EventEmitter<NbpAccordionEvent>}  nbpOnToggle
         */
        this.nbpOnToggle = new EventEmitter();
    }
    //     Methods
    /**
     * metodo utilizzabile per l’apertura programmatica
     */
    NbpAccordionButtonComponent.prototype.open = function () {
        this.nbpCollapsed = false;
    };
    /**
     * metodo utilizzabile per la chiusura programmatica
     */
    NbpAccordionButtonComponent.prototype.close = function () {
        this.nbpCollapsed = true;
    };
    /**
     * metodo utilizzabile per verificare se il componente è collassato
     */
    NbpAccordionButtonComponent.prototype.isCollapsed = function () {
        return this.nbpCollapsed;
    };
    NbpAccordionButtonComponent.prototype.ngOnInit = function () {
    };
    NbpAccordionButtonComponent.prototype._toggleCollapse = function () {
        this.nbpCollapsed = !this.nbpCollapsed;
        this._onToggle();
    };
    NbpAccordionButtonComponent.prototype._onToggle = function () {
        var acordionEvent = new NbpAccordionEvent();
        acordionEvent.source = this;
        acordionEvent.isCollapsed = this.nbpCollapsed;
        this.nbpOnToggle.emit(acordionEvent);
    };
    return NbpAccordionButtonComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionButtonComponent.prototype, "id", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionButtonComponent.prototype, "nbpTitle", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpAccordionButtonComponent.prototype, "nbpCollapsed", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionButtonComponent.prototype, "nbpIcon", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpAccordionButtonComponent.prototype, "nbpPullRight", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpAccordionButtonComponent.prototype, "nbpOnToggle", void 0);
NbpAccordionButtonComponent = __decorate([
    Component({
        selector: 'nbp-accordion-button',template: "<div class=\"panel nbp-panel-button\"><div class=\"panel-heading\" role=\"button\" [ngClass]=\"{'panel-heading-round': !isCollapsed(), 'pull-right': nbpPullRight}\" role=\"tab\" [id]=\"id\" (click)=\"_toggleCollapse()\"><h4 class=\"panel-title\"><span class=\"icon accordion-icon\" [ngClass]=\"nbpIcon\" *ngIf=\"!!nbpIcon\"></span> <span class=\"title\" [ngClass]=\"{'accordion-title-icon':!!nbpIcon}\">{{nbpTitle}}</span><br><span [ngClass]=\"{'triangolo-bottom-accordion': isCollapsed(), 'triangolo-top-accordion': !isCollapsed()}\"></span></h4></div><div class=\"panel-collapse\" [ngClass]=\"{'in panel-open':!isCollapsed()}\"><div class=\"panel-body\"><ng-content></ng-content></div></div></div>",
        animations: [],
        styles: [".panel{box-shadow:none}.panel .panel-open{background-color:#f2f2f2!important;border:.0625em!important;border-color:#6f6f6f;margin-top:.625em}.panel-heading{text-align:center;border:.0625em solid #258900!important;box-shadow:none;color:#258900;background-color:#fff;padding:1em 1.25em 1em 1.25em!important;height:3.125em;width:18.75em;border-radius:1.5625em}.panel-heading:hover{cursor:pointer;background-color:#258900!important}.panel-heading:hover .panel-title{color:#fff}.panel-heading:hover .triangolo-top-accordion{border-color:transparent transparent #fff transparent}.panel-heading:hover .triangolo-bottom-accordion{border-color:#fff transparent transparent transparent}.panel-heading.panel-heading-round{background-color:#258900!important}.panel-heading.panel-heading-round .panel-title{color:#fff}.panel-heading.panel-heading-round .triangolo-top-accordion{border-color:transparent transparent #fff transparent}.panel-heading.panel-heading-round .triangolo-bottom-accordion{border-color:#fff transparent transparent transparent}.panel-heading .panel-title{color:#258900;font-weight:700;font-size:.75em!important;text-transform:uppercase}.panel-heading .accordion-arrow{color:#000;position:relative;font-size:1.625em!important;margin-top:1.156em}.panel-heading .triangolo-top-accordion{border-style:solid;border-width:0 .4375em .4375em .4375em;border-color:transparent transparent #258900 transparent;display:inline-block!important}.panel-heading .triangolo-bottom-accordion{border-style:solid;border-width:.4375em .4375em 0 .4375em;border-color:#258900 transparent transparent transparent;display:inline-block!important}.panel-heading .stringa{color:#fff;font-size:.75em!important;margin-right:1.875em;font-weight:700;font-family:arial}.panel-heading h4{margin:0;font-size:1em}.panel-body{border:.0625em!important;box-shadow:none;background-color:#f2f2f2!important}.panel-collapse{width:100%;border:.0625em!important;box-shadow:none;background-color:#f2f2f2!important;max-height:0;overflow:hidden}.panel-collapse.in{max-height:1000em}.panelTitle{font-size:.875em;font-weight:700;text-transform:uppercse}.accordionIcon{font-size:2em;position:absolute;top:.5625em}.nbp-panel-button>.panel-heading{background-color:#fff;position:relative}.nbp-panel-button>.panel-heading .accordion-icon{font-size:2em;position:absolute;top:.5625em}.nbp-panel-button>.panel-heading>.panel-title{font-size:.875em;font-weight:700;text-transform:uppercse}.nbp-panel-button>.panel-heading>.panel-title>.title{padding:.125em .125em .125em 0}.nbp-panel-button>.panel-heading>.panel-title .accordion-title-icon{padding:.125em .125em .125em 3.25em!important;margin-bottom:.625em}.nbp-panel-button>.panel-heading>.panel-open{background-color:#5ca740!important}:host-context(.contrasto-on) .panel .panel-open{background-color:#fff!important;color:#000}:host-context(.contrasto-on) .panel .panel-open .panel-title{color:#000}:host-context(.contrasto-on) .panel .panel-open .stringa{color:#000}:host-context(.contrasto-on) .panel .panel-open .accordion-arrow{color:#000}:host-context(.contrasto-on) .panel-heading{background-color:#000!important}:host-context(.contrasto-on) .panel-heading:hover{background-color:#fff!important;color:#000;border:.0625em solid #000!important}:host-context(.contrasto-on) .panel-heading:hover .panel-title{color:#000}:host-context(.contrasto-on) .panel-heading:hover .stringa{color:#000}:host-context(.contrasto-on) .panel-heading:hover .accordion-arrow{color:#000}:host-context(.contrasto-on) .panel-heading:hover .triangolo-top-accordion{border-color:transparent transparent #000 transparent}:host-context(.contrasto-on) .panel-heading:hover .triangolo-bottom-accordion{border-color:#000 transparent transparent transparent}:host-context(.contrasto-on) .panel-heading .panel-title{color:#fff}:host-context(.contrasto-on) .panel-heading .accordion-arrow{color:#fff}:host-context(.contrasto-on) .panel-heading .stringa{color:#fff}:host-context(.contrasto-on) .panel-body{background-color:#fff!important;border-top:none!important;color:#000}:host-context(.contrasto-on) .accordion-green-header>.panel-heading{background-color:#000!important}:host-context(.contrasto-on) .accordion-green-header>.panel-heading:hover{background-color:#fff!important;color:#000;border:.0625em solid #000}:host-context(.contrasto-on) .accordion-green-header>.panel-open{background-color:#fff!important;color:#000;border:.0625em solid #000!important}:host-context(.contrasto-on) .triangolo-top-accordion{border-color:transparent transparent #fff transparent}:host-context(.contrasto-on) .triangolo-bottom-accordion{border-color:#fff transparent transparent transparent} /*# sourceMappingURL=nbp-accordion-button.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpAccordionButtonComponent);
export { NbpAccordionButtonComponent };
