// cjlib/nbp/nbp-customer-info-generic/nbp-customer-info-generic.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input } from '@angular/core';
import { NbpBurgerMenuGenericComponent } from '../nbp-burger-menu-generic/nbp-burger-menu-generic.component';
import { NbpMenuOperationComponent } from '../nbp-menu-operation/nbp-menu-operation.component';
var NbpCustomerInfoGenericComponent = (function () {
    function NbpCustomerInfoGenericComponent() {
        /**
         * Titolo del pannello.
         */
        this.clientTitle = "INFORMAZIONI UTENTE";
        /**
         * Booleano che indica se mostrare o meno la freccia che apre il customer-info.
         * Default è true.
         */
        this.showArrow = true;
        /**
         * Booleano che indica se allineare o meno a destra il contenuto del customer-info.
         * Default è false.
         */
        this.alignRight = false;
        /**
         * Booleano che pilota la visualizzazione iniziale dell’accordion collassato o espanso.
         */
        this._isCollapsed = true;
        this._key = [];
        this._clientInfoKey = [];
        this._clientInfoValue = [];
        this._clientInfoHtml = [];
        this.selectedClient = 0;
    }
    NbpCustomerInfoGenericComponent.prototype._toggleCollapse = function () {
        this._isCollapsed = !this._isCollapsed;
        this.burger.close();
        if (!!this.operation) {
            this.operation.close();
        }
    };
    NbpCustomerInfoGenericComponent.prototype.closeCustomer = function () {
        this._isCollapsed = true;
    };
    NbpCustomerInfoGenericComponent.prototype.ngOnInit = function () {
        // for (var i = 0; i < Object.keys(this.clientsInfo).length; i++) {
        for (var i = 0; i < this.clientsInfo.length; i++) {
            // bisogna ricordare indici del json per usarli in view
            //this.indexClientsInfo.push(i);
            this._clientInfoKey[i] = [];
            this._clientInfoValue[i] = [];
            for (var key in this.clientsInfo[i]) {
                this._clientInfoKey[i].push(key);
                this._clientInfoValue[i].push(this.clientsInfo[i][key]);
            }
        }
    };
    NbpCustomerInfoGenericComponent.prototype.selectClient = function (indexClient) {
        this.selectedClient = indexClient;
    };
    return NbpCustomerInfoGenericComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", NbpBurgerMenuGenericComponent)
], NbpCustomerInfoGenericComponent.prototype, "burger", void 0);
__decorate([
    Input(),
    __metadata("design:type", NbpMenuOperationComponent)
], NbpCustomerInfoGenericComponent.prototype, "operation", void 0);
__decorate([
    Input("clientsInfo"),
    __metadata("design:type", Object)
], NbpCustomerInfoGenericComponent.prototype, "clientsInfo", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpCustomerInfoGenericComponent.prototype, "clientTitle", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpCustomerInfoGenericComponent.prototype, "showArrow", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpCustomerInfoGenericComponent.prototype, "alignRight", void 0);
NbpCustomerInfoGenericComponent = __decorate([
    Component({
        selector: 'nbp-customer-info-generic',template: "<div [ngClass]=\"{'align-right': alignRight, 'full-width': !showArrow}\" class=\"container-ident-cliente\" nowrap=\"nowrap\"><span class=\"voce-ident-cliente\">{{_clientInfoKey[selectedClient][0] | uppercase}}</span> <span class=\"testo-ident-cliente\">{{_clientInfoValue[selectedClient][0]}}</span><br><span class=\"voce-ident-cliente\">{{_clientInfoKey[selectedClient][1] | uppercase}}</span> <span class=\"testo-ident-cliente\">{{_clientInfoValue[selectedClient][1]}}</span><br><span class=\"voce-ident-cliente\">{{_clientInfoKey[selectedClient][2] | uppercase}}</span> <span class=\"testo-ident-cliente\">{{_clientInfoValue[selectedClient][2]}}</span></div><div *ngIf=\"showArrow\" class=\"container-arrow\" style=\"margin-top: 6px;\"><span class=\"accordion-arrow\" [ngClass]=\"{'icon ico-comuni-freccia-accordion-apri':_isCollapsed , 'icon ico-comuni-freccia-accordion-chiudi':!_isCollapsed}\" (click)=\"_toggleCollapse()\"></span></div><div class=\"container-row\" style=\"margin-top: -4px;\"><div [ngClass]=\"{'triangolo':!_isCollapsed,'triangolo-hide':_isCollapsed}\"></div><div [ngClass]=\"{'triangolo-border':!_isCollapsed,'triangolo-hide':_isCollapsed}\"></div><div class=\"info-cliente-open\" [hidden]=\"_isCollapsed\" [ngClass]=\"{'border-pannel':!_isCollapsed}\"><div class=\"row info-cliente-title collapse\" [ngClass]=\"{'in':!_isCollapsed}\">{{clientTitle}}</div><div class=\"container-row\" style=\"width:100%!important\"><div class=\"collapse\" [ngClass]=\"{'in':!_isCollapsed}\" style=\"width:100%!important\"><div *ngFor=\"let item of clientsInfo; let i = index\" style=\"width:100%!important\"><div *ngIf=\"item.html\" [innerHTML]=\"item.html\"></div><div *ngIf=\"!item.html\" class=\"panel-body\" (click)=\"selectClient(i)\" [ngClass]=\"{'selected-client': i === selectedClient}\"><span>{{_clientInfoKey[i][1] | uppercase}}</span> <span>{{_clientInfoValue[i][1]}}</span><br><span>{{_clientInfoValue[i][0]}}</span><ng-container *ngIf=\"i === selectedClient\"><span class=\"icomoon-Simboli_Risorsa-182 icon-select\"></span></ng-container></div></div></div></div></div></div>",
        styles: [".accordion-arrow{cursor:pointer;font-size:1.6em}.align-right{text-align:right}.container-row{display:table-row;float:left!important}.container-ident-cliente{margin-top:0;margin-bottom:0;font-size:.875em;color:inherit;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;width:78%;font-weight:400;float:left;padding-top:6px;padding-left:7px;padding-right:7px;padding-bottom:7px;margin-left:-3.3px}.full-width{width:100%}.voce-ident-cliente{margin-top:0;margin-bottom:0;font-size:1em;color:inherit;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;font-weight:400}.border-pannel{border:1px solid #797979!important}.testo-ident-cliente{margin-top:0;margin-bottom:0;font-size:1em;color:inherit;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;font-weight:700}.panel-body{font-family:Arial;font-weight:400;font-style:normal;font-size:.81em;width:100%;border-top:1px solid gray;vertical-align:middle;z-index:90;cursor:pointer;text-align:left;color:#ccc}.panel-body:hover{background-color:gray;color:#fff}.info-cliente-title{font-family:Arial;font-style:normal;font-size:14px;color:#ccc;padding:10px 0 10px 0;border-bottom:0 solid gray;margin-left:auto;margin-right:auto}.info-cliente-open{position:relative;background-color:#323232;z-index:90;border:0;border-radius:0;text-align:center;margin-top:.5625em;display:table;color:#fff;width:22.25em;margin-left:-6px}.container-arrow{float:right;padding-top:.9em;padding-right:1.375em}.triangolo{width:0;height:0;z-index:91;border-left:.563em solid transparent;border-right:.563em solid transparent;border-bottom:.563em solid #323232;float:right;margin-right:1.96em;position:relative;top:1px}.triangolo-border{width:0;height:0;z-index:90;border-left:.625em solid transparent;border-right:.625em solid transparent;border-bottom:.625em solid #797979;float:right;margin-right:0;position:relative;top:0;left:19px;margin-bottom:-1px}.triangolo-hide{width:0;height:0;z-index:90;border-left:.563em solid transparent;border-right:.563em solid transparent;border-bottom:.563em solid transparent;float:right;margin-right:1.75em;position:relative;top:1px}.icon-select{color:#1ed31e;display:block;float:right;position:relative;top:-.56em}.selected-client{background-color:#444;color:#fff;font-weight:700}:host-context(.contrasto-on) .border-pannel{border:1px solid #fff!important}:host-context(.contrasto-on) .panel-body{border-top:1px solid #fff;color:#fff}:host-context(.contrasto-on) .panel-body:hover{background-color:#fff;color:#000;border-color:#000}:host-context(.contrasto-on) .info-cliente-title{color:#fff;border-bottom:0 solid #fff}:host-context(.contrasto-on) .info-cliente-open{background-color:#000}:host-context(.contrasto-on) .triangolo{border-bottom-color:#000}:host-context(.contrasto-on) .triangolo-border{border-bottom:.625em solid #fff}:host-context(.contrasto-on) .icon-select{color:#000}:host-context(.contrasto-on) .selected-client{background-color:#fff;color:#000;font-weight:700}:host-context(.contrasto-on) .selected-client.panel-body{border-color:#000} /*# sourceMappingURL=nbp-customer-info-generic.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpCustomerInfoGenericComponent);
export { NbpCustomerInfoGenericComponent };
