// cjlib/nbp/nbp-pagination-letters/nbp-pagination-letters.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Output, EventEmitter, Input } from '@angular/core';
var NbpPaginationLettersComponent = (function () {
    function NbpPaginationLettersComponent() {
        /**
         * Evento di notifica per passaggio ad una pagina qualsiasi
         */
        this.nbpSelectChar = new EventEmitter();
    }
    NbpPaginationLettersComponent.prototype.ngOnInit = function () {
    };
    NbpPaginationLettersComponent.prototype.onNbpSelectChar = function (char) {
        this.chars.forEach(function (item) { item.selected = (item.label === char); });
        this.nbpSelectChar.emit(char);
    };
    return NbpPaginationLettersComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpPaginationLettersComponent.prototype, "chars", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpPaginationLettersComponent.prototype, "nbpSelectChar", void 0);
NbpPaginationLettersComponent = __decorate([
    Component({
        selector: 'nbp-pagination-letters',
        template: "<div class=\"contenitor\"><ng-container *ngFor=\"let char of chars; index as i;\"><div class=\"letter\" (click)=\"onNbpSelectChar(char.label)\" [ngClass]=\"{ 'enabled': !char.disabled, 'disabled': char.disabled, 'selected': char.selected }\">{{char.label}}</div></ng-container></div>",
        styles: [".contenitor{display:table;border-collapse:separate;border-spacing:.312em;margin:0 auto}.letter{text-align:center}.enabled{background-color:transparent;font-size:.937em;font-family:Arial,Regular;margin:1px;height:2em;width:2em;color:#258900;border:1px solid #258900;border-radius:1.562em;display:table-cell;vertical-align:middle;cursor:pointer}.enabled:hover{color:#fff;background-color:#258900}.selected{background-color:transparent;font-size:.937em;font-family:Arial,Regular;margin:1px;height:2em;width:2em;color:#fff;background-color:#258900;border-radius:1.562em;display:table-cell;vertical-align:middle;cursor:pointer}.disabled{background-color:transparent;font-size:.937em;font-family:Arial,Regular;margin:1px;height:2em;width:2em;border-radius:1.562em;display:table-cell;vertical-align:middle;pointer-events:none;opacity:.5;color:#ccc;border:1px solid #ccc;cursor:none}:host-context(.contrasto-on) .enabled{background-color:transparent;color:#000;border:1px solid #000}:host-context(.contrasto-on) .enabled:hover{color:#fff;background-color:#000}:host-context(.contrasto-on) .selected{color:#fff;background-color:#000}:host-context(.contrasto-on) .disabled{opacity:.5;color:#000;border:1px solid #000} /*# sourceMappingURL=nbp-pagination-letters.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpPaginationLettersComponent);
export { NbpPaginationLettersComponent };
