// cjlib/nbp/nbp-chart/nbp-chart.module.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NbpBarChartComponent } from './nbp-bar-chart.component';
export { NbpBarChartComponent } from './nbp-bar-chart.component';
import { NbpRingChartComponent } from './nbp-ring-chart.component';
export { NbpRingChartComponent } from './nbp-ring-chart.component';
import { NbpModule } from '../nbp.module';
var NbpChartModule = (function () {
    /** Module containing charting components */
    function NbpChartModule() {
    }
    return NbpChartModule;
}());
NbpChartModule = __decorate([
    NgModule({
        imports: [
            CommonModule,
            NbpModule
        ],
        declarations: [
            NbpRingChartComponent,
            NbpBarChartComponent
        ],
        providers: [],
        exports: [
            NbpRingChartComponent,
            NbpBarChartComponent
        ]
    })
    /** Module containing charting components */
], NbpChartModule);
export { NbpChartModule };
