// cjlib/nbp/nbp-slider-keypoints/nbp-slider-keypoints.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, forwardRef, Output, EventEmitter, ViewChild, ElementRef } from '@angular/core';
import { NG_VALUE_ACCESSOR, } from '@angular/forms';
/** intero autogenerato usato per id unici */
var nextId = 0;
/**
 * Provider Expression che permette la registrazione come ControlValueAccessor.
 * Permette di gestire [(ngModel)].
 * @docs-private
 */
export var NBP_SLIDER_CONTROL_VALUE_ACCESSOR = {
    provide: NG_VALUE_ACCESSOR,
    useExisting: forwardRef(function () { return NbpSliderKeyPointsComponent; }),
    multi: true
};
var NbpSliderKeyPointsComponent = (function () {
    function NbpSliderKeyPointsComponent() {
        this.propagateChange = function (_) { };
        /**
         * Identificativo univoco per l'id dello slider. Se non viene passato un valore, name viene autogenerato.
         */
        this.id = "nbp-slider-" + ++nextId;
        /**
         * Identificativo univoco per il name dello slider. Se non viene passato un valore, name viene autogenerato.
         */
        this.name = "nbp-slider-" + ++nextId;
        /**
        * Tipo di slider desiderato. Al momento è disponibile solo quello semplice.
        * Di default è il tipo 0.
        */
        this.type = 0;
        /**
         * Valore boolean che definisce la disabilitazione del componente.
         * Di default il valore è false.
         */
        this.disabled = false;
        this.range = null;
        this.config = {};
        this.keyboard = null;
        this.min = -10;
        this.max = 10;
        this.step = 1;
        this.limit = 10;
        this.connect = false;
        this.behavior = null;
        /**
         * Valore iniziale dello slider.
         * Di default il valore è 0.
         */
        this._sliderValue = 5;
        /**
         * Evento al cambio di valore dello slider.
         */
        this.nbpChangeValue = new EventEmitter();
    }
    /**
     * Implementazione interfaccia ControlValueAccessor.
     */
    NbpSliderKeyPointsComponent.prototype.writeValue = function (value) {
        if (value !== undefined) {
            this.sliderValue = value;
            this.nbpChangeValue.emit(value);
        }
    };
    NbpSliderKeyPointsComponent.prototype.registerOnChange = function (fn) {
        this.propagateChange = fn;
    };
    NbpSliderKeyPointsComponent.prototype.registerOnTouched = function () { };
    Object.defineProperty(NbpSliderKeyPointsComponent.prototype, "sliderValue", {
        get: function () {
            return this._sliderValue;
        },
        set: function (val) {
            this._sliderValue = val;
            this.propagateChange(this._sliderValue);
        },
        enumerable: true,
        configurable: true
    });
    NbpSliderKeyPointsComponent.prototype.ngAfterViewInit = function () {
        this.noUiValues = this.sliderRef.nativeElement.querySelectorAll('.noUi-value-large');
        // this.sliderRef.nativeElement.querySelectorAll('.noUi-tooltip')[0].insertAdjacentHTML('afterend', '<div class="two">two</div>')
        var buffer;
        for (var i = 0; i < this.noUiValues.length; i++) {
            buffer = this.noUiValues[i].getAttribute("data-value");
            this.noUiValues[i].addEventListener('click', this.clickOnPip.bind(this, buffer), false);
        }
    };
    NbpSliderKeyPointsComponent.prototype.clickOnPip = function (currentValueTip, mouseEvent) {
        this.sliderValue = currentValueTip;
    };
    return NbpSliderKeyPointsComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSliderKeyPointsComponent.prototype, "id", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSliderKeyPointsComponent.prototype, "name", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderKeyPointsComponent.prototype, "type", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpSliderKeyPointsComponent.prototype, "disabled", void 0);
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpSliderKeyPointsComponent.prototype, "range", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderKeyPointsComponent.prototype, "config", void 0);
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpSliderKeyPointsComponent.prototype, "keyboard", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderKeyPointsComponent.prototype, "min", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderKeyPointsComponent.prototype, "max", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderKeyPointsComponent.prototype, "step", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderKeyPointsComponent.prototype, "limit", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpSliderKeyPointsComponent.prototype, "connect", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderKeyPointsComponent.prototype, "behavior", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderKeyPointsComponent.prototype, "_sliderValue", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpSliderKeyPointsComponent.prototype, "nbpChangeValue", void 0);
__decorate([
    ViewChild('sliderRef', { read: ElementRef }),
    __metadata("design:type", ElementRef)
], NbpSliderKeyPointsComponent.prototype, "sliderRef", void 0);
NbpSliderKeyPointsComponent = __decorate([
    Component({
        selector: 'nbp-slider-keypoints',
        template: "<div class=\"row\" style=\"width:27.562em; padding-top: 4.375em;\"><nouislider id=\"\" #sliderRef *ngIf=\"type==0\" [name]=\"name\" [config]=\"config\" [(ngModel)]=\"sliderValue\"></nouislider></div><div class=\"row\" style=\"width:27.562em;margin-top:13px; margin-left:-9px;\"><div style=\"width:20px; float:left; margin-left:-5px;\">{{config.range.min}}</div><div style=\"width:20px; float: right; margin-right:-10px;\">{{config.range.max}}</div></div>",
        styles: [":host /deep/ :focus{outline:0!important;border:0}:host /deep/ .noUi-handle:before{font-family:icomoon!important;font-style:normal!important;font-weight:400!important;font-variant:normal!important;text-transform:none!important;line-height:0;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;font-size:2.187em;background-color:#fff;border-radius:2.187em;margin-left:-8px;cursor:pointer;color:#258900}:host /deep/ .noUi-handle:after{left:17px}.noUi-target *,:host /deep/ .noUi-target{-webkit-touch-callout:none;-webkit-tap-highlight-color:transparent;-webkit-user-select:none;-ms-touch-action:none;touch-action:none;-ms-user-select:none;-moz-user-select:none;user-select:none;-moz-box-sizing:border-box;box-sizing:border-box}:host /deep/ .noUi-target{position:relative;direction:ltr}.noUi-connects,:host /deep/ .noUi-base{width:100%;height:100%;position:relative;z-index:1}:host /deep/ .noUi-connects{overflow:hidden;z-index:0}.noUi-origin,:host /deep/ .noUi-connect{will-change:transform;position:absolute;z-index:1;top:0;left:0;height:100%;width:100%;-ms-transform-origin:0 0;-webkit-transform-origin:0 0;transform-origin:0 0}:host /deep/ html:not([dir=rtl]) .noUi-horizontal .noUi-origin{left:auto;right:0}:host /deep/ .noUi-vertical .noUi-origin{width:0}:host /deep/ .noUi-horizontal .noUi-origin{height:0}:host /deep/ .noUi-handle{position:absolute}.noUi-state-tap .noUi-origin,:host /deep/ .noUi-state-tap .noUi-connect{-webkit-transition:transform .3s;transition:transform .3s}:host /deep/ .noUi-state-drag *{cursor:inherit!important}:host /deep/ .noUi-horizontal{height:18px}:host /deep/ .noUi-horizontal .noUi-handle{width:0;height:0;left:-.625em;top:0;margin-left:100%}:host /deep/ .noUi-vertical{width:18px}:host /deep/ .noUi-vertical .noUi-handle{width:0;height:0;left:-6px;top:-17px}:host /deep/ html:not([dir=rtl]) .noUi-horizontal .noUi-handle{right:-.625em;left:auto}:host /deep/ .noUi-target{background:#fafafa}:host /deep/ .noUi-connects{border-radius:3px}:host /deep/ .noUi-connect{background:#3fb8af}:host /deep/ .noUi-draggable{cursor:ew-resize}:host /deep/ .noUi-vertical .noUi-draggable{cursor:ns-resize}:host /deep/ .noUi-handle{width:0;height:0;left:0;top:3px;margin-left:98%;cursor:pointer}.noUi-vertical .noUi-handle:after,:host /deep/ .noUi-vertical .noUi-handle:before{width:14px;height:1px;left:6px;top:14px}:host /deep/ .noUi-vertical .noUi-handle:after{top:17px}:host /deep/ [disabled] .noUi-connect{background:#b8b8b8}:host /deep/ [disabled].noUi-target,[disabled] .noUi-handle,[disabled].noUi-handle{cursor:not-allowed}.noUi-pips *,:host /deep/ .noUi-pips{-moz-box-sizing:border-box;box-sizing:border-box}:host /deep/ .noUi-pips{position:absolute;color:#999}:host /deep/ .noUi-value{position:absolute;white-space:nowrap;text-align:center}:host /deep/ .noUi-value-sub{color:#ccc;font-size:10px}:host /deep/ .noUi-marker{position:absolute;background:#ccc}:host /deep/ .noUi-marker-sub{background:#aaa}:host /deep/ .noUi-marker-large{background:#aaa}:host /deep/ .noUi-pips-horizontal{padding:.625em 0;height:5em;top:100%;left:0;width:100%;margin-top:-1.687em}:host /deep/ .noUi-value-horizontal{-webkit-transform:translate(-50%,50%);transform:translate(-50%,50%)}:host /deep/ .noUi-rtl .noUi-value-horizontal{-webkit-transform:translate(50%,50%);transform:translate(50%,50%)}:host /deep/ .noUi-marker-horizontal.noUi-marker{margin-left:-1px;width:.125em;height:5px}:host /deep/ .noUi-marker-horizontal.noUi-marker-sub{height:10px}:host /deep/ .noUi-marker-horizontal.noUi-marker-large{height:15px}:host /deep/ .noUi-pips-vertical{padding:0 10px;height:100%;top:0;left:100%}:host /deep/ .noUi-value-vertical{-webkit-transform:translate(0,-50%);transform:translate(0,-50%,0);padding-left:25px}:host /deep/ .noUi-rtl .noUi-value-vertical{-webkit-transform:translate(0,50%);transform:translate(0,50%)}:host /deep/ .noUi-marker-vertical.noUi-marker{width:5px;height:.125em;margin-top:-1px}:host /deep/ .noUi-marker-vertical.noUi-marker-sub{width:10px}:host /deep/ .noUi-marker-vertical.noUi-marker-large{width:15px}:host /deep/ .noUi-horizontal .noUi-tooltip{-webkit-transform:translate(-50%,0);transform:translate(-50%,0);left:50%;bottom:120%}:host /deep/ .noUi-vertical .noUi-tooltip{-webkit-transform:translate(0,-50%);transform:translate(0,-50%);top:50%;right:120%}:host /deep/ .noUi-value-horizontal.noUi-value-large{background-color:#6f6f6f;color:transparent;height:1em;width:1em;border-radius:50%;cursor:pointer}:host /deep/ .noUi-value-horizontal.noUi-value-large:hover{background-color:#258900;color:transparent;height:1em;width:1em;border-radius:50%;cursor:pointer}:host /deep/ .noUi-marker-horizontal.noUi-marker{margin-left:-1px;width:.125em;height:0}:host /deep/ .noUi-marker-horizontal.noUi-marker-sub{height:0}:host /deep/ .noUi-marker-horizontal.noUi-marker-large{height:0}:host /deep/ .noUi-horizontal{height:.125em;padding:0!important;top:0!important;width:100%!important;background-color:#6f6f6f!important}:host /deep/ .noUi-tooltip{display:block;padding:.375em .812em .375em .812em;white-space:nowrap;border-width:0;position:absolute;left:0;width:4.375em;height:2em;background:inherit;background-color:#6f6f6f;border:none;border-radius:0;box-shadow:none;font-family:'Arial Bold',Arial;font-weight:700;font-style:normal;font-size:1em;color:#fff;text-align:center;line-height:1.25em;margin-left:.5em;margin-bottom:1.875em}:host /deep/ .noUi-tooltip::after{content:\"\";position:absolute;top:100%;right:37%;border-top-width:.375em;border-right-width:.5em;border-bottom-width:.375em;border-left-width:.5em;border-style:solid;border-color:#6f6f6f transparent transparent transparent}:host /deep/ .noUi-horizontal .noUi-tooltip{-webkit-transform:translate(-50%,0);transform:translate(-50%,0);left:63%;bottom:190%}:host /deep/ .noUi-vertical .noUi-tooltip{-webkit-transform:translate(0,-50%);transform:translate(0,-50%);top:50%;right:120%}.contrasto-on .noUi-target *,.contrasto-on :host /deep/ .noUi-target{-webkit-tap-highlight-color:#000!important}.contrasto-on :host /deep/ .noUi-target{background:#000!important}.contrasto-on :host /deep/ .noUi-connect{background:#000!important}.contrasto-on :host /deep/ .icomoon-noUi-startIcon:before{background-color:#fff!important;color:#000}.contrasto-on :host /deep/ .icomoon-noUi-endIcon:before{background-color:#fff!important;color:#000}.contrasto-on :host /deep/ .noUi-handle:before{background-color:#fff!important;color:#000}.contrasto-on :host /deep/ [disabled] .noUi-connect{background:#000}.contrasto-on :host /deep/ .noUi-pips{color:#000}.contrasto-on :host /deep/ .noUi-value{color:transparent}.contrasto-on :host /deep/ .noUi-value-sub{color:transparent}.contrasto-on :host /deep/ .noUi-marker{background:#000}.contrasto-on :host /deep/ .noUi-marker-sub{background:#000}.contrasto-on :host /deep/ .noUi-marker-large{background:#000}.contrasto-on :host /deep/ .noUi-tooltip{background-color:#000;color:#fff}.contrasto-on :host /deep/ .noUi-tooltip::after{border-color:#000 transparent transparent transparent}.contrasto-on :host /deep/ .noUi-value-horizontal.noUi-value-large{background-color:#000}.contrasto-on :host /deep/ .noUi-value-horizontal.noUi-value-large:hover{background-color:#fff;border:1px solid #000} /*# sourceMappingURL=nbp-slider-keypoints.component.css.map */ "],
        // encapsulation: ViewEncapsulation.Native,
        providers: [NBP_SLIDER_CONTROL_VALUE_ACCESSOR]
    }),
    __metadata("design:paramtypes", [])
], NbpSliderKeyPointsComponent);
export { NbpSliderKeyPointsComponent };
