// cjlib/nbp/nbp-menu-operation/nbp-menu-operation.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { Router, ActivatedRoute } from '@angular/router';
import { TranslateService } from '@ngx-translate/core';
var NbpMenuOperationComponent = (function () {
    function NbpMenuOperationComponent(router, activatedRoute, translate) {
        this.router = router;
        this.activatedRoute = activatedRoute;
        this.translate = translate;
        /**
         * Permette di disabilitare la traduzzione delle voci di menu
         */
        this.nbpTranslate = true;
        /**
         * Evento emesso sulla pressione di ogni voce di menu'.
         * @param {EventEmitter<INbpItemMenu>}  nbpOnItemClick
         */
        this.nbpOnItemClick = new EventEmitter();
        /**
         * Evento emesso all'apertura del menu operazioni.
         * @param {EventEmitter<any>}  nbpOnOpen
         */
        this.nbpOnOpen = new EventEmitter();
        /**
         * Evento emesso alla chiusure del menu operazioni.
         * @param {EventEmitter<any>}  nbpOnOpen
         */
        this.nbpOnClose = new EventEmitter();
        this._isOpen = false;
        this.maxItemsSections = 0;
        this.reverse = "";
    }
    NbpMenuOperationComponent.prototype._translate = function (label) {
        if (!this.nbpTranslate) {
            return label;
        }
        return this.translate.instant(label);
    };
    NbpMenuOperationComponent.prototype._changeRoute = function (routeValue) {
        if (routeValue.external) {
            if (routeValue.link.search('http://') > -1) {
                window.location.href = routeValue.link;
            }
            else {
                window.location.href = 'http://' + routeValue.link;
            }
        }
        else {
            this.router.navigate([routeValue.link]);
        }
    };
    NbpMenuOperationComponent.prototype._onClick = function (item) {
        // console.log('click item menu: ' + JSON.stringify(item));
        this.reverse = "";
        this.nbpOnItemClick.emit(item);
        this.toggle();
    };
    /**
     * Permette apertura programmatica del BurgerMenu
     */
    NbpMenuOperationComponent.prototype.open = function () {
        var _this = this;
        this._isOpen = true;
        this.nbpOnOpen.emit();
        setTimeout(function () {
            var height = document.getElementsByClassName('first-level-menu-operation')[0].clientHeight + 20;
            _this.height = height;
        }, 100);
    };
    /**
     * Permette la chiusura programmatica del BurgerMenu
     */
    NbpMenuOperationComponent.prototype.close = function () {
        var _this = this;
        this._isOpen = false;
        this.nbpOnClose.emit();
        setTimeout(function () {
            _this.height = 0;
        }, 100);
    };
    NbpMenuOperationComponent.prototype.setMoveFirstLevel = function () {
        if (this._isOpen) {
            return 'first-level-menu-operation-open';
        }
        else {
            return 'first-level-menu-operation-close';
        }
    };
    /**
     * Permette di eseguire il toggle del BurgerMenu
     */
    NbpMenuOperationComponent.prototype.toggle = function () {
        this._isOpen ? this.close() : this.open();
    };
    NbpMenuOperationComponent.prototype.ngOnInit = function () {
        this.title = (this.nbpMenu) ? this.nbpMenu.title : "Operazione e pagamenti";
    };
    return NbpMenuOperationComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpMenuOperationComponent.prototype, "nbpMenu", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpMenuOperationComponent.prototype, "nbpTranslate", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpMenuOperationComponent.prototype, "nbpOnItemClick", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpMenuOperationComponent.prototype, "nbpOnOpen", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpMenuOperationComponent.prototype, "nbpOnClose", void 0);
NbpMenuOperationComponent = __decorate([
    Component({
        selector: 'nbp-menu-operation',template: "<div class=\"menu\" [ngClass]=\"{'closed': !_isOpen, 'open': _isOpen}\" [style.height.px]=\"height\"><div class=\"container first-level-menu-operation open\"><div class=\"row text-center\"><span class=\"title-operation\" style=\"color: #ffffff;\">{{_translate(title)}}</span></div><div class=\"row\"><div class=\"col-sm-24\"><div *ngIf=\"nbpMenu\"><div class=\"col-sm-8 section-menu p-0\" *ngFor=\"let section of nbpMenu.sections; let i = index\"><ul class=\"main-menu\"><li><a *ngFor=\"let item of section.items\" (click)=\"_onClick(item);_changeRoute(item)\"><span *ngIf=\"item.icon\" class=\"icon {{item.icon}}\" style=\"padding-top:0.25em\"></span> <span>{{_translate(item.label)}}</span></a></li></ul></div></div></div></div></div></div>",
        styles: [".menu{position:relative;width:100%;height:0;overflow-x:hidden;overflow-y:hidden;background-color:#323232;-moz-transition:height .25s ease-out;-o-transition:height .25s ease-out;transition:height .25s ease-out}.menu .icon{margin-right:.313em;font-size:1.422em!important;vertical-align:middle}.title-operation{color:#fff;font-size:1.85em}.menu.closed{height:0}.main-menu-container{padding:0}.section-menu:nth-child(2n+3) a{color:#fa9600;border:1px solid #fa9600;border-radius:4px}.section-menu:nth-child(2n+3) a:hover{background-color:#fa9600;color:#323232;border-color:#323233}.p-0{padding:0!important}.main-menu{list-style:none;padding:0 2.187em 0 2.187em;margin:0}.main-menu a{color:#ccc;display:table;width:100%;padding:0 1.25em 0 1.25em;margin:.62em 0 .62em 0;text-decoration:none;font-weight:700;cursor:pointer;height:2.813em;border:1px solid #6f6f6f;border-radius:4px}.main-menu a:active,.main-menu a:focus,.main-menu a:hover{background-color:#258900;color:#fff;border:1px solid #258900}.main-menu a span{display:table-cell;vertical-align:middle;text-align:center;text-transform:uppercase;font-size:.75em}.main-menu a span:nth-child(1){width:1.563em}.main-menu .select{background-color:#258900;color:#fff;border:.0625em solid #258900}.upper-bar{border-bottom:.0625em solid #6f6f6f}.menu-principale{color:#ccc;font-size:1em;padding:1em .625em .625em .625em;cursor:pointer}.menu-principale .icon{font-size:.875em}.first-level-menu-operation{position:absolute;left:0;visibility:hidden;right:0;opacity:0;left:-6.25em;-moz-transition:all .35s ease-out;-o-transition:all .35s ease-out;transition:all .35s ease-out}.first-level-menu-operation.open{opacity:1;left:0;visibility:visible;margin-top:.625em;-moz-transition:all .35s ease-out;-o-transition:all .35s ease-out;transition:all .35s ease-out}@keyframes example2{from{opacity:1}to{opacity:0}}@media (max-width:992px){.main-menu a{width:10.375em}}:host-context(.contrasto-on) .menu{background-color:#000}:host-context(.contrasto-on) .menu.open{border-color:#fff}:host-context(.contrasto-on) .main-menu a{color:#fff;border-color:#fff}:host-context(.contrasto-on) .main-menu a:active,:host-context(.contrasto-on) .main-menu a:focus,:host-context(.contrasto-on) .main-menu a:hover{background-color:#fff;color:#000;border-color:#fff}:host-context(.contrasto-on) .main-menu .select{background-color:#fff;color:#000;border-color:#fff}:host-context(.contrasto-on) .upper-bar{border-color:#fff}:host-context(.contrasto-on) .menu-principale{color:#fff}:host-context(.contrasto-on) .selected-first-level-menu-operation{color:#fff}.moveInTitle{opacity:0;margin-left:300px;animation-name:example;animation-duration:1s;animation-fill-mode:forwards;animation-delay:0s}.moveIn1{opacity:0;margin-left:300px;animation-name:example;animation-duration:1s;animation-fill-mode:forwards;animation-delay:.25s}.moveIn2{opacity:0;margin-left:300px;animation-name:example;animation-duration:1s;animation-fill-mode:forwards;animation-delay:.5s}.moveIn3{opacity:0;margin-left:300px;animation-name:example;animation-duration:1s;animation-fill-mode:forwards;animation-delay:.75s}.moveIn4{opacity:0;margin-left:300px;animation-name:example;animation-duration:1s;animation-fill-mode:forwards;animation-delay:1s}.moveIn5{opacity:0;margin-left:300px;animation-name:example;animation-duration:1s;animation-fill-mode:forwards;animation-delay:1.25s}.moveIn6{opacity:0;margin-left:300px;animation-name:example;animation-duration:1s;animation-fill-mode:forwards;animation-delay:1.5s}.moveIn7{opacity:0;margin-left:300px;animation-name:example;animation-duration:1s;animation-fill-mode:forwards;animation-delay:1.75s}.moveIn8{opacity:0;margin-left:300px;animation-name:example;animation-duration:1s;animation-fill-mode:forwards;animation-delay:2s}.moveIn9{opacity:0;margin-left:300px;animation-name:example;animation-duration:1s;animation-fill-mode:forwards;animation-delay:2.25s}.moveInTitleR{opacity:1;margin-left:0;animation-name:exampleR;animation-duration:1s;animation-fill-mode:forwards;animation-delay:0s}.moveInR1{opacity:1;margin-left:0;animation-name:exampleR;animation-duration:1s;animation-fill-mode:forwards;animation-delay:.25s}.moveInR2{opacity:1;margin-left:0;animation-name:exampleR;animation-duration:1s;animation-fill-mode:forwards;animation-delay:.5s}.moveInR3{opacity:1;margin-left:0;animation-name:exampleR;animation-duration:1s;animation-fill-mode:forwards;animation-delay:.75s}.moveInR4{opacity:1;margin-left:0;animation-name:exampleR;animation-duration:1s;animation-fill-mode:forwards;animation-delay:1s}.moveInR5{opacity:1;margin-left:0;animation-name:exampleR;animation-duration:1s;animation-fill-mode:forwards;animation-delay:1.25s}.moveInR6{opacity:1;margin-left:0;animation-name:exampleR;animation-duration:1s;animation-fill-mode:forwards;animation-delay:1.5s}.moveInR7{opacity:1;margin-left:0;animation-name:exampleR;animation-duration:1s;animation-fill-mode:forwards;animation-delay:1.75s}.moveInR8{opacity:1;margin-left:0;animation-name:exampleR;animation-duration:1s;animation-fill-mode:forwards;animation-delay:2s}.moveInR9{opacity:1;margin-left:0;animation-name:exampleR;animation-duration:1s;animation-fill-mode:forwards;animation-delay:2.25s}@keyframes example{from{opacity:0;margin-left:600px}to{opacity:1;margin-left:0}}@keyframes exampleR{from{opacity:1;margin-left:0}to{opacity:0;margin-left:600px}}@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none){[class*=' icon-']::before{font-size:1.375em!important}} /*# sourceMappingURL=nbp-menu-operation.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [Router,
        ActivatedRoute,
        TranslateService])
], NbpMenuOperationComponent);
export { NbpMenuOperationComponent };
