// cjlib/nbp/nbp-slider-range/nbp-slider-range-bar/nbp-slider-range-bar.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, ViewChild } from '@angular/core';
import { NbpSliderRangeBarDirective } from './nbp-slider-range-bar.directive';
/**
 * Componente slider che permette di settare un range di tempo tra due date
 */
var NbpSliderRangeBarComponent = (function () {
    function NbpSliderRangeBarComponent() {
        this.min = 0;
        this.max = 100;
        this.labels = [];
        //range slider
        this.rangeDateDays = 100;
        this.isInit = false;
    }
    NbpSliderRangeBarComponent.prototype.ngOnChanges = function () {
        this.rangeDateDays = this.daysDifference(this.min, this.max);
        this.labels = this.getLabels();
    };
    NbpSliderRangeBarComponent.prototype.setPallinoClass = function (value) {
        return (this.model.value[0] < value && value < this.model.value[1]) ? "pallino-verde" : "pallino-nero";
    };
    NbpSliderRangeBarComponent.prototype.getLabels = function () {
        var list = [];
        var mesi = ["GEN", "FEB", "MAR", "APR", "MAG", "GIU", "LUG", "AGO", "SET", "OTT", "NOV", "DIC"];
        var tmpWidth, buffer1, buffer2;
        var oldDate, buffer, buffer3, dayTmp, monthTmp, yearTmp, tmpDate;
        var diffDays;
        tmpDate = this.min;
        while (tmpDate <= this.max) {
            //calcolo label
            buffer = new Date(tmpDate);
            dayTmp = buffer.getDate();
            monthTmp = buffer.getMonth();
            yearTmp = buffer.getFullYear();
            //la label va messa solo se siamo ad inizio mese
            buffer1 = dayTmp == 1 ? mesi[monthTmp] : "";
            buffer2 = dayTmp == 1 ? yearTmp + "" : "";
            buffer3 = tmpDate;
            //calcolo data successiva
            if (monthTmp == 11) {
                monthTmp = 0;
                yearTmp++;
            }
            else {
                monthTmp++;
            }
            //calcolo width
            oldDate = tmpDate;
            tmpDate = (new Date(yearTmp, monthTmp, 1)).getTime();
            diffDays = this.daysDifference(oldDate, tmpDate);
            tmpWidth = (diffDays / this.rangeDateDays) * 100 + "%";
            list.push({
                label1: buffer1,
                label2: buffer2,
                value: buffer3,
                width: tmpWidth
            });
        }
        //sistema i valori sbagliati del loop. Il loop li setta in maniera errata
        diffDays = this.daysDifference(oldDate, this.max);
        list[list.length - 1].width = (diffDays / this.rangeDateDays) * 100 + "%";
        return list;
    };
    NbpSliderRangeBarComponent.prototype.daysDifference = function (d0, d1) {
        var diff = new Date(+d1).setHours(12) - new Date(+d0).setHours(12);
        return Math.round(diff / 8.64e7);
    };
    return NbpSliderRangeBarComponent;
}());
__decorate([
    ViewChild(NbpSliderRangeBarDirective),
    __metadata("design:type", NbpSliderRangeBarDirective)
], NbpSliderRangeBarComponent.prototype, "bar", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderRangeBarComponent.prototype, "min", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderRangeBarComponent.prototype, "max", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangeBarComponent.prototype, "model", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangeBarComponent.prototype, "opts", void 0);
NbpSliderRangeBarComponent = __decorate([
    Component({
        selector: 'nbp-slider-range-bar',
        template: "<div class=\"slider-bar-container\"><div class=\"slider-bar\"><div nbpSliderRangeBar class=\"slider-bar-highlight\" [min]=\"min\" [max]=\"max\" [opts]=\"opts\" [model]=\"model\"></div></div><div class=\"slider-points\"><ng-container *ngFor=\"let item of labels; let first = first\"><div class=\"slider-label-item margin-color\" [ngStyle]=\"{'min-width': item.width }\"><div *ngIf=\"item.label1+item.label2\" [ngClass]=\"setPallinoClass(item.value)\"><span>X</span></div></div></ng-container></div><div class=\"slider-labels\"><ng-container *ngFor=\"let item of labels; let first = first\"><div class=\"slider-label-item margin-color\" [ngStyle]=\"{'min-width': item.width }\"><div><span>{{item.label1}}</span></div><div><span>{{item.label2}}</span></div></div></ng-container></div></div>",
        styles: [".margin-color{color:#6f6f6f;font-weight:700}.pallino-nero{background-color:#6f6f6f;color:transparent;font-size:0;height:16px;width:16px;border-radius:50%}.pallino-verde{background-color:#39ba2a;color:transparent;font-size:0;height:16px;width:16px;border-radius:50%}.slider-bar-container{position:relative}.slider-bar-container .slider-bar{height:4px;width:100%;background-color:#6f6f6f;position:relative;border-radius:2px}.slider-bar-container .slider-bar .slider-bar-highlight{position:absolute;left:0;width:50%;height:100%;background-color:#39ba2a;border-radius:2px}.slider-bar-container .slider-labels{margin-top:19px;margin-left:0;width:99.3%;display:flex}.slider-bar-container .slider-labels::after{display:block;position:absolute;content:'';height:1px;width:1px;background-color:#bbb;top:6px;right:0}.slider-bar-container .slider-labels .slider-label-item{text-align:left}.slider-bar-container .slider-labels .slider-label-item span{transform:translate(-20%,0);position:relative;display:inline-block}.slider-bar-container .slider-labels .slider-label-item:first-child span{transform:translate(-20%,0)}.slider-bar-container .slider-labels .slider-label-item:last-child span{transform:translate(-20%,0)}.slider-bar-container .slider-labels .slider-label-item::after{display:block;position:absolute;content:'';height:1px;width:1px;background-color:#bbb;top:6px}.slider-bar-container .slider-points{margin-top:-10px;width:99.3%;display:flex}.slider-bar-container .slider-points::after{display:block;position:absolute;content:'';height:1px;width:1px;background-color:#bbb;top:6px;right:0}.slider-bar-container .slider-points .slider-label-item{text-align:left}.slider-bar-container .slider-points .slider-label-item span{transform:translate(-90%,0);position:relative;display:inline-block}.slider-bar-container .slider-points .slider-label-item:first-child span{transform:translate(-20%,0)}.slider-bar-container .slider-points .slider-label-item:last-child span{transform:translate(20%,0)}.slider-bar-container .slider-points .slider-label-item::after{display:block;position:absolute;content:'';height:1px;width:1px;background-color:#bbb;top:6px} /*# sourceMappingURL=nbp-slider-range-bar.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpSliderRangeBarComponent);
export { NbpSliderRangeBarComponent };
