// cjlib/nbp/nbp-slider-range/nbp-slider-range-bar/nbp-slider-range-bar.directive.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Directive, Input, ElementRef } from '@angular/core';
var NbpSliderRangeBarDirective = (function () {
    // parent: any;
    function NbpSliderRangeBarDirective(el) {
        this.el = el;
        this.range = 100;
    }
    NbpSliderRangeBarDirective.prototype.ngOnInit = function () {
        var _this = this;
        this.range = this.max - this.min;
        this.setBarPosition();
        /* Rileva Model Changes */
        this
            .model
            .valueChanges
            .subscribe(function (value) {
            /* Ridisegna Bar */
            _this.setBarPosition();
        });
    };
    // public reRenderBar() {
    //   this.range = this.max - this.min;
    //   this.setBarPosition();
    // }
    NbpSliderRangeBarDirective.prototype.setBarPosition = function () {
        var value = this.model.value;
        var min = value[0];
        var max = value[1];
        var leftPos = this.calculateNextPosFromValue(min);
        var rightPos = this.calculateNextPosFromValue(max);
        var leftPosPerc = leftPos / this.getParent().width * 100;
        var widthPerc = (rightPos - leftPos) / this.getParent().width * 100;
        this.el.nativeElement.style.left = leftPosPerc + "%";
        this.el.nativeElement.style.width = widthPerc + "%";
    };
    NbpSliderRangeBarDirective.prototype.calculateNextPosFromValue = function (value) {
        return ((value - this.min) / this.range) * this.getParent().width;
    };
    NbpSliderRangeBarDirective.prototype.getParent = function () {
        return this.el.nativeElement.parentElement.getBoundingClientRect();
    };
    return NbpSliderRangeBarDirective;
}());
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangeBarDirective.prototype, "model", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangeBarDirective.prototype, "opts", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangeBarDirective.prototype, "min", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangeBarDirective.prototype, "max", void 0);
NbpSliderRangeBarDirective = __decorate([
    Directive({
        selector: '[nbpSliderRangeBar]'
    }),
    __metadata("design:paramtypes", [ElementRef])
], NbpSliderRangeBarDirective);
export { NbpSliderRangeBarDirective };
