// cjlib/nbp/nbp-etichetta-check/nbp-etichetta-check.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, ElementRef, forwardRef, Input, Output, EventEmitter, ViewChild } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
/**
 * Componente etichetta dal funzionamento analogo a quello di una checkbox.
 */
var NbpEtichettaCheckComponent = NbpEtichettaCheckComponent_1 = (function () {
    function NbpEtichettaCheckComponent() {
        var _this = this;
        /**
         * input che determina il valore quando il bottone è selezionato
         */
        this.trueValue = true;
        /**
        * input che determina il valore quando il bottone non è selezionato
        */
        this.falseValue = false;
        /**
         * evento per la rimozione del bottone
         */
        this.removeButtonEvt = new EventEmitter();
        /**
         * evento lanciato al cambiamento di stato del bottone
         */
        this.change = new EventEmitter();
        this.propagateChange = function (_) { };
        this._overIcon = false;
        /**
         * evento per la rimozione del bottone
         */
        this.removeButton = function () {
            _this.removeButtonEvt.emit(_this.key);
        };
    }
    NbpEtichettaCheckComponent.prototype.writeValue = function (value) {
        if ((value !== undefined) && (value !== null)) {
            this.checked = value;
        }
    };
    NbpEtichettaCheckComponent.prototype.registerOnChange = function (fn) {
        this.propagateChange = fn;
    };
    NbpEtichettaCheckComponent.prototype.registerOnTouched = function (fn) {
        this.onTouchedCallback = fn;
    };
    ;
    /**
     *
     * funzione innescata da un cambiamento esterno del valore del bottone/check
     */
    NbpEtichettaCheckComponent.prototype.onChange = function (event) {
        this.checked = this.checkValRef.nativeElement.checked;
        this.checked = !this.checked;
        var sendVal = (this.checked) ? this.trueValue : this.falseValue;
        this.propagateChange(sendVal);
        this.change.emit(this.checked);
    };
    return NbpEtichettaCheckComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpEtichettaCheckComponent.prototype, "key", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpEtichettaCheckComponent.prototype, "trueValue", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpEtichettaCheckComponent.prototype, "falseValue", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NbpEtichettaCheckComponent.prototype, "removeButtonEvt", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NbpEtichettaCheckComponent.prototype, "change", void 0);
__decorate([
    ViewChild('checkVal'),
    __metadata("design:type", ElementRef)
], NbpEtichettaCheckComponent.prototype, "checkValRef", void 0);
NbpEtichettaCheckComponent = NbpEtichettaCheckComponent_1 = __decorate([
    Component({
        selector: 'nbp-etichetta-check',
        template: "<div class=\"etic-btn\" (click)=\"onChange($event)\" [ngClass]=\"{'etic-btn-check'   :checked, 'etic-btn-no-over' :(!_overIcon && !checked)}\"><input type=\"checkbox\" #checkVal style=\"display: none;\" [checked]=\"checked\" [hidden]=\"true\" (change)=\"onChange($event)\"> <i *ngIf=\"checked\" class=\"icomoon-Simboli_Risorsa-182 pull-left etic-icon-check\"></i> <span class=\"etic-btn-span\">{{key|uppercase }} </span><button class=\"icomoon-Simboli_Risorsa-22 pull-right etic-icon-rem\" (mouseenter)=\"_overIcon=true\" (mouseleave)=\"_overIcon=false\" (click)=\"removeButton();$event.stopPropagation();\"></button></div>",
        styles: [".etic-btn{border:0;min-width:130px;height:30px;display:inline-block;font-family:'Arial Bold',Arial;font-size:1em;font-weight:700;color:#6f6f6f;text-decoration:none;text-transform:uppercase;text-align:center;line-height:27px;position:relative;cursor:pointer;background:#f8f8f8;-webkit-box-shadow:0 3px 3px 0 rgba(0,0,0,.25);-moz-box-shadow:0 3px 3px 0 rgba(0,0,0,.25);box-shadow:0 3px 3px 0 rgba(0,0,0,.25);text-align:left}.etic-btn-span{margin-left:5px;font-size:.75em;line-height:30px;vertical-align:middle}.etic-btn-check{color:#fff;background:#258900;-webkit-box-shadow:inset 0 0 4px 1px #164602;-moz-box-shadow:inset 0 0 4px 1px #164602;box-shadow:inset 0 0 4px 1px #164602}.etic-icon-rem{font-size:20px;vertical-align:middle;font-weight:700;line-height:30px;background-color:transparent;border:none}.etic-icon-check{font-size:12px;vertical-align:middle;font-weight:700;line-height:30px;padding-left:5px;padding-top:1px}button:active div,button:active i,button:active span,button:focus div,button:focus i,button:focus span{position:relative;top:0;left:0} /*# sourceMappingURL=nbp-etichetta-check.component.css.map */ "],
        providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(function () { return NbpEtichettaCheckComponent_1; }),
                multi: true
            }
        ]
    }),
    __metadata("design:paramtypes", [])
], NbpEtichettaCheckComponent);
export { NbpEtichettaCheckComponent };
var NbpEtichettaCheckComponent_1;
