// cjlib/nbp/nbp-accordion-showmore-circle/nbp-accordion-showmore-circle.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { NbpAccordionEvent } from '../nbp-accordion/nbp-accordion.interfaces';
var nextId = 0;
var NbpAccordionShowmoreCircleComponent = (function () {
    function NbpAccordionShowmoreCircleComponent() {
        this.id = "nbp-accordion-showmore-circle-" + ++nextId;
        /**
         * (Opzionale) Booleano che pilota la visualizzazione iniziale dell’accordion collassato o espanso.
         *Default accordion collassato.
         */
        this.nbpCollapsed = true;
        /**
         * (Opzionale) Stringa che indica l'allineamento del Title desiderato.
         *  Possibili valori : right, left, center
         *  Default allineamento : center.
         */
        this.nbpTitleAlign = "center";
        /**
         * (Opzionale) Stringa che indica l'allineamento del Body desiderato.
         *  Possibili valori : right, left, center
         *  Default allineamento : right.
         */
        this.nbpBodyAlign = "right";
        /**
       * Evento emesso sul toggle del componente.
       * @param {EventEmitter<NbpAccordionEvent>}  nbpOnToggle
       */
        this.nbpOnToggle = new EventEmitter();
    }
    //     Methods
    /**
     * metodo utilizzabile per l’apertura programmatica
     */
    NbpAccordionShowmoreCircleComponent.prototype.open = function () {
        this.nbpCollapsed = false;
    };
    /**
     * metodo utilizzabile per la chiusura programmatica
     */
    NbpAccordionShowmoreCircleComponent.prototype.close = function () {
        this.nbpCollapsed = true;
    };
    /**
     * metodo utilizzabile per verificare se il componente è collassato
     */
    NbpAccordionShowmoreCircleComponent.prototype.isCollapsed = function () {
        return this.nbpCollapsed;
    };
    NbpAccordionShowmoreCircleComponent.prototype._toggleCollapse = function () {
        this.nbpCollapsed = !this.nbpCollapsed;
        this._onToggle();
    };
    NbpAccordionShowmoreCircleComponent.prototype._onToggle = function () {
        var acordionEvent = new NbpAccordionEvent();
        // acordionEvent.source = this;
        acordionEvent.isCollapsed = this.nbpCollapsed;
        this.nbpOnToggle.emit(acordionEvent);
    };
    return NbpAccordionShowmoreCircleComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionShowmoreCircleComponent.prototype, "id", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionShowmoreCircleComponent.prototype, "nbpTitle", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpAccordionShowmoreCircleComponent.prototype, "nbpCollapsed", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionShowmoreCircleComponent.prototype, "nbpTitleAlign", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionShowmoreCircleComponent.prototype, "nbpBodyAlign", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpAccordionShowmoreCircleComponent.prototype, "nbpOnToggle", void 0);
NbpAccordionShowmoreCircleComponent = __decorate([
    Component({
        selector: 'nbp-accordion-showmore-circle',template: "<div><div class=\"panel-heading\" role=\"tab\" [id]=\"id\" [style.text-align]=\"nbpTitleAlign\" (click)=\"_toggleCollapse()\"><h4 class=\"panel-title\"><span>{{nbpTitle}}</span> <a role=\"button\" class=\"collapsed\"><span class=\"accordion-arrow-round\" [ngClass]=\"{'icomoon-Simboli_Risorsa-15':isCollapsed() , 'icomoon-Simboli_Risorsa-14':!isCollapsed()}\"></span></a></h4></div><div class=\"panel-collapse\" [style.text-align]=\"nbpBodyAlign\" [ngClass]=\"{'in panel-open':!isCollapsed()}\"><div class=\"panel-body\"><ng-content></ng-content></div></div></div>",
        animations: [],
        styles: ["a:focus,a:hover{text-decoration:none}.panel{border:1px!important;border-color:#fff;box-shadow:none}.panel .panel-open{background-color:#f2f2f2!important;border:1px!important;border-color:#6f6f6f}.panel-heading{border:1px!important;box-shadow:none;color:#258900;border-color:#6f6f6f;background-color:#fff;padding:1em 1.25em 1em 0!important;height:50px}.panel-heading:hover{cursor:pointer}.panel-heading h4{margin:0;font-size:1em}.panel-heading .panel-title{color:green;font-weight:700;font-size:.75em!important;text-transform:uppercase}.panel-heading .accordion-arrow-round{color:#000;position:relative;font-size:30px;vertical-align:middle;margin-left:8px}.panel-body{margin-top:5px;border:1px!important;box-shadow:none;background-color:#f2f2f2!important;border-radius:0 0 2px 2px!important}.panel-collapse{width:100%;margin-top:5px;border:1px!important;box-shadow:none;background-color:#f2f2f2!important;border-radius:0 0 2px 2px!important;max-height:0;transition:max-height .15s ease-out;overflow:hidden}.panel-collapse.in{max-height:1000em;transition:max-height .25s ease-in}.panelTitle{font-size:.875em;font-weight:700}:host-context(.contrasto-on) .panel .panel-open{background-color:#fff!important;color:#000}:host-context(.contrasto-on) .panel .panel-open .panel-title{color:#000}:host-context(.contrasto-on) .panel .panel-open .stringa{color:#000}:host-context(.contrasto-on) .panel .panel-open .accordion-arrow{color:#000}:host-context(.contrasto-on) .panel-heading{background-color:#fff!important}:host-context(.contrasto-on) .panel-heading .panel-title{color:#000}:host-context(.contrasto-on) .panel-heading .accordion-arrow{color:#000}:host-context(.contrasto-on) .panel-heading .stringa{color:#000}:host-context(.contrasto-on) .panel-body{background-color:#fff!important;border-top:none!important;color:#000}:host-context(.contrasto-on) .accordion-green-header>.panel-heading{background-color:#000!important}:host-context(.contrasto-on) .accordion-green-header>.panel-heading:hover{background-color:#fff!important;color:#000;border:1px solid #000}:host-context(.contrasto-on) .accordion-green-header>.panel-open{background-color:#fff!important;color:#000;border:1px solid #000!important}:host-context(.contrasto-on) .panel-collapse{background-color:#fff!important} /*# sourceMappingURL=nbp-accordion-showmore-circle.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpAccordionShowmoreCircleComponent);
export { NbpAccordionShowmoreCircleComponent };
