// cjlib/nbp/nbp-bottom-page-button/nbp-bottom-page-button.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input } from '@angular/core';
import { TranslateService } from '@ngx-translate/core';
/**
 * componente per la visualizzazione di una fascia di bottoni a fondo pagina
 */
var NbpBottomPageButtonComponent = (function () {
    function NbpBottomPageButtonComponent(translate) {
        this.translate = translate;
        /**
         * Permette di disabilitare la traduzzione delle voci di menu
         */
        this.nbpTranslate = true;
        this.rowNumber = [];
        this._buildStyle = function (type, elem) {
            return (elem && elem.css && elem.css[type]) ? elem.css[type] : {};
        };
    }
    NbpBottomPageButtonComponent.prototype._translate = function (label) {
        if (!this.nbpTranslate) {
            return label;
        }
        return this.translate.instant(label);
    };
    NbpBottomPageButtonComponent.prototype.countRow = function (items) {
        var res = 0;
        if (items && items > 0) {
            var intVal = Math.floor(items / 3);
            ((items % 3) > 0) ? res = intVal + 1 : res = intVal;
        }
        return Array(res).fill(0).map(function (x, i) { return i; });
    };
    // }
    NbpBottomPageButtonComponent.prototype.ngOnInit = function () {
        if (this.nbpFixedWidht === undefined) {
            this.nbpFixedWidht = '960px';
        }
        var rowCount = 0;
        this.rowNumber = [];
        if (this.nbpButtonActionList && this.nbpButtonActionList.length > 0) {
            this.rowNumber = this.countRow(this.nbpButtonActionList.length);
        }
    };
    NbpBottomPageButtonComponent.prototype.ngOnChanges = function (changes) {
        var buttonActionList = changes.nbpButtonActionList;
        if (buttonActionList.currentValue && buttonActionList.currentValue.length > 0) {
            this.rowNumber = this.countRow(buttonActionList.currentValue.length);
        }
    };
    return NbpBottomPageButtonComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpBottomPageButtonComponent.prototype, "nbpButtonActionList", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpBottomPageButtonComponent.prototype, "nbpTitleBottom", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpBottomPageButtonComponent.prototype, "nbpFixedWidht", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpBottomPageButtonComponent.prototype, "nbpTranslate", void 0);
NbpBottomPageButtonComponent = __decorate([
    Component({
        selector: 'nbp-bottom-page-button',
        template: "<div class=\"sfondo-div\"><div class=\"u2056_div container\"><div class=\"row no-margin\"><div class=\"text-center title-button\">{{_translate(nbpTitleBottom)}}</div></div><div class=\"row no-margin row-button-line text-center\" *ngFor=\"let row of rowNumber; let rowIndex = index\"><ng-container *ngFor=\"let item of [0,1,2]; let indexButton = index\"><div class=\"col-width\" *ngIf=\"!!nbpButtonActionList[((rowIndex * 3) + indexButton)]\"><button class=\"btn-bottom\" (click)=\"nbpButtonActionList[((rowIndex * 3) + indexButton)].action($event)\"><div class=\"btn-boottom-container\"><div class=\"btn-boottom-icon\" [ngStyle]=\"_buildStyle('icon',nbpButtonActionList[((rowIndex * 3) + indexButton)])\"><span class=\"{{nbpButtonActionList[((rowIndex * 3) + indexButton)].icon}}\"></span></div><div class=\"btn-boottom-label\" [ngStyle]=\"_buildStyle('text',nbpButtonActionList[((rowIndex * 3) + indexButton)])\"><span>{{_translate(nbpButtonActionList[((rowIndex * 3) + indexButton)].name)}}</span></div></div></button></div></ng-container></div></div></div>",
        styles: [".sfondo-div{width:100%;background-color:#258900}.u2056_div{border-width:0;margin-left:auto;margin-right:auto;left:0;right:0;bottom:0;min-width:960px;background:inherit;background-color:#258900;border:none;border-radius:0;-moz-box-shadow:none;-webkit-box-shadow:none;box-shadow:none;font-size:1.5em;color:#fff}.title-button{padding-top:20px;padding-bottom:25px}.col-width{padding-left:4.64%;padding-right:4.64%;width:33.33333333%;float:left}.no-margin{margin-left:0;margin-right:0}.row-button-line{padding-bottom:20px}.btn-bottom{border-radius:2px!important;border-style:solid;border-width:1px;width:250px;height:50px;text-align:center;background-color:#258900;font:'Arial Bold';font-size:.75em;color:#fff}.btn-bottom:hover{background-color:#323232;color:#fff;border-color:#323232}.btn-boottom-container{display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex;flex-flow:row nowrap;align-items:center;justify-content:center;width:100%}.btn-boottom-icon{-webkit-box-flex:0 0 42px;-moz-box-flex:0 0 42px;-webkit-flex:0 0 42px;-ms-flex:0 0 42px;flex:0 0 42px;font-size:32px;margin-top:4px}.btn-boottom-label{padding-left:10px;font:'Arial Bold';font-size:.75em;color:#fff;max-width:12.33em;overflow:hidden;text-overflow:clip} /*# sourceMappingURL=nbp-bottom-page-button.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [TranslateService])
], NbpBottomPageButtonComponent);
export { NbpBottomPageButtonComponent };
