// cjlib/nbp/nbp-sequenziale-sottostep-generic/nbp-sequenziale-sottostep-generic.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter, HostListener } from '@angular/core';
import { NbpSize } from './../nbp-commons/nbp-commons.enums';
var NbpSequenzialeSottostepGenericComponent = (function () {
    function NbpSequenzialeSottostepGenericComponent() {
        /**
         * Array di Json degli step.
         *
         * esempio formato:
         * [
         *   { active: true, fillPercent: 25, label: 'step 1', stepDimension: NbpStepDimension.GRANDE },
         *   { active: false, fillPercent: 0, label: 'step 2', stepDimension: NbpStepDimension.GRANDE },
         *   { active: false, fillPercent: 0, label: 'step 3', stepDimension: NbpStepDimension.GRANDE },
         *   { active: false, fillPercent: 0, label: 'step 4', stepDimension: NbpStepDimension.GRANDE }
         * ];
         *
         */
        this.steps = [];
        /**
         * Array di Json dei sottostep di ciascun step.
         *
         * esempio formato:
         *
         * [{
         *   currentParentLabel: 'Step 1',
         *   pages: [{label: 'Sottostep 1'},
         *           {label: 'Sottostep 2'},
         *           {label: 'Sottostep 3'},
         *           {label: 'Sottostep 4'},
         *           {label: 'Sottostep 5'}]
         *  },
         *
         *  ........................
         *  ........................
         *
         * }]
         *
         */
        this.pagingSteps = [];
        /**
         * Evento generato al cambio step.
         * Fornisce in output json costituito da :
         * {
         *  step: valore step corrente
         *  sottoStep: valore sottoStep corrente
         *  disablePrev: stato pulsante Prev
         *  disableNext: stato pulsante Next
         * }
         */
        this.stepEvent = new EventEmitter();
        //dati uso interno
        this.disablePrev = true;
        this.disableNext = false;
        //dati da fornire in output su event emitter
        this.currentStep = 0;
        this.currentPagingStep = 0;
        //variabili locali da non toccare
        this.lg = NbpSize.LG;
        this.sticky = false;
    }
    NbpSequenzialeSottostepGenericComponent.prototype.ngOnInit = function () {
        this.stepperUpdate();
    };
    Object.defineProperty(NbpSequenzialeSottostepGenericComponent.prototype, "navigateOperation", {
        /**
         * Operazione di navigazione richiesta.
         *
         * valori ammessi: "NEXT, PREV"
         *
         */
        set: function (value) {
            if (!value)
                return;
            value = value.toUpperCase();
            if (value.indexOf("NEXT") != -1)
                this.Next();
            if (value.indexOf("PREV") != -1)
                this.Prev();
        },
        enumerable: true,
        configurable: true
    });
    NbpSequenzialeSottostepGenericComponent.prototype.Prev = function () {
        if (this.currentStep <= 0 && this.currentPagingStep <= 0) {
            return;
        }
        var allPagingSteps = this.getAllPagingSteps(this.currentStep);
        if (allPagingSteps.length > 0 && this.currentPagingStep > 0) {
            this.currentPagingStep--;
            this.steps[this.currentStep].fillPercent = this.getFillPercent(this.currentPagingStep, allPagingSteps.length);
        }
        else {
            this.changeStep(this.currentStep, this.currentStep - 1);
        }
        this.manageButtons();
    };
    NbpSequenzialeSottostepGenericComponent.prototype.Next = function () {
        var allPagingSteps = this.getAllPagingSteps(this.currentStep);
        if (this.currentStep >= this.steps.length - 1 && this.currentPagingStep >= allPagingSteps.length) {
            this.steps[this.currentStep].fillPercent = 100;
            // this.currentStep++;
            // this.manageButtons();
            this.manageButtons();
            return;
        }
        if (allPagingSteps.length > 0 && this.currentPagingStep < allPagingSteps.length) {
            this.currentPagingStep++;
            this.steps[this.currentStep].fillPercent = this.getFillPercent(this.currentPagingStep, allPagingSteps.length);
            if (this.currentPagingStep == allPagingSteps.length) {
                this.changeStep(this.currentStep, this.currentStep + 1);
            }
        }
        else {
            this.changeStep(this.currentStep, this.currentStep + 1);
        }
        this.manageButtons();
    };
    NbpSequenzialeSottostepGenericComponent.prototype.getAllPagingSteps = function (stepIndex) {
        if (this.pagingSteps[stepIndex] && this.pagingSteps[stepIndex].pages) {
            return this.pagingSteps[stepIndex].pages;
        }
        else {
            return [];
        }
    };
    NbpSequenzialeSottostepGenericComponent.prototype.getFillPercent = function (pageIndex, pagesLength) {
        return (100 * (pageIndex + 1) / pagesLength);
    };
    NbpSequenzialeSottostepGenericComponent.prototype.changeStep = function (oldStep, newStep) {
        if (newStep >= this.steps.length) {
            return;
        }
        this.steps[oldStep].fillPercent = oldStep < newStep ? 100 : 0;
        this.steps[oldStep].active = false;
        this.currentStep = newStep;
        this.steps[newStep].active = true;
        var allPagingStepsNew = this.getAllPagingSteps(newStep);
        if (allPagingStepsNew.length > 0) {
            if (oldStep > newStep) {
                this.currentPagingStep = allPagingStepsNew.length - 1;
            }
            else {
                this.currentPagingStep = 0;
            }
            this.steps[newStep].fillPercent = this.getFillPercent(this.currentPagingStep, allPagingStepsNew.length);
        }
        else {
            this.currentPagingStep = 0;
            this.steps[newStep].fillPercent = 100;
        }
    };
    NbpSequenzialeSottostepGenericComponent.prototype.manageButtons = function () {
        var allPagingSteps = this.getAllPagingSteps(this.currentStep);
        // if(this.currentStep >= this.steps.length - 1 && this.currentPagingStep >= allPagingSteps.length - 1) {
        if (this.currentStep >= this.steps.length - 1 && this.steps[this.currentStep].fillPercent == 100) {
            this.disableNext = true;
        }
        else {
            this.disableNext = false;
        }
        if (this.currentStep <= 0 && this.currentPagingStep <= 0) {
            this.disablePrev = true;
        }
        else {
            this.disablePrev = false;
        }
        this.stepperUpdate();
    };
    NbpSequenzialeSottostepGenericComponent.prototype.stepperUpdate = function () {
        var result = {
            step: this.currentStep,
            sottoStep: this.currentPagingStep,
            disablePrev: this.disablePrev,
            disableNext: this.disableNext
        };
        this.stepEvent.emit(result);
    };
    NbpSequenzialeSottostepGenericComponent.prototype.onWindowScroll = function () {
        var number = window.pageYOffset || document.documentElement.scrollTop || document.body.scrollTop || 0;
        if (number > 70) {
            this.sticky = true;
        }
        else {
            this.sticky = false;
        }
    };
    return NbpSequenzialeSottostepGenericComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpSequenzialeSottostepGenericComponent.prototype, "steps", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSequenzialeSottostepGenericComponent.prototype, "pagingSteps", void 0);
__decorate([
    Input(),
    __metadata("design:type", String),
    __metadata("design:paramtypes", [String])
], NbpSequenzialeSottostepGenericComponent.prototype, "navigateOperation", null);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NbpSequenzialeSottostepGenericComponent.prototype, "stepEvent", void 0);
__decorate([
    HostListener("window:scroll", []),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NbpSequenzialeSottostepGenericComponent.prototype, "onWindowScroll", null);
NbpSequenzialeSottostepGenericComponent = __decorate([
    Component({
        selector: 'nbp-sequenziale-sottostep-generic',template: "<div style=\"margin-top: -0.8em;\"><nbp-progress [nbpConfiguration]=\"steps\"></nbp-progress></div><div class=\"line-separator\" [ngClass]=\"{'sticky-line' : sticky}\"></div><div class=\"container main-container\" style=\"min-height: 100%\"><div class=\"page-steps-container col-md-24 border-top-white\" [ngClass]=\"{'sticky' : sticky}\" *ngIf=\"pagingSteps[currentStep] && pagingSteps[currentStep].pages && pagingSteps[currentStep].pages.length > 0\"><div class=\"container pad-t-12 pad-b-12 pad-l-10 pad-r-10\"><ul class=\"page-steps pull-left\"><ng-container *ngFor=\"let page of pagingSteps[currentStep].pages; let pageIndex= index\"><li class=\"page-step\" *ngIf=\"pageIndex <= currentPagingStep\"><div class=\"col\"><span class=\"page-steps-number\" [ngClass]=\"{'active': pageIndex === currentPagingStep, 'done': pageIndex < currentPagingStep}\">{{pageIndex + 1}}</span><div class=\"row\"><label class=\"page-step-label\" *ngIf=\"pageIndex === currentPagingStep\">{{pagingSteps[currentStep].currentParentLabel}}</label><br><label class=\"page-steps-label\" *ngIf=\"pageIndex === currentPagingStep\">{{page.label}}</label></div></div></li></ng-container></ul><ul class=\"page-steps pull-right\"><ng-container *ngFor=\"let page of pagingSteps[currentStep].pages; let pageIndex = index\"><li class=\"page-step\" *ngIf=\"pageIndex > currentPagingStep\"><span class=\"page-steps-number\">{{pageIndex + 1}}</span></li></ng-container></ul></div></div></div>",
        styles: [".sticky{position:fixed;background-color:#fff;z-index:9040;left:10px;top:93px;width:100%;-moz-transition:all .25s ease-out;-o-transition:all .25s ease-out;transition:all .25s ease-out}.line-separator{z-index:9030;position:absolute;width:100%;height:3.75em;border-bottom:1px solid rgba(204,204,204,.5)}.line-separator.sticky-line{position:fixed!important;background-color:#fff;left:0;top:93px;-moz-transition:all .25s ease-out;-o-transition:all .25s ease-out;transition:all .25s ease-out}.page-steps{list-style:none;padding:0;margin:0;height:2em}.page-steps li{display:inline-block;height:2em;line-height:2em}.page-steps li .row{line-height:0}.page-steps .page-step .col{display:inline-table}.page-steps .page-step .row{display:table-cell}.page-steps .page-steps-number{display:inline-block;border:1px solid #6f6f6f;font-weight:700;color:#6f6f6f;text-align:center;-webkit-border-radius:50%;-moz-border-radius:50%;border-radius:50%;width:2em;height:2em;font-size:.785em;line-height:2em;margin:0 .125em 0 .125em}.page-steps .page-steps-number.active{color:#258900;border:1px solid #258900}.page-steps .page-steps-number.done{color:#fff;background-color:#258900;border-color:#258900}.page-steps .page-step-label{font-weight:400;color:#258900;font-size:.6875em!important;line-height:.5em;text-transform:uppercase;margin-left:.5em;margin-top:.4em}.page-steps .page-steps-label{font-weight:700;color:#258900;font-size:.785em!important;line-height:1em;text-transform:uppercase;margin-left:.5em;margin-top:.1em!important}.page-steps-container{background-color:#fff}@media (min-width:768px){.page-steps .page-steps-label{font-size:.938em}.page-steps .page-steps-number{font-size:.938em}}@-moz-document url-prefix(){.page-steps .page-step .col{display:inline-flex}}@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none){.page-steps .page-step .col .row{top:-10px}}:host-context(.contrasto-on) .page-steps .page-steps-number{color:#000}:host-context(.contrasto-on) .page-steps .page-steps-number.active{color:#000;border-color:#000}:host-context(.contrasto-on) .page-steps .page-steps-number.done{color:#fff;background-color:#000;border-color:#000}:host-context(.contrasto-on) .page-steps .page-step-label{color:#000}:host-context(.contrasto-on) .page-steps .page-steps-label{color:#000}:host-context(.contrasto-on) .page-steps-container{background-color:#fff} /*# sourceMappingURL=nbp-sequenziale-sottostep-generic.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpSequenzialeSottostepGenericComponent);
export { NbpSequenzialeSottostepGenericComponent };
