// cjlib/nbp/nbp-calendar-generic/@ng-bootstrap/datepicker/datepicker-reply-month-view.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
var NgbDatepickerReplyMonthView = (function () {
    function NgbDatepickerReplyMonthView() {
    }
    NgbDatepickerReplyMonthView.prototype.isMuted = function () { return false; };
    ;
    NgbDatepickerReplyMonthView.prototype.abilitaMesi = function (data) {
        if (this.startDateEnabled.month <= data.number && data.number <= this.endDateEnabled.month && (this.startDateEnabled.year <= this.year && this.year <= this.endDateEnabled.year)) {
            return true;
        }
        return false;
    };
    return NgbDatepickerReplyMonthView;
}());
__decorate([
    Input(),
    __metadata("design:type", Number)
], NgbDatepickerReplyMonthView.prototype, "currentMonth", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NgbDatepickerReplyMonthView.prototype, "mese", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NgbDatepickerReplyMonthView.prototype, "year", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NgbDatepickerReplyMonthView.prototype, "disabled", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NgbDatepickerReplyMonthView.prototype, "focused", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NgbDatepickerReplyMonthView.prototype, "selected", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NgbDatepickerReplyMonthView.prototype, "startDateEnabled", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NgbDatepickerReplyMonthView.prototype, "endDateEnabled", void 0);
NgbDatepickerReplyMonthView = __decorate([
    Component({
        selector: '[ngbDatepickerReplyMonthView]',
        changeDetection: ChangeDetectionStrategy.OnPush,
        host: {
            'class': 'btn-secondary-month',
            '[class.bg-primary]': 'selected',
            '[class.text-white]': 'selected',
            '[class.text-muted]': 'isMuted()',
            '[class.outside]': 'isMuted()',
            '[class.active]': 'focused'
        },
        template: "<div [ngClass]=\" abilitaMesi(mese) ? 'ngbdatepickerreplymonthview' : 'ngbdatepickerreplymonthviewdisabled disabledView'\">{{ mese.name }}</div>",
        styles: [":host{text-align:center;width:2rem;height:2rem;line-height:2rem;border-radius:.25rem}:host.outside{opacity:.5} /*# sourceMappingURL=datepicker-reply-month-view.css.map */ "],
    })
], NgbDatepickerReplyMonthView);
export { NgbDatepickerReplyMonthView };
