// cjlib/nbp/nbp-calendar-generic/@ng-bootstrap/datepicker/datepicker-reply-years-view.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
var NgbDatepickerReplyYearsView = (function () {
    function NgbDatepickerReplyYearsView() {
    }
    NgbDatepickerReplyYearsView.prototype.isMuted = function () { return false; };
    ;
    NgbDatepickerReplyYearsView.prototype.abilitaAnni = function (data) {
        if (this.startDateEnabled.year <= data.number && data.number <= this.endDateEnabled.year) {
            return true;
        }
        return false;
    };
    return NgbDatepickerReplyYearsView;
}());
__decorate([
    Input(),
    __metadata("design:type", Number)
], NgbDatepickerReplyYearsView.prototype, "currentMonth", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NgbDatepickerReplyYearsView.prototype, "year", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NgbDatepickerReplyYearsView.prototype, "disabled", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NgbDatepickerReplyYearsView.prototype, "focused", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NgbDatepickerReplyYearsView.prototype, "selected", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NgbDatepickerReplyYearsView.prototype, "startDateEnabled", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NgbDatepickerReplyYearsView.prototype, "endDateEnabled", void 0);
NgbDatepickerReplyYearsView = __decorate([
    Component({
        selector: '[ngbDatepickerReplyYearsView]',
        changeDetection: ChangeDetectionStrategy.OnPush,
        host: {
            'class': 'btn-secondary-years',
            '[class.bg-primary]': 'selected',
            '[class.text-white]': 'selected',
            '[class.text-muted]': 'isMuted()',
            '[class.outside]': 'isMuted()',
            '[class.active]': 'focused'
        },
        template: "<div [ngClass]=\" abilitaAnni(year) ? 'ngbDatepickerReplyYearsView' : 'ngbDatepickerReplyYearsViewdisabled disabledView'\">{{ year.name }}</div>",
        styles: [":host{text-align:center;width:2rem;height:2rem;line-height:2rem;border-radius:.25rem}:host.outside{opacity:.5} /*# sourceMappingURL=datepicker-reply-years-view.css.map */ "],
    })
], NgbDatepickerReplyYearsView);
export { NgbDatepickerReplyYearsView };
