// cjlib/nbp/nbp-calendar-generic/@ng-bootstrap/datepicker/nbp-calendar-date-model.js
/**
 * Classe Model per l'inizializzazione del model del componente nbp-calendar-generic
 */
var NbpCalendarDateModel = (function () {
    function NbpCalendarDateModel(d) {
        if (d != null) {
            if (d instanceof Date) {
                this.year = d.getFullYear();
                this.month = d.getMonth() + 1;
                this.day = d.getDate();
            }
            else if (this.instanceOfNgbDateStruct(d)) {
                this.year = d.year;
                this.month = d.month;
                this.day = d.day;
            }
        }
    }
    /**
     * per verificare che implementi l'interface NgbDateStruct non posso usare instanceof,
     * quindi controllo esistano i campi al nuo interno e non siano nulli
     * @param d
     */
    NbpCalendarDateModel.prototype.instanceOfNgbDateStruct = function (d) {
        return !this.isNill(d.day)
            && !this.isNill(d.month)
            && !this.isNill(d.year);
    };
    NbpCalendarDateModel.prototype.isNill = function (value) {
        return value === null || value === undefined;
    };
    return NbpCalendarDateModel;
}());
export { NbpCalendarDateModel };
