// cjlib/nbp/nbp-slider-range/nbp-slider-range-picker/nbp-slider-range-picker.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, ElementRef, ViewChild } from '@angular/core';
import { NbpSliderRangePickerDirective } from './nbp-slider-range-picker.directive';
import Utils from '../utils';
var NbpSliderRangePickerComponent = (function () {
    // parent: any;
    function NbpSliderRangePickerComponent(el) {
        this.el = el;
        this.min = 0;
        this.max = 100;
        this.type = 'range';
        // this.parent = this.el.nativeElement.parentElement.getBoundingClientRect()
    }
    NbpSliderRangePickerComponent.prototype.ngOnInit = function () {
    };
    NbpSliderRangePickerComponent.prototype.onChange = function (_a) {
        var pos = _a.pos, value = _a.value, index = _a.index;
        if (index == undefined) {
            this.model.setValue(value);
        }
        else {
            var existValue = this.model.value;
            existValue[index] = value;
            this.model.setValue(existValue);
        }
    };
    NbpSliderRangePickerComponent.prototype.onFinish = function (_a) {
        var pos = _a.pos, value = _a.value, index = _a.index;
        if (index == undefined) {
            this.model.setValue(value);
            this.sModel.setValue(value);
        }
        else {
            var existValue = this.model.value;
            existValue[index] = value;
            this.model.setValue(existValue);
            this.sModel.setValue(existValue);
        }
    };
    NbpSliderRangePickerComponent.prototype.clickOnBar = function ($event) {
        var nextVal = this.calculateClickPos($event.clientX);
        var trigger = this.opts.trigger || false;
        /* Trigger Mode */
        if (trigger) {
            // nextVal = this.calculateTriggerValue(nextVal)
        }
        this.model.setValue([this.model.value[0], this.model.value[1]]);
        this.sModel.setValue([this.model.value[0], this.model.value[1]]);
    };
    NbpSliderRangePickerComponent.prototype.calculateTriggerValue = function (value) {
        return Utils.calculateTriggerValue(value, this.opts);
    };
    NbpSliderRangePickerComponent.prototype.calculateClickPos = function (xPos) {
        var percentage = ((xPos - this.getParent().left) / this.getParent().width) * 100;
        var range = this.max - this.min;
        var nextValue = ((percentage / 100) * range) + this.min;
        if (nextValue < this.min) {
            return this.min;
        }
        else if (nextValue > this.max) {
            return this.max;
        }
        else {
            var range_1 = this.max - this.min;
            return nextValue;
        }
    };
    NbpSliderRangePickerComponent.prototype.getParent = function () {
        return this.el.nativeElement.parentElement.getBoundingClientRect();
    };
    return NbpSliderRangePickerComponent;
}());
__decorate([
    ViewChild(NbpSliderRangePickerDirective),
    __metadata("design:type", Object)
], NbpSliderRangePickerComponent.prototype, "picker", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderRangePickerComponent.prototype, "min", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderRangePickerComponent.prototype, "max", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangePickerComponent.prototype, "opts", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangePickerComponent.prototype, "model", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangePickerComponent.prototype, "sModel", void 0);
NbpSliderRangePickerComponent = __decorate([
    Component({
        selector: 'nbp-slider-range-picker',
        host: {
            '(click)': 'clickOnBar($event)'
        },
        template: "<div class=\"slider-pickers-container\"><div class=\"slider-picker-inner\"><div nbpSliderRangePicker class=\"slider-picker-left rotate icomoon-Simboli_Risorsa-35\" [defaultValue]=\"model.value[0]\" [max]=\"max\" [min]=\"min\" [index]=\"0\" [opts]=\"opts\" [maxValue]=\"model.value[1]\" (onChange)=\"onChange($event)\" (onFinish)=\"onFinish($event)\"><nbp-slider-range-popover [value]=\"model.value[0]\" [opts]=\"opts\"></nbp-slider-range-popover></div><div nbpSliderRangePicker class=\"slider-picker-right icomoon-Simboli_Risorsa-33\" [defaultValue]=\"model.value[1]\" [max]=\"max\" [min]=\"min\" [opts]=\"opts\" [minValue]=\"model.value[0]\" [index]=\"1\" (onChange)=\"onChange($event)\" (onFinish)=\"onFinish($event)\"><nbp-slider-range-popover [value]=\"model.value[1]\" [opts]=\"opts\"></nbp-slider-range-popover></div></div></div>",
        styles: [".slider-pickers-container{position:absolute;padding:0 0;display:block;top:2px;left:0;width:99%}.slider-pickers-container .slider-picker-inner{position:relative;display:block;width:100%}.slider-pickers-container .slider-picker-inner .slider-picker-left{position:absolute;width:0;height:0;background-color:#39ba2a;border:1px solid #fff;box-shadow:0 2px 1px #ccc;transform:translate(-50%,-50%);border-radius:50%;top:0;z-index:10;cursor:pointer}.slider-pickers-container .slider-picker-inner .slider-picker-left:before{font-family:icomoon!important;font-style:normal!important;font-weight:400!important;font-variant:normal!important;text-transform:none!important;line-height:0;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;font-size:35px;background-color:#fff;border-radius:35px;margin-left:-8px;margin-top:30px;cursor:pointer;color:#258900}.slider-pickers-container .slider-picker-inner .slider-picker-right{position:absolute;width:0;height:0;background-color:#39ba2a;border:1px solid #fff;box-shadow:0 2px 1px #ccc;transform:translate(-50%,-50%);border-radius:50%;top:0;z-index:10;cursor:pointer}.slider-pickers-container .slider-picker-inner .slider-picker-right:before{font-family:icomoon!important;font-style:normal!important;font-weight:400!important;font-variant:normal!important;text-transform:none!important;line-height:0;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;font-size:35px;background-color:#fff;border-radius:35px;margin-left:-8px;margin-top:30px;cursor:pointer;color:#258900} /*# sourceMappingURL=nbp-slider-range-picker.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [ElementRef])
], NbpSliderRangePickerComponent);
export { NbpSliderRangePickerComponent };
