// cjlib/nbp/nbp-search-box/nbp-search-box.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, IterableDiffers, Output, Input, EventEmitter } from '@angular/core';
import { NbpStyle } from '../nbp-commons/nbp-commons.enums';
/**
 * Componente per la ricerca testuale. E' in grado di mostrare suggerimenti per le ricerche
 */
var NbpSearchBoxComponent = (function () {
    /**
     *
     * @param differs  inizializza l' iteratore che serve controlla l'input in ingresso nbpResults
     */
    function NbpSearchBoxComponent(differs) {
        var _this = this;
        /**
        * nome dei campi personalizzabili dell'oggetto nbpResults. Default: "group"
        */
        this.nbpGroup = "group";
        /**
        * nome dei campi personalizzabili dell'oggetto nbpResults. Default: "label"
        */
        this.nbpLabel = "label";
        /**
        * nome dei campi personalizzabili dell'oggetto nbpResults. Default: "subtitle"
        */
        this.nbpSubtitle = "subtitle";
        /**
         * nome dei campi personalizzabili dell'oggetto nbpResults. Default: "description"
         */
        this.nbpDescription = "description";
        /**
         * nome dei campi personalizzabili dell'oggetto nbpResults. Default: "type"
         */
        this.nbpType = "type";
        /**
         * numero di altri risultati da mostrare. Default: 0 (numero non mostrato).
         * Valore speciale "-1": non mostra la scritta
         */
        this.nbpOtherResultsNumber = 0;
        /**
         * testo bottone ricerca
         */
        this.nbpSearchButtonText = "cerca";
        /**
         * testo placeholder casella ricerca
         * @type {string}
         */
        this.nbpPlaceholder = '';
        /**
         * evento che innesca la ricerca dei suggerimenti
         */
        this.change = new EventEmitter();
        /**
        * evento che innesca la ricerca dei risultati
        */
        this.search = new EventEmitter();
        /**
         * evento sollevato al click sul testo "mostra altri risultati"
         */
        this.showMoreResults = new EventEmitter();
        /**
         * Object che conterrà i raggruppamenti dei risultato
         */
        this.groups = {};
        /**
         * lista dei raggruppamenti
         */
        this.groupsList = [];
        /**
         * variabile che gestisce la visulizzazione dei risultati
         */
        this.showResults = false;
        /**
         * variabile per vedere se l'imput della ricerca dei risultati è cambiato
         */
        this.eventPass = '';
        /**
         * variabile in output per la ricerca
         */
        this.searchText = '';
        this.primary = NbpStyle.PRIMARY;
        /**
         * funzione che raggruppa i suggerimenti
         */
        this.formatResults = function () {
            _this.groups = [];
            _this.groupsList = [];
            var self = _this;
            _this.nbpResults.forEach(function (elem) {
                if (self.groups[elem[self.nbpGroup]]) {
                    self.groups[elem[self.nbpGroup]].push(elem);
                }
                else {
                    self.groups[elem[self.nbpGroup]] = [elem];
                    self.groupsList.push(elem[self.nbpGroup]);
                }
            });
            if (_this.nbpResults.length > 0) {
                _this.showResults = true;
            }
        };
        this.differ = differs.find([]).create(null);
    }
    NbpSearchBoxComponent.prototype.ngOnInit = function () { };
    /**
     * funzione che determina se vi è stato un cambiamento in ingresso
     */
    NbpSearchBoxComponent.prototype.ngDoCheck = function () {
        var change = this.differ.diff(this.nbpResults);
        if (change && change.length > 0) {
            if (this.searchText !== this.eventPass) {
                this.formatResults();
                this.eventPass = this.searchText;
            }
        }
    };
    /**
     *
     * @param elem input selezionato dai risultati
     * funzione che imposta la varibile searchText in base al risultato in cui ho cliccato
     */
    NbpSearchBoxComponent.prototype.selectSearch = function (elem) {
        this.searchText = elem.label;
        this.showResults = false;
    };
    /**
     * Funzione che rimuove i suggerimenti
     */
    NbpSearchBoxComponent.prototype.clearResults = function () {
        this.searchText = '';
        this.nbpResults = [];
        this.showResults = false;
    };
    /**
     *
     * funzione che lancia l'evento per la ricerca dei suggerimenti
     */
    NbpSearchBoxComponent.prototype.changeStr = function ($event) {
        if (this.searchText !== this.eventPass) {
            this.change.emit(this.searchText);
        }
    };
    /**
     * funzione che lancia l'evento di ricerca
     */
    NbpSearchBoxComponent.prototype.sendSearch = function () {
        this.search.emit(this.searchText);
    };
    /**
     * funzione che lancia evento di click su "mostra più risultati
     */
    NbpSearchBoxComponent.prototype.showMoreClick = function () {
        this.showMoreResults.emit();
    };
    return NbpSearchBoxComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpSearchBoxComponent.prototype, "nbpResults", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSearchBoxComponent.prototype, "nbpGroup", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSearchBoxComponent.prototype, "nbpLabel", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSearchBoxComponent.prototype, "nbpSubtitle", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSearchBoxComponent.prototype, "nbpDescription", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSearchBoxComponent.prototype, "nbpType", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSearchBoxComponent.prototype, "nbpOtherResultsNumber", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSearchBoxComponent.prototype, "nbpSearchButtonText", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSearchBoxComponent.prototype, "nbpPlaceholder", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpSearchBoxComponent.prototype, "change", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpSearchBoxComponent.prototype, "search", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpSearchBoxComponent.prototype, "showMoreResults", void 0);
NbpSearchBoxComponent = __decorate([
    Component({
        selector: 'nbp-search-box',
        template: "<div class=\"col-sm-20 search-rettangolo\"><div class=\"input-container\"><input tabindex=\"0\" class=\"col-sm-22 input-text-search\" [(ngModel)]=\"searchText\" (keyup)=\"changeStr($event)\" [placeholder]=\"nbpPlaceholder\"><div class=\"span-close\"><img *ngIf=\"searchText\" style=\"cursor: pointer;\" (click)=\"clearResults();\" src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABoAAAAaCAQAAAADQ4RFAAAABGdBTUEAALGPC/xhBQAAACBjSFJN AAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAAAmJLR0QA/4ePzL8AAAAJcEhZ cwAADsMAAA7DAcdvqGQAAAAHdElNRQfiBRkGDi40cdu/AAAByklEQVQ4y42UsWsTURzHP7+7B0Zs tw4ZbsxgIWOiLSik0JAMHrSQQcE1f0BHcbCbQ4cOujk4umSoNINiIBUsHCSFDgEzdBB0yJDNE1Ry 9xxyvdzl3tV8p/d+9/u8e7/v770nLMlz7KY0tCNFYKonch52743SOZKcXJTCI9kjI+3JQcUzQoND eY4iT69/HuzMUlBfrb/lKTerV9gv+wDWfL4CAru/T/sqhgaHKyAAtfXjaHsXJf31uhZ9JSVD8jec OGOnemZBeBSXf6mq+kvGu/HsoY69kxcgnqO+JzI6dnt2Kg+SSFBXx7QWkaCq7GZq2VaAchdYFgHr kZLG0m4SmAkBGko7ghljw4ggjpKiwa1WgHL/rJkQwLHMDdHlv4Vbvi6av1pMDcg4qNuvgjfKzTYA 9FRloUX5aSfjmiYWn/MQoGX6m/SssJuL5GDhe4HhgEp+K4GO3Q4+xTlX/qbAoCb9KOCLx67Brp7e krVovF85EYDhOx6zknS36kb3qdDmciVmdPvJvE9A2Z+5jP6PBPXUdd/6UdjWJzcincL2/UlkejI+ 3NMv5a6hkrE8qySWXDrifXWnZjd0UxdlA5gwlQ/Bx19n14/XXP8Ai1y4DKW+PFcAAAAldEVYdGRh dGU6Y3JlYXRlADIwMTgtMDUtMjVUMDY6MTQ6NDYtMDc6MDCXQfmnAAAAJXRFWHRkYXRlOm1vZGlm eQAyMDE4LTA1LTI1VDA2OjE0OjQ2LTA3OjAw5hxBGwAAAABJRU5ErkJggg==\"><div *ngIf=\"!searchText\" class=\"icomoon-Ricerca_zoom1\" style=\"margin-top: 2px;\"></div></div><div class=\"container-result\" *ngIf=\"showResults\" (clickOutside)=\"showResults=false\"><div class=\"title-results\"><p><span>RICERCHE SUGGERITE</span></p></div><div *ngFor=\"let item of groupsList\" class=\"group\"><div class=\"group-title\">{{item}}</div><table><ng-container *ngFor=\"let info of groups[item]\"><tr class=\"row-line\" (click)=\"selectSearch(info)\"><td><div class=\"info\"><div class=\"info-label-data\"><span>{{info[nbpLabel] | uppercase}}</span></div><div class=\"info-label-desc\"><span>{{info[nbpSubtitle]}}</span>&nbsp;/&nbsp; <span>{{info[nbpDescription] }}</span></div></div><div class=\"label-acc\"><div><span>{{info[nbpType]|uppercase}}</span></div></div></td></tr></ng-container></table></div><div *ngIf=\"nbpOtherResultsNumber !== -1\" class=\"bottom-result text-center\" (click)=\"showMoreClick()\"><p><span style=\"cursor: pointer\">VEDI ALTRI {{nbpOtherResultsNumber == 0 ? '' : nbpOtherResultsNumber}} RISULTATI DI RICERCA</span></p></div></div></div></div><nbp-button class=\"col-sm-4\" [nbpLabel]=\"nbpSearchButtonText\" [nbpStyle]=\"primary\" (nbpClick)=\"sendSearch()\"></nbp-button>",
        styles: [".search-rettangolo{border-width:0;position:relative;height:3.125em;background:inherit;background-color:#fff;box-sizing:border-box;border-width:.0625em;border-style:solid;border-color:#258900;border-radius:2.5em;-moz-box-shadow:none;-webkit-box-shadow:none;box-shadow:none;z-index:1}.input-text-search{margin-left:12px;margin-top:1px;width:90%;height:2.88em;font-family:Arial;font-weight:400;font-style:normal;font-size:1em;text-decoration:none;color:#323232;text-align:left;border:0;outline-style:none}.input-text-search::-ms-clear{display:none;width:0;height:0}.span-close{float:right;vertical-align:middle;line-height:1.77em;padding-right:.75em;font-size:1.63em;color:#258900}.search-icon{font-size:1.63em;color:#258900}.title-cont{border-width:0;position:absolute;left:1.25em;padding-top:.625em;word-wrap:break-word}.btn-margin{margin-left:6.25em}.container-result{margin-top:3.125em;border-width:0;position:absolute;left:1.6em;width:93%;padding-top:.625em;min-height:6.25em;background:inherit;background-color:#fff;box-sizing:border-box;border-width:.06em;border-style:solid;border-color:#258900;border-top:0;border-radius:0;-moz-box-shadow:none;-webkit-box-shadow:none;box-shadow:none;font-size:1em;color:#323232;z-index:10000}.title-results{font-size:.75em;color:#323232;height:.875em}.bottom-result{font-family:'Arial Bold',Arial;font-weight:700;font-style:normal;font-size:.688em;color:#258900;text-align:center;bottom:.625em;margin-bottom:1.4375em}.bottom-result p{margin-top:.562em}.group{padding:0 1.25em .875em 1.25em}.group .group-title{padding:.313em .938em .562em .938em;vertical-align:middle;text-align:left;line-height:.938em;font-family:'Arial Bold',Arial;font-weight:700;font-style:normal;font-size:.75em;color:#323232}.group .row-line{height:2.625em;background-color:#f8f8f8;cursor:pointer}.group table{width:100%}.group tr{border-top:.063em solid #ccc}.group tr:last-child{border-bottom:.063em solid #ccc}.group .info{float:left;width:80%;padding-left:.938em;padding-top:.313em;text-align:left}.group .info-label-data{font-family:'Arial Bold',Arial;font-weight:700;font-style:normal;font-size:.75em;color:#323232}.group .info-label-desc{font-family:'Arial Bold',Arial;font-weight:400;font-style:normal;font-size:.688em;color:#323232}.group .label-acc{float:right;width:20%;text-align:right;line-height:2.625em;vertical-align:middle;padding-right:.625em;font-family:'Arial Bold',Arial;font-weight:700;font-style:normal;font-size:.75em;color:#323232} /*# sourceMappingURL=nbp-search-box.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [IterableDiffers])
], NbpSearchBoxComponent);
export { NbpSearchBoxComponent };
