// cjlib/nbp/nbp-datasource/nbp-datasource.interfaces.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Subject } from 'rxjs/Subject';
var INbpDataSource = (function () {
    function INbpDataSource() {
        this.pageDataSubject = new Subject();
    }
    INbpDataSource.prototype.sendDataEvent = function (pageNumber, sortFields, pageSize) {
        var _this = this;
        var page = this.getPage(pageNumber, sortFields, pageSize);
        page.subscribe(function (iDataPage) { return _this.pageDataSubject.next(iDataPage); });
    };
    INbpDataSource.prototype.getDataStream = function () {
        return this.pageDataSubject.asObservable();
    };
    return INbpDataSource;
}());
export { INbpDataSource };
var INbpAutocompleteDataSource = (function () {
    function INbpAutocompleteDataSource() {
        this.pageDataSubject = new Subject();
    }
    INbpAutocompleteDataSource.prototype.sendDataEvent = function (filterFields, filter) {
        var _this = this;
        var page = this.getPage(filterFields, filter);
        page.subscribe(function (iDataPage) { return _this.pageDataSubject.next(iDataPage); });
    };
    INbpAutocompleteDataSource.prototype.getDataStream = function () {
        return this.pageDataSubject.asObservable();
    };
    return INbpAutocompleteDataSource;
}());
export { INbpAutocompleteDataSource };
var INbpArrayDataSource = (function (_super) {
    __extends(INbpArrayDataSource, _super);
    function INbpArrayDataSource() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return INbpArrayDataSource;
}(INbpDataSource));
export { INbpArrayDataSource };
