// cjlib/nbp/nbp-datasource/nbp-datasource.service.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { INbpArrayDataSource } from './nbp-datasource.interfaces';
import { SortingDirection, CompareStatus } from './nbp-datasource.enums';
import { getPropertyValueByPath } from './../../utility/lib-utility';
import { Observable } from 'rxjs/Observable';
import 'rxjs/add/operator/map';
import 'rxjs/add/observable/of';
var NbpDataSource = (function (_super) {
    __extends(NbpDataSource, _super);
    function NbpDataSource(arrayData, defaultPageSize) {
        var _this = _super.call(this) || this;
        _this.defaultPageSize = defaultPageSize;
        if (!arrayData) {
            console.log('Errore - Array passato: "Undefined"');
        }
        _this.arrayData = arrayData ? arrayData : [];
        return _this;
    }
    NbpDataSource.prototype.getSortingDirections = function (sortFields) {
        var sortingDirections = sortFields.map(function (element) {
            var sortingDirection;
            if (element.charAt(0) === '-') {
                sortingDirection = SortingDirection.DESCENDING;
            }
            else {
                sortingDirection = SortingDirection.ASCENDING;
            }
            return sortingDirection;
        });
        return sortingDirections;
    };
    NbpDataSource.prototype.dataSortComparator = function (sortDirection, sortField) {
        return function (a, b) {
            var result;
            var field = sortDirection === SortingDirection.DESCENDING ? sortField.substr(1) : sortField;
            var currentTmp = getPropertyValueByPath(a, field);
            var nextTmp = getPropertyValueByPath(b, field);
            var current = currentTmp instanceof Date ? currentTmp.getTime() : currentTmp;
            var next = nextTmp instanceof Date ? nextTmp.getTime() : nextTmp;
            if (current === null || current === undefined) {
                result = CompareStatus.LOWER;
            }
            else if (next === null || next === undefined) {
                result = CompareStatus.HIGHER;
            }
            else {
                result = (current < next) ? CompareStatus.LOWER : ((current > next) ? CompareStatus.HIGHER : CompareStatus.EQUAL);
            }
            return result * sortDirection;
        };
    };
    NbpDataSource.prototype.sortData = function (data, sortFields) {
        var sortingDirections = this.getSortingDirections(sortFields);
        var sortField = sortFields[0];
        var sortDirection = sortingDirections[0];
        var sortedData = data.sort(this.dataSortComparator(sortDirection, sortField));
        return sortedData;
    };
    NbpDataSource.prototype.getPage = function (pageNumber, sorting, pageSize) {
        var shadowData = this.arrayData.slice();
        var currentPageSize = pageSize ? pageSize : (this.defaultPageSize ? this.defaultPageSize : this.arrayData.length);
        var pageToreturn = pageNumber ? pageNumber : 1;
        var start = (pageToreturn - 1) * currentPageSize;
        var end = start + currentPageSize;
        if (sorting && sorting.length) {
            shadowData = this.sortData(this.arrayData.slice(), sorting);
        }
        var data = shadowData.slice(start, end);
        var dataPage = {
            data: data,
            status: {
                pageNumber: pageNumber,
                pageSize: currentPageSize,
                totalRecords: this.arrayData.length
            }
        };
        return Observable.of(dataPage);
    };
    NbpDataSource.prototype.setData = function (newArrayData, pageNumber, sortFields, pageSize) {
        if (!newArrayData) {
            console.log('Errore - Array passato: "Undefined"');
        }
        this.arrayData = newArrayData ? newArrayData : [];
        this.sendDataEvent(pageNumber, sortFields, pageSize);
    };
    return NbpDataSource;
}(INbpArrayDataSource));
export { NbpDataSource };
