// cjlib/nbp/nbp-tooltip/nbp-tooltip.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, TemplateRef, HostListener, ElementRef } from '@angular/core';
import { NbpSize } from './../nbp-commons/nbp-commons.enums';
import { coerceBooleanProperty } from './../../utility/lib-utility';
var NbpTooltipComponent = (function () {
    function NbpTooltipComponent(element) {
        this.element = element;
        this._tooltipSize = NbpSize;
        this._isVisible = false;
        /**
          * Strategia di apertura tooltip, 'click' o 'hover':
          * click -> comportamento popover con apertura e chiusura su click (più pulsante x all'interno del tooltip)
          * hover -> comportamento tooltip classico apertura su hover , chiusura sulla perdita dell'hover
          * Valore di default: hover
         */
        this.nbpVisibilityStrategy = 'hover';
        /**
         * (Opzionale) Enum che definisce il la posizione del tooltip rispetto all’oggetto al quale è ancorato:
         * “Top” | “Right” | “Bottom” | “Left” | “Top-Left” | “Top-Right” | “Bottom-Left” | “Bottom-Right”
         * Default: “Top”
         */
        this.nbpPosition = 'Top'; // enum;
        /**
         * (Opzionale) Enum che definisce il tipo di icona da visualizzare nel tooltip:
         * “Warning” | “Help” | ”Info”
         * Default: nessuna icona
         * Funzionante solo se non presente nbpTemplate
         */
        // TODO NON VIENE MAI USATA QUESTA PROPERTY -- COMMENTATA PER EVITARE REFUSI
        /*@Input() nbpType: object; // enum; Non presente nelle nuove StyleGuide: nuovo componente icona (?)/(!)/(i) -> tooltip?*/
        /**
         * Proprietà che definisce la grandezza del tooltip, tramite l'enum NbpSize
         * valore di default: NbpSize.SM
         */
        this.nbpSize = NbpSize.SM;
        this._nbpHidden = false;
    }
    Object.defineProperty(NbpTooltipComponent.prototype, "hidden", {
        /**
         * Booleano che abilita/disabilita la visualizzazione del tooltip.
        * Attributo "hidden" standard di HTML5 element
         */
        get: function () { return this._nbpHidden; },
        set: function (value) { this._nbpHidden = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    ;
    ;
    //---modificaReply
    NbpTooltipComponent.prototype.onClick = function (target) {
        event.stopPropagation();
        if (!this.element.nativeElement.contains(target)) {
            this._closePopover(target);
        }
        if (!this._isVisibilityStrategyHover()) {
            this._closePopover(target);
        }
    };
    //---------------------
    NbpTooltipComponent.prototype._open = function () {
        var _this = this;
        setTimeout(function () {
            _this._isVisible = true;
        }, 0);
    };
    NbpTooltipComponent.prototype._close = function () {
        this._isVisible = false;
    };
    NbpTooltipComponent.prototype._closePopover = function (target) {
        var parentFound = false;
        while (target != null && !parentFound) {
            if (target === this.element.nativeElement) {
                parentFound = true;
            }
            target = target.parentElement;
        }
        if (!parentFound) {
            this._isVisible = false;
        }
    };
    NbpTooltipComponent.prototype._toggleVisibility = function () {
        if (this.nbpVisibilityStrategy === 'hover') {
            return;
        }
        if (this._isVisible) {
            this._close();
        }
        else {
            this._open();
        }
    };
    NbpTooltipComponent.prototype._isVisibilityStrategyHover = function () {
        return this.nbpVisibilityStrategy === 'hover';
    };
    return NbpTooltipComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpTooltipComponent.prototype, "name", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpTooltipComponent.prototype, "nbpVisibilityStrategy", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpTooltipComponent.prototype, "nbpPosition", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpTooltipComponent.prototype, "nbpSize", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpTooltipComponent.prototype, "nbpText", void 0);
__decorate([
    Input(),
    __metadata("design:type", TemplateRef)
], NbpTooltipComponent.prototype, "nbpTemplate", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpTooltipComponent.prototype, "nbpTemplateData", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpTooltipComponent.prototype, "hidden", null);
__decorate([
    HostListener('document: click', ['$event.target']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [HTMLElement]),
    __metadata("design:returntype", void 0)
], NbpTooltipComponent.prototype, "onClick", null);
NbpTooltipComponent = __decorate([
    Component({
        selector: 'nbp-tooltip',template: "<span #div [ngClass]=\"{ 'nbp-tooltip': _isVisibilityStrategyHover(), 'nbp-popover': !_isVisibilityStrategyHover() }\"><span (click)=\"_toggleVisibility();\" [class.clickable]=\"!_isVisibilityStrategyHover()\"><ng-content></ng-content></span><span [hidden]=\"hidden\" [ngClass]=\"{ 'tooltip-content': _isVisibilityStrategyHover(), 'popover-content': !_isVisibilityStrategyHover(), 'top':nbpPosition=='Top', 'bottom':nbpPosition=='Bottom', 'right':nbpPosition=='Right', 'left':nbpPosition=='Left', 'top-left':nbpPosition=='Top-Left', 'top-right':nbpPosition=='Top-Right', 'bottom-left':nbpPosition=='Bottom-Left', 'bottom-right':nbpPosition=='Bottom-Right',  'sm':nbpSize === _tooltipSize.SM,  'md':nbpSize === _tooltipSize.MD,  'lg':nbpSize === _tooltipSize.LG,  'opened': !_isVisibilityStrategyHover() && _isVisible }\"><span *ngIf=\"!nbpTemplate; else nbpTemplateContainer\" [ngClass]=\"{ 'tooltip-text': _isVisibilityStrategyHover() ,  'popover-text': !_isVisibilityStrategyHover() }\"><div class=\"row\"><div class=\"col-md-24\" [ngClass]=\"{ 'text-padding-hover': _isVisibilityStrategyHover(), 'text-padding-click': !_isVisibilityStrategyHover() }\">{{nbpText}}</div></div></span><ng-template #nbpTemplateContainer [ngTemplateOutlet]=\"nbpTemplate\" [ngTemplateOutletContext]=\"nbpTemplateData\"></ng-template><div class=\"icon-position clickable\" *ngIf=\"!_isVisibilityStrategyHover()\"><span class=\"ico-azioni-elimina\" style=\"font-size: 19px\" (click)=\"_close(); $event.stopPropagation()\"></span></div></span></span>",
        styles: [".bottom .icon-position,.bottom-left .icon-position,.bottom-right .icon-position,.right .icon-position,.top .icon-position,.top-left .icon-position,.top-right .icon-position{position:absolute;top:1.3125em;right:1.3125em;color:#fa9600;font-weight:700}.text-padding-click{padding:20px 80px 20px 20px;text-align:left;font-size:14px;font-family:Arial}.text-padding-hover{padding:.5em 1em .5em 1em}.left .icon-position{position:absolute;top:.3125em;left:.3125em;color:#fa9600;font-weight:700}.nbp-popover,.nbp-tooltip{position:relative;display:inline-block;opacity:.9;z-index:1000}.clickable{cursor:pointer}.nbp-popover .popover-content,.nbp-tooltip .tooltip-content{visibility:hidden;background-color:#323232;color:#fff;padding:.3125em 0;position:absolute;z-index:1;display:inline-block!important}.nbp-popover .popover-content .tooltip-text,.nbp-tooltip .tooltip-content .tooltip-text{text-align:center;font-weight:400}.nbp-popover .popover-content .popover-text,.nbp-tooltip .tooltip-content .popover-text{text-align:center;font-weight:400}.nbp-popover .popover-content.md,.nbp-tooltip .tooltip-content.md{width:21.875em}.nbp-popover .popover-content.sm,.nbp-tooltip .tooltip-content.sm{width:14.5em}.nbp-popover .popover-content.lg,.nbp-tooltip .tooltip-content.lg{width:31.25em}.nbp-popover .popover-content.right,.nbp-tooltip .tooltip-content.right{top:50%!important;left:100%!important;margin-left:.625em;transform:translateY(-50%)}.nbp-popover .popover-content.right::after,.nbp-tooltip .tooltip-content.right::after{content:\"\";position:absolute;top:50%;right:100%;transform:translateY(-50%);border-top-width:.625em;border-right-width:.5625em;border-bottom-width:.625em;border-left-width:.5625em;border-style:solid;border-color:transparent #323232 transparent transparent}.nbp-popover .popover-content.left,.nbp-tooltip .tooltip-content.left{top:50%;right:100%;margin-right:.625em;transform:translateY(-50%)}.nbp-popover .popover-content.left::after,.nbp-tooltip .tooltip-content.left::after{content:\"\";position:absolute;top:50%;left:100%;transform:translateY(-50%);border-top-width:.625em;border-right-width:.5625em;border-bottom-width:.625em;border-left-width:.5625em;border-style:solid;border-color:transparent transparent transparent #323232}.nbp-popover .popover-content.top,.nbp-tooltip .tooltip-content.top{bottom:100%;left:62%;margin-bottom:.8125em;transform:translateX(-50%)}.nbp-popover .popover-content.top::after,.nbp-tooltip .tooltip-content.top::after{content:\"\";position:absolute;top:100%;left:50%;transform:translateX(-50%);border-top-width:.625em;border-right-width:.75em;border-bottom-width:.625em;border-left-width:.75em;border-style:solid;border-color:#323232 transparent transparent transparent}.nbp-popover .popover-content.bottom,.nbp-tooltip .tooltip-content.bottom{top:100%;left:50%;margin-top:.9375em;transform:translateX(-50%)}.nbp-popover .popover-content.bottom::after,.nbp-tooltip .tooltip-content.bottom::after{content:\"\";position:absolute;bottom:100%;left:50%;transform:translateX(-50%);border-top-width:.625em;border-right-width:.75em;border-bottom-width:.625em;border-left-width:.75em;border-style:solid;border-color:transparent transparent #323232 transparent}.nbp-popover .popover-content.top-left,.nbp-tooltip .tooltip-content.top-left{bottom:100%;left:-4px;margin-bottom:.9375em}.nbp-popover .popover-content.top-left::after,.nbp-tooltip .tooltip-content.top-left::after{content:\"\";position:absolute;top:100%;left:5%;border-top-width:.625em;border-right-width:.75em;border-bottom-width:.625em;border-left-width:.75em;border-style:solid;border-color:#323232 transparent transparent transparent}.nbp-popover .popover-content.top-right,.nbp-tooltip .tooltip-content.top-right{bottom:100%;right:-14px;margin-bottom:.9375em}.nbp-popover .popover-content.top-right::after,.nbp-tooltip .tooltip-content.top-right::after{content:\"\";position:absolute;top:100%;right:5%;border-top-width:.625em;border-right-width:.75em;border-bottom-width:.625em;border-left-width:.75em;border-style:solid;border-color:#323232 transparent transparent transparent}.nbp-popover .popover-content.bottom-left,.nbp-tooltip .tooltip-content.bottom-left{top:100%;left:-4px;margin-top:.9375em}.nbp-popover .popover-content.bottom-left::after,.nbp-tooltip .tooltip-content.bottom-left::after{content:\"\";position:absolute;bottom:100%;left:5%;border-top-width:.625em;border-right-width:.75em;border-bottom-width:.625em;border-left-width:.75em;border-style:solid;border-color:transparent transparent #323232 transparent}.nbp-popover .popover-content.bottom-right,.nbp-tooltip .tooltip-content.bottom-right{top:100%;right:-14px;margin-top:.9375em}.nbp-popover .popover-content.bottom-right::after,.nbp-tooltip .tooltip-content.bottom-right::after{content:\"\";position:absolute;bottom:100%;right:5%;border-top-width:.625em;border-right-width:.75em;border-bottom-width:.625em;border-left-width:.75em;border-style:solid;border-color:transparent transparent #323232 transparent}.nbp-popover .popover-content.opened,.nbp-tooltip:hover .tooltip-content{visibility:visible}:host-context(.contrasto-on) .bottom .icon-position,:host-context(.contrasto-on) .bottom-left .icon-position,:host-context(.contrasto-on) .bottom-right .icon-position,:host-context(.contrasto-on) .right .icon-position,:host-context(.contrasto-on) .top .icon-position,:host-context(.contrasto-on) .top-left .icon-position,:host-context(.contrasto-on) .top-right .icon-position{color:#fff}:host-context(.contrasto-on) .left .icon-position{color:#fff}:host-context(.contrasto-on) .nbp-popover .popover-content,:host-context(.contrasto-on) .nbp-tooltip .tooltip-content{background-color:#000}:host-context(.contrasto-on) .nbp-popover .popover-content.right::after,:host-context(.contrasto-on) .nbp-tooltip .tooltip-content.right::after{border-color:transparent #000 transparent transparent}:host-context(.contrasto-on) .nbp-popover .popover-content.left::after,:host-context(.contrasto-on) .nbp-tooltip .tooltip-content.left::after{border-color:transparent transparent transparent #000}:host-context(.contrasto-on) .nbp-popover .popover-content.top::after,:host-context(.contrasto-on) .nbp-tooltip .tooltip-content.top::after{border-color:#000 transparent transparent transparent}:host-context(.contrasto-on) .nbp-popover .popover-content.bottom::after,:host-context(.contrasto-on) .nbp-tooltip .tooltip-content.bottom::after{border-color:transparent transparent #000 transparent}:host-context(.contrasto-on) .nbp-popover .popover-content.top-left::after,:host-context(.contrasto-on) .nbp-tooltip .tooltip-content.top-left::after{border-color:#000 transparent transparent transparent}:host-context(.contrasto-on) .nbp-popover .popover-content.top-right::after,:host-context(.contrasto-on) .nbp-tooltip .tooltip-content.top-right::after{border-color:#000 transparent transparent transparent}:host-context(.contrasto-on) .nbp-popover .popover-content.bottom-left::after,:host-context(.contrasto-on) .nbp-tooltip .tooltip-content.bottom-left::after{border-color:transparent transparent #000 transparent}:host-context(.contrasto-on) .nbp-popover .popover-content.bottom-right::after,:host-context(.contrasto-on) .nbp-tooltip .tooltip-content.bottom-right::after{border-color:transparent transparent #000 transparent} /*# sourceMappingURL=nbp-tooltip.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [ElementRef])
], NbpTooltipComponent);
export { NbpTooltipComponent };
