// cjlib/nbp/nbp-slider-storico/nbp-slider-storico-popover/nbp-slider-storico-popover.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input } from '@angular/core';
import Utils from '../utils';
var NbpSliderStoricoPopoverComponent = (function () {
    function NbpSliderStoricoPopoverComponent() {
        this.value = 0;
        this.formatted = 0;
    }
    NbpSliderStoricoPopoverComponent.prototype.ngOnInit = function () {
        this.getValue();
    };
    NbpSliderStoricoPopoverComponent.prototype.ngOnChanges = function () {
        this.getValue();
    };
    NbpSliderStoricoPopoverComponent.prototype.getValue = function () {
        var format = Utils.getDecimalFormatter(this.opts.label.format);
        this.formatted = Utils.formatValue(this.value, format);
    };
    return NbpSliderStoricoPopoverComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderStoricoPopoverComponent.prototype, "value", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderStoricoPopoverComponent.prototype, "opts", void 0);
NbpSliderStoricoPopoverComponent = __decorate([
    Component({
        selector: 'nbp-slider-storico-popover',
        template: "<div class=\"nbp-slider-storico-popover\"><div class=\"nbp-slider-storico-popover-content\">{{formatted}}</div></div>",
        styles: [".nbp-slider-storico-popover{width:80px;position:absolute;bottom:100%;margin-bottom:31px;left:50%;margin-left:10px;transform:translate(-50%,0)}.nbp-slider-storico-popover .nbp-slider-storico-popover-content{background-color:#f9f9f9;border:1px solid #fff;padding:8px 12px;border-radius:3px;box-shadow:0 1px 1px #ccc}.nbp-slider-storico-popover .nbp-slider-storico-popover-content:after,.nbp-slider-storico-popover .nbp-slider-storico-popover-content:before{top:100%;left:50%;border:solid transparent;content:\" \";height:0;width:0;position:absolute;pointer-events:none}.nbp-slider-storico-popover .nbp-slider-storico-popover-content:after{border-color:rgba(136,183,213,0);border-top-color:#fff;border-width:6px;margin-left:-6px}.nbp-slider-storico-popover .nbp-slider-storico-popover-content:before{border-color:rgba(194,225,245,0);border-top-color:#ddd;border-width:7px;margin-left:-7px} /*# sourceMappingURL=nbp-slider-storico-popover.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpSliderStoricoPopoverComponent);
export { NbpSliderStoricoPopoverComponent };
