// cjlib/cj/cj-error-modal/cj-error-modal.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, ViewChild, NgZone } from '@angular/core';
import { ApplicationStatusService } from './../../utility/application-status/application-status.service';
import { NbpSize } from './../../nbp/nbp-commons/nbp-commons.enums';
import { NbpModalComponent } from './../../nbp/nbp-modal/nbp-modal.component';
var CjErrorModalComponent = (function () {
    function CjErrorModalComponent(applicationStatus, zone) {
        this.applicationStatus = applicationStatus;
        this.zone = zone;
        this._errorTitle = 'Global Error!';
        this._size = NbpSize.MD;
        this._openModal = false;
    }
    CjErrorModalComponent.prototype.ngOnInit = function () {
        var _this = this;
        this._errorSubscription = this.applicationStatus.getGlobalErrorStream().filter(function (error) { return error !== null; })
            .subscribe(function (error) {
            console.log('CjErrorModal dentro subscribe');
            _this.zone.run(function () {
                if (_this._cjErrorModal) {
                    _this._cjErrorModal.open();
                    _this._cjGlobalError = error;
                }
                //  this._openModal = true;
            });
        }, function (err) { return console.error('Observer got an error: ' + err); }, function () { return console.log('Observer got a complete notification'); });
    };
    CjErrorModalComponent.prototype.ngOnDestroy = function () {
        this._errorSubscription.unsubscribe();
    };
    return CjErrorModalComponent;
}());
__decorate([
    ViewChild('errorModal'),
    __metadata("design:type", NbpModalComponent)
], CjErrorModalComponent.prototype, "_cjErrorModal", void 0);
CjErrorModalComponent = __decorate([
    Component({
        selector: 'cj-error-modal',template: "<nbp-modal #errorModal [(nbpOpened)]=\"_openModal\" [nbpSize]=\"_size\"><div class=\"container-fluid\"><cj-error-view [cjClientError]=\"_cjGlobalError\"></cj-error-view></div></nbp-modal>",
        styles: [".error-view-content{max-height:500px;overflow:auto} /*# sourceMappingURL=cj-error-modal.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [ApplicationStatusService, NgZone])
], CjErrorModalComponent);
export { CjErrorModalComponent };
