// cjlib/nbp/nbp-datepicker-generic/@ng-bootstrap/datepicker/datepicker-navigation.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter, ChangeDetectionStrategy } from '@angular/core';
import { NavigationEvent } from './datepicker-view-model';
import { NgbDate } from './ngb-date';
import { NgbDatepickerI18n } from './datepicker-i18n';
import { NgbCalendar } from './ngb-calendar';
var NgbDatepickerNavigation = (function () {
    function NgbDatepickerNavigation(i18n, _calendar) {
        this.i18n = i18n;
        this._calendar = _calendar;
        this.navigation = NavigationEvent;
        this.navigate = new EventEmitter();
        this.select = new EventEmitter();
    }
    NgbDatepickerNavigation.prototype.doNavigate = function (event) {
        this.navigate.emit(event);
    };
    NgbDatepickerNavigation.prototype.nextDisabled = function () {
        return this.disabled || (this.maxDate && this._calendar.getNext(this.date, 'm').after(this.maxDate));
    };
    NgbDatepickerNavigation.prototype.prevDisabled = function () {
        var prevDate = this._calendar.getPrev(this.date, 'm');
        return this.disabled || (this.minDate && prevDate.year <= this.minDate.year && prevDate.month < this.minDate.month);
    };
    NgbDatepickerNavigation.prototype.selectDate = function (date) { this.select.emit(date); };
    return NgbDatepickerNavigation;
}());
__decorate([
    Input(),
    __metadata("design:type", NgbDate)
], NgbDatepickerNavigation.prototype, "date", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NgbDatepickerNavigation.prototype, "disabled", void 0);
__decorate([
    Input(),
    __metadata("design:type", NgbDate)
], NgbDatepickerNavigation.prototype, "maxDate", void 0);
__decorate([
    Input(),
    __metadata("design:type", NgbDate)
], NgbDatepickerNavigation.prototype, "minDate", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NgbDatepickerNavigation.prototype, "months", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NgbDatepickerNavigation.prototype, "showSelect", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NgbDatepickerNavigation.prototype, "showWeekNumbers", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NgbDatepickerNavigation.prototype, "navigate", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NgbDatepickerNavigation.prototype, "select", void 0);
NgbDatepickerNavigation = __decorate([
    Component({
        selector: 'ngb-datepicker-navigation',
        changeDetection: ChangeDetectionStrategy.OnPush,
        host: { 'class': 'd-flex justify-content-between', '[class.collapsed]': '!showSelect' },
        template: "<span><button type=\"button\" class=\"btn btn-sm btn-link camillo\" style=\"width:45px\" (click)=\"!!doNavigate(navigation.PREVYEAR)\" [disabled]=\"prevDisabled()\" tabindex=\"-1\"><span class=\"ngb-dp-navigation-chevron camillo\" style=\"width:50%\"></span> <span class=\"ngb-dp-navigation-chevron camillo\" style=\"margin-left:-0.8em!important\"></span></button> <button type=\"button\" class=\"btn btn-sm btn-link camillo\" style=\"width:45px; margin-left:-0.2em!important\" (click)=\"!!doNavigate(navigation.PREV)\" [disabled]=\"prevDisabled()\" tabindex=\"-1\"><span class=\"ngb-dp-navigation-chevron camillo\"></span></button></span><ngb-datepicker-navigation-select *ngIf=\"showSelect\" class=\"d-block\" [style.width.rem]=\"months * 9\" [date]=\"date\" [minDate]=\"minDate\" [maxDate]=\"maxDate\" [disabled]=\"disabled\" (select)=\"selectDate($event)\"></ngb-datepicker-navigation-select><span><button type=\"button\" style=\"width:45px; margin-right:-0.23em!important;\" class=\"btn btn-sm btn-link camillo\" (click)=\"!!doNavigate(navigation.NEXT)\" [disabled]=\"nextDisabled()\" tabindex=\"-1\"><span class=\"ngb-dp-navigation-chevron right camillo\"></span></button> <button type=\"button\" style=\"width:45px\" class=\"btn btn-sm btn-link camillo\" (click)=\"!!doNavigate(navigation.NEXTYEAR)\" [disabled]=\"nextDisabled()\" tabindex=\"-1\"><span class=\"ngb-dp-navigation-chevron right camillo\" style=\"width:50%; margin-right: -0.8em!important;\"></span> <span class=\"ngb-dp-navigation-chevron right camillo\"></span></button></span>",
        styles: [":host{height:2rem;line-height:1.85rem}:host.collapsed{margin-bottom:-2rem}.btn-link{width:2.8rem;min-height:100%!important}caption,table,tbody,td,tfoot,th,thead,tr{margin:0;padding:0;border:0;outline:0;font-size:100%;vertical-align:baseline;background:0 0}.camillo{display:inline-block;margin:0;padding:0;border:none}.ngb-dp-navigation-chevron::before{border-style:solid;border-width:.2em .2em 0 0;content:'';display:inline-block;height:.75em;transform:rotate(-135deg);-webkit-transform:rotate(-135deg);-ms-transform:rotate(-135deg);width:.75em;margin:0 0 0 .5rem}.ngb-dp-navigation-chevron.right:before{-webkit-transform:rotate(45deg);-ms-transform:rotate(45deg);transform:rotate(45deg);margin:0 .5rem 0 0} /*# sourceMappingURL=datepicker-navigation.css.map */ "],
    }),
    __metadata("design:paramtypes", [NgbDatepickerI18n, NgbCalendar])
], NgbDatepickerNavigation);
export { NgbDatepickerNavigation };
