// cjlib/cj/cj-application-warning/cj-application-alert.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, NgZone } from '@angular/core';
import { ApplicationStatusService } from './../../utility/application-status/application-status.service';
import { NbpAlertType } from './../../nbp/nbp-commons/nbp-commons.enums';
var CjApplicationAlertComponent = (function () {
    function CjApplicationAlertComponent(applicationStatus, zone) {
        this.applicationStatus = applicationStatus;
        this.zone = zone;
        /**
         * (Opzionale) Array di alert applicativi che si pososno passare al componente
         * istanziato in pagina
         */
        this.cjApplicationAlerts = undefined;
    }
    CjApplicationAlertComponent.prototype.ngOnInit = function () {
        var _this = this;
        if (this.cjApplicationAlerts == undefined) {
            this.cjApplicationAlerts = [];
            this._alertSubscription = this.applicationStatus.getApplicationAlertStream().subscribe(function (alert) {
                // TODO in seguito anche il tipo di alert potrebbe essere derivato dal contenuto dell'alert
                // errorLevel
                var appAlert = {
                    type: NbpAlertType.WARNING_BLOCCANTE,
                    content: alert.errorMessage,
                    show: true
                };
                _this.zone.run(function () {
                    _this.cjApplicationAlerts.push(appAlert);
                });
            });
        }
    };
    CjApplicationAlertComponent.prototype.ngOnDestroy = function () {
        if (this._alertSubscription) {
            this._alertSubscription.unsubscribe();
        }
    };
    CjApplicationAlertComponent.prototype._onCloseAlert = function (alertIndex) {
        this.cjApplicationAlerts.splice(alertIndex, 1);
        // this.cjApplicationAlerts[alertIndex].show = false;
    };
    // metodi pubblici esposti dal componente
    CjApplicationAlertComponent.prototype.closeAllAlert = function () {
        this.cjApplicationAlerts.splice(0, this.cjApplicationAlerts.length);
    };
    return CjApplicationAlertComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Array)
], CjApplicationAlertComponent.prototype, "cjApplicationAlerts", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], CjApplicationAlertComponent.prototype, "id", void 0);
CjApplicationAlertComponent = __decorate([
    Component({
        selector: 'cj-application-alert',template: "<div class=\"pad-b-12\" *ngFor=\"let alert of cjApplicationAlerts; let alertIndex = index;\"><nbp-alert-box *ngIf=\"alert.show\" (nbpClose)=\"_onCloseAlert(alertIndex)\" [nbpAlertType]=\"alert.type\">{{alert.content}}</nbp-alert-box></div>",
        styles: [":host {\n    display: inline-block;\n  }"]
    }),
    __metadata("design:paramtypes", [ApplicationStatusService, NgZone])
], CjApplicationAlertComponent);
export { CjApplicationAlertComponent };
