// cjlib/nbp/nbp-error-advice/nbp-error-advice.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { NbpStyle } from './../nbp-commons/nbp-commons.enums';
var NbpErrorAdviceComponent = (function () {
    function NbpErrorAdviceComponent() {
        this._nbpButtonStyle = NbpStyle.PRIMARY;
        this.onError = new EventEmitter();
    }
    NbpErrorAdviceComponent.prototype.ngOnInit = function () {
        if (!this.nbpAdvice) {
            this.nbpAdvice = 'Attenzione, c\'è stato un errore durante la procedura, si prega di annullare e riprovare più tardi.';
        }
        if (!this.nbpActionLabel) {
            this.nbpActionLabel = 'Annulla';
        }
    };
    NbpErrorAdviceComponent.prototype._onActionButtonClick = function () {
        this.onError.emit();
    };
    return NbpErrorAdviceComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpErrorAdviceComponent.prototype, "nbpAdvice", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpErrorAdviceComponent.prototype, "nbpActionLabel", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpErrorAdviceComponent.prototype, "onError", void 0);
NbpErrorAdviceComponent = __decorate([
    Component({
        selector: 'nbp-error-advice',template: "<div class=\"content\"><p class=\"advice-container alert alert-danger\">{{nbpAdvice}}</p><div class=\"button-container\"><nbp-button [nbpLabel]=\"nbpActionLabel\" [nbpStyle]=\"_nbpButtonStyle\" (nbpClick)=\"_onActionButtonClick()\"></nbp-button></div></div>",
        styles: [".content{margin-top:10px;margin-right:auto;margin-bottom:10px;margin-left:auto}.button-container{text-align:center;padding-top:10px;padding-left:10px;padding-bottom:10px;padding-right:10px}.advice-container{text-align:center;padding-top:10px;padding-left:10px;padding-bottom:10px;padding-right:10px} /*# sourceMappingURL=nbp-error-advice.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpErrorAdviceComponent);
export { NbpErrorAdviceComponent };
