// cjlib/nbp/nbp-progress/nbp-progress.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
import { Component, Input, ViewChild, ElementRef, Renderer2, QueryList, HostListener, Inject } from '@angular/core';
import { NbpStepDimension } from './../nbp-commons/nbp-commons.enums';
import { coerceBooleanProperty } from './../../utility/lib-utility';
import { DOCUMENT } from '@angular/platform-browser';
var NbpProgressComponent = (function () {
    function NbpProgressComponent(document, renderer2) {
        var _this = this;
        this.document = document;
        this.renderer2 = renderer2;
        /**
         * Proprietà che indica lo step corrente
         * Oggetto strutturato che contiene stepId e pageId
         */
        // @Input() nbpCurrentStep: object;
        this._nbpWidth = 'col-xs-8';
        this._nbpOffset = 'col-xs-offset-0';
        this._nbpStepType = NbpStepDimension; // :NbpStepDimension;
        this._nbpLeftOffsetForCenter = 24;
        this.sticky = false;
        this.stickyActive = false;
        this.initializeProgressBar = function () {
            // setup Step Dimension to Large if not defined
            if (_this.nbpConfiguration) {
                _this.nbpConfiguration.forEach(function (stepConf, index) {
                    // set nbpConfiguration stepDimension(optional) if not declared
                    if (!stepConf.stepDimension) {
                        stepConf.stepDimension = _this._nbpStepType.GRANDE;
                    }
                    // calculate bootstrap col-__-offset depending on stepDimension
                    // this._nbpLeftOffsetForCenter = 24;
                    // switch (stepConf.stepDimension) {
                    //   case this._nbpStepType.GRANDE:
                    //     this._nbpLeftOffsetForCenter = this._nbpLeftOffsetForCenter - 3;
                    //     break;
                    //   case this._nbpStepType.MEDIO:
                    //     this._nbpLeftOffsetForCenter = this._nbpLeftOffsetForCenter - 2;
                    //     break;
                    //   case this._nbpStepType.PICCOLO:
                    //     this._nbpLeftOffsetForCenter = this._nbpLeftOffsetForCenter - 1;
                    //     break;
                    // }
                    // remove old and add new bootstrap col- class depending of stepDimension
                    // if ( this.stepList.length > 0 ) {
                    //   let currentElement = this.stepList.toArray()[index].nativeElement;
                    //   if ( currentElement.classList.contains(/\bcol-.{2}-[1-9]{1}[0-9]{0,1}\b/g) ) {
                    //     currentElement.classList.removeClass(/\bcol-.{2}-[1-9]{1}[0-9]{0,1}\b/g);
                    //   }
                    // }
                });
            }
            // calculate the offset for center the step
            // if ( this._nbpLeftOffsetForCenter > 0 ) { // && this._nbpLeftOffsetForCenter < 24
            //   let colOffset = Math.round((this._nbpLeftOffsetForCenter / 2));
            //   this._nbpOffset = `col-xs-offset-${colOffset} col-md-offset-${colOffset}`;
            // }
            // OLD CODE TO DELETE
            // if (this.nbpConfiguration) {
            //   switch (this.nbpConfiguration.length) {
            //     case 1:
            //       this._nbpWidth = 'col-xs-4 col-md-4';
            //       this._nbpOffset = 'col-xs-offset-12 col-md-offset-12';
            //       break;
            //     case 2:
            //       this._nbpWidth = 'col-xs-4 col-md-4';
            //       this._nbpOffset = 'col-xs-offset-8 col-md-offset-8';
            //       break;
            //     case 3:
            //       this._nbpWidth = 'col-xs-4 col-md-4';
            //       this._nbpOffset = 'col-xs-offset-6 col-md-offset-6';
            //       break;
            //     case 4:
            //       this._nbpWidth = 'col-xs-4 col-md-4';
            //       this._nbpOffset = 'col-xs-offset-4 col-md-offset-4';
            //       break;
            //     case 5:
            //       this._nbpWidth = 'col-xs-4 col-md-4';
            //       this._nbpOffset = 'col-xs-offset-2 col-md-offset-2';
            //       break;
        };
    }
    Object.defineProperty(NbpProgressComponent.prototype, "nbpDisableSticky", {
        /**
         * Booleano che abilita/disabilita il posizionamento sticky
         * al top della pagina
         */
        get: function () { return this._nbpDisableSticky; },
        set: function (value) { this._nbpDisableSticky = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    ;
    ;
    NbpProgressComponent.prototype.ngOnInit = function () {
        // console.log('nbpProgressInit');
        // console.log('progress: ', this.progress);
        this.renderer2.addClass(this.progress.nativeElement, 'header');
        // console.log('progress: ', this.progress);
        this.renderer2.addClass(this.steps.nativeElement, 'header');
    };
    NbpProgressComponent.prototype.ngOnDestroy = function () {
        /*
        this.renderer2.removeClass(this.progress.nativeElement, 'header');
    
        this.renderer2.removeClass(this.step.nativeElement, 'header');
    
        console.log('progress destroyed', this.progress);
        */
    };
    NbpProgressComponent.prototype.ngDoCheck = function () {
        this.initializeProgressBar();
    };
    NbpProgressComponent.prototype.onWindowScroll = function () {
        var number = window.pageYOffset || document.documentElement.scrollTop || document.body.scrollTop || 0;
        if (number > 70) {
            this.sticky = true;
        }
        else {
            this.sticky = false;
        }
    };
    return NbpProgressComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpProgressComponent.prototype, "nbpConfiguration", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpProgressComponent.prototype, "nbpDisableSticky", null);
__decorate([
    ViewChild('progress'),
    __metadata("design:type", ElementRef)
], NbpProgressComponent.prototype, "progress", void 0);
__decorate([
    ViewChild('steps'),
    __metadata("design:type", ElementRef)
], NbpProgressComponent.prototype, "steps", void 0);
__decorate([
    ViewChild('step'),
    __metadata("design:type", QueryList)
], NbpProgressComponent.prototype, "stepList", void 0);
__decorate([
    HostListener("window:scroll", []),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NbpProgressComponent.prototype, "onWindowScroll", null);
NbpProgressComponent = __decorate([
    Component({
        selector: 'nbp-progress',template: "<div #progress id=\"progress-container\" class=\"nbp-progress-container\" [ngClass]=\"{'sticky':sticky && !_nbpDisableSticky}\"><div class=\"progress-container container\"><div #steps id=\"step-container\" class=\"progress-steps-container\"><div #step *ngFor=\"let stepConf of nbpConfiguration; let stepIndex = index\" class=\"step pad-l-5 pad-r-5\" [ngClass]=\"{'step-grande': stepConf.stepDimension === _nbpStepType.GRANDE, 'step-medio': stepConf.stepDimension === _nbpStepType.MEDIO, 'step-piccolo': stepConf.stepDimension === _nbpStepType.PICCOLO } \"><div class=\"progress\" [ngClass]=\"{'active': stepConf.active || stepConf.fillPercent === 100}\"><div class=\"progress-bar\" [ngStyle]=\"{ width: stepConf.fillPercent + '%' }\"></div></div><div class=\"progress-step-name\" [ngClass]=\"{'active': stepConf.active || stepConf.fillPercent === 100}\">{{stepConf.label}}</div></div></div></div><ng-content></ng-content></div><div *ngIf=\"stickyActive\" style=\"height:5.375em\"></div>",
        styles: ["#progress-container.header{border-top:none!important}#step-container.header{height:3.125em!important}.nbp-progress-container{background-color:#f8f8f8;margin-left:0;margin-right:0;z-index:9040!important;border-bottom:1px solid rgba(204,204,204,.5)}.nbp-progress-container.sticky{position:fixed!important;top:42px;left:0;width:100%!important;z-index:9040!important;-moz-transition:all .25s ease-out;-o-transition:all .25s ease-out;transition:all .25s ease-out}:host(.sticky.menu-opened) .nbp-progress-container.sticky{top:calc(112px + 22em)!important}.progress-container{display:table}.progress-steps-container{height:5em;display:table-cell;padding-top:5px;padding-bottom:5px;vertical-align:middle;text-align:center}.progress-steps-container>.step{display:inline-block;float:none!important}.progress-steps-container>.step.step-grande>.progress{width:6.875em}.progress-steps-container>.step.step-medio>.progress{width:5em}.progress-steps-container>.step.step-piccolo>.progress{width:2.5em}.progress{height:.625em;background-color:transparent;border-radius:.625em;border:.125em solid #ccc;margin-bottom:.25em;box-shadow:none;overflow:hidden}.progress.active{border:.125em solid #258900}.progress-bar{border-radius:.625em;background-color:#258900;box-shadow:none}.progress-step-name{position:relative;text-align:center;font-size:.625em;font-weight:700;text-transform:uppercase;color:#6f6f6f}.progress-step-name.arrowStep:after{position:absolute;-webkit-transform:rotate(45deg);transform:rotate(45deg);content:\" \";width:15px;height:15px;border:0 solid #ccc;border-width:1px 0 0 1px;background:#fff;left:0;right:0;bottom:-32px;margin:0 auto}.progress-step-name.active{color:#258900}:host-context(.contrasto-on) .progress{background-color:#fff;border:.125em solid #000}:host-context(.contrasto-on) .nbp-progress-container{background-color:#fff;border-bottom:1px solid #000}:host-context(.contrasto-on) .progress.active{border:.125em solid #000}:host-context(.contrasto-on) .progress-bar{background-color:#000}:host-context(.contrasto-on) .progress-step-name{color:#000}:host-context(.contrasto-on) .progress-step-name.arrowStep:after{transform:rotate(45deg);content:\" \";width:15px;height:15px;border:0 solid #000;border-width:1px 0 0 1px;background:#fff;left:0;right:0;bottom:-32px;margin:0 auto}:host-context(.contrasto-on) .progress-step-name.active{color:#000}@media (max-width:767px){.progress-container{width:80%}}:host-context(.contrasto-on) .progress.active{border:.125em solid #000}:host-context(.contrasto-on) .progress-bar{background-color:#000}:host-context(.contrasto-on) .progress-step-name{color:#000}:host-context(.contrasto-on) .progress-step-name.active{color:#000} /*# sourceMappingURL=nbp-progress.component.css.map */ "]
    }),
    __param(0, Inject(DOCUMENT)),
    __metadata("design:paramtypes", [Object, Renderer2])
], NbpProgressComponent);
export { NbpProgressComponent };
