// cjlib/nbp/nbp-burger-menu-generic-no-translate/nbp-burger-menu-generic-no-translate.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { Router, ActivatedRoute } from '@angular/router';
var NbpBurgerMenuGenericNoTranslateComponent = (function () {
    function NbpBurgerMenuGenericNoTranslateComponent(router, activatedRoute) {
        this.router = router;
        this.activatedRoute = activatedRoute;
        /**
         * (Opzionale) booleano per disabilitare la possibilità di fare logout.
         * Valore default false
         */
        this.nbpDisableLogout = false;
        /**
         * Evento emesso sulla pressione del logout.
         * @param {EventEmitter<INbpUser>}  nbpOnLogout
         */
        this.nbpOnLogout = new EventEmitter();
        /**
         * Evento emesso sulla pressione di ogni voce di menu'.
         * @param {EventEmitter<INbpItemMenu>}  nbpOnItemClick
         */
        this.nbpOnItemClick = new EventEmitter();
        /**
         * Evento emesso all'apertura del burgerMenu.
         * @param {EventEmitter<any>}  nbpOnOpen
         */
        this.nbpOnOpen = new EventEmitter();
        /**
         * Evento emesso alla chiusure del burgerMenu.
         * @param {EventEmitter<any>}  nbpOnOpen
         */
        this.nbpOnClose = new EventEmitter();
        this._isOpen = false; // true in test phase
        this._showSecondLevelMenu = false; // false in test phase
    }
    NbpBurgerMenuGenericNoTranslateComponent.prototype._translate = function (label) {
        return label;
    };
    NbpBurgerMenuGenericNoTranslateComponent.prototype._changeRoute = function (routeValue) {
        if (routeValue.external) {
            if (routeValue.link.search('http://') > -1) {
                window.location.href = routeValue.link;
            }
            else {
                window.location.href = 'http://' + routeValue.link;
            }
        }
        else {
            this.router.navigate([routeValue.link]);
        }
    };
    NbpBurgerMenuGenericNoTranslateComponent.prototype._onClick = function (item) {
        var _this = this;
        // console.log('click item menu: ' + JSON.stringify(item));
        this.nbpOnItemClick.emit(item);
        if (!item.secondLevelMenu) {
            this.toggle();
        }
        if (item.secondLevelMenu && item.secondLevelMenu.sections && item.secondLevelMenu.sections.length > 0) {
            this._selectedItem = item;
            this._showSecondLevelMenu = true;
            setTimeout(function () {
                var height = document.getElementsByClassName('second-level-menu')[0].clientHeight + 20;
                _this.height = height;
            }, 100);
        }
    };
    NbpBurgerMenuGenericNoTranslateComponent.prototype._goBack = function () {
        var _this = this;
        this._showSecondLevelMenu = false;
        this._selectedItem = null;
        setTimeout(function () {
            var height = document.getElementsByClassName('first-level-menu')[0].clientHeight + 20;
            _this.height = height;
        }, 100);
    };
    // Metodi Pubblici del componente
    /**
     * Permette apertura programmatica del BurgerMenu
     */
    NbpBurgerMenuGenericNoTranslateComponent.prototype.open = function () {
        var _this = this;
        this._isOpen = true;
        this.nbpOnOpen.emit();
        setTimeout(function () {
            if (!_this._showSecondLevelMenu) {
                var height = document.getElementsByClassName('first-level-menu')[0].clientHeight + 20;
            }
            else {
                var height = document.getElementsByClassName('second-level-menu')[0].clientHeight + 20;
            }
            _this.height = height;
        }, 100);
    };
    /**
     * Permette la chiusura programmatica del BurgerMenu
     */
    NbpBurgerMenuGenericNoTranslateComponent.prototype.close = function () {
        var _this = this;
        this._isOpen = false;
        this.nbpOnClose.emit();
        setTimeout(function () {
            _this.height = 0;
        }, 100);
    };
    /**
     * Permette di eseguire il toggle del BurgerMenu
     */
    NbpBurgerMenuGenericNoTranslateComponent.prototype.toggle = function () {
        this._isOpen ? this.close() : this.open();
    };
    NbpBurgerMenuGenericNoTranslateComponent.prototype.onLogout = function () {
        alert('logOut');
    };
    NbpBurgerMenuGenericNoTranslateComponent.prototype.ngOnInit = function () {
    };
    return NbpBurgerMenuGenericNoTranslateComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpBurgerMenuGenericNoTranslateComponent.prototype, "nbpMenu", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpBurgerMenuGenericNoTranslateComponent.prototype, "nbpUser", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpBurgerMenuGenericNoTranslateComponent.prototype, "nbpDisableLogout", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpBurgerMenuGenericNoTranslateComponent.prototype, "nbpOnLogout", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpBurgerMenuGenericNoTranslateComponent.prototype, "nbpOnItemClick", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpBurgerMenuGenericNoTranslateComponent.prototype, "nbpOnOpen", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpBurgerMenuGenericNoTranslateComponent.prototype, "nbpOnClose", void 0);
NbpBurgerMenuGenericNoTranslateComponent = __decorate([
    Component({
        selector: 'nbp-burger-menu-generic-no-translate',template: "<div class=\"menu\" [ngClass]=\"{'closed': !_isOpen, 'open': _isOpen}\" [style.height.px]=\"height\"><div class=\"container first-level-menu\" [ngClass]=\"{'open': !_showSecondLevelMenu}\"><div class=\"row\"><div class=\"col-sm-16\"><div><div class=\"col-sm-12 section-menu\" *ngFor=\"let section of nbpMenu.sections; let i = index\"><ul class=\"main-menu\"><li><a *ngFor=\"let item of section.items\" (click)=\"_onClick(item);_changeRoute(item)\"><span *ngIf=\"item.icon\" class=\"icon {{item.icon}}\" style=\"padding-top:0.25em\"></span> <span>{{_translate(item.label)}}</span></a></li></ul></div></div></div><div class=\"col-sm-8\"><ng-content selector=\"app-logged-user\"></ng-content></div></div></div><div class=\"container second-level-menu\" [ngClass]=\"{'open': _showSecondLevelMenu}\" *ngIf=\"_showSecondLevelMenu\"><div class=\"row upper-bar\"><div class=\"menu-principale pull-left\"><span class=\"glyphicon icon icon-arrow-left\"></span> <span class=\"menu-principale-title\" (click)=\"_goBack();\">Menu principale</span></div></div><div class=\"row second-level-buttons\"><div class=\"selected-first-level-menu\"><span class=\"icon {{_selectedItem.icon}}\"></span> <span class=\"selected-first-level-label\">{{_translate(_selectedItem.label)}}</span></div><div class=\"col-sm-8\" *ngFor=\"let section of _selectedItem.secondLevelMenu.sections; let i = index\" [ngClass]=\"{'main-menu-container-pad-l': i > 0, 'main-menu-container-pad-r': i !== _selectedItem.secondLevelMenu.sections.length - 1}\"><ul class=\"main-menu\"><li><a *ngFor=\"let item of section.items\" (click)=\"_onClick(item);_changeRoute(item)\"><span *ngIf=\"item.icon\" class=\"icon {{item.icon}}\"></span> <span>{{_translate(item.label)}}</span></a></li></ul></div></div></div></div>",
        styles: [".menu{position:relative;width:100%;height:0;overflow-x:hidden;overflow-y:hidden;background-color:#323232;-moz-transition:height,opacity .7s ease-out;-o-transition:height,opacity .7s ease-out;transition:height,opacity .7s ease-out}.menu .icon{margin-right:.313em;font-size:1.563em;vertical-align:middle}.menu.closed{height:0}.main-menu-container{padding:0}.section-menu:nth-child(2n+3){clear:left}.main-menu{list-style:none;padding:0 .313em 0 .313em;margin:0}.main-menu a{color:#ccc;display:table;width:19.375em;padding:0 1.25em 0 1.25em;margin:.5em 0 .5em 0;text-decoration:none;cursor:pointer;height:2.813em}.main-menu a:active,.main-menu a:focus,.main-menu a:hover{background-color:#258900;color:#fff}.main-menu a span{display:table-cell;vertical-align:middle;text-align:left}.main-menu a span:nth-child(1){width:1.563em}.main-menu .select{background-color:#258900;color:#fff;border:.0625em solid #258900}.upper-bar{border-bottom:.0625em solid #6f6f6f}.menu-principale{color:#ccc;font-size:1em;padding:1em .625em .625em .625em;cursor:pointer}.menu-principale .icon{font-size:.875em}.menu-principale-title{vertical-align:top;-moz-animation:fadein 3.7s ease-out;animation:fadein 3.7s ease-out}@-moz-keyframes fadein{0%{opacity:0}100%{opacity:1}}@keyframes fadein{0%{opacity:0}100%{opacity:1}}.selected-first-level-menu{color:#ccc;padding:1.625em .625em 1.62em .625em}.selected-first-level-menu .selected-first-level-label{font-size:1.125em}.first-level-menu{position:absolute;left:0;visibility:hidden;right:0;opacity:0;left:-6.25em;padding:0 .625em 0 .625em;-moz-transition:all .7s ease-out;-o-transition:all .7s ease-out;transition:all .7s ease-out}.first-level-menu.open{opacity:1;left:0;visibility:visible;margin-top:.625em}.second-level-menu{position:absolute;left:0;visibility:hidden;right:0;opacity:0;left:6.25em;-moz-transition:all .7s ease-out;-o-transition:all .7s ease-out;transition:all .7s ease-out}.second-level-menu.open{opacity:1;left:0;visibility:visible}@media (max-width:992px){.main-menu a{width:14.375em}}:host-context(.contrasto-on) .menu{background-color:#000}:host-context(.contrasto-on) .menu.open{border-color:#fff}:host-context(.contrasto-on) .main-menu a{color:#fff;border-color:#fff}:host-context(.contrasto-on) .main-menu a:active,:host-context(.contrasto-on) .main-menu a:focus,:host-context(.contrasto-on) .main-menu a:hover{background-color:#fff;color:#000;border-color:#fff}:host-context(.contrasto-on) .main-menu .select{background-color:#fff;color:#000;border-color:#fff}:host-context(.contrasto-on) .upper-bar{border-color:#fff}:host-context(.contrasto-on) .menu-principale{color:#fff}:host-context(.contrasto-on) .selected-first-level-menu{color:#fff}@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none){[class*=' icon-']::before{font-size:1.375em!important}} /*# sourceMappingURL=nbp-burger-menu-generic-no-translate.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [Router,
        ActivatedRoute])
], NbpBurgerMenuGenericNoTranslateComponent);
export { NbpBurgerMenuGenericNoTranslateComponent };
