// cjlib/cj/cj-signature/cj-signature.interfaces.js
/**
 * Elenco Sign Description censite
 */
export var SignDescription;
(function (SignDescription) {
    SignDescription[SignDescription["ALTRO"] = 0] = "ALTRO";
    SignDescription[SignDescription["ALTRO_REMOTA"] = 1] = "ALTRO_REMOTA";
    SignDescription[SignDescription["CARTACEA"] = 2] = "CARTACEA";
    SignDescription[SignDescription["DIGITALE_REMOTA"] = 3] = "DIGITALE_REMOTA";
    SignDescription[SignDescription["GRAFOMETRICA"] = 4] = "GRAFOMETRICA";
})(SignDescription || (SignDescription = {}));
/**
 * Elenco Sign Type censiti
 */
export var SignType;
(function (SignType) {
    SignType[SignType["CARTA"] = 2] = "CARTA";
    SignType[SignType["FEA"] = 1] = "FEA";
    SignType[SignType["FDR"] = 0] = "FDR";
    SignType[SignType["FEA_RIATTIVAZIONE"] = 3] = "FEA_RIATTIVAZIONE";
    // FIRMA_ELETTRONICA_AVANZATA = 'FEA',
    // FIRMA_DIGITALE_REMOTA = 'FDR',
    // RIATTIVAZIONE_FEA = 'RIATTIVAZIONE_FIRMA_FEA',
    // FIRMA_CARTACEA = 'FIRMA_CARTACEA'
})(SignType || (SignType = {}));
