// cjlib/nbp/nbp-slider-storico/nbp-slider-storico-bar/nbp-slider-storico-bar.directive.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Directive, Input, ElementRef } from '@angular/core';
var NbpSliderStoricoBarDirective = (function () {
    // parent: any;
    function NbpSliderStoricoBarDirective(el) {
        this.el = el;
        this.range = 100;
    }
    NbpSliderStoricoBarDirective.prototype.ngOnInit = function () {
        var _this = this;
        // this.parent = this.el.nativeElement.parentElement.getBoundingClientRect();
        this.range = this.max - this.min;
        this.setBarPosition();
        /* Rileva Model Changes */
        this
            .model
            .valueChanges
            .subscribe(function (value) {
            /* Re-render Bar */
            _this.setBarPosition();
        });
    };
    // public reRenderBar() {
    //   this.range = this.max - this.min;
    //   this.setBarPosition();
    // }
    NbpSliderStoricoBarDirective.prototype.setBarPosition = function () {
        var value = this.model.value;
        var rightPos = this.calculateNextPosFromValue(value);
        var rightPosPerc = rightPos / this.getParent().width * 100;
        this.el.nativeElement.style.left = 0 + "px";
        this.el.nativeElement.style.width = rightPosPerc + "%";
    };
    NbpSliderStoricoBarDirective.prototype.calculateNextPosFromValue = function (value) {
        return ((value - this.min) / this.range) * this.getParent().width;
    };
    NbpSliderStoricoBarDirective.prototype.getParent = function () {
        return this.el.nativeElement.parentElement.getBoundingClientRect();
    };
    return NbpSliderStoricoBarDirective;
}());
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderStoricoBarDirective.prototype, "model", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderStoricoBarDirective.prototype, "opts", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderStoricoBarDirective.prototype, "min", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderStoricoBarDirective.prototype, "max", void 0);
NbpSliderStoricoBarDirective = __decorate([
    Directive({
        selector: '[nbpSliderStoricoBar]'
    }),
    __metadata("design:paramtypes", [ElementRef])
], NbpSliderStoricoBarDirective);
export { NbpSliderStoricoBarDirective };
