// cjlib/nbp/nbp-slider-storico/nbp-slider-storico-bar/nbp-slider-storico-bar.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, ViewChild } from '@angular/core';
import { NbpSliderStoricoBarDirective } from './nbp-slider-storico-bar.directive';
var NbpSliderStoricoBarComponent = (function () {
    function NbpSliderStoricoBarComponent() {
        this.min = 0;
        this.max = 100;
        //range slider
        this.rangeDateDays = 100;
        this.labels = [];
        this.style = {
            label: {
                "width": "100%"
            }
        };
        this.range = 0;
        this.isInit = false;
    }
    NbpSliderStoricoBarComponent.prototype.ngOnChanges = function () {
        this.rangeDateDays = this.daysDifference(this.min, this.max);
        this.labels = this.getLabels();
    };
    NbpSliderStoricoBarComponent.prototype.setWidthLblPre = function () {
        var diffDays = this.daysDifference(this.min, this.model.value);
        var tmpWidth = (diffDays / this.rangeDateDays) * 100;
        tmpWidth = 100 - tmpWidth;
        if (tmpWidth < 10)
            tmpWidth = 10;
        return tmpWidth + "%";
    };
    NbpSliderStoricoBarComponent.prototype.setWidthLblSto = function () {
        var diffDays = this.daysDifference(this.min, this.model.value);
        var tmpWidth = (diffDays / this.rangeDateDays) * 100;
        if (tmpWidth > 87.4751)
            tmpWidth = 87.4751;
        return tmpWidth + "%";
    };
    NbpSliderStoricoBarComponent.prototype.setOpacityLblPre = function () {
        var diffDays = this.daysDifference(this.min, this.model.value);
        var tmpWidth = (diffDays / this.rangeDateDays) * 100;
        tmpWidth = 100 - tmpWidth;
        var tmpOpacity;
        if (tmpWidth < 10)
            tmpOpacity = tmpWidth / 10;
        else
            tmpOpacity = 1;
        return tmpOpacity;
    };
    NbpSliderStoricoBarComponent.prototype.setOpacityLblSto = function () {
        var diffDays = this.daysDifference(this.min, this.model.value);
        var tmpWidth = (diffDays / this.rangeDateDays) * 100;
        var tmpOpacity;
        if (tmpWidth < 10)
            tmpOpacity = tmpWidth / 10;
        else
            tmpOpacity = 1;
        return tmpOpacity;
    };
    NbpSliderStoricoBarComponent.prototype.setPallinoClass = function (value) {
        return !(value < this.model.value) ? "pallino-verde" : "pallino-nero";
    };
    NbpSliderStoricoBarComponent.prototype.getLabels = function () {
        var list = [];
        var mesi = ["GEN", "FEB", "MAR", "APR", "MAG", "GIU", "LUG", "AGO", "SET", "OTT", "NOV", "DIC"];
        var tmpWidth, buffer1, buffer2;
        var oldDate, buffer, buffer3, monthTmp, yearTmp, tmpDate;
        var diffDays;
        tmpDate = this.min;
        while (tmpDate < this.max) {
            //calcolo label
            buffer = new Date(tmpDate);
            monthTmp = buffer.getMonth();
            yearTmp = buffer.getFullYear();
            buffer1 = mesi[monthTmp];
            buffer2 = yearTmp + "";
            buffer3 = tmpDate;
            //calcolo data successiva
            if (monthTmp == 11) {
                monthTmp = 0;
                yearTmp++;
            }
            else {
                monthTmp++;
            }
            //calcolo width
            oldDate = tmpDate;
            tmpDate = (new Date(yearTmp, monthTmp, 1)).getTime();
            diffDays = this.daysDifference(oldDate, tmpDate);
            tmpWidth = (diffDays / this.rangeDateDays) * 100 + "%";
            list.push({
                label1: buffer1,
                label2: buffer2,
                value: buffer3,
                width: tmpWidth
            });
        }
        //sistema i valori sbagliati del loop. Il loop li setta in maniera errata
        diffDays = this.daysDifference(oldDate, this.max);
        list[list.length - 1].width = (diffDays / this.rangeDateDays) * 100 + "%";
        // list[list.length - 1].value = this.max;
        return list;
    };
    NbpSliderStoricoBarComponent.prototype.daysDifference = function (d0, d1) {
        var diff = new Date(+d1).setHours(12) - new Date(+d0).setHours(12);
        return Math.round(diff / 8.64e7);
    };
    return NbpSliderStoricoBarComponent;
}());
__decorate([
    ViewChild(NbpSliderStoricoBarDirective),
    __metadata("design:type", NbpSliderStoricoBarDirective)
], NbpSliderStoricoBarComponent.prototype, "bar", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderStoricoBarComponent.prototype, "min", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderStoricoBarComponent.prototype, "max", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderStoricoBarComponent.prototype, "model", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderStoricoBarComponent.prototype, "opts", void 0);
NbpSliderStoricoBarComponent = __decorate([
    Component({
        selector: 'nbp-slider-storico-bar',
        template: "<div class=\"slider-bar-container\"><div class=\"slider-bar\"><div nbpSliderStoricoBar class=\"slider-bar-highlight\" [min]=\"min\" [max]=\"max\" [opts]=\"opts\" [model]=\"model\"></div></div><div class=\"slider-labels\" [ngStyle]=\"style.label\"><ng-container *ngFor=\"let item of labels; let first = first\"><div class=\"slider-label-item margin-color\" [ngStyle]=\"{'min-width': item.width }\"><div *ngIf=\"!first\"><span>{{item.label1}}</span></div><div *ngIf=\"!first\"><span>{{item.label2}}</span></div></div></ng-container></div><div class=\"slider-points\"><ng-container *ngFor=\"let item of labels; let first = first\"><div class=\"slider-label-item margin-color\" [ngStyle]=\"{'min-width': item.width }\"><div *ngIf=\"!first\" [ngClass]=\"setPallinoClass(item.value)\"><span>X</span></div></div></ng-container></div><div class=\"slider-labels-principal\"><div class=\"storico\" [ngStyle]=\"{'width': setWidthLblSto(), 'opacity': setOpacityLblSto() }\"><div class=\"label\">STORICO</div></div><div class=\"previsionale\" [ngStyle]=\"{'width': setWidthLblPre(), 'opacity': setOpacityLblPre() }\"><div class=\"label\">PREVISIONALE</div></div></div></div>",
        styles: [".margin-color{margin-top:-70px;color:#6f6f6f;font-weight:700}.pallino-nero{background-color:#6f6f6f;color:transparent;font-size:0;height:16px;width:16px;border-radius:50%}.pallino-verde{background-color:#39ba2a;color:transparent;font-size:0;height:16px;width:16px;border-radius:50%}.slider-bar-container{position:relative}.slider-bar-container .slider-bar{height:4px;width:100%;background-color:#39ba2a;position:relative;border-radius:2px}.slider-bar-container .slider-bar .slider-bar-highlight{position:absolute;left:0;width:50%;height:100%;background-color:#6f6f6f;border-radius:2px}.slider-bar-container .slider-labels-principal{overflow:hidden;margin-top:-34px;width:100%;display:flex;font-size:18px;font-weight:900}.slider-bar-container .slider-labels-principal .storico{text-align:center;width:50%}.slider-bar-container .slider-labels-principal .storico .label{border:1px solid #6f6f6f;padding-left:18px;padding-right:18px;background-color:#6f6f6f;color:#fff;border-radius:0}.slider-bar-container .slider-labels-principal .previsionale{text-align:center;width:50%;min-width:140px}.slider-bar-container .slider-labels-principal .previsionale .label{border:1px solid #6f6f6f;padding-left:18px;padding-right:18px;background-color:green;color:#fff;border-radius:0}.slider-bar-container .slider-labels{margin-top:0;width:100%;display:flex}.slider-bar-container .slider-labels::after{display:block;position:absolute;content:'';height:1px;width:1px;background-color:#bbb;top:6px;right:0}.slider-bar-container .slider-labels .slider-label-item{text-align:left}.slider-bar-container .slider-labels .slider-label-item span{transform:translate(-25%,0);position:relative;display:inline-block}.slider-bar-container .slider-labels .slider-label-item::after{display:block;position:absolute;content:'';height:1px;width:1px;background-color:#bbb;top:6px}.slider-bar-container .slider-points{margin-top:60px;width:100%;display:flex}.slider-bar-container .slider-points::after{display:block;position:absolute;content:'';height:1px;width:1px;background-color:#bbb;top:6px;right:0}.slider-bar-container .slider-points .slider-label-item{flex:1;text-align:left}.slider-bar-container .slider-points .slider-label-item span{transform:translate(-50%,0);position:relative;display:inline-block}.slider-bar-container .slider-points .slider-label-item:first-child span{transform:translate(-20%,0)}.slider-bar-container .slider-points .slider-label-item:last-child span{transform:translate(20%,0)}.slider-bar-container .slider-points .slider-label-item::after{display:block;position:absolute;content:'';height:1px;width:1px;background-color:#bbb;top:6px} /*# sourceMappingURL=nbp-slider-storico-bar.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpSliderStoricoBarComponent);
export { NbpSliderStoricoBarComponent };
