// cjlib/nbp/nbp-sel-mercato/nbp-sel-mercato.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, ElementRef, forwardRef, Input, Output, EventEmitter, ViewChild } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
/**
 * Componente per la selezione/deselezione del mercato. Funzionamento analogo ad una checkbox
 */
var NbpSelMercatoComponent = NbpSelMercatoComponent_1 = (function () {
    function NbpSelMercatoComponent() {
        /**
         * valore emesso dal componente sull'evento onChange quando il componente è checked
         * @type {boolean}
         */
        this.trueValue = true;
        /**
         * valore emesso dal componente sull'evento onChange quando il componente è unchecked
         * @type {boolean}
         */
        this.falseValue = false;
        /**
         * evento al cambio di stato del componente. Trasporta i valori definiti in trueValue e falseValue (true e false di default)
         * @type {EventEmitter<any>}
         */
        this.change = new EventEmitter();
        this.propagateChange = function (_) { };
    }
    NbpSelMercatoComponent.prototype.writeValue = function (value) {
        if ((value !== undefined) && (value !== null)) {
            this.checked = value;
        }
    };
    NbpSelMercatoComponent.prototype.registerOnChange = function (fn) {
        this.propagateChange = fn;
    };
    NbpSelMercatoComponent.prototype.registerOnTouched = function (fn) {
        this.onTouchedCallback = fn;
    };
    ;
    NbpSelMercatoComponent.prototype.onChange = function (event) {
        this.checked = this.checkValRef.nativeElement.checked;
        this.checked = !this.checked;
        var sendVal = (this.checked) ? this.trueValue : this.falseValue;
        this.propagateChange(sendVal);
        this.change.emit(this.checked);
    };
    // _objIcon = {};
    NbpSelMercatoComponent.prototype.ngOnInit = function () {
        // if(this.iconIn){
        //   this._objIcon[this.iconIn]=true;
        // }
    };
    NbpSelMercatoComponent.prototype.getIconClass = function () {
        return this.checked ? this.iconIn + ' circle-check' : this.iconIn;
    };
    return NbpSelMercatoComponent;
}());
NbpSelMercatoComponent.idCounter = 0;
__decorate([
    ViewChild('checkVal'),
    __metadata("design:type", ElementRef)
], NbpSelMercatoComponent.prototype, "checkValRef", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSelMercatoComponent.prototype, "key", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSelMercatoComponent.prototype, "trueValue", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSelMercatoComponent.prototype, "falseValue", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSelMercatoComponent.prototype, "iconIn", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NbpSelMercatoComponent.prototype, "change", void 0);
NbpSelMercatoComponent = NbpSelMercatoComponent_1 = __decorate([
    Component({
        selector: 'nbp-sel-mercato',
        template: "<button class=\"merc-btn\" [ngClass]=\"{'btn-merc-check':checked}\" (click)=\"onChange($event)\"><input type=\"checkbox\" style=\"display: none;\" #checkVal [checked]=\"checked\" [hidden]=\"true\" (change)=\"onChange($event)\"><div class=\"icona circle icon-merc\" [ngClass]=\"getIconClass()\"></div><span class=\"label-key\" [ngClass]=\"{'label-key-check':checked}\">{{key|uppercase }}</span></button>",
        styles: [".merc-btn{border:0;min-width:130px;height:48px;border-radius:24px;display:inline-block;font-family:'Arial Bold',Arial;font-size:1em;font-weight:700;color:#6f6f6f;text-decoration:none;text-transform:uppercase;text-align:center;line-height:27px;position:relative;cursor:pointer;background:#f8f8f8;text-align:left}.circle{height:40px;width:40px;background-color:#fff;border-radius:50%;display:inline-table;vertical-align:middle;margin-top:.5px;margin-bottom:2.5px;margin-left:-2px;color:#ccc}.label-key{vertical-align:middle;line-height:48px;margin-left:10px;text-align:center;width:800%;margin-right:10px}.icon-merc{vertical-align:middle}.btn-merc-check{background:#c4dbbc}.circle-check:before{color:#258900!important}.label-key-check{color:#258900}button:active div,button:active span,button:focus div,button:focus span{position:relative;top:0;left:0}.icona:before{font-family:icomoon!important;font-style:normal!important;font-weight:400!important;font-variant:normal!important;text-transform:none!important;font-size:40px} /*# sourceMappingURL=nbp-sel-mercato.component.css.map */ "],
        providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(function () { return NbpSelMercatoComponent_1; }),
                multi: true
            }
        ]
    }),
    __metadata("design:paramtypes", [])
], NbpSelMercatoComponent);
export { NbpSelMercatoComponent };
var NbpSelMercatoComponent_1;
