// cjlib/nbp/nbp-etichetta-filtro/nbp-etichetta-filtro.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
/**
 * Componente etichetta con possibilità di chiusura. Utile per rappresentare e rimuovere filtri.
 */
var NbpEtichettaFiltroComponent = (function () {
    function NbpEtichettaFiltroComponent() {
        var _this = this;
        /**
         * evento in output al click sull'icona di rimozione
         */
        this.removeFilterEvt = new EventEmitter();
        this.removeFilter = function () {
            _this.removeFilterEvt.emit(_this.key);
        };
    }
    NbpEtichettaFiltroComponent.prototype.ngOnInit = function () {
    };
    return NbpEtichettaFiltroComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpEtichettaFiltroComponent.prototype, "key", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NbpEtichettaFiltroComponent.prototype, "removeFilterEvt", void 0);
NbpEtichettaFiltroComponent = __decorate([
    Component({
        selector: 'nbp-etichetta-filtro',
        template: "<div class=\"etic-cont\"><div class=\"cont-text\"><span class=\"over-text\">{{key|uppercase }}</span></div><i class=\"icomoon-Simboli_Risorsa-20 pull-right icon-rem\" (click)=\"removeFilter();$event.stopPropagation();\"></i></div>",
        styles: [".etic-cont{font-family:'Arial Bold',Arial;font-size:1em;font-weight:700;color:#323232;background-color:#ccc;min-width:90px;border-radius:10px;min-height:25px;padding-left:10px;text-decoration:none;text-align:left;margin-left:10px;margin-right:10px;display:table!important;vertical-align:middle;line-height:1.6em}.etic-cont .icon-rem{padding-right:10px;font-size:.625em;outline:0!important;color:#258900;padding-left:.625em;padding-top:calc(.25em);display:table-cell!important;vertical-align:middle;line-height:1.852em;float:left;cursor:pointer}button:active div,button:active i,button:active span,button:focus div,button:focus i,button:focus span{position:relative;top:0;left:0}.cont-text{display:table-cell!important;vertical-align:middle} /*# sourceMappingURL=nbp-etichetta-filtro.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpEtichettaFiltroComponent);
export { NbpEtichettaFiltroComponent };
