// cjlib/nbp/nbp-spinner/nbp-spinner.component.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
import { Component, EventEmitter, forwardRef, HostBinding, Inject, Injector, Input, Optional, Output, ViewChild, } from '@angular/core';
import { NG_ASYNC_VALIDATORS, NG_VALIDATORS, NG_VALUE_ACCESSOR, NgModel, NgControl } from '@angular/forms';
import { NbpInputElementBase } from './../nbp-commons/nbp-input-element-base';
import { NbpFormContainerComponent } from './../nbp-form-container/nbp-form-container.component';
import { coerceBooleanProperty } from './../../utility/lib-utility';
export var NBP_SPINNER_CONTROL_VALUE_ACCESSOR = {
    provide: NG_VALUE_ACCESSOR,
    useExisting: forwardRef(function () { return NbpSpinnerComponent; }),
    multi: true
};
export var NBP_INPUT_ELEMENT_BASE = {
    provide: NbpInputElementBase,
    useExisting: forwardRef(function () { return NbpSpinnerComponent; }),
    multi: true
};
/** intero autogenerato usato per id unici */
var nextId = 0;
var NbpSpinnerComponent = (function (_super) {
    __extends(NbpSpinnerComponent, _super);
    function NbpSpinnerComponent(validatori, asyncValidatori, parentFormContainer, inj) {
        var _this = _super.call(this, validatori, asyncValidatori) || this;
        _this.validatori = validatori;
        _this.asyncValidatori = asyncValidatori;
        _this.parentFormContainer = parentFormContainer;
        _this.inj = inj;
        /**
         * Identificativo univoco per il componente NbpSpinner. Se non viene passato un valore, l'id viene autogenerato.
         */
        _this.id = "nbp-spinner-" + ++nextId;
        _this._nbpDisabled = false;
        _this._nbpRequired = false;
        _this._nbpShowValidation = false;
        _this._nbpNegoziazione = false;
        /**
         * testo del placeholder
         */
        _this.placeholder = '';
        /**
         * Evento sollevato quando il valore del componente cambia
         */
        _this.nbpValueChange = new EventEmitter();
        /**
         * (Opzionale) Funzione custom di validazione del campo
         */
        _this.nbpValidation = new EventEmitter();
        return _this;
    }
    Object.defineProperty(NbpSpinnerComponent.prototype, "disabled", {
        /**
         * (Opzionale)
         * Flag che abilita/disabilita il componente.
         * Valore di default false
         */
        get: function () {
            if (this.parentFormContainer && this.parentFormContainer.disabled) {
                this._nbpDisabled = true;
            }
            return this._nbpDisabled;
        },
        set: function (value) { this._nbpDisabled = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NbpSpinnerComponent.prototype, "required", {
        /**
         * (Opzionale)
         * Flag che abilita/disabilita il required sul componente.
         * Valore di default false
         */
        get: function () { return this._nbpRequired; },
        set: function (value) { this._nbpRequired = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NbpSpinnerComponent.prototype, "nbpShowValidation", {
        /**
         * (Opzionale)
         * Booleano che consente di visualizzare i messaggi di feedback relativi alla validazione.
         * Valore di default false
         */
        get: function () {
            if (this.parentFormContainer) {
                this._nbpShowValidation = this.parentFormContainer.nbpShowValidation;
            }
            return this._nbpShowValidation;
        },
        set: function (value) { this._nbpShowValidation = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    ;
    Object.defineProperty(NbpSpinnerComponent.prototype, "nbpNegoziazione", {
        /**
         * (Opzionale)
         * Se impostato, modifica l'aspetto dello spinner. Di default è
         */
        get: function () {
            return this._nbpNegoziazione;
        },
        set: function (value) {
            this._nbpNegoziazione = value;
        },
        enumerable: true,
        configurable: true
    });
    NbpSpinnerComponent.prototype._onSubtractClick = function (event) {
        if (this.value !== undefined && this.value > this.min) {
            this.value--;
            this._emitChangeEvent(event);
        }
        else {
            this.value = this.min;
        }
    };
    NbpSpinnerComponent.prototype._onAddClick = function (event) {
        if (this.value !== undefined && this.value < this.max) {
            this.value++;
            this._emitChangeEvent(event);
        }
        else {
            this.value = this.min;
        }
    };
    NbpSpinnerComponent.prototype._onTouchedEvent = function (event) {
        this.touch();
    };
    NbpSpinnerComponent.prototype._emitChangeEvent = function (event) {
        this.nbpValueChange.emit(this.value);
        this.nbpValidation.emit(this.model);
    };
    NbpSpinnerComponent.prototype.ngOnInit = function () {
        this.ngControl = this.inj.get(NgControl);
        /*if (!this.nbpShowValidation) {
          this.nbpShowValidation = false;
        }*/
        if (this.min === undefined) {
            this.min = 0;
        }
        if (this.max === undefined) {
            this.max = 99;
        }
    };
    return NbpSpinnerComponent;
}(NbpInputElementBase));
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSpinnerComponent.prototype, "id", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSpinnerComponent.prototype, "name", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSpinnerComponent.prototype, "min", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSpinnerComponent.prototype, "max", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpSpinnerComponent.prototype, "disabled", null);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpSpinnerComponent.prototype, "required", null);
__decorate([
    HostBinding('class.nbp-show-validation'),
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpSpinnerComponent.prototype, "nbpShowValidation", null);
__decorate([
    HostBinding('class.negotiation'),
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpSpinnerComponent.prototype, "nbpNegoziazione", null);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSpinnerComponent.prototype, "placeholder", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSpinnerComponent.prototype, "nbpErrorMessage", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpSpinnerComponent.prototype, "nbpValueChange", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpSpinnerComponent.prototype, "nbpValidation", void 0);
__decorate([
    ViewChild(NgModel),
    __metadata("design:type", NgModel)
], NbpSpinnerComponent.prototype, "model", void 0);
NbpSpinnerComponent = __decorate([
    Component({
        selector: 'nbp-spinner',
        template: "<div class=\"spinner-container\"><div class=\"left\"><button class=\"spinner-button\" (click)=\"_onSubtractClick($event)\" [disabled]=\"value <= min || disabled\" (blur)=\"_onTouchedEvent($event)\"><span>-</span></button></div><div class=\"spinner-input-container\"><input type=\"number\" class=\"spinner-input\" name=\"spinnerValue\" [placeholder]=\"placeholder\" [(ngModel)]=\"value\" [min]=\"min\" [max]=\"max\" (blur)=\"_onTouchedEvent($event)\"></div><div class=\"right\"><button class=\"spinner-button\" (click)=\"_onAddClick($event)\" [disabled]=\"value >= max || disabled\" (blur)=\"_onTouchedEvent($event)\"><span>+</span></button></div></div>",
        styles: [":host(.ng-touched.ng-invalid,.nbp-show-validation.ng-invalid) .spinner-container{background-color:#fff;border:2px solid #c63200;border-radius:2px}:host(.negotiation) .spinner-input{font-size:14px!important;font-weight:400!important;color:#6f6f6f!important}:host(.negotiation) .spinner-button{width:30px!important;height:40px!important;background-color:#258900!important}.spinner-container{background-color:#fff;border:2px solid #323232;border-radius:2px;padding:4px;height:auto;overflow:auto;display:table;table-layout:fixed;width:100%}.spinner-container:disabled{background-color:#ccc}.spinner-container .left{width:30px;text-align:left;display:table-cell;vertical-align:middle}.spinner-container .left span{vertical-align:3px}.spinner-container .right{width:30px;text-align:right;display:table-cell;vertical-align:middle}.spinner-container .spinner-button{width:30px;height:32px;background-color:#258900;font-size:24px;line-height:24px;color:#fff;border:none;border-radius:2px}.spinner-container .spinner-button:disabled{background-color:#ccc}.spinner-container .spinner-input{border:none;height:32px;text-align:center;font-size:16px;color:#323232;width:100%}.spinner-container input[type=number]::-webkit-input-placeholder{text-overflow:ellipsis!important}.spinner-container input[type=number]:placeholder-shown{text-overflow:ellipsis!important}.spinner-container input:-ms-input-placeholder{-ms-text-overflow:ellipsis;text-overflow:ellipsis!important}.spinner-container input[type=number]::-moz-placeholder{text-overflow:ellipsis!important}.spinner-container input[type=number]::placeholder{text-overflow:ellipsis!important}.spinner-container .spinner-input-container{display:table-cell;vertical-align:middle}.spinner-container input[type=number]::-webkit-inner-spin-button,.spinner-container input[type=number]::-webkit-outer-spin-button{-webkit-appearance:none}.spinner-container input[type=number]{-moz-appearance:textfield}.spinner-container input[type=number]::-ms-clear{display:none} /*# sourceMappingURL=nbp-spinner.component.css.map */ "],providers: [NBP_SPINNER_CONTROL_VALUE_ACCESSOR, NBP_INPUT_ELEMENT_BASE]
    }),
    __param(0, Optional()), __param(0, Inject(NG_VALIDATORS)),
    __param(1, Optional()), __param(1, Inject(NG_ASYNC_VALIDATORS)),
    __param(2, Optional()),
    __metadata("design:paramtypes", [Array, Array, NbpFormContainerComponent,
        Injector])
], NbpSpinnerComponent);
export { NbpSpinnerComponent };
