// cjlib/nbp/nbp-header-generic/nbp-header-generic.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, ViewChild, Input } from '@angular/core';
import { NbpBurgerMenuGenericComponent } from '../nbp-burger-menu-generic/nbp-burger-menu-generic.component';
import { NbpCustomerInfoGenericComponent } from '../nbp-customer-info-generic/nbp-customer-info-generic.component';
import { Http } from '@angular/http';
import { Router } from '@angular/router';
import { NbpMenuOperationComponent } from '../nbp-menu-operation/nbp-menu-operation.component';
var NbpHeaderGenericComponent = (function () {
    function NbpHeaderGenericComponent(http, router) {
        var _this = this;
        this.http = http;
        this.router = router;
        /**
         * Stringa da visualizzare come titolo del customer-info. Viene passato al componente nbp-customer-info-generic.
         */
        this.clientTitle = "INFORMAZIONI UTENTE";
        /**
         * Proprietà che indica il nome della funzione da eseguire al click su logged-user
         */
        this.userClick = function () { };
        /**
         * Proprietà che indica il nome della funzione da eseguire al click sull'icona Intesa dell'header.
         * Di default è la navigazione alla home.
         */
        this.iconClick = function () { _this.router.navigate(['home']); };
        /**
         * Proprietà tipo string indicante l'url di un file json usato per settare un icona alternativa a quella di default.
         * Il programmatore ha la possibilità di definire la nomenclatura di suo gradimento con la condizione che termini in .json.
         * Il file prescelto dovrà essere raggiungibile da una chiamata HTTP di tipo GET che provvederà alla lettura.
         *
         * Esempi:
         *
         * 1) assets/logo.json
         * 2) assets/_logo.json
         * 3) assets/_logoCustom.json
         * 4) xxxxxx/yyyyyy.json
         * 4) .......................
         *
         * Il file json di configurazione immagine conterrà i campi
         *  1) urlImage,
         *  2) width,
         *  3) height.
         *
         * Se il file viene trovato verra inserita l'icona definita in urlImage sopra all'icona Intesa San Paolo.
         * In caso contrario verrà visualizzata la label passata nella proprietà [name] del componente header.
         *
         * esempio:
         *
         *  denominazione File di Configurazione Logo : assets/logo.json
         *  {
         *  "urlImage": "assets/images/logoProva.png",
         *  "width": "30px",
         *  "height": "30px"
         *   }
         *
         *
         */
        this.urlCustomLogo = "";
        /**
        * tradurre le voci di menu
        */
        this.nbpTranslate = true;
        /**
         * Funzione custom lanciata al click sul tasto di uscita del burger menu.
         */
        this.exitClick = function () { };
        /**
         * Booleano che indica se mostrare o meno il tasto di uscita del burger menu.
         * Default è false.
         */
        this.showExit = false;
        /**
         * Booleano che indica se mostrare o meno la freccia che apre il customer-info.
         * Default è true.
         */
        this.showClientArrow = true;
        /**
         * Booleano che indica se allineare o meno a destra il contenuto del customer-info.
         * Default è false.
         */
        this.alignClientRight = false;
        /**
         * File Configurazione Alternativa Logo : assets/logo.json.
         * Il file dovrà definire i campi urlImage, width e height.
         *
         * Se il file viene trovato verra inserita l'icona definita sopra all'icona Intesa San Paolo.
         * In caso contrario verrà visualizzata la label passata nella proprietà [name] del componente header.
         *
         * esempio:
         *
         *  denominazione File di Configurazione Logo : assets/logo.json
         *  {
         *  "urlImage": "assets/images/logoProva.png",
         *  "width": "30px",
         *  "height": "30px"
         *   }
         *
         *
         */
        this.logoJson = null;
    }
    NbpHeaderGenericComponent.prototype.openTab = function () {
        if (!!this.ChildMO) {
            this.ChildMO.close();
        }
        if (!!this.ChildBM) {
            this.ChildBM.toggle();
        }
        if (!!this.ChildCC) {
            this.ChildCC.closeCustomer();
        }
    };
    NbpHeaderGenericComponent.prototype.openOperation = function () {
        if (!!this.ChildBM) {
            this.ChildBM.close();
        }
        if (!!this.ChildMO) {
            this.ChildMO.toggle();
        }
        if (!!this.ChildCC) {
            this.ChildCC.closeCustomer();
        }
    };
    NbpHeaderGenericComponent.prototype.isMenuOpen = function () {
        return this.ChildBM ? this.ChildBM._isOpen : false;
    };
    NbpHeaderGenericComponent.prototype.isMenuOperationOpen = function () {
        if (!!this.ChildMO) {
            return this.ChildMO._isOpen;
        }
        else {
            return false;
        }
    };
    NbpHeaderGenericComponent.prototype.onLogout = function () {
    };
    NbpHeaderGenericComponent.prototype.ngOnInit = function () {
        var _this = this;
        if (this.urlCustomLogo)
            this.http.get(this.urlCustomLogo)
                .subscribe(function (res) {
                _this.logoJson = res.json();
            });
    };
    NbpHeaderGenericComponent.prototype.customClickAndClose = function () {
        this.userClick();
        if (!!this.ChildBM) {
            this.ChildBM.close();
        }
        if (!!this.ChildMO) {
            this.ChildMO.close();
        }
    };
    return NbpHeaderGenericComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpHeaderGenericComponent.prototype, "burger", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpHeaderGenericComponent.prototype, "operation", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpHeaderGenericComponent.prototype, "user", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpHeaderGenericComponent.prototype, "name", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpHeaderGenericComponent.prototype, "clients", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpHeaderGenericComponent.prototype, "clientTitle", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpHeaderGenericComponent.prototype, "userClick", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpHeaderGenericComponent.prototype, "iconClick", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpHeaderGenericComponent.prototype, "urlCustomLogo", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpHeaderGenericComponent.prototype, "nbpTranslate", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpHeaderGenericComponent.prototype, "exitClick", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpHeaderGenericComponent.prototype, "showExit", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpHeaderGenericComponent.prototype, "showClientArrow", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpHeaderGenericComponent.prototype, "alignClientRight", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpHeaderGenericComponent.prototype, "nbpLogo", void 0);
__decorate([
    ViewChild(NbpCustomerInfoGenericComponent),
    __metadata("design:type", NbpCustomerInfoGenericComponent)
], NbpHeaderGenericComponent.prototype, "ChildCC", void 0);
__decorate([
    ViewChild('ChildBM'),
    __metadata("design:type", NbpBurgerMenuGenericComponent)
], NbpHeaderGenericComponent.prototype, "ChildBM", void 0);
__decorate([
    ViewChild('ChildMO'),
    __metadata("design:type", NbpMenuOperationComponent)
], NbpHeaderGenericComponent.prototype, "ChildMO", void 0);
NbpHeaderGenericComponent = __decorate([
    Component({
        selector: 'nbp-header-generic',template: "<div class=\"header\"><div class=\"navbar\" id=\"header-navbar\"><div class=\"container\" style=\"display: table;\"><div *ngIf=\"logoJson; then logoAlternativo else logoTradizionale\"></div><ng-template #logoTradizionale><div class=\"navbar-logo pull-left col-md-14 col-sm-4 col-xs-4\" style=\"top: 8px;\"><label class=\"nome-applicativo\">{{name}}</label> <a *ngIf=\"!nbpLogo\" class=\"navbar-logo-brand-small\" (click)=\"iconClick()\"><span class=\"sr-only\">Intesa San Paolo</span> </a><img *ngIf=\"nbpLogo\" src=\"{{nbpLogo}}\" class=\"navbar-logo-brand-small-custom\" (click)=\"iconClick()\"></div></ng-template><ng-template #logoAlternativo><div class=\"navbar-logo pull-left col-md-14 col-sm-4 col-xs-4\" style=\"top: 8px; left:-14px\"><img src=\"{{logoJson.urlImage}}\" [ngStyle]=\"{ 'height': logoJson.height, 'width': logoJson.width}\"> <a *ngIf=\"!nbpLogo\" class=\"navbar-logo-brand-small\" (click)=\"iconClick()\"><span class=\"sr-only\">Intesa San Paolo</span> </a><img *ngIf=\"nbpLogo\" src=\"{{nbpLogo}}\" class=\"navbar-logo-brand-small-custom\" (click)=\"iconClick()\"></div></ng-template><div class=\"navbar-toggle pull-right col-md-4 col-sm-8 col-xs-8\" (click)=\"openTab()\" [ngClass]=\"{'navbar-toggle-open': isMenuOpen()}\"><button type=\"button\"><span class=\"icon {{isMenuOpen() ? 'ico-comuni-chiudi' : 'ico-navbar-menu'}}\"></span><div [ngClass]=\"{'menuMargin': !isMenuOpen()}\"><label style=\"color: #f8f8f8 !important; margin: 0 0 0 0.0em !important; cursor: pointer; font-size: 0.75em !important; padding-top:10px\">Menu</label></div></button></div><div *ngIf=\"operation\" class=\"navbar-toggle pull-right col-md-4 col-sm-8 col-xs-8\" (click)=\"openOperation()\" [ngClass]=\"{'navbar-toggle-open': isMenuOperationOpen()}\"><button type=\"button\"><span class=\"icon {{isMenuOperationOpen() ? 'ico-comuni-chiudi' : 'icomoon-Soldi_Risorsa-45'}}\"></span><div [ngClass]=\"{'operationMargin': !isMenuOperationOpen(), 'menuMargin': isMenuOperationOpen()}\"><label style=\"color: #f8f8f8 !important; margin: 0 0 0 0.0em !important; cursor: pointer; font-size: 0.75em !important; padding-top:10px\">Operazioni</label></div></button></div><div *ngIf=\"clients\" class=\"navbar-info-cliente col-md-6 col-sm-10 col-xs-10 pull-right\"><div><nbp-customer-info-generic [showArrow]=\"showClientArrow\" [alignRight]=\"alignClientRight\" [burger]=\"ChildBM\" [operation]=\"ChildMO\" [clientsInfo]=\"clients\" [clientTitle]=\"clientTitle\"></nbp-customer-info-generic></div></div><ng-content></ng-content></div><nbp-burger-menu-generic #ChildBM [nbpMenu]=\"burger\" [nbpTranslate]=\"nbpTranslate\"><nbp-logged-user-generic [nbpUser]=\"user\" (nbpLogout)=\"onLogout()\" (click)=\"customClickAndClose()\" disableLogout></nbp-logged-user-generic><div *ngIf=\"showExit\" class=\"exit-button-container\"><span class=\"exit-button\"><a (click)=\"exitClick()\"><table><tr><th class=\"exit-image-container\"><div class=\"icon icomoon-Sicurezza_Risorsa-19\"></div></th><th class=\"exit-info\"><span>Esci</span></th></tr></table></a></span></div></nbp-burger-menu-generic><nbp-menu-operation #ChildMO [nbpMenu]=\"operation\" [nbpTranslate]=\"nbpTranslate\"></nbp-menu-operation></div></div>",
        styles: [".header{width:100%;background-color:#f2f2f2;box-shadow:0 0 1.25em -.625em rgba(0,0,0,.8)}.header .navbar{position:relative;border:0!important;border-radius:0!important;margin:0!important;left:0;right:0;width:100%;background-color:#258900}.header .navbar .container{display:table;position:relative;padding-right:0;background-color:#258900;height:3em}.header .navbar .container-slim{height:3.125em!important}.header .row{margin:0}.header-overlay{position:relative;top:0;left:0;width:100%;height:100%;background:#f2f2f2}.navbar .container .navbar-logo{display:table-cell;vertical-align:middle;width:17em}.navbar .container .navbar-toggle{display:table-cell;vertical-align:middle;width:5em;height:4.375em;background-color:transparent;border:0;border-radius:0;margin:0;padding:0;text-align:center;cursor:pointer}.navbar .container .navbar-toggle:hover{background:#323232}.navbar .container .navbar-toggle:focus{background:#323232}.navbar .container .navbar-toggle:active{background:#323232}.navbar .container .navbar-info-cliente{display:table-cell;vertical-align:middle;width:22.25em;height:4.375em;background-color:transparent;border-left:1px solid #fff;border-right:1px solid #fff;text-align:left;color:#fff;line-height:1em;padding-top:4px}.navbar .container .navbar-info-cliente .container-arrow{float:right;padding-top:1.25em;padding-right:1.5625em}.navbar .container .navbar-info-cliente .accordion-arrow{cursor:pointer;font-size:1.6em}.navbar .container .navbar-info-cliente .info-cliente-open{position:absolute;background-color:#323232;z-index:1000;width:22.25em;height:9.8125em;border:0;border-radius:0;text-align:center;margin-top:.5625em;display:table}.navbar .container .navbar-info-cliente .info-cliente-open .container-nsg-title{vertical-align:middle;height:2.3125em;display:table-cell}.navbar .container .navbar-info-cliente .info-cliente-open .container-nsg-cliente{vertical-align:middle;height:3.75em;border-top:1px solid #6f6f6f;display:table-cell;cursor:pointer;text-align:left}.navbar .container .navbar-info-cliente .info-cliente-open .container-nsg-cliente-select{vertical-align:middle;height:3.75em;border-top:1px solid #6f6f6f;display:table-cell;cursor:pointer;text-align:left;background-color:#3f3f3f;font-weight:700}.navbar .container .navbar-info-cliente .info-cliente-open .container-nsg-cliente-select .icon{float:right;color:#258900;padding-right:10px}.navbar .container .navbar-info-cliente .info-cliente-open .title-nsg-label{color:#ccc;text-transform:uppercase}.navbar .container .navbar-info-cliente .info-cliente-open .nsg-cliente{margin-left:1em;color:#ccc}.navbar .container .navbar-info-cliente .info-cliente-open .nsg-cliente-selected{margin-left:1em;color:#fff}.navbar .container .navbar-hole{display:table-cell;vertical-align:middle;margin-top:5px;margin-bottom:5px}.navbar-logo-brand{display:block;width:8.9375em;height:1.563em;background-image:url(/cjlib/assets/images/sprite2.png);background-repeat:no-repeat}.navbar-logo-brand-small{cursor:pointer;display:block;background-image:url(/cjlib/assets/images/sprite.png);background-repeat:no-repeat;width:8.9375em;height:1.563em;background-size:8.9375em;margin-top:3px}.navbar-logo-brand-small-custom{display:block;height:25px;background-repeat:no-repeat}.navbar-toggle button{position:relative;top:52%;transform:translateY(-50%);border:none;background:0 0;outline:0;padding-bottom:3px}.navbar-toggle .icon-bar{width:1.875em;height:.125em;margin:.313em auto;background-color:#fff!important}.navbar-toggle .ico-navbar-menu{color:#fff;width:32px;height:28px;font-size:1.65em!important}.navbar-toggle .ico-navbar-menu:before{font-size:32px}.navbar-toggle .icomoon-Soldi_Risorsa-45{color:#fff;height:28px;font-size:1.65em!important}.navbar-toggle .icomoon-Soldi_Risorsa-45:before{font-size:32px}.navbar-toggle .ico-comuni-chiudi{color:#fff;font-size:1.65em!important}.navbar-toggle .menuMargin{margin-top:-.2em}.navbar-toggle .operationMargin{margin-top:-.69em}.navbar-toggle-operation button{position:relative;top:50%;transform:translateY(-50%);border:none;background:0 0;outline:0}.navbar-toggle-operation .icon-bar{width:1.875em;height:.125em;margin:.313em auto;background-color:#fff!important}.navbar-toggle-operation .icomoon-Modifica_matita{color:#fff;font-size:1.5em;margin:.313em auto}.navbar-toggle-operation span.labelOperazioni{color:#fff;font-size:.75em}.navbar-toggle-open{background:#323232!important}.nome-applicativo{position:relative;font-size:1.5em;margin-bottom:-.417em;color:#fff;font-weight:400}.header-paging-container,.header-progress-container{border-bottom:1px solid #ccc}@media (max-width:768px){.navbar .container{width:100%}}@media (-webkit-min-device-pixel-ratio:2){.navbar-logo-brand{background-image:url(/cjlib/assets/images/sprite.png)}}@media (min-resolution:192dpi){.navbar-logo-brand{background-image:url(/cjlib/assets/images/sprite.png)}}.triangolo{width:0;height:0;z-index:100;border-left:.625em solid transparent;border-right:.625em solid transparent;border-bottom:.625em solid #323232;float:right;margin-right:1.75em}.triangolo-hide{width:0;height:0;z-index:100;border-left:.625em solid transparent;border-right:.625em solid transparent;border-bottom:.625em solid transparent;float:right}.container-row{display:table-row}.container-ident-cliente{margin-top:0;margin-bottom:0;font-size:.875em;color:inherit;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;width:78%;padding-left:.625em;font-weight:400;float:left}.voce-ident-cliente{margin-top:0;margin-bottom:0;font-size:.875em;color:inherit;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;font-weight:400}.testo-ident-cliente{margin-top:0;margin-bottom:0;font-size:.875em;color:inherit;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;font-weight:700}@media (max-width:992px){.nome-applicativo{font-size:1.2em}.navbar .container .navbar-logo{width:13em}}.exit-button-container{padding:0}[class*=\" ico-\"]:before{vertical-align:middle}.exit-button{margin:0}.exit-button a{color:#fa9600;display:table;padding:0 1.25em 0 1.25em;margin:.5em 0 .5em 0;text-decoration:none;cursor:pointer;width:100%!important;height:2.813em;width:19.375em}.exit-button a:active,.exit-button a:focus,.exit-button a:hover{background-color:#fa9600;color:#323232}.exit-button a table{width:100%;display:table-cell;vertical-align:middle}.exit-image-container div{vertical-align:middle;font-size:1.563em!important;margin:0!important}.exit-image-container img{position:relative;border-radius:50%;width:100%}.exit-info{padding-left:.625em;vertical-align:middle;line-height:1;font-weight:400}:host-context(.contrasto-on) .header{background-color:#fff}:host-context(.contrasto-on) .header .navbar{background-color:#000}:host-context(.contrasto-on) .header .navbar .container{background-color:#000}:host-context(.contrasto-on) .header .navbar-tape{background-color:#000}:host-context(.contrasto-on) .header-overlay{background:#fff}:host-context(.contrasto-on) .exit-button a{color:#fff}:host-context(.contrasto-on) .exit-button a:active,:host-context(.contrasto-on) .exit-button a:focus,:host-context(.contrasto-on) .exit-button a:hover{background-color:#fff;color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle{background-color:transparent}:host-context(.contrasto-on) .navbar .container .navbar-toggle:hover{background:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle:focus{background:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle:active{background:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation{background-color:transparent}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation:hover{background:#fff;color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation:hover .icomoon-Modifica_matita{color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation:hover span{color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation:focus{background:#fff;color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation:focus .icomoon-Modifica_matita{color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation:focus span{color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation:active{background:#fff;color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation:active .icomoon-Modifica_matita{color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation:active span{color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation.navbar-toggle-open{background:#fff!important;color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation.navbar-toggle-open .icomoon-Modifica_matita{color:#000}:host-context(.contrasto-on) .navbar .container .navbar-toggle-operation.navbar-toggle-open span{color:#000}:host-context(.contrasto-on) .navbar .container .navbar-info-cliente{background-color:transparent;border-color:#fff;color:#fff}:host-context(.contrasto-on) .navbar-toggle .icon-bar{background-color:#fff!important}:host-context(.contrasto-on) .navbar-toggle .ico-comuni-chiudi,:host-context(.contrasto-on) .navbar-toggle .ico-navbar-menu,:host-context(.contrasto-on) .navbar-toggle .icomoon-Modifica_matita,:host-context(.contrasto-on) .navbar-toggle .icomoon-Soldi_Risorsa-45{color:#fff}:host-context(.contrasto-on) .navbar-toggle-open{background:#000!important}:host-context(.contrasto-on) .prev-page-btn-container .prev-page-btn{color:#000}:host-context(.contrasto-on) .page-title-container{border-color:#fff}:host-context(.contrasto-on) .nome-applicativo{color:#fff}:host-context(.contrasto-on) .header-paging-container,:host-context(.contrasto-on) .header-progress-container{border-color:#fff}:host-context(.contrasto-on) .triangolo{border-color:#fff;background:#000}:host-context(.contrasto-on) .container-ident-cliente{color:inherit}:host-context(.contrasto-on) .voce-ident-cliente{color:inherit}:host-context(.contrasto-on) .testo-ident-cliente{color:inherit} /*# sourceMappingURL=nbp-header-generic.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [Http, Router])
], NbpHeaderGenericComponent);
export { NbpHeaderGenericComponent };
