// cjlib/nbp/nbp-floater/nbp-floater.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input } from '@angular/core';
// import { TooltipComponent } from '../tooltip/tooltip.component';
import { NbpSize } from './../nbp-commons/nbp-commons.enums';
var NbpFloaterComponent = (function () {
    function NbpFloaterComponent() {
        /**
         * Box Principale Floater.
         *
         * Tipo: IBoxFloater è costituito da un array di oggetti Json.
         *
         * Ciascun oggetto Json permette di implementare  un pulsante del paginatore ed è cosi definito:
         *
         * { icon: l'icona visualizzata nel pulsante del Floater,
         *   iconText: testo visualizzato immediatamente sotto l'icona,
         *   tooltip: tooltip pulsante,
         *   scrollTo: Id del codice Html a cui il Floater si muoverà alla digitazione del pulsante,
         *   callbackBefore: funzione chiamata prima del movimento verso l'Id,
         *   callbackAfter: funzione chiamata dopo il movimento verso l'Id
         * };
         *
         * E' il box principale visualizzato dal Floater. Nella visualizzazione si alterna in modalità toggle
         * con il Box Secondario. O appare l'uno o l'altro. Lo switching avviene attraverso la digitazione del
         * pulsante di help oppure le freccie di espansione
         *
         */
        this.boxPrincipaleContents = [];
        /**
         * Icona apertura Box Principale
         *
         * Tipo: String.
         *
         * Rappesenta l'icona visualizzata quando il Box Principale è Chiuso
         *
         */
        this.iconBoxPrincipale = "icomoon-FloaterMenu";
        /**
         * Icona apertura Box Secondario
         *
         * Tipo: String.
         *
         * Rappesenta l'icona visualizzata quando il Box Principale è Chiuso
         *
         */
        this.iconBoxSecondario = "icomoon-Messagi_messagio3";
        /**
         * Box secondario Floater
         *
         * Tipo: IBoxFloater è costituito da un array di oggetti Json.
         *
         * Ciascun oggetto Json permette di implementare  un pulsante del paginatore ed è cosi definito:
         *
         * { icon: l'icona visualizzata nel pulsante del Floater,
         *   iconText: testo visualizzato immediatamente sotto l'icona,
         *   tooltip: tooltip pulsante,
         *   scrollTo: Id del codice Html a cui il Floater si muoverà alla digitazione del pulsante,
         *   callbackBefore: funzione chiamata prima del movimento verso l'Id,
         *   callbackAfter: funzione chiamata dopo il movimento verso l'Id
         * };
         *
         * E' il box secondario visualizzato dal Floater in alternativa al Box Principale.
         *
         */
        this.boxSecondarioContents = [];
        this.helpTooltipMessage = "";
        this.helpTooltip = false;
        this.sizeTooltip = NbpSize.MD;
        this.toggleOperazioni = true;
        this.toggleAssistenza = false;
        this.activeTooltip = -1;
    }
    /**
     * setActiveToolTip
     *
     *  Funzione di selezione del tooltip attivo.
     *  I tooltip sono numerizzati da un ciclo for di visualizzazione
     *  sulla base della posizione nell'array del loro pulsante di attivazione.
     *  Questa funzione riceve in input il numero del tooltip e lo setta come attivo
     */
    NbpFloaterComponent.prototype.setActiveTooltip = function (numberTooltip) {
        this.activeTooltip = numberTooltip;
    };
    /**
     * selectTooltipClassMouseOver
     *
     * ritorna valore boolean connesso al mouse over del tooltip
     *
     */
    NbpFloaterComponent.prototype.selectTooltipClassMouseOver = function (p) {
        if (this.activeTooltip == p)
            return true;
        return false;
    };
    return NbpFloaterComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpFloaterComponent.prototype, "boxPrincipaleContents", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpFloaterComponent.prototype, "iconBoxPrincipale", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpFloaterComponent.prototype, "iconBoxSecondario", void 0);
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpFloaterComponent.prototype, "boxSecondarioContents", void 0);
__decorate([
    Input("helpTooltip"),
    __metadata("design:type", String)
], NbpFloaterComponent.prototype, "helpTooltipMessage", void 0);
NbpFloaterComponent = __decorate([
    Component({
        selector: 'nbp-floater',
        template: "<div fxLayout=\"column\" fxLayoutAlign=\"center start\"><div class=\"posizioneFissaFloater\"><div *ngIf=\"toggleOperazioni\"><div class=\"boxFrecciaOperazioni\" style=\"margin-bottom:-1px\" (mouseenter)=\"helpTooltip=false; activeTooltip = -1\" (click)=\"toggleOperazioni=!toggleOperazioni;toggleAssistenza=false;helpTooltip=false\"><div class=\"icomoon-Simboli_Risorsa-10 iconFreccia\"></div></div><div class=\"box-operazioni\" *ngFor=\"let item of boxPrincipaleContents; let p = index\"><div fxLayout=\"row\" fxLayoutAlign=\"start stretch\"><div (mouseenter)=\"helpTooltip=false; activeTooltip = p\" (mouseleave)=\"helpTooltip = false; activeTooltip = -1\" (click)=\"item.function()\" [scrollTo]=\"item.scrollTo\" [callbackBefore]=\"item.callbackBefore\" [callbackAfter]=\"item.callbackAfter\" class=\"boxPrincipale floater-container\" fxLayout=\"column\" fxLayoutAlign=\"space-evenly center\" fxLayoutGap=\"-4px\"><div class=\"{{item.icon}} iconParlaConNoi iconMargin\"></div></div><div *ngIf=\"activeTooltip == p && item.tooltip\" [ngClass]=\"item.tooltip ? 'floaterTooltip' : 'floaterTooltipEmpty'\" class=\"floaterTooltip floater-tooltip\" fxLayout=\"column\" fxLayoutAlign=\"space-evenly center\">{{item.tooltip}}</div></div></div></div><div *ngIf=\"!toggleOperazioni\" fxLayout=\"row\" fxLayoutAlign=\"start stretch\"><div style=\"border-bottom: 0px solid #258900\" (mouseenter)=\"helpTooltip=false; activeTooltip = -1\" class=\"boxPrincipale\" fxLayout=\"column\" fxLayoutAlign=\"space-evenly center\" (click)=\"toggleOperazioni=!toggleOperazioni;toggleAssistenza=false;activeTooltip = -1\" [ngClass]=\"!toggleOperazioni ? 'boxPrimarioGreen' : 'boxPrimarioGreen'\"><div class=\"icomoon-FloaterMenu iconHelp iconMargin\"></div></div></div><div *ngIf=\"!toggleAssistenza\" fxLayout=\"row\" fxLayoutAlign=\"start stretch\"><div style=\"border-bottom: 0px solid #333333\" (mouseenter)=\"helpTooltip = true\" (mouseleave)=\"helpTooltip = false\" fxLayout=\"column\" fxLayoutAlign=\"space-evenly center\" (click)=\"toggleAssistenza=!toggleAssistenza;toggleOperazioni=false\" [ngClass]=\"helpTooltip ? 'boxSecondarioGray' : 'boxSecondarioBlack'\"><div class=\"iconHelp iconMargin\" [ngClass]=\"iconBoxSecondario\"></div></div><div *ngIf=\"helpTooltip\" class=\"floaterTooltip\" fxLayout=\"column\" fxLayoutAlign=\"space-evenly\">{{helpTooltipMessage}}</div></div><div *ngIf=\"toggleAssistenza\"><div *ngFor=\"let item of boxSecondarioContents; let p = index\"><div fxLayout=\"row\" fxLayoutAlign=\"start stretch\"><div (mouseenter)=\"activeTooltip = 100 + p\" (mouseleave)=\"activeTooltip = -1\" (click)=\"item.function()\" [scrollTo]=\"item.scrollTo\" [callbackBefore]=\"item.callbackBefore\" [callbackAfter]=\"item.callbackAfter\" [ngClass]=\"activeTooltip == 100 + p ? 'boxSecondarioBlack' : 'boxSecondarioGray'\" fxLayout=\"column\" fxLayoutAlign=\"space-evenly center\" fxLayoutGap=\"-4px\"><div class=\"{{item.icon}} iconAssistenza iconMarginAssistenza\"></div><div class=\"textIcon\">{{item.iconText}}</div></div><div *ngIf=\"activeTooltip == 100 + p\" (click)=\"item.function()\" class=\"floaterTooltip\" fxLayout=\"column\" fxLayoutAlign=\"space-evenly\">{{item.tooltip}}</div></div></div></div><div (click)=\"toggleAssistenza=!toggleAssistenza;toggleOperazioni=false;helpTooltip=false\"><div *ngIf=\"toggleAssistenza\" class=\"boxFrecciaAssistenzaGray\"><div class=\"icomoon-Simboli_Risorsa-11 iconFreccia\" style=\"margin-top:-1px\"></div></div></div></div></div>",
        styles: [".floaterTooltip{min-height:100%;background-color:#333;color:#fff;font-size:.813em;font-family:Arial;padding:1em;cursor:pointer;width:12.14em}.floater-container:hover{background-color:#5aa942!important}.floater-tooltip{background-color:#5aa942!important;font-family:Arial,Helvetica,sans-serif;font-weight:700;font-size:.875em}.iconMargin{margin-top:7.5px;margin-left:-1px!important}.iconMarginAssistenza{margin-top:3.5px;margin-left:-1px!important}.floaterTooltipEmpty{height:0%;width:0%;padding:0}.boxPrincipale{margin-top:-1px;font-weight:400;font-family:Arial;font-size:13px;background-color:#258900;padding:0;width:50px;height:50px;text-align:center;cursor:pointer;border-bottom:0 solid #258900}.boxPrimarioGreen{margin-top:-1px;font-weight:400;font-family:Arial;font-size:13px;background-color:#258900;padding:0;width:50px;height:50px;text-align:center;cursor:pointer;border-bottom:0 solid #258900}.boxSecondarioBlack{margin-top:-1px;font-weight:400;font-family:Arial;font-size:13px;background-color:#333;padding:0;width:50px;height:50px;text-align:center;cursor:pointer;border-bottom:0 solid #333}.boxSecondarioGray{margin-top:-1px;font-weight:400;font-family:Arial;font-size:13px;background-color:#333;border:0;padding:0;width:50px;height:50px;text-align:center;cursor:pointer;border-bottom:0 solid #333}.boxSecondarioBlack1{font-weight:400;font-family:Arial;font-size:13px;background-color:#333;padding:0;width:50px;height:45px;text-align:center;cursor:pointer;border-bottom:0 solid #333}.boxSecondarioGray1{font-weight:400;font-family:Arial;font-size:13px;background-color:#333;border:0;padding:0;width:50px;height:50px;text-align:center;cursor:pointer;border-bottom:0 solid #333}.boxFrecciaOperazioni{background-color:#258900;width:50px;height:20px;text-align:center;cursor:pointer;border-top:0 solid #258900;border-bottom:1px solid #258900}.boxFrecciaAssistenzaBlack{background-color:#333;width:50px;height:20px;text-align:center;cursor:pointer;border-top:0 solid #333;border-bottom:0 solid #333}.boxFrecciaAssistenzaGray{background-color:#333;width:50px;height:20px;text-align:center;cursor:pointer;border-top:0 solid #333}.iconParlaConNoi{font-size:30px;color:#fff;font-weight:600}.iconAssistenza{font-size:30px;color:#fff;font-weight:400}.iconHelp{font-size:30px;color:#fff;font-weight:400}.iconFreccia{font-size:20px;color:#fff;font-weight:600}.textIcon{font-weight:400;font-family:Arial;font-size:13px;font-size:10px;color:#fff}.posizioneFissaFloater{position:fixed;top:50%;transform:translatey(-50%)}:host-context(.contrasto-on) .floater-container:hover{background-color:#fff!important}:host-context(.contrasto-on) .floater-tooltip{background-color:#fff!important;color:#000!important;border:1px solid #000;border-left:none}:host-context(.contrasto-on) .floaterTooltip{background-color:#000;color:#fff}:host-context(.contrasto-on) .boxPrincipale{background-color:#fff}:host-context(.contrasto-on) .boxPrimarioGreen{background-color:#fff;border-right:1px solid #000;border-top:1px solid #000}:host-context(.contrasto-on) .boxPrimarioGreen .iconHelp{color:#000}:host-context(.contrasto-on) .box-operazioni .boxPrincipale{border-right:1px solid #000}:host-context(.contrasto-on) .box-operazioni .boxPrincipale.floater-container:hover{border-right:none}:host-context(.contrasto-on) .boxSecondarioBlack{background-color:#000}:host-context(.contrasto-on) .boxSecondarioGray{background-color:#000}:host-context(.contrasto-on) .boxSecondarioBlack1{background-color:#000}:host-context(.contrasto-on) .boxSecondarioGray1{background-color:#000}:host-context(.contrasto-on) .boxFrecciaOperazioni{background-color:#fff;border-bottom:1px solid #000}:host-context(.contrasto-on) .boxFrecciaOperazioni .iconFreccia{color:#000;border-right:1px solid #000;border-top:1px solid #000}:host-context(.contrasto-on) .boxFrecciaAssistenzaBlack{background-color:#000}:host-context(.contrasto-on) .boxFrecciaAssistenzaGray{background-color:#000}:host-context(.contrasto-on) .iconParlaConNoi{color:#000}:host-context(.contrasto-on) .iconAssistenza{color:#fff}:host-context(.contrasto-on) .iconHelp{color:#fff}:host-context(.contrasto-on) .textIcon{color:#fff} /*# sourceMappingURL=nbp-floater.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpFloaterComponent);
export { NbpFloaterComponent };
