// cjlib/nbp/nbp-wrapper/nbp-wrapper.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
import { Component, Inject } from '@angular/core';
import { NbpViewSettingsService } from './../nbp-view-settings/nbp-view-settings.service';
import { DOCUMENT } from '@angular/platform-browser';
var nextId = 0;
var NbpWrapperComponent = (function () {
    function NbpWrapperComponent(nbpViewSettingsService, document) {
        this.nbpViewSettingsService = nbpViewSettingsService;
        this.document = document;
    }
    NbpWrapperComponent.prototype.ngOnInit = function () {
        this.receiveContrastoEvent();
        this.receiveFontSizeEvent();
    };
    NbpWrapperComponent.prototype.receiveContrastoEvent = function () {
        var _this = this;
        this.nbpViewSettingsService.getContrastoEventStream()
            .subscribe(function (isContrastoOn) { return _this._isContrastoOn = isContrastoOn; });
    };
    NbpWrapperComponent.prototype.receiveFontSizeEvent = function () {
        var _this = this;
        this.nbpViewSettingsService.getFontSizeEventStream()
            .subscribe(function (fontSize) { return _this.document.body.style.fontSize = fontSize; });
    };
    return NbpWrapperComponent;
}());
NbpWrapperComponent = __decorate([
    Component({
        selector: 'nbp-wrapper',template: '<div id="nbp-wrapper-container" class="container-fluid"><ng-content></ng-content></div>',
        styles: [],
        host: { '[class.contrasto-on]': '_isContrastoOn' }
    }),
    __param(1, Inject(DOCUMENT)),
    __metadata("design:paramtypes", [NbpViewSettingsService, Object])
], NbpWrapperComponent);
export { NbpWrapperComponent };
