// cjlib/nbp/nbp-sticky-tab-generic/nbp-sticky-tab-generic.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, HostListener, Input, Output, EventEmitter } from '@angular/core';
var NbpStickyTabGenericComponent = (function () {
    function NbpStickyTabGenericComponent() {
        // items: ITabItem<any>[];
        //index: number;
        /**
         * headerTab: array degli elementi da visualizzare nei Tab.
         */
        this.items = [];
        /**
         * headerTab: Tab selezionato.
         */
        this.selectedTab = 1;
        /**
         * posStickyOn: number. Valore di offset dello scrolling a partire dal quale viene attivato la modalità Sticky
         */
        this.posStickyOn = 70;
        /**
         * tabChange: Evento generato al cambio tab.
         */
        this.tabChange = new EventEmitter();
        // variable locale uso interno 
        this.sticky = false;
    }
    NbpStickyTabGenericComponent.prototype.selectTab = function (num) {
        this.selectedTab = num;
    };
    NbpStickyTabGenericComponent.prototype.onTabChange = function (idx) {
        this.tabChange.emit(idx);
    };
    NbpStickyTabGenericComponent.prototype.onWindowScroll = function () {
        var number = window.pageYOffset || document.documentElement.scrollTop || document.body.scrollTop || 0;
        // console.log(number)
        if (number > this.posStickyOn) {
            this.sticky = true;
        }
        else {
            this.sticky = false;
        }
        //we'll do some stuff here when the window is scrolled
    };
    return NbpStickyTabGenericComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpStickyTabGenericComponent.prototype, "items", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpStickyTabGenericComponent.prototype, "selectedTab", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpStickyTabGenericComponent.prototype, "posStickyOn", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NbpStickyTabGenericComponent.prototype, "tabChange", void 0);
__decorate([
    HostListener("window:scroll", []),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NbpStickyTabGenericComponent.prototype, "onWindowScroll", null);
NbpStickyTabGenericComponent = __decorate([
    Component({
        selector: 'nbp-sticky-tab-generic',
        template: "<div style=\"min-height: 100%; overflow-x: hidden;\"><div class=\"row\"><nbp-tab-bar [ngClass]=\"{'sticky':sticky}\" [nbpItems]=\"items\" [nbpSelectedIndex]=\"selectedTab\" (nbpSelectedIndexChange)=\"onTabChange($event)\"></nbp-tab-bar></div></div>",
        styles: [":host /deep/ .tabs-container .tab-title{font-family:Arial;font-weight:Bold;font-size:.75em!important}:host /deep/ .tabs-container i{font-size:32px!important}:host /deep/ .tabs-container .tab-container{min-width:184px!important;min-height:5.375em!important;max-height:5.375em!important;background-color:#eee!important}:host /deep/ .tabs-container .tab-container.selected.enabledTab{background-color:#fff!important;border:.0625em solid transparent!important}:host /deep/ .tabs-container .tab-container.enabledTab{border:.0625em solid #eee!important}:host /deep/ .tabs-container .tab-container.disabledTab{border:.0625em solid #eee!important}.sticky{top:55px;z-index:1030} /*# sourceMappingURL=nbp-sticky-tab-generic.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpStickyTabGenericComponent);
export { NbpStickyTabGenericComponent };
