// cjlib/nbp/nbp-commons/nbp-commons.enums.js
/**
 * Enum per la definizione del colore dei componenti della libreria
 * PRIMARY - Green
 * SECONDARY - White
 * DEFAULT - Grey
 * OUTLINE - Background transparent
 * DEFAULT_L - Grey + Float left
 * DEFAULT_R - Grey + Float right
 * PRIMARY_L - Green + Float left
 * PRIMARY_R - Green + Float right
 */
/**
 * Enum per la definizione del colore dei componenti della libreria
 * PRIMARY - Green
 * SECONDARY - White
 * DEFAULT - Grey
 * OUTLINE - Background transparent
 * DEFAULT_L - Grey + Float left
 * DEFAULT_R - Grey + Float right
 * PRIMARY_L - Green + Float left
 * PRIMARY_R - Green + Float right
 */ export var NbpStyle;
(function (NbpStyle) {
    NbpStyle[NbpStyle["PRIMARY"] = 0] = "PRIMARY";
    NbpStyle[NbpStyle["SECONDARY"] = 1] = "SECONDARY";
    NbpStyle[NbpStyle["DEFAULT"] = 2] = "DEFAULT";
    NbpStyle[NbpStyle["OUTLINE"] = 3] = "OUTLINE";
    //modificareply
    NbpStyle[NbpStyle["BACKGROUNDGREEN"] = 4] = "BACKGROUNDGREEN";
    NbpStyle[NbpStyle["BACKGROUNDGRAY"] = 5] = "BACKGROUNDGRAY";
    NbpStyle[NbpStyle["BACKGROUNDGRAYPRIMARY"] = 6] = "BACKGROUNDGRAYPRIMARY";
    NbpStyle[NbpStyle["CHECK"] = 7] = "CHECK";
    NbpStyle[NbpStyle["PROFILE"] = 8] = "PROFILE";
    NbpStyle[NbpStyle["DEFAULT_L"] = 9] = "DEFAULT_L";
    NbpStyle[NbpStyle["DEFAULT_R"] = 10] = "DEFAULT_R";
    NbpStyle[NbpStyle["PRIMARY_L"] = 11] = "PRIMARY_L";
    NbpStyle[NbpStyle["PRIMARY_R"] = 12] = "PRIMARY_R";
})(NbpStyle || (NbpStyle = {}));
/**
 * Enum per la definizione della dimensione dei componenti della libreria
 * LG - Large
 * MD - Medium
 */
export var NbpSize;
(function (NbpSize) {
    NbpSize[NbpSize["SM"] = 0] = "SM";
    NbpSize[NbpSize["LG"] = 1] = "LG";
    NbpSize[NbpSize["MD"] = 2] = "MD";
})(NbpSize || (NbpSize = {}));
/**
 * Enum per la definizione della grandezza del font da utilizzare
 */
export var NbpFontSize;
(function (NbpFontSize) {
    NbpFontSize[NbpFontSize["SMALL"] = 0] = "SMALL";
    NbpFontSize[NbpFontSize["MEDIUM"] = 1] = "MEDIUM";
    NbpFontSize[NbpFontSize["LARGE"] = 2] = "LARGE";
})(NbpFontSize || (NbpFontSize = {}));
export var NbpPipe;
(function (NbpPipe) {
    NbpPipe[NbpPipe["CURRENCY"] = 0] = "CURRENCY";
    NbpPipe[NbpPipe["DATE"] = 1] = "DATE";
    NbpPipe[NbpPipe["DECIMAL"] = 2] = "DECIMAL";
    NbpPipe[NbpPipe["LOWERCASE"] = 3] = "LOWERCASE";
    NbpPipe[NbpPipe["PERCENT"] = 4] = "PERCENT";
    NbpPipe[NbpPipe["UPPERCASE"] = 5] = "UPPERCASE";
    NbpPipe[NbpPipe["CONTO"] = 6] = "CONTO";
})(NbpPipe || (NbpPipe = {}));
/**
 * Enum per la definizione del canale applicativo
 */
export var NbpChannel;
(function (NbpChannel) {
    NbpChannel[NbpChannel["ABC"] = 0] = "ABC";
    NbpChannel[NbpChannel["INTRANET"] = 1] = "INTRANET";
    NbpChannel[NbpChannel["IB"] = 2] = "IB";
})(NbpChannel || (NbpChannel = {}));
export var NbpLoadingStatus;
(function (NbpLoadingStatus) {
    NbpLoadingStatus[NbpLoadingStatus["START"] = 0] = "START";
    NbpLoadingStatus[NbpLoadingStatus["STOP"] = 1] = "STOP";
})(NbpLoadingStatus || (NbpLoadingStatus = {}));
export var NbpAlertType;
(function (NbpAlertType) {
    NbpAlertType[NbpAlertType["INFORMATIVA"] = 0] = "INFORMATIVA";
    NbpAlertType[NbpAlertType["INFORMATIVA_POSITIVA"] = 1] = "INFORMATIVA_POSITIVA";
    NbpAlertType[NbpAlertType["INFORMATIVA_PROMOZIONALE"] = 2] = "INFORMATIVA_PROMOZIONALE";
    NbpAlertType[NbpAlertType["WARNING"] = 3] = "WARNING";
    NbpAlertType[NbpAlertType["WARNING_BLOCCANTE"] = 4] = "WARNING_BLOCCANTE";
})(NbpAlertType || (NbpAlertType = {}));
export var NbpStepDimension;
(function (NbpStepDimension) {
    NbpStepDimension[NbpStepDimension["PICCOLO"] = 1] = "PICCOLO";
    NbpStepDimension[NbpStepDimension["MEDIO"] = 2] = "MEDIO";
    NbpStepDimension[NbpStepDimension["GRANDE"] = 3] = "GRANDE";
})(NbpStepDimension || (NbpStepDimension = {}));
/**
 * Enum dei tipi di Eventi/Messaggi gestiti per NDCE
 */
export var NbpEmbeddingMessageType;
(function (NbpEmbeddingMessageType) {
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["CJ_READY"] = 1] = "CJ_READY";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["CJ_STEP"] = 2] = "CJ_STEP";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["ERROR"] = 3] = "ERROR";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["FLOATING_MENU_ON"] = 4] = "FLOATING_MENU_ON";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["FLOATING_MENU_OFF"] = 5] = "FLOATING_MENU_OFF";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["GOTO"] = 6] = "GOTO";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["LOADING_WINDOW_START"] = 7] = "LOADING_WINDOW_START";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["LOADING_WINDOW_STOP"] = 8] = "LOADING_WINDOW_STOP";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["MODAL_WINDOW_SHOW"] = 9] = "MODAL_WINDOW_SHOW";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["MODAL_WINDOW_HIDE"] = 10] = "MODAL_WINDOW_HIDE";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["PAGE_SCROLL"] = 11] = "PAGE_SCROLL";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["PAGE_SIZE"] = 12] = "PAGE_SIZE";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["SET_CONTRAST_OFF"] = 13] = "SET_CONTRAST_OFF";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["SET_CONTRAST_ON"] = 14] = "SET_CONTRAST_ON";
    NbpEmbeddingMessageType[NbpEmbeddingMessageType["SET_FONT_SIZE"] = 15] = "SET_FONT_SIZE";
})(NbpEmbeddingMessageType || (NbpEmbeddingMessageType = {}));
