// cjlib/nbp/nbp-accordion-showmore/nbp-accordion-showmore.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { NbpAccordionEvent } from '../nbp-accordion/nbp-accordion.interfaces';
var nextId = 0;
var NbpAccordionShowmoreComponent = (function () {
    function NbpAccordionShowmoreComponent() {
        this.id = "nbp-accordion-showmore-" + ++nextId;
        /**
         * (Opzionale) Booleano che pilota la visualizzazione iniziale dell’accordion collassato o espanso.
         *Default accordion collassato.
         */
        this.nbpCollapsed = true;
        /**
         * (Opzionale) Booleano che indica se allineare a destra l'heading del componente.
         * Default false.
         */
        this.nbpPullRight = false;
        /**
         * Evento emesso sul toggle del componente.
         * @param {EventEmitter<NbpAccordionEvent>}  nbpOnToggle
         */
        this.nbpOnToggle = new EventEmitter();
    }
    //     Methods
    /**
     * metodo utilizzabile per l’apertura programmatica
     */
    NbpAccordionShowmoreComponent.prototype.open = function () {
        this.nbpCollapsed = false;
    };
    /**
     * metodo utilizzabile per la chiusura programmatica
     */
    NbpAccordionShowmoreComponent.prototype.close = function () {
        this.nbpCollapsed = true;
    };
    /**
     * metodo utilizzabile per verificare se il componente è collassato
     */
    NbpAccordionShowmoreComponent.prototype.isCollapsed = function () {
        return this.nbpCollapsed;
    };
    NbpAccordionShowmoreComponent.prototype.ngOnInit = function () {
    };
    NbpAccordionShowmoreComponent.prototype._toggleCollapse = function () {
        this.nbpCollapsed = !this.nbpCollapsed;
        this._onToggle();
    };
    NbpAccordionShowmoreComponent.prototype._onToggle = function () {
        var acordionEvent = new NbpAccordionEvent();
        acordionEvent.source = this;
        acordionEvent.isCollapsed = this.nbpCollapsed;
        this.nbpOnToggle.emit(acordionEvent);
    };
    return NbpAccordionShowmoreComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionShowmoreComponent.prototype, "id", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionShowmoreComponent.prototype, "nbpTitle", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpAccordionShowmoreComponent.prototype, "nbpCollapsed", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionShowmoreComponent.prototype, "nbpIcon", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpAccordionShowmoreComponent.prototype, "nbpPullRight", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpAccordionShowmoreComponent.prototype, "nbpOnToggle", void 0);
NbpAccordionShowmoreComponent = __decorate([
    Component({
        selector: 'nbp-accordion-showmore',template: "<div class=\"panel nbp-panel-showmore\"><div class=\"panel-heading\" [ngClass]=\"{'panel-heading-round': isCollapsed(), 'pull-right': nbpPullRight}\" role=\"tab\" [id]=\"id\" (click)=\"_toggleCollapse()\"><h4 class=\"panel-title\"><span class=\"icon accordion-icon\" [ngClass]=\"nbpIcon\" *ngIf=\"!!nbpIcon\"></span> <span class=\"title\" [ngClass]=\"{'accordion-title-icon':!!nbpIcon}\">{{nbpTitle}}</span> <a role=\"button\" class=\"collapsed pull-right\"><span style=\"font-size: 1.6em\" [ngClass]=\"{'icon ico-comuni-freccia-accordion-apri':isCollapsed() , 'icon ico-comuni-freccia-accordion-chiudi':!isCollapsed()}\"></span></a></h4></div><div class=\"panel-collapse\" [ngClass]=\"{'in panel-open':!isCollapsed()}\"><div class=\"panel-body\"><ng-content></ng-content></div></div></div>",
        animations: [],
        styles: [".panel{border:1px!important;border-color:#fff;box-shadow:none}.panel .panel-open{background-color:#f2f2f2!important;border:1px!important;border-color:#6f6f6f}.panel-heading{border:1px!important;box-shadow:none;color:#258900;border-color:#6f6f6f;background-color:#fff;padding:1em 1.25em 1em 0!important;height:50px;width:18.75em}.panel-heading:hover{cursor:pointer}.panel-heading .panel-title{color:green;font-weight:700;font-size:.75em!important;text-transform:uppercase;text-align:center}.panel-heading .accordion-arrow{color:#000;position:relative;font-size:26px!important;margin-top:1.156em}.panel-heading .stringa{color:#fff;font-size:.75em!important;margin-right:1.875em;font-weight:700;font-family:arial}.panel-heading h4{margin:0;font-size:1em}.panel-body{border:1px!important;box-shadow:none;background-color:#f2f2f2!important;border-radius:0 0 2px 2px!important}.panel-collapse{width:100%;border:1px!important;box-shadow:none;background-color:#f2f2f2!important;border-radius:0 0 2px 2px!important;max-height:0;transition:max-height .15s ease-out;overflow:hidden}.panel-collapse.in{max-height:1000em;transition:max-height .25s ease-in}.panelTitle{font-size:.875em;font-weight:700}.accordionIcon{font-size:32px;position:absolute;top:9px}.nbp-panel-showmore a{text-decoration:none!important;margin-top:-3px;margin-right:4.962em}.nbp-panel-showmore>.panel-heading{background-color:#fff!important;position:relative}.nbp-panel-showmore>.panel-heading .accordion-icon{font-size:32px;position:absolute;top:9px}.nbp-panel-showmore>.panel-heading>.panel-title{font-size:.875em;font-weight:700;text-align:left!important}.nbp-panel-showmore>.panel-heading>.panel-title>.title{padding:2px 20px 2px 0}.nbp-panel-showmore>.panel-heading>.panel-title .accordion-title-icon{padding:2px 2px 2px 52px!important;margin-bottom:10px}.nbp-panel-showmore>.panel-heading>.panel-open{background-color:#5ca740!important}:host-context(.contrasto-on) .panel .panel-open{background-color:#fff!important;color:#000}:host-context(.contrasto-on) .panel .panel-open .panel-title{color:#000}:host-context(.contrasto-on) .panel .panel-open .stringa{color:#000}:host-context(.contrasto-on) .panel .panel-open .accordion-arrow{color:#000}:host-context(.contrasto-on) .panel-heading{background-color:#fff!important}:host-context(.contrasto-on) .panel-heading .panel-title{color:#000}:host-context(.contrasto-on) .panel-heading .accordion-arrow{color:#000}:host-context(.contrasto-on) .panel-heading .stringa{color:#000}:host-context(.contrasto-on) .panel-body{background-color:#fff!important;border-top:none!important;color:#000}:host-context(.contrasto-on) .accordion-green-header>.panel-heading{background-color:#000!important}:host-context(.contrasto-on) .accordion-green-header>.panel-heading:hover{background-color:#fff!important;color:#000;border:1px solid #000}:host-context(.contrasto-on) .accordion-green-header>.panel-open{background-color:#fff!important;color:#000;border:1px solid #000!important} /*# sourceMappingURL=nbp-accordion-showmore.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpAccordionShowmoreComponent);
export { NbpAccordionShowmoreComponent };
