// cjlib/nbp/nbp-slider/nbp-slider.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, forwardRef, Output, EventEmitter } from '@angular/core';
import { NG_VALUE_ACCESSOR, } from '@angular/forms';
/** intero autogenerato usato per id unici */
var nextId = 0;
/**
 * Provider Expression che permette la registrazione come ControlValueAccessor.
 * Permette di gestire [(ngModel)].
 * @docs-private
 */
export var NBP_SLIDER_CONTROL_VALUE_ACCESSOR = {
    provide: NG_VALUE_ACCESSOR,
    useExisting: forwardRef(function () { return NbpSliderComponent; }),
    multi: true
};
var NbpSliderComponent = (function () {
    function NbpSliderComponent() {
        this.propagateChange = function (_) { };
        /**
         * Identificativo univoco per il name dello slider. Se non viene passato un valore, name viene autogenerato.
         */
        this.id = "nbp-slider-" + ++nextId;
        /**
         * Identificativo univoco per l'Id dello slider. Se non viene passato un valore, name viene autogenerato.
         */
        this.name = "nbp-slider-" + ++nextId;
        /**
        * Tipo di slider desiderato. Al momento è disponibile solo quello semplice.
        * Di default è il tipo 0.
        */
        this.type = 0;
        /**
         * Valore corrente dello slider.
         * Di default il valore è 0.
         */
        this.value = 0;
        /**
         * Valore boolean che definisce la disabilitazione del componente.
         * Di default il valore è false.
         */
        this.disabled = false;
        /**
       * Evento al cambio di valore dello slider.
       */
        this.nbpChangeValue = new EventEmitter();
        this.range = null;
        this.config = {};
        this.keyboard = null;
        this.min = -10;
        this.max = 10;
        this.step = 1;
        this.limit = 10;
        this.connect = false;
        this.behavior = null;
        this._sliderValue = 5;
    }
    /**
     * Implementazione interfaccia ControlValueAccessor.
     */
    // https://blog.thoughtram.io/angular/2016/07/27/custom-form-controls-in-angular-2.html#understanding-controlvalueaccessor
    NbpSliderComponent.prototype.writeValue = function (value) {
        if (value !== undefined) {
            this.sliderValue = value;
            this.nbpChangeValue.emit(value);
        }
    };
    NbpSliderComponent.prototype.registerOnChange = function (fn) {
        this.propagateChange = fn;
    };
    NbpSliderComponent.prototype.registerOnTouched = function () { };
    Object.defineProperty(NbpSliderComponent.prototype, "sliderValue", {
        get: function () {
            return this._sliderValue;
        },
        set: function (val) {
            this._sliderValue = val;
            this.propagateChange(this._sliderValue);
        },
        enumerable: true,
        configurable: true
    });
    return NbpSliderComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSliderComponent.prototype, "id", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpSliderComponent.prototype, "name", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderComponent.prototype, "type", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderComponent.prototype, "value", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpSliderComponent.prototype, "disabled", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpSliderComponent.prototype, "nbpChangeValue", void 0);
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpSliderComponent.prototype, "range", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderComponent.prototype, "config", void 0);
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpSliderComponent.prototype, "keyboard", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderComponent.prototype, "min", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderComponent.prototype, "max", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderComponent.prototype, "step", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderComponent.prototype, "limit", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpSliderComponent.prototype, "connect", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderComponent.prototype, "behavior", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpSliderComponent.prototype, "_sliderValue", void 0);
NbpSliderComponent = __decorate([
    Component({
        selector: 'nbp-slider',
        template: "<div class=\"row\" style=\"width:27.562em; padding-top: 4.375em;\"><nouislider *ngIf=\"type==0\" [name]=\"name\" [config]=\"config\" [(ngModel)]=\"value\"></nouislider><div class=\"icomoon-noUi-startIcon\"></div><div class=\"icomoon-noUi-endIcon\"></div></div>",
        styles: [".noUi-target *,:host /deep/ .noUi-target{-webkit-touch-callout:none;-webkit-tap-highlight-color:transparent;-webkit-user-select:none;-ms-touch-action:none;touch-action:none;-ms-user-select:none;-moz-user-select:none;user-select:none;-moz-box-sizing:border-box;box-sizing:border-box}:host /deep/ .noUi-target{position:relative;direction:ltr}.noUi-connects,:host /deep/ .noUi-base{width:100%;height:100%;position:relative;z-index:1}:host /deep/ .noUi-connects{overflow:hidden;z-index:0}.noUi-origin,:host /deep/ .noUi-connect{will-change:transform;position:absolute;z-index:1;top:0;left:0;height:100%;width:100%;-ms-transform-origin:0 0;-webkit-transform-origin:0 0;transform-origin:0 0}:host /deep/ html:not([dir=rtl]) .noUi-horizontal .noUi-origin{left:auto;right:0}:host /deep/ .noUi-vertical .noUi-origin{width:0}:host /deep/ .noUi-horizontal .noUi-origin{height:0}:host /deep/ .noUi-handle{position:absolute}.noUi-state-tap .noUi-origin,:host /deep/ .noUi-state-tap .noUi-connect{-webkit-transition:transform .3s;transition:transform .3s}:host /deep/ .noUi-state-drag *{cursor:inherit!important}:host /deep/ .noUi-horizontal{height:.25em}:host /deep/ .noUi-horizontal .noUi-handle{width:1.75em;height:1.75em;left:0;top:0;margin-left:98%}:host /deep/ .noUi-vertical{width:.25em}:host /deep/ .noUi-vertical .noUi-handle{width:1.75em;height:34px;left:-6px;top:-17px}:host /deep/ html:not([dir=rtl]) .noUi-horizontal .noUi-handle{right:-17px;left:auto}:host /deep/ .noUi-target{background:#258900}:host /deep/ .noUi-connects{border-radius:3px}:host /deep/ .noUi-connect{background:#3fb8af}:host /deep/ .noUi-draggable{cursor:ew-resize}:host /deep/ .noUi-vertical .noUi-draggable{cursor:ns-resize}:host /deep/ .noUi-handle{cursor:pointer}.icomoon-noUi-startIcon:before{font-family:icomoon!important;font-style:normal!important;font-weight:400!important;font-variant:normal!important;text-transform:none!important;line-height:0;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;font-size:1.5625em;background-color:#fff;color:green;border-radius:1.5625em;margin-left:-100%;position:relative;top:-.665em}.icomoon-noUi-endIcon:before{font-family:icomoon!important;font-style:normal!important;font-weight:400!important;font-variant:normal!important;text-transform:none!important;line-height:0;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;font-size:1.5625em;background-color:#fff;color:green;border-radius:1.5625em;margin-left:97.5%;position:relative;top:-1.5em}:host /deep/ .noUi-handle:before{font-family:icomoon!important;font-style:normal!important;font-weight:400!important;font-variant:normal!important;text-transform:none!important;line-height:0;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;font-size:2.187em;background-color:#fff;border-radius:2.187em;margin-left:-.5em;cursor:pointer;color:#258900}:host /deep/ .noUi-handle:after{left:17px}.noUi-vertical .noUi-handle:after,:host /deep/ .noUi-vertical .noUi-handle:before{width:14px;height:1px;left:6px;top:14px}:host /deep/ .noUi-vertical .noUi-handle:after{top:17px}:host /deep/ [disabled] .noUi-connect{background:#b8b8b8}:host /deep/ [disabled].noUi-target,[disabled] .noUi-handle,[disabled].noUi-handle{cursor:not-allowed}.noUi-pips *,:host /deep/ .noUi-pips{-moz-box-sizing:border-box;box-sizing:border-box}:host /deep/ .noUi-pips{position:absolute;color:#999}:host /deep/ .noUi-value{position:absolute;white-space:nowrap;text-align:center;color:#000}:host /deep/ .noUi-value-sub{color:#ccc;font-size:10px;color:transparent}:host /deep/ .noUi-marker{position:absolute;background:#ccc}:host /deep/ .noUi-marker-sub{background:#aaa}:host /deep/ .noUi-marker-large{background:#aaa}:host /deep/ .noUi-pips-horizontal{padding:10px 0;height:80px;top:100%;left:0;width:100%}:host /deep/ .noUi-value-horizontal{-webkit-transform:translate(-50%,50%);transform:translate(-50%,50%)}:host /deep/ .noUi-rtl .noUi-value-horizontal{-webkit-transform:translate(50%,50%);transform:translate(50%,50%)}:host /deep/ .noUi-marker-horizontal.noUi-marker{margin-left:-1px;width:2px;height:5px;width:0}:host /deep/ .noUi-marker-horizontal.noUi-marker-sub{height:10px}:host /deep/ .noUi-marker-horizontal.noUi-marker-large{height:15px}:host /deep/ .noUi-pips-vertical{padding:0 10px;height:100%;top:0;left:100%}:host /deep/ .noUi-value-vertical{-webkit-transform:translate(0,-50%);transform:translate(0,-50%,0);padding-left:1.5625em}:host /deep/ .noUi-rtl .noUi-value-vertical{-webkit-transform:translate(0,50%);transform:translate(0,50%)}:host /deep/ .noUi-marker-vertical.noUi-marker{width:5px;height:2px;margin-top:-1px}:host /deep/ .noUi-marker-vertical.noUi-marker-sub{width:10px}:host /deep/ .noUi-marker-vertical.noUi-marker-large{width:15px}:host /deep/ .noUi-tooltip{display:block;padding:.312em;white-space:nowrap;border-width:0;position:absolute;left:0;width:4.375em;height:2.75em;background:inherit;background-color:#6f6f6f;border:none;border-radius:0;box-shadow:none;font-family:'Arial Bold',Arial;font-weight:700;font-style:normal;font-size:1em;color:#fff;text-align:center;line-height:1.25em;padding-top:.75em;margin-left:-.7em}:host /deep/ .noUi-horizontal .noUi-tooltip{-webkit-transform:translate(-50%,0);transform:translate(-50%,0);left:63%;bottom:190%}:host /deep/ .noUi-vertical .noUi-tooltip{-webkit-transform:translate(0,-50%);transform:translate(0,-50%);top:50%;right:120%}.contrasto-on .noUi-target *,.contrasto-on :host /deep/ .noUi-target{-webkit-tap-highlight-color:#000!important}.contrasto-on :host /deep/ .noUi-target{background:#000!important}.contrasto-on :host /deep/ .noUi-connect{background:#000!important}.contrasto-on :host /deep/ .icomoon-noUi-startIcon:before{background-color:#fff!important;color:#000}.contrasto-on :host /deep/ .icomoon-noUi-endIcon:before{background-color:#fff!important;color:#000}.contrasto-on :host /deep/ .noUi-handle:before{background-color:#fff!important;color:#000}.contrasto-on :host /deep/ [disabled] .noUi-connect{background:#000}.contrasto-on :host /deep/ .noUi-pips{color:#000}.contrasto-on :host /deep/ .noUi-value{color:#000}.contrasto-on :host /deep/ .noUi-value-sub{color:transparent}.contrasto-on :host /deep/ .noUi-marker{background:#000}.contrasto-on :host /deep/ .noUi-marker-sub{background:#000}.contrasto-on :host /deep/ .noUi-marker-large{background:#000}.contrasto-on :host /deep/ .noUi-tooltip{background-color:#000;color:#fff} /*# sourceMappingURL=nbp-slider.component.css.map */ "],
        // encapsulation: ViewEncapsulation.None,
        providers: [NBP_SLIDER_CONTROL_VALUE_ACCESSOR]
    }),
    __metadata("design:paramtypes", [])
], NbpSliderComponent);
export { NbpSliderComponent };
