// cjlib/nbp/nbp-accordion/nbp-accordion.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { NbpAccordionEvent } from './nbp-accordion.interfaces';
import { NbpStyle, NbpSize } from './../nbp-commons/nbp-commons.enums';
var nextId = 0;
var NbpAccordionComponent = (function () {
    function NbpAccordionComponent() {
        this.id = "nbp-accordion-" + ++nextId;
        /**
         * (Opzionale) Booleano che pilota la visualizzazione iniziale dell’accordion collassato o espanso.
         *Default accordion collassato.
         */
        this.nbpCollapsed = true;
        this.nbpStyle = NbpStyle.DEFAULT;
        /**
       * Booleano che indica se mostrare icona e tooltip nell'intestazione del componente
       */
        this.showTooltip = false;
        /**
         * (Opzionale) Enum che definisce il la posizione del tooltip rispetto all’oggetto al quale è ancorato:
         * “Top” | “Right” | “Bottom” | “Left” | “Top-Left” | “Top-Right” | “Bottom-Left” | “Bottom-Right”
         * Default: “Top”
         */
        this.tooltipPosition = 'Top';
        /**
         * Proprietà che definisce la grandezza del tooltip, tramite l'enum NbpSize
         * valore di default: NbpSize.SM
         */
        this.tooltipSize = NbpSize.SM;
        /**
         * Stringa che definisce la classe dell'icona da mostrare
         */
        this.tooltipIcon = 'icomoon-Simboli_Risorsa-1';
        /**
         * Evento emesso sul toggle del componente.
         * @param {EventEmitter<NbpAccordionEvent>}  nbpOnToggle
         */
        this.nbpOnToggle = new EventEmitter();
        this._style = NbpStyle;
    }
    //     Methods
    /**
     * metodo utilizzabile per l’apertura programmatica
     */
    NbpAccordionComponent.prototype.open = function () {
        this.nbpCollapsed = false;
    };
    /**
     * metodo utilizzabile per la chiusura programmatica
     */
    NbpAccordionComponent.prototype.close = function () {
        this.nbpCollapsed = true;
    };
    /**
     * metodo utilizzabile per verificare se il componente è collassato
     */
    NbpAccordionComponent.prototype.isCollapsed = function () {
        return this.nbpCollapsed;
    };
    NbpAccordionComponent.prototype.ngOnInit = function () {
    };
    NbpAccordionComponent.prototype._toggleCollapse = function () {
        this.nbpCollapsed = !this.nbpCollapsed;
        this._onToggle();
    };
    NbpAccordionComponent.prototype._onToggle = function () {
        var acordionEvent = new NbpAccordionEvent();
        acordionEvent.source = this;
        acordionEvent.isCollapsed = this.nbpCollapsed;
        this.nbpOnToggle.emit(acordionEvent);
    };
    return NbpAccordionComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpAccordionComponent.prototype, "id", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionComponent.prototype, "nbpTitle", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionComponent.prototype, "nbpSubtitle", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionComponent.prototype, "nbpLabel", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpAccordionComponent.prototype, "nbpCollapsed", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpAccordionComponent.prototype, "nbpStyle", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionComponent.prototype, "nbpIcon", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpAccordionComponent.prototype, "showTooltip", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionComponent.prototype, "tooltipPosition", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionComponent.prototype, "tooltipVisibility", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionComponent.prototype, "tooltipText", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpAccordionComponent.prototype, "tooltipSize", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpAccordionComponent.prototype, "tooltipIcon", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpAccordionComponent.prototype, "nbpOnToggle", void 0);
NbpAccordionComponent = __decorate([
    Component({
        selector: 'nbp-accordion',template: "<div class=\"panel nbp-panel-primary panel-default\"><div class=\"panel-heading\" [ngClass]=\"{'panel-heading-round': isCollapsed()}\" role=\"tab\" [id]=\"id\" (click)=\"_toggleCollapse()\"><div class=\"panel-title\" [ngClass]=\"{'panel-title-tooltip': showTooltip}\"><span class=\"icon accordion-icon\" [ngClass]=\"nbpIcon\" *ngIf=\"!!nbpIcon\"></span><div class=\"title-container\"><div class=\"title\">{{nbpTitle}}</div><span *ngIf=\"nbpSubtitle\" style=\"margin-right: 5px\" class=\"subtitle\">{{nbpSubtitle}}</span><nbp-tooltip *ngIf=\"showTooltip\" [nbpSize]=\"tooltipSize\" [nbpPosition]=\"tooltipPosition\" [nbpVisibilityStrategy]=\"tooltipVisibility\" [nbpText]=\"tooltipText\"><span class=\"{{tooltipIcon}}\" style=\"font-size:1.429em;vertical-align: middle;\"></span></nbp-tooltip></div><span class=\"label\">{{nbpLabel}}</span> <span class=\"icon accordion-arrow\" [ngClass]=\"{'ico-comuni-freccia-accordion-apri':isCollapsed() , 'ico-comuni-freccia-accordion-chiudi':!isCollapsed()}\"></span></div></div><div class=\"panel-collapse\" [ngClass]=\"{'in panel-open':!isCollapsed()}\"><div class=\"panel-body\"><ng-content></ng-content></div></div></div>",
        animations: [],
        styles: [".panel{border:none!important;border-radius:0!important;box-shadow:none}.panel .panel-open{background-color:#6f6f6f!important;overflow:visible}.panel-heading{height:3.75em;border-radius:.125em .125em 0 0!important;box-shadow:none;background-color:#f8f8f8!important;cursor:pointer}.panel-heading.panel-heading-round{border-radius:.125em .125em .125em .125em!important}.panel-heading .panel-title{color:#000;font-size:.875em!important}.panel-heading .panel-title-tooltip{margin-top:-.3125em}.panel-heading .accordion-arrow{color:#000;position:relative;font-size:2em}.panel-heading .accordion-arrow .icon{vertical-align:middle}.panel-heading .label{color:#000!important}.panel-body{border:none!important;box-shadow:none;background-color:#f2f2f2!important;border-radius:0 0 .125em .125em!important}.panel-collapse{border:none!important;box-shadow:none;background-color:#f2f2f2!important;border-radius:0 0 .125em .125em!important;max-height:0;transition:max-height .15s ease-out;overflow:hidden}.panel-collapse.in{max-height:1000em;transition:max-height .25s ease-in}.panelTitleLayout{display:table;table-layout:fixed;width:100%;height:3em}.panelTitleItemLayout{display:table-cell;vertical-align:middle}.titlefont{overflow:hidden;white-space:nowrap;text-overflow:ellipsis}.nbp-panel-primary>.panel-heading{position:relative}.nbp-panel-primary>.panel-heading>.panel-title{display:table;table-layout:fixed;width:100%;height:3em}.nbp-panel-primary>.panel-heading>.panel-title>.accordion-icon{display:table-cell;vertical-align:middle;text-align:center;font-size:2em;width:2.3125em;padding-right:.3125em}.nbp-panel-primary>.panel-heading>.panel-title>.accordion-icon ::before{vertical-align:middle}.nbp-panel-primary>.panel-heading>.panel-title>.title-container{display:table-cell;vertical-align:middle}.nbp-panel-primary>.panel-heading>.panel-title>.title-container>.title{overflow:hidden;white-space:nowrap;text-overflow:ellipsis;font-size:1.14em;font-weight:400}.nbp-panel-primary>.panel-heading>.panel-title>.title-container>.subtitle{overflow:hidden;white-space:nowrap;text-overflow:ellipsis;font-size:1.14em;font-weight:700}.nbp-panel-primary>.panel-heading>.panel-title>.label{display:table-cell;vertical-align:middle;overflow:hidden;white-space:nowrap;text-overflow:ellipsis;font-size:1em;font-weight:400;text-align:right}.nbp-panel-primary>.panel-heading>.panel-title .accordion-arrow{display:table-cell;vertical-align:middle;text-align:center;font-size:1.5em;width:25px}.panel-default>.panel-heading{background-color:#f8f8f8!important}.panel-default>.panel-open{background-color:#6f6f6f!important}:host-context(.contrasto-on) .panel .panel-open{background-color:#fff!important;color:#000;border:none!important}:host-context(.contrasto-on) .panel .panel-open .panel-title{color:#000}:host-context(.contrasto-on) .panel .panel-open .stringa{color:#000}:host-context(.contrasto-on) .panel .panel-open .accordion-arrow{color:#000}:host-context(.contrasto-on) .panel-heading{background-color:#000!important}:host-context(.contrasto-on) .panel-heading .panel-title{color:#fff}:host-context(.contrasto-on) .panel-heading .accordion-arrow{color:#fff}:host-context(.contrasto-on) .panel-heading .label{color:#fff!important}:host-context(.contrasto-on) .panel-body{background-color:#fff!important;border:none!important;color:#000}:host-context(.contrasto-on) .accordion-green-header>.panel-heading{background-color:#000!important}:host-context(.contrasto-on) .accordion-green-header>.panel-open{background-color:#fff!important;color:#000;border:1px solid #000!important} /*# sourceMappingURL=nbp-accordion.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpAccordionComponent);
export { NbpAccordionComponent };
