// cjlib/nbp/nbp-sign-type-select/nbp-sign-type-select.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from '@angular/core';
var NbpSignTypeSelectComponent = (function () {
    function NbpSignTypeSelectComponent() {
        /**
         * Evento emesso al click sui radio button di scelta del primo Step
         * @param {EventEmitter<ISignTypeData>}  nbpOnFirstStepSelect
         */
        this.nbpOnFirstStepSelect = new EventEmitter();
        /**
         * Evento emesso al click sui radio button di scelta del secondo step
         * @param {EventEmitter<ISignTypeData>}  nbpOnSecondStepSelect
         */
        this.nbpOnSecondStepSelect = new EventEmitter();
    }
    NbpSignTypeSelectComponent.prototype.ngOnInit = function () {
        if (!this.nbpData) {
            this.nbpData = {
                firstStepValue: '',
                secondStepValue: ''
            };
        }
    };
    NbpSignTypeSelectComponent.prototype._onFirstStepSelect = function (event) {
        this.nbpData.firstStepValue = event;
        console.log('NbpSignTypeSelectComponent - Emitting nbpOnFirstStepSelect');
        console.dir(this.nbpData);
        this.nbpOnFirstStepSelect.emit(this.nbpData);
    };
    NbpSignTypeSelectComponent.prototype._onSecondStepSelect = function (event) {
        this.nbpData.secondStepValue = event;
        console.log('NbpSignTypeSelectComponent - Emitting nbpOnSecondStepSelect');
        console.dir(this.nbpData);
        this.nbpOnSecondStepSelect.emit(this.nbpData);
    };
    return NbpSignTypeSelectComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSignTypeSelectComponent.prototype, "nbpCustomer", void 0);
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpSignTypeSelectComponent.prototype, "nbpFirstStepValues", void 0);
__decorate([
    Input(),
    __metadata("design:type", Array)
], NbpSignTypeSelectComponent.prototype, "nbpSecondStepValues", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSignTypeSelectComponent.prototype, "nbpData", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpSignTypeSelectComponent.prototype, "nbpOnFirstStepSelect", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpSignTypeSelectComponent.prototype, "nbpOnSecondStepSelect", void 0);
NbpSignTypeSelectComponent = __decorate([
    Component({
        selector: 'nbp-sign-type-select ',template: "<div class=\"row sign-container\"><div class=\"col-xs-24 col-sm-8 col-md-8 sign-customer-section\"><div class=\"row\"><div class=\"col-xs-offset-2 col-sm-offset-3 col-md-offset-3\"><div class=\"customer-circle-info\">{{nbpCustomer.circleInfo}}</div><div class=\"customer-section-title\">Cliente:</div><div class=\"customer-section-info\">{{nbpCustomer.info}}</div></div></div></div><div class=\"col-xs-24 col-sm-7 col-md-7 sign-place-section\"><nbp-radio-selection [nbpRadioList]=\"nbpFirstStepValues\" [nbpData]=\"nbpData.firstStepValue\" (nbpOnSelection)=\"_onFirstStepSelect($event)\"></nbp-radio-selection></div><div class=\"col-xs-24 col-sm-9 col-md-9 sign-type-section\"><nbp-radio-selection [nbpRadioList]=\"nbpSecondStepValues\" [nbpData]=\"nbpData.secondStepValue\" (nbpOnSelection)=\"_onSecondStepSelect($event)\"></nbp-radio-selection></div></div>",
        styles: [".sign-container{padding-top:20px;padding-bottom:20px}.sign-customer-section{background-color:#fff;min-height:145px;border:1px solid #ccc}.customer-section-title{font-size:1em;font-weight:700;padding-top:24px}.customer-section-info{font-size:1em;padding-top:6px}.customer-circle-info{width:40px;height:40px;border:1px solid #ccc;border-radius:40px;background:#fff;position:absolute;top:-20px;text-align:center;font-weight:700;padding-top:8px;color:#89bc72}.sign-place-section{background-color:#f2f2f2;min-height:145px;border-top:1px solid #ccc;border-bottom:1px solid #ccc;border-right:1px solid #ccc;border-left:0 solid transparent;padding-top:10px}.sign-place-section:before{content:'';display:block;width:20px;height:20px;background-color:#f2f2f2;border-top:1px solid #ccc;border-left:1px solid #ccc;position:absolute;transform-origin:top left;transform:rotate(-45deg) translateX(-20px);top:34px;left:0}.sign-type-section{background-color:#e3e3e3;min-height:145px;border-top:1px solid #ccc;border-bottom:1px solid #ccc;border-right:1px solid #ccc;border-left:0 solid transparent;padding-top:10px}.sign-type-section:before{content:'';display:block;width:20px;height:20px;background-color:#e3e3e3;border-top:1px solid #ccc;border-left:1px solid #ccc;position:absolute;transform-origin:top left;transform:rotate(-45deg) translateX(-20px);top:34px;left:0}::ng-deep nbp-radio .radio-label{font-weight:300!important;font-size:1em!important;padding-top:10px}@media screen and (max-width:767px){.sign-place-section{border-top:0 solid transparent;border-left:1px solid #ccc!important}.sign-place-section:before{border-top:1px solid #ccc;border-right:1px solid #ccc;border-left:0 solid transparent;top:-15px;left:34px}.sign-type-section{border-top:0 solid transparent;border-left:1px solid #ccc!important}.sign-type-section:before{border-top:1px solid #ccc;border-right:1px solid #ccc;border-left:0 solid transparent;top:-15px;left:34px}} /*# sourceMappingURL=nbp-sign-type-select.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpSignTypeSelectComponent);
export { NbpSignTypeSelectComponent };
