// cjlib/cj/cj-error-view/cj-error-view.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input } from '@angular/core';
import { NbpFeedbackType } from './../../nbp/nbp-feedback/nbp-feedback.enums';
var CjErrorViewComponent = (function () {
    function CjErrorViewComponent() {
        this._feedbackType = NbpFeedbackType;
    }
    CjErrorViewComponent.prototype.ngOnInit = function () { };
    CjErrorViewComponent.prototype.ngOnChanges = function (changes) {
        var _this = this;
        this._isDebugEnv = this.evaluateDebugEnvironment();
        setTimeout(function () { _this._mainMessage = !_this._isDebugEnv ? 'Si prega di contattare il servizio di supporto tecnico' : null; });
    };
    CjErrorViewComponent.prototype.evaluateDebugEnvironment = function () {
        var isDebugEnv = false;
        if (this.cjClientError && this.cjClientError.environment) {
            switch (this.cjClientError.environment) {
                case 'PRODUZIONE':
                    isDebugEnv = false;
                    break;
                default:
                    isDebugEnv = true;
                    break;
            }
        }
        return isDebugEnv;
    };
    return CjErrorViewComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Object)
], CjErrorViewComponent.prototype, "cjClientError", void 0);
CjErrorViewComponent = __decorate([
    Component({
        selector: 'cj-error-view',template: "<nbp-feedback nbpTitle=\"Si è verificato un errore\" [nbpMainMessage]=\"_mainMessage\" [nbpErrorCode]=\"cjClientError?.errorCode\" [nbpFeedbackType]=\"_feedbackType.ERROR\" [nbpShowDebugInfo]=\"_isDebugEnv\"><nbp-debug-info [nbpError]=\"cjClientError\"></nbp-debug-info></nbp-feedback>",
        styles: [" /*# sourceMappingURL=cj-error-view.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], CjErrorViewComponent);
export { CjErrorViewComponent };
