// cjlib/nbp/nbp-combo-filter/nbp-combo-filter.component.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
import { Component, Input, Output, Optional, Inject, EventEmitter, forwardRef, ViewChild, HostBinding, HostListener, ElementRef, Injector } from '@angular/core';
import { NG_VALUE_ACCESSOR, NG_VALIDATORS, NG_ASYNC_VALIDATORS, NgControl } from '@angular/forms';
import { NgModel } from '@angular/forms';
import { coerceBooleanProperty, getPropertyValueByPath, generateObjectFromStringPath } from './../../utility/lib-utility';
import { NbpPipe } from './../nbp-commons/nbp-commons.enums';
import { NbpFormatterService } from './../../utility/nbp-formatter.service';
import { INbpDataSource } from './../nbp-datasource/nbp-datasource.interfaces';
import { NbpInputElementBase } from './../nbp-commons/nbp-input-element-base';
import { trigger, state, style, animate, transition } from '@angular/animations';
/**
* Provider Expression that allows nbp-combo to register as a ControlValueAccessor.
* This allows it to support [(ngModel)].
* @docs-private
*/
export var NBP_COMBO_CONTROL_VALUE_ACCESSOR = {
    provide: NG_VALUE_ACCESSOR,
    useExisting: forwardRef(function () { return NbpComboFilterComponent; }),
    multi: true
};
export var NBP_INPUT_ELEMENT_BASE = {
    provide: NbpInputElementBase,
    useExisting: forwardRef(function () { return NbpComboFilterComponent; }),
    multi: true
};
/** intero autogenerato usato per id unici per nbpCombo component */
var nextId = 0;
/**
* Componente per la implementazione della combobox, supporta tutti gli attributi standard html per il tag input.
* Personalizzabile per la gestione di currency grazie all'applicaizone di opportuni pipes
* definiti dalla libreria.
*/
var NbpComboFilterComponent = (function (_super) {
    __extends(NbpComboFilterComponent, _super);
    function NbpComboFilterComponent(validatori, asyncValidatori, formatterService, element, inj) {
        var _this = _super.call(this, validatori, asyncValidatori) || this;
        _this.validatori = validatori;
        _this.asyncValidatori = asyncValidatori;
        _this.formatterService = formatterService;
        _this.element = element;
        _this.inj = inj;
        /**
        * Identificativo univoco per il text field.
        * Se nessun id è fornito e' auto-generato.
        */
        _this.id = "nbp-combo-filter-" + ++nextId;
        //modificaReply
        /**
        * Elementi del DropBox:visualizza tutti o solo quelli filtrati
        */
        _this.showAllItems = false;
        /**
        * (Opzionale)
        * Booleano che consente di visualizzare i messaggi di feedback relativi alla validazione.
        * Valore di default false
        */
        _this.nbpShowValidation = false;
        _this.showErrorLabel = false;
        /**
        * Il testo del placeholder.
        */
        _this.placeholder = '';
        /**
        * Evento emesso quando è selezionato un elemento della combobox.
        */
        _this.select = new EventEmitter();
        /**
        * Evento emesso quando la dropdown viene chiuda
        */
        _this.dropdownClosed = new EventEmitter();
        /**
        * Evento emesso quando la dropdown viene aperta
        */
        _this.dropdownOpened = new EventEmitter();
        /**
        * (Opzionale) Funzione custom di validazione del campo
        */
        _this.nbpValidation = new EventEmitter();
        /**
         * (Opzionale) Paramentro che indica l'ordine di spostamento da un campo all'altro attraverso il tab.
         */
        _this.tabindex = 0;
        _this._dropdownVisible = false;
        _this._arrowClicked = false;
        _this._nbpPipe = NbpPipe;
        return _this;
    }
    Object.defineProperty(NbpComboFilterComponent.prototype, "disabled", {
        /**
        * Flag che abilita/disabilita le funzionalità associate alla combobox
        */
        get: function () { return this._nbpDisabled; },
        set: function (value) { this._nbpDisabled = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NbpComboFilterComponent.prototype, "required", {
        /**
        * Flag che imposta come obbligatorio o meno il campo di input (disabilitato) associato alla combobox
        */
        get: function () { return this._nbpRequired; },
        set: function (value) { this._nbpRequired = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NbpComboFilterComponent.prototype, "nbpShowEmptyValue", {
        /**
        * Flag che imposta come obbligatorio o meno il campo di input (disabilitato) associato alla combobox
        */
        get: function () { return this._nbpShowEmptyValue; },
        set: function (value) { this._nbpShowEmptyValue = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NbpComboFilterComponent.prototype, "nbpHighlightedDefault", {
        /**
        * (opzionale ) Booleano che gestisce l’attributo  bordo giallo della combobox.
        * Al primo click/focus tornerà alla versione di default.
        * valore default false
        */
        get: function () { return this._nbpHighlightedDefault; },
        set: function (value) { this._nbpHighlightedDefault = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    ;
    ;
    Object.defineProperty(NbpComboFilterComponent.prototype, "nbpBindingToKeyField", {
        /**
        * Definisce quando il binding del ngModel deve essere fatto sull'attributo nbpKeyField
        *
        */
        get: function () { return this._nbpBindingToKeyField; },
        set: function (value) { this._nbpBindingToKeyField = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    ;
    ;
    NbpComboFilterComponent.prototype._getDropdownState = function () {
        return this._dropdownVisible ? 'expanded' : 'collapsed';
    };
    NbpComboFilterComponent.prototype._getModelValue = function (item) {
        if (!item || item && item.emptyValue) {
            return;
        }
        return (this._nbpBindingToKeyField && this.nbpKeyField) ? getPropertyValueByPath(item, this.nbpKeyField) : item;
    };
    NbpComboFilterComponent.prototype._evaluateCurrentSelectionIndex = function () {
        var _this = this;
        var currentSelection;
        if (this.value) {
            if (this.nbpKeyField && this.nbpBindingToKeyField) {
                currentSelection = this._nbpShadowData.findIndex(function (item) { return getPropertyValueByPath(item, _this.nbpKeyField) === _this.value; });
            }
            else {
                currentSelection = this._nbpShadowData.findIndex(function (item) { return item === _this.value; });
            }
            return currentSelection;
        }
        return 0;
    };
    NbpComboFilterComponent.prototype._onPreSelectEvent = function (item) {
        this._nbpViewValue = this._getViewModelElement(item);
        this.value = this._getModelValue(item);
        this.nbpValidation.emit(this.model);
        // TODO validazione anche su set programmatica del value
        event.stopPropagation();
        this.select.emit(item);
    };
    NbpComboFilterComponent.prototype._manageDropdownState = function (event) {
        switch (event.key.toLowerCase()) {
            case 'enter':
                event.preventDefault();
                if (this._dropdownVisible) {
                    if (!!this._preSelectedElementIndex || this._preSelectedElementIndex === 0) {
                        this._onPreSelectEvent(this._nbpShadowData[this._preSelectedElementIndex]);
                    }
                }
                this._dropdownVisible = !this._dropdownVisible;
                break;
            case 'tab':
                if (this._dropdownVisible) {
                    event.preventDefault();
                    if (!!this._preSelectedElementIndex || this._preSelectedElementIndex === 0) {
                        this._onPreSelectEvent(this._nbpShadowData[this._preSelectedElementIndex]);
                    }
                }
                this._dropdownVisible = false;
                break;
            default:
                this._dropdownVisible = false;
                break;
        }
        if (this._evaluateCurrentSelectionIndex() >= 0) {
            this.scrollToSelectedIndex(this._evaluateCurrentSelectionIndex());
        }
        else {
            this._preSelectedElementIndex = null;
        }
    };
    NbpComboFilterComponent.prototype._isAlphaNumericCharacter = function (keyCode) {
        return (keyCode >= 48 && keyCode <= 57) || (keyCode >= 65 && keyCode <= 90) || (keyCode >= 95 && keyCode <= 122);
    };
    NbpComboFilterComponent.prototype._isOptionPreselected = function (item) {
        var currentPreSelection = this._nbpShadowData.findIndex(function (element) { return element === item; });
        return currentPreSelection === this._preSelectedElementIndex;
    };
    NbpComboFilterComponent.prototype.scrollToSelectedIndex = function (elementIndex) {
        if (this._dropdownVisible) {
            setTimeout(function () {
                document.getElementById('combo-filter-element_' + elementIndex).scrollIntoViewIfNeeded();
            });
        }
    };
    NbpComboFilterComponent.prototype._moveNext = function (event) {
        if (this._nbpDisabled) {
            return;
        }
        event.preventDefault();
        var currentSelection = this._preSelectedElementIndex !== null && this._preSelectedElementIndex !== undefined ?
            this._preSelectedElementIndex : (this.nbpShowEmptyValue) ? 0 : -1;
        var nextSelection = currentSelection + 1;
        if (nextSelection >= this._nbpShadowData.length) {
            nextSelection = currentSelection;
        }
        this.scrollToSelectedIndex(nextSelection);
        // this._onPreSelectEvent(this._nbpShadowData[nextSelection]);
        this._preSelectedElementIndex = nextSelection;
    };
    NbpComboFilterComponent.prototype._movePrev = function (event) {
        if (this._nbpDisabled) {
            return;
        }
        event.preventDefault();
        var currentSelection = this._preSelectedElementIndex !== null && this._preSelectedElementIndex !== undefined ?
            this._preSelectedElementIndex : 0;
        var nextSelection = currentSelection < 0 ? 0 : currentSelection - 1;
        if (nextSelection < 0) {
            nextSelection = currentSelection;
        }
        this.scrollToSelectedIndex(nextSelection);
        // this._onPreSelectEvent(this._nbpShadowData[nextSelection]);
        this._preSelectedElementIndex = nextSelection;
    };
    NbpComboFilterComponent.prototype.ngOnInit = function () {
        this.ngControl = this.inj.get(NgControl);
        this._prepareDataSource();
        //modificareply
        if (this.firstFocus) {
            this.elementRef.nativeElement.focus();
            this.elementRef.nativeElement.click();
        }
    };
    NbpComboFilterComponent.prototype._prepareDataSource = function () {
        var _this = this;
        if (this.nbpDataSource) {
            var currentPageStream = this.nbpDataSource.getPage();
            this.datasourceSubscription = currentPageStream.subscribe(function (value) {
                if (value.data && value.data.length) {
                    _this._nbpShadowData = value.data.slice();
                    if (_this.nbpShowEmptyValue) {
                        if (_this.nbpViewField) {
                            var emptyObject = { emptyValue: true };
                            emptyObject = generateObjectFromStringPath(emptyObject, _this.nbpViewField);
                            if (_this.nbpKeyField) {
                                emptyObject = generateObjectFromStringPath(emptyObject, _this.nbpKeyField);
                            }
                            _this._nbpShadowData.unshift(emptyObject);
                        }
                        else {
                            _this._nbpShadowData.unshift(null);
                        }
                    }
                    // console.log('Resolve Subscribe');
                }
                else {
                    _this._nbpShadowData = new Array();
                }
                ;
                _this._nbpShadowDataFull = _this._nbpShadowData.slice();
            }, function (error) {
                console.log(JSON.stringify(error));
            });
        }
    };
    NbpComboFilterComponent.prototype._manageDropdown = function (event) {
        if (!this._dropdownVisible && this.containsFilterValue !== "") {
            this._dropdownVisible = true;
        }
    };
    NbpComboFilterComponent.prototype._getViewValue = function (item) {
        if (!item) {
            return null;
        }
        return this.formatterService.displayValueTransformation(this._getViewModelElement(item), this.nbpFilter, this.nbpFilterArg);
    };
    NbpComboFilterComponent.prototype._getValue = function (item) {
        return (this._nbpBindingToKeyField && this.nbpKeyField) ? getPropertyValueByPath(item, this.nbpKeyField) : item;
    };
    NbpComboFilterComponent.prototype._getDropDownLabel = function (item) {
        return this.formatterService.displayValueTransformation((this.nbpViewField ? getPropertyValueByPath(item, this.nbpViewField) : item), this.nbpFilter, this.nbpFilterArg);
    };
    NbpComboFilterComponent.prototype._onSelectEvent = function (item) {
        if (item) {
            // Alla selezione, il filtro contains viene settato uguale all'item
            this.containsFilterValue = this._getDropDownLabel(item);
            this.onContainsFilterChange();
            this._nbpViewValue = this._getViewModelElement(item);
            this.value = this._getValue(item);
            this.nbpValidation.emit(this.model);
            this._dropdownVisible = !this._dropdownVisible;
            // TODO validazione anche su set programmatica del value
            event.stopPropagation();
            this.select.emit(item);
        }
    };
    NbpComboFilterComponent.prototype._getViewModelElement = function (item) {
        if (this._nbpBindingToKeyField) {
            return this.nbpKeyField ? this._getSelectedElement(item) : item;
        }
        return this.nbpViewField ? getPropertyValueByPath(item, this.nbpViewField) : item;
    };
    NbpComboFilterComponent.prototype._getSelectedElement = function (item) {
        var _this = this;
        var elems = this._nbpShadowData.filter(function (elem) {
            return getPropertyValueByPath(elem, _this.nbpKeyField) === item;
        });
        if (elems.length > 0) {
            var selectedElem = elems[0];
            return this.nbpViewField ? getPropertyValueByPath(selectedElem, this.nbpViewField) : item;
        }
        else {
            return item;
        }
    };
    NbpComboFilterComponent.prototype._isOptionSelected = function (item) {
        if (!item || !this.value) {
            return false;
        }
        if (this.nbpKeyField && item && getPropertyValueByPath(item, this.nbpKeyField)) {
            var currentValue = this.value;
            if (this._nbpBindingToKeyField) {
                return getPropertyValueByPath(item, this.nbpKeyField) === currentValue;
            }
            else {
                return getPropertyValueByPath(item, this.nbpKeyField) === getPropertyValueByPath(currentValue, this.nbpKeyField);
            }
        }
        if (item) {
            return item === this.value;
        }
        return false;
    };
    NbpComboFilterComponent.prototype._onInputFocus = function (event) {
        if (!this._nbpDisabled
            && this._nbpShadowData
            && this._nbpShadowData.length > 0 && !this._arrowClicked) {
            this._dropdownVisible = !this._dropdownVisible;
            this._dropdownVisible ? this.dropdownOpened.emit() : this.dropdownClosed.emit();
        }
    };
    NbpComboFilterComponent.prototype._onInputClick = function (event) {
        //modificaReply
        if (this.showAllItems) {
            this._nbpShadowData = this._nbpShadowDataFull.slice();
        }
        //--------------------
        if (!this._nbpDisabled
            && this._nbpShadowData
            && this._nbpShadowData.length > 0) {
            this._dropdownVisible = !this._dropdownVisible;
            this._arrowClicked = !this._arrowClicked;
        }
    };
    NbpComboFilterComponent.prototype._onTouchedEvent = function (event) {
        this.touch();
        this.showErrorLabel = true;
    };
    NbpComboFilterComponent.prototype._closeDropDown = function (target) {
        if (!this._dropdownVisible) {
            return;
        }
        var parentFound = false;
        while (target != null && !parentFound) {
            if (target === this.element.nativeElement) {
                parentFound = true;
            }
            target = target.parentElement;
        }
        if (!parentFound) {
            this._dropdownVisible = false;
            this._arrowClicked = false;
            this.dropdownClosed.emit();
        }
    };
    NbpComboFilterComponent.prototype.onFocusout = function (target) {
        this._closeDropDown(target);
    };
    NbpComboFilterComponent.prototype.onClick = function (target) {
        this._closeDropDown(target);
    };
    /**
    * Listener per cambio del filtro "contains" -> si occupa di filtrare le opzioni in base a quanto inserito
    */
    NbpComboFilterComponent.prototype.onContainsFilterChange = function () {
        var _this = this;
        if (this.containsFilterValue) {
            var checkString_1 = this.containsFilterValue.toLowerCase();
            var startingIndex_1 = this.nbpShowEmptyValue ? 1 : 0;
            this._nbpShadowData = this._nbpShadowDataFull.filter(function (elem, index) {
                if (index >= startingIndex_1) {
                    var elemLabel = _this._getDropDownLabel(elem);
                    return elemLabel && elemLabel.toLowerCase().includes(checkString_1);
                }
                else {
                    return true;
                }
            });
        }
        else {
            this._nbpShadowData = this._nbpShadowDataFull.slice();
        }
    };
    return NbpComboFilterComponent;
}(NbpInputElementBase));
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpComboFilterComponent.prototype, "name", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpComboFilterComponent.prototype, "id", void 0);
__decorate([
    Input("showallitems"),
    __metadata("design:type", Boolean)
], NbpComboFilterComponent.prototype, "showAllItems", void 0);
__decorate([
    HostBinding('class.nbp-show-validation'),
    Input(),
    __metadata("design:type", Boolean)
], NbpComboFilterComponent.prototype, "nbpShowValidation", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpComboFilterComponent.prototype, "nbpErrorMessage", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpComboFilterComponent.prototype, "showErrorLabel", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboFilterComponent.prototype, "disabled", null);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboFilterComponent.prototype, "required", null);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpComboFilterComponent.prototype, "placeholder", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboFilterComponent.prototype, "nbpShowEmptyValue", null);
__decorate([
    Input(),
    __metadata("design:type", INbpDataSource)
], NbpComboFilterComponent.prototype, "nbpDataSource", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpComboFilterComponent.prototype, "nbpViewField", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpComboFilterComponent.prototype, "nbpKeyField", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboFilterComponent.prototype, "nbpHighlightedDefault", null);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpComboFilterComponent.prototype, "nbpFilter", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpComboFilterComponent.prototype, "nbpFilterArg", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboFilterComponent.prototype, "nbpBindingToKeyField", null);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpComboFilterComponent.prototype, "select", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NbpComboFilterComponent.prototype, "dropdownClosed", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NbpComboFilterComponent.prototype, "dropdownOpened", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpComboFilterComponent.prototype, "nbpValidation", void 0);
__decorate([
    ViewChild(NgModel),
    __metadata("design:type", NgModel)
], NbpComboFilterComponent.prototype, "model", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpComboFilterComponent.prototype, "firstFocus", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpComboFilterComponent.prototype, "tabindex", void 0);
__decorate([
    ViewChild('autoFocusComboFilter'),
    __metadata("design:type", ElementRef)
], NbpComboFilterComponent.prototype, "elementRef", void 0);
__decorate([
    HostListener('document: focusout', ['$event.target']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [HTMLElement]),
    __metadata("design:returntype", void 0)
], NbpComboFilterComponent.prototype, "onFocusout", null);
__decorate([
    HostListener('document: click', ['$event.target']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [HTMLElement]),
    __metadata("design:returntype", void 0)
], NbpComboFilterComponent.prototype, "onClick", null);
NbpComboFilterComponent = __decorate([
    Component({
        selector: 'nbp-combo-filter',template: "<div class=\"dropdown-container\"><input type=\"hidden\" [(ngModel)]=\"value\"><div class=\"dropdown-inner-container\" (keydown.arrowdown)=\"_moveNext($event)\" (keydown.arrowup)=\"_movePrev($event)\" (keydown.enter)=\"_manageDropdownState($event)\" (keydown.esc)=\"_manageDropdownState($event)\" (keydown.tab)=\"_manageDropdownState($event)\" (keyup.tab)=\"_manageDropdownState($event)\" (keydown.shift.tab)=\"_manageDropdownState($event)\" (keyup.shift.tab)=\"_manageDropdownState($event)\" (focusout)=\"_onTouchedEvent($event)\"><input type=\"text\" [id]=\"id\" [name]=\"name\" [placeholder]=\"placeholder\" class=\"dropdown-input-box form-control\" [ngClass]=\"{ 'dropdown-input-box-green': _dropdownVisible || _focusActive, 'highlight-default':nbpHighlightedDefault, 'input-box-disabled': disabled }\" (input)=\"_manageDropdown($event)\" [value]=\"_getViewValue(value)\" [required]=\"required\" [(ngModel)]=\"containsFilterValue\" (ngModelChange)=\"onContainsFilterChange()\" autocomplete=\"off\" (focus)=\"_focusActive = true;\" (focusout)=\"_focusActive = false;\" [tabindex]=\"tabindex\" #autoFocusComboFilter> <button class=\"dropdown-button\" (click)=\"_onInputClick($event)\" [ngClass]=\"{'collapse-disabled' : disabled}\" tabindex=\"-1\"><span class=\"dropdown-button-border\"></span> <span [ngClass]=\"{'triangolo-bottom' : !_dropdownVisible, 'triangolo-top' : _dropdownVisible}\"></span></button></div><ul [@dropdownAnimationState]=\"_getDropdownState()\" [style.visibility]=\"_dropdownVisible ? 'visible' : 'hidden'\"><li *ngFor=\"let item of _nbpShadowData; let i = index\" (click)=\"_onSelectEvent(item)\" [ngClass]=\"{'active': _isOptionSelected(item), 'preselected':_isOptionPreselected(item), 'last': i === _nbpShadowData.length - 1  }\" [id]=\"'combo-filter-element_'+i\"><span>{{_getDropDownLabel(item) }}</span></li></ul></div>",
        styles: [":host{position:relative;display:block}:host(.ng-invalid.nbp-show-validation) .dropdown-input-box{border:2px solid #c63200!important;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075);box-shadow:inset 0 1px 1px rgba(0,0,0,.075)}:host(.ng-invalid.nbp-show-validation) .dropdown-input-box:focus{border:2px solid #c63200!important}.highlight-default{border:2px solid #fce07f!important;outline:0;box-shadow:none!important}.dropdown-container{display:inline-block;position:relative;width:100%}.dropdown-container ul{opacity:1;background:#fff;border-top-width:0;border-right-color:#323232;border-right-style:solid;border-right-width:1px;border-bottom-color:#323232;border-bottom-style:solid;border-bottom-width:1px;border-left-color:#323232;border-left-style:solid;border-left-width:1px;border-radius:0 0 2px 2px;list-style:none;margin:0 0 .625em 0;padding:0;position:absolute;width:100%;z-index:1000;overflow-y:auto}.dropdown-container ul>li{cursor:pointer;min-height:3.125em;padding:17px 12px;border-bottom:1px solid #f2f2f2;text-align:left}.dropdown-container ul>li:hover{background:#c4dbbc}.dropdown-container ul>li.preselected{background:#c4dbbc}.dropdown-container ul>li.preselected.active{box-shadow:inset 0 0 1em #258900}.dropdown-container ul>li.active{background:#89bc72}.dropdown-container ul>li.last{border-bottom:none}.dropdown-container ul.ng-hide-add{transition:all ease-out .25s}.dropdown-container ul.ng-hide-remove{transition:all ease-out .25s}.dropdown-container ul.ng-hide{opacity:0;max-height:0}.dropdown-inner-container{position:relative;z-index:999}.dropdown-label{color:#6f6f6f;font-size:1em;margin:0 0 0 .75em!important}.dropdown-input-box{margin:0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;padding:.375em 2.3em .375em 1em;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;color:#323232;border:1px solid #323232;border-radius:2px;font-size:1em!important;height:3.125em}.dropdown-input-box:focus{border:2px solid #89bc72;outline:0;box-shadow:none!important}.dropdown-input-box::-ms-clear{display:none}.dropdown-input-box:disabled{background-color:#fff!important;cursor:default;box-shadow:none!important}.dropdown-input-box.input-box-disabled{background-color:#fff!important;border:1px solid #6f6f6f;opacity:.5!important;cursor:not-allowed}::placeholder{color:#6f6f6f;opacity:1}:-ms-input-placeholder{color:#6f6f6f}::-ms-input-placeholder{color:#6f6f6f}.dropdown-input-box-green{border:2px solid #89bc72;outline:0;box-shadow:none!important}.dropdown-input-box-disabled{background-color:#fff!important;opacity:.5!important}.dropdown-button{position:absolute;top:0;right:0;width:2.5em;height:3.125em;text-align:center;color:#ccc;border-color:transparent;background-color:transparent!important}.dropdown-button[role=\"\"]{cursor:default;opacity:.5}.dropdown-button.collapse-disabled{pointer-events:none}.dropdown-button.collapse-disabled .dropdown-button-border{border-left:1px solid #6f6f6f!important}.dropdown-button.collapse-disabled .triangolo-bottom{border-style:solid;border-width:8px 6.5px 0 6.5px;border-color:#6f6f6f transparent transparent transparent}.dropdown-button.collapse-disabled .triangolo-top{border-style:solid;border-width:0 6.5px 8px 6.5px;border-color:transparent transparent #6f6f6f transparent}.dropdown-button-border{position:absolute;display:inline-block;height:60%;top:20%;left:0;border-left:1px solid #323232}.has-error .dropdown-input-box{border:2px solid #c63200;-webkit-box-shadow:inset 0 1px 1px rgba(0,0,0,.075);box-shadow:inset 0 1px 1px rgba(0,0,0,.075)}.has-error .dropdown-input-box:focus{border:2px solid #c63200}.dropdown-error-message{color:#c63200;font-size:1em;margin-top:.625em;margin-bottom:.625em;opacity:1}.dropdown-error-message.ng-hide-add{transition:all ease-out .25s}.dropdown-error-message.ng-hide-remove{transition:all ease-out .25s}.dropdown-error-message.ng-hide{opacity:0;margin-top:-.625em}.triangolo-top{width:0;height:0;border-style:solid;border-width:0 .625em .625em .625em;border-color:transparent transparent #258900 transparent;display:inline-block!important}.triangolo-bottom{width:0;height:0;border-style:solid;border-width:.625em .625em 0 .625em;border-color:#258900 transparent transparent transparent;display:inline-block!important}.yellow-border{border:2px solid #fce07f}:host-context(.contrasto-on) :host(.ng-invalid.nbp-show-validation) .dropdown-input-box{border-color:#000!important}:host-context(.contrasto-on) :host(.ng-invalid.nbp-show-validation) .dropdown-input-box:focus{border-color:#000!important}:host-context(.contrasto-on) .highlight-default{border-color:#000!important}:host-context(.contrasto-on) .dropdown-container ul{background:#fff;border-right-color:#000;border-bottom-color:#000;border-left-color:#000}:host-context(.contrasto-on) .dropdown-container ul>li{border-bottom-color:#fff}:host-context(.contrasto-on) .dropdown-container ul>li:hover{background:#000;color:#fff}:host-context(.contrasto-on) .dropdown-container ul>li.active{background:#000;color:#fff}:host-context(.contrasto-on) .dropdown-label{color:#000}:host-context(.contrasto-on) .dropdown-input-box{color:#000;border-color:#000}:host-context(.contrasto-on) .dropdown-input-box:focus{border-color:#000}:host-context(.contrasto-on) .dropdown-input-box:disabled{background-color:#fff!important}:host-context(.contrasto-on) .dropdown-input-box.input-box-disabled{background-color:#fff!important;border-color:#000}:host-context(.contrasto-on) .dropdown-input-box-green{border-color:#000}:host-context(.contrasto-on) .dropdown-input-box-disabled{background-color:#fff!important}:host-context(.contrasto-on) .dropdown-button{color:#000}:host-context(.contrasto-on) .dropdown-button.collapse-disabled .dropdown-button-border{border-left-color:#000!important}:host-context(.contrasto-on) .dropdown-button.collapse-disabled .triangolo-bottom{border-top-color:#000}:host-context(.contrasto-on) .dropdown-button.collapse-disabled .triangolo-top{border-bottom-color:#000}:host-context(.contrasto-on) .dropdown-button-border{border-left-color:#000}:host-context(.contrasto-on) .has-error .dropdown-input-box{border-color:#000}:host-context(.contrasto-on) .has-error .dropdown-input-box:focus{border-color:#000}:host-context(.contrasto-on) .dropdown-error-message{color:#000}:host-context(.contrasto-on) .triangolo-top{border-bottom-color:#000}:host-context(.contrasto-on) .triangolo-bottom{border-top-color:#000}:host-context(.contrasto-on) .yellow-border{border-color:#fff} /*# sourceMappingURL=nbp-combo-filter.component.css.map */ "],
        viewProviders: [NbpFormatterService],
        providers: [NBP_COMBO_CONTROL_VALUE_ACCESSOR, NBP_INPUT_ELEMENT_BASE],
        animations: [
            trigger('dropdownAnimationState', [
                state('expanded', style({ opacity: 1, visibility: 'visible' })),
                state('collapsed, void', style({ opacity: 0, visibility: 'hidden' })),
                transition('expanded => collapsed', animate('100ms ease-in')),
                transition('collapsed => expanded', animate('100ms ease-out'))
            ])
        ]
    }),
    __param(0, Optional()), __param(0, Inject(NG_VALIDATORS)),
    __param(1, Optional()), __param(1, Inject(NG_ASYNC_VALIDATORS)),
    __metadata("design:paramtypes", [Array, Array, NbpFormatterService,
        ElementRef, Injector])
], NbpComboFilterComponent);
export { NbpComboFilterComponent };
