// cjlib/nbp/nbp-slider-range/nbp-slider-range-popover/nbp-slider-range-popover.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input } from '@angular/core';
import Utils from '../utils';
var NbpSliderRangePopoverComponent = (function () {
    function NbpSliderRangePopoverComponent() {
        this.value = 0;
        this.formatted = "";
    }
    NbpSliderRangePopoverComponent.prototype.ngOnInit = function () {
        this.getValue();
    };
    NbpSliderRangePopoverComponent.prototype.ngOnChanges = function () {
        this.getValue();
    };
    NbpSliderRangePopoverComponent.prototype.getValue = function () {
        var format = Utils.getDecimalFormatter(this.opts.label.format);
        this.formatted = Utils.formatValue(this.value, format);
    };
    return NbpSliderRangePopoverComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangePopoverComponent.prototype, "value", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpSliderRangePopoverComponent.prototype, "opts", void 0);
NbpSliderRangePopoverComponent = __decorate([
    Component({
        selector: 'nbp-slider-range-popover',
        template: "<div class=\"nbp-slider-range-popover\"><div class=\"nbp-slider-range-popover-content\">{{formatted}}</div></div>",
        styles: [".nbp-slider-range-popover{width:80px;position:absolute;bottom:100%;margin-bottom:27px;margin-left:9px;left:50%;transform:translate(-50%,0)} /*# sourceMappingURL=nbp-slider-range-popover.component.css.map */ "]
    }),
    __metadata("design:paramtypes", [])
], NbpSliderRangePopoverComponent);
export { NbpSliderRangePopoverComponent };
