// cjlib/cj/cj-privacy/cj-privacy-client.service.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
import { Injectable, Inject } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { APP_CONFIG } from '../../module.config';
import 'rxjs/add/operator/catch';
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/toPromise';
/**
 * Client Http per recuperare le informazioni di privacy
 */
var CjPrivacyClientService = (function () {
    function CjPrivacyClientService(config, http) {
        this.config = config;
        this.http = http;
        // da usare quando avremo gli indirizzi veri dei servizi rest
        // private cjBaseUrl = '';
        this.headers = new HttpHeaders().set('Content-Type', 'application/json');
    }
    // Lettura:     GET   /clienti/{ndg}/{abi}/{codUnitaOperativa}/privacy
    // Dispositiva: POST  /clienti/{ndg}/{abi}/{codUnitaOperativa}/privacy
    // Stampa:      POST  /clienti/{ndg}/{abi}/{codUnitaOperativa}/privacy/stampa
    CjPrivacyClientService.prototype.getPrivacyByNdg = function (ndg, abi, unitaOperativa) {
        var _this = this;
        return this.config.getNpuServicesBaseUrl()
            .concatMap(function (url) {
            var indirizzo = "clienti/" + ndg + "/" + abi + "/" + unitaOperativa + "/privacy";
            return _this.http.get(url + indirizzo, { headers: _this.headers, withCredentials: true });
        });
    };
    CjPrivacyClientService.prototype.savePrivacy = function (data, ndg, abi, unitaOperativa) {
        var _this = this;
        return this.config.getNpuServicesBaseUrl()
            .concatMap(function (url) {
            var indirizzo = "clienti/" + ndg + "/" + abi + "/" + unitaOperativa + "/privacy";
            return _this.http.post(url + indirizzo, data, { headers: _this.headers, withCredentials: true });
        });
    };
    CjPrivacyClientService.prototype.printPrivacy = function (ndg, abi, unitaOperativa, tipoOfferta) {
        var _this = this;
        console.log('Stampa in corso...');
        return this.config.getNpuServicesBaseUrl()
            .concatMap(function (url) {
            var indirizzo = "clienti/" + ndg + "/" + abi + "/" + unitaOperativa + "/" + tipoOfferta + "/privacy/stampa";
            return _this.http.post(url + indirizzo, { headers: _this.headers, withCredentials: true });
        });
    };
    return CjPrivacyClientService;
}());
CjPrivacyClientService = __decorate([
    Injectable(),
    __param(0, Inject(APP_CONFIG)),
    __metadata("design:paramtypes", [Object, HttpClient])
], CjPrivacyClientService);
export { CjPrivacyClientService };
