// cjlib/cj/cj-privacy/cj-privacy.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
import { Component, Optional, Input, Output, EventEmitter, ViewChild } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import 'rxjs/add/operator/toPromise';
import { CjPrivacyService } from './cj-privacy.service';
import { ApplicationStatusService } from '../../utility/application-status/application-status.service';
import { CjPrivacyFormComponent } from '../cj-privacy-form/cj-privacy-form.component';
var CjPrivacyComponent = (function () {
    function CjPrivacyComponent(privacyService, route, appStatus) {
        this.privacyService = privacyService;
        this.route = route;
        this.appStatus = appStatus;
        // TODO recuperare dall'url anche gli altri 2 parametri ABI e unitaOperativa
        /**
         * Evento emesso al click dei pulsanti
         * @param {EventEmitter<ICjDettaglioPrivacy>}  cjOnDone
         */
        this.cjOnDone = new EventEmitter();
        /**
         * Evento emesso al click sul pulsante del componente nbp-error-advise
         * @param {EventEmitter<any>}  cjOnError
         */
        this.cjOnError = new EventEmitter();
        /**
         * Evento emesso al termine della chiamata di stampa
         * @param {EventEmitter<any>} cjOnPrint
         */
        this.cjOnPrint = new EventEmitter();
        this._viewErrorAdvise = false;
    }
    CjPrivacyComponent.prototype.ngOnInit = function () {
        this._getPrivacyParams();
    };
    CjPrivacyComponent.prototype._getPrivacyParams = function () {
        var _this = this;
        if (this.route) {
            if (!this.cjNdg || !this.cjAbi || !this.cjUnitaOperativa) {
                // controllo che esista l'url param
                if (this.cjUrlParamsName) {
                    // recupero dall'url il parametro da passare al servizio
                    this.route.params.subscribe(function (params) {
                        _this.cjNdg = params[_this.cjUrlParamsName.ndgUrlParam];
                        _this.cjAbi = params[_this.cjUrlParamsName.abiUrlParam];
                        _this.cjUnitaOperativa = params[_this.cjUrlParamsName.unitaOperativaUrlParam];
                        _this.cjTipoOfferta = params[_this.cjUrlParamsName.tipoOffertaUrlParam];
                        _this._getDettaglioPrivacy();
                    });
                }
                else {
                    this._viewErrorAdvise = true;
                    console.error('CjPrivacy - parametri configurazione componente non corretti');
                }
            }
            else {
                this._getDettaglioPrivacy();
            }
        }
        else {
            if (!this.cjNdg || !this.cjAbi || !this.cjUnitaOperativa) {
                this._viewErrorAdvise = true;
                console.error('CjPrivacy - parametri configurazione componente non corretti');
            }
            else {
                this._getDettaglioPrivacy();
            }
        }
    };
    CjPrivacyComponent.prototype._getDettaglioPrivacy = function () {
        var _this = this;
        if (this.cjNdg && this.cjAbi && this.cjUnitaOperativa) {
            this.privacyService.getPrivacy(this.cjNdg, this.cjAbi, this.cjUnitaOperativa).subscribe(function (next) {
                _this._privacyData = next;
            }, function (error) {
                _this._viewErrorAdvise = true;
                throw error;
            });
        }
    };
    CjPrivacyComponent.prototype._onPrint = function (event) {
        var _this = this;
        // Chiamata al servizio di salvataggio e stampa consensi
        this.privacyService.savePrivacy(event, this.cjNdg, this.cjAbi, this.cjUnitaOperativa)
            .toPromise().then(function (savedData) {
            // console.log('CjPrivacy - Risultato del salvataggio: ');
            console.dir(savedData);
            _this.privacyService.printPrivacy(_this.cjNdg, _this.cjAbi, _this.cjUnitaOperativa, _this.cjTipoOfferta)
                .toPromise().then(function (printData) {
                console.log('CjPrivacy - Risultato della stampa: ');
                console.dir(printData);
                if ((printData.esito.toUpperCase() === 'OK') && printData.xmlToPrint) {
                    // Lancio l'evento di stampa che conterrà l'xml da inviare all'activex
                    _this._onSendPrintEvent(printData.xmlToPrint);
                    _this.privacyFormComponent.setButtonConfigurationAfterPrintDone();
                }
            });
        });
    };
    CjPrivacyComponent.prototype._onSendPrintEvent = function (xmlToPrint) {
        // TODO - Qualcosa tipo:
        // this.cjOnPrint.emit(xmlToPrint.xmlDocument);
        // per ora:
        this.cjOnPrint.emit(xmlToPrint);
    };
    CjPrivacyComponent.prototype._onContinue = function (event) {
        // Chiusura della pagina
        console.log('CjPrivacy - Click su pulsante Continua');
        this.cjOnDone.emit(event);
    };
    CjPrivacyComponent.prototype._onCancel = function (event) {
        // Chiusura della pagina
        console.log('CjPrivacy - Click su pulsante Annulla');
        this.cjOnDone.emit(event);
    };
    CjPrivacyComponent.prototype._onErrorAdvise = function () {
        console.log('CjPrivacy - emitting cjOnError');
        this.cjOnError.emit();
    };
    return CjPrivacyComponent;
}());
__decorate([
    Input(),
    __metadata("design:type", String)
], CjPrivacyComponent.prototype, "cjNdg", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], CjPrivacyComponent.prototype, "cjAbi", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], CjPrivacyComponent.prototype, "cjUnitaOperativa", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], CjPrivacyComponent.prototype, "cjTipoOfferta", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], CjPrivacyComponent.prototype, "cjUrlParamsName", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], CjPrivacyComponent.prototype, "cjOnDone", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], CjPrivacyComponent.prototype, "cjOnError", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], CjPrivacyComponent.prototype, "cjOnPrint", void 0);
__decorate([
    ViewChild(CjPrivacyFormComponent),
    __metadata("design:type", CjPrivacyFormComponent)
], CjPrivacyComponent.prototype, "privacyFormComponent", void 0);
CjPrivacyComponent = __decorate([
    Component({
        selector: 'cj-privacy',template: "<div class=\"row\"><div class=\"col-md-offset-2 col-md-20\"><nbp-error-advice *ngIf=\"_viewErrorAdvise\" (onError)=\"_onErrorAdvise()\"></nbp-error-advice></div></div><div class=\"row\"><div class=\"col-xs-offset-2 col-sm-offset-2 col-md-offset-2 col-xs-20 col-sm-20 col-md-20\"><cj-privacy-form *ngIf=\"!_viewErrorAdvise\" [cjPrivacy]=\"_privacyData\" (cjOnStampa)=\"_onPrint($event)\" (cjOnContinua)=\"_onContinue($event)\" (cjOnAnnulla)=\"_onCancel($event)\"></cj-privacy-form></div></div>"
    }),
    __param(1, Optional()),
    __metadata("design:paramtypes", [CjPrivacyService, ActivatedRoute,
        ApplicationStatusService])
], CjPrivacyComponent);
export { CjPrivacyComponent };
