// cjlib/nbp/nbp-combo/nbp-combo.component.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
import { Component, Input, Output, Optional, Inject, Injector, EventEmitter, forwardRef, ViewChild, HostBinding, HostListener, ElementRef, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
import { NG_VALUE_ACCESSOR, NG_VALIDATORS, NG_ASYNC_VALIDATORS } from '@angular/forms';
import { NgModel, NgControl } from '@angular/forms';
import { coerceBooleanProperty, getPropertyValueByPath, generateObjectFromStringPath } from './../../utility/lib-utility';
import { NbpPipe } from './../nbp-commons/nbp-commons.enums';
import { NbpFormatterService } from './../../utility/nbp-formatter.service';
import { INbpDataSource } from './../nbp-datasource/nbp-datasource.interfaces';
import { NbpFormContainerComponent } from './../nbp-form-container/nbp-form-container.component';
import { NbpInputElementBase } from './../nbp-commons/nbp-input-element-base';
/**
 * Provider Expression that allows nbp-combo to register as a ControlValueAccessor.
 * This allows it to support [(ngModel)].
 * @docs-private
 */
export var NBP_COMBO_CONTROL_VALUE_ACCESSOR = {
    provide: NG_VALUE_ACCESSOR,
    useExisting: forwardRef(function () { return NbpComboComponent; }),
    multi: true
};
export var NBP_INPUT_ELEMENT_BASE = {
    provide: NbpInputElementBase,
    useExisting: forwardRef(function () { return NbpComboComponent; }),
    multi: true
};
/** intero autogenerato usato per id unici per nbpCombo component */
var nextId = 0;
/**
 * Componente per la implementazione della combobox, supporta tutti gli attributi standard html per il tag input.
 * Personalizzabile per la gestione di currency grazie all'applicaizone di opportuni pipes
 * definiti dalla libreria.
 */
var NbpComboComponent = (function (_super) {
    __extends(NbpComboComponent, _super);
    function NbpComboComponent(validatori, asyncValidatori, parentFormContainer, formatterService, element, inj, cd) {
        var _this = _super.call(this, validatori, asyncValidatori) || this;
        _this.validatori = validatori;
        _this.asyncValidatori = asyncValidatori;
        _this.parentFormContainer = parentFormContainer;
        _this.formatterService = formatterService;
        _this.element = element;
        _this.inj = inj;
        _this.cd = cd;
        /**
         * Identificativo univoco per il text field.
         * Se nessun id è fornito e' auto-generato.
         */
        _this.id = "nbp-combo-" + ++nextId;
        _this._nbpShowValidation = false;
        _this.showErrorLabel = false;
        /**
         * Il testo del placeholder.
         */
        _this.placeholder = '';
        /**
         * Evento emesso quando è selezionato un elemento della combobox.
         */
        _this.select = new EventEmitter();
        /**
         * Evento emesso quando la dropdown viene chiuda
         */
        _this.dropdownClosed = new EventEmitter();
        /**
         * Evento emesso quando la dropdown viene aperta
         */
        _this.dropdownOpened = new EventEmitter();
        /**
         * (Opzionale) Funzione custom di validazione del campo
         */
        _this.nbpValidation = new EventEmitter();
        /**
         * (Opzionale) Paramentro che indica l'ordine di spostamento da un campo all'altro attraverso il tab.
         */
        _this.tabindex = 0;
        _this._dropdownVisible = false;
        _this._arrowClicked = false;
        _this._nbpPipe = NbpPipe;
        _this._multi = false;
        _this.initPolyfillScroll();
        return _this;
    }
    Object.defineProperty(NbpComboComponent.prototype, "nbpShowValidation", {
        /**
         * (Opzionale)
         * Booleano che consente di visualizzare i messaggi di feedback relativi alla validazione.
         * Valore di default false
         */
        get: function () {
            if (this.parentFormContainer) {
                this._nbpShowValidation = this.parentFormContainer.nbpShowValidation;
            }
            return this._nbpShowValidation;
        },
        set: function (value) { this._nbpShowValidation = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    ;
    Object.defineProperty(NbpComboComponent.prototype, "disabled", {
        /**
         * Flag che abilita/disabilita le funzionalità associate alla combobox
         */
        get: function () {
            if (this.parentFormContainer && this.parentFormContainer.disabled) {
                this._nbpDisabled = true;
            }
            return this._nbpDisabled;
        },
        set: function (value) { this._nbpDisabled = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NbpComboComponent.prototype, "required", {
        /**
         * Flag che imposta come obbligatorio o meno il campo di input (disabilitato) associato alla combobox
         */
        get: function () { return this._nbpRequired; },
        set: function (value) { this._nbpRequired = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NbpComboComponent.prototype, "nbpShowEmptyValue", {
        /**
         * Flag che imposta come obbligatorio o meno il campo di input (disabilitato) associato alla combobox
         */
        get: function () {
            // alert("get------->" + this._nbpShowEmptyValue + "<--------")
            //camillocamillo
            return this._nbpShowEmptyValue;
        },
        set: function (value) {
            // alert("set input ------->" + value + "<--------")
            this._nbpShowEmptyValue = coerceBooleanProperty(value);
            // alert("set output ------->" + this._nbpShowEmptyValue + "<--------")
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NbpComboComponent.prototype, "nbpHighlightedDefault", {
        /**
         * (opzionale ) Booleano che gestisce l’attributo  bordo giallo della combobox.
         * Al primo click/focus tornerà alla versione di default.
         * valore default false
         */
        get: function () { return this._nbpHighlightedDefault; },
        set: function (value) { this._nbpHighlightedDefault = coerceBooleanProperty(value); },
        enumerable: true,
        configurable: true
    });
    ;
    ;
    Object.defineProperty(NbpComboComponent.prototype, "nbpBindingToKeyField", {
        /**
         * Definisce quando il binding del ngModel deve essere fatto sull'attributo nbpKeyField
         *
         */
        get: function () { return this._nbpBindingToKeyField; },
        set: function (value) {
            if (!this._multi) {
                this._nbpBindingToKeyField = coerceBooleanProperty(value);
            }
            else {
                this._nbpBindingToKeyField = false;
            }
        },
        enumerable: true,
        configurable: true
    });
    ;
    ;
    NbpComboComponent.prototype.initPolyfillScroll = function () {
        if (!Element.prototype['scrollIntoViewIfNeeded']) {
            Element.prototype.scrollIntoViewIfNeeded = function (centerIfNeeded) {
                centerIfNeeded = arguments.length === 0 ? true : !!centerIfNeeded;
                var parent = this.parentNode, parentComputedStyle = window.getComputedStyle(parent, null), parentBorderTopWidth = parseInt(parentComputedStyle.getPropertyValue('border-top-width'), 10), parentBorderLeftWidth = parseInt(parentComputedStyle.getPropertyValue('border-left-width'), 10), overTop = this.offsetTop - parent.offsetTop < parent.scrollTop, overBottom = (this.offsetTop - parent.offsetTop + this.clientHeight - parentBorderTopWidth) > (parent.scrollTop + parent.clientHeight), overLeft = this.offsetLeft - parent.offsetLeft < parent.scrollLeft, overRight = (this.offsetLeft - parent.offsetLeft + this.clientWidth - parentBorderLeftWidth) > (parent.scrollLeft + parent.clientWidth), alignWithTop = overTop && !overBottom;
                if ((overTop || overBottom) && centerIfNeeded) {
                    parent.scrollTop = this.offsetTop - parent.offsetTop - parent.clientHeight / 2 - parentBorderTopWidth + this.clientHeight / 2;
                }
                if ((overLeft || overRight) && centerIfNeeded) {
                    parent.scrollLeft = this.offsetLeft - parent.offsetLeft - parent.clientWidth / 2 - parentBorderLeftWidth + this.clientWidth / 2;
                }
                if ((overTop || overBottom || overLeft || overRight) && !centerIfNeeded) {
                    this.scrollIntoView(alignWithTop);
                }
            };
        }
    };
    NbpComboComponent.prototype.writeValue = function (value) {
        _super.prototype.writeValue.call(this, value);
        this.cd.markForCheck();
    };
    NbpComboComponent.prototype._getDropdownState = function () {
        return this._dropdownVisible ? 'expanded' : 'collapsed';
    };
    NbpComboComponent.prototype._manageDropdownState = function (event) {
        switch (event.key.toLowerCase()) {
            case 'enter':
                event.preventDefault();
                if (this._dropdownVisible) {
                    this._onPreSelectEvent(this._nbpShadowData[this._preSelectedElementIndex]);
                }
                this._dropdownVisible = !this._dropdownVisible;
                break;
            case 'tab':
                if (this._dropdownVisible) {
                    event.preventDefault();
                    this._onPreSelectEvent(this._nbpShadowData[this._preSelectedElementIndex]);
                }
                this._dropdownVisible = false;
                break;
            default:
                this._dropdownVisible = false;
                break;
        }
        if (this._evaluateCurrentSelectionIndex() >= 0) {
            this.scrollToSelectedIndex(this._evaluateCurrentSelectionIndex());
        }
        else {
            this._preSelectedElementIndex = null;
        }
    };
    NbpComboComponent.prototype._isOptionPreselected = function (item) {
        var currentPreSelection = this._nbpShadowData.findIndex(function (element) { return element === item; });
        return currentPreSelection === this._preSelectedElementIndex;
    };
    NbpComboComponent.prototype._onSelectByKeyboard = function (event) { };
    NbpComboComponent.prototype._showCheckbox = function (item) { };
    NbpComboComponent.prototype._isAlphaNumericCharacter = function (keyCode) {
        return (keyCode >= 48 && keyCode <= 57) || (keyCode >= 65 && keyCode <= 90) || (keyCode >= 95 && keyCode <= 122);
    };
    NbpComboComponent.prototype._searchByKey = function (event) {
        var _this = this;
        if (this._nbpDisabled) {
            return;
        }
        //console.log(event.key);
        if (this._isAlphaNumericCharacter(event.keyCode)) {
            var currentSelection = void 0;
            var listByKey_1;
            if (this.nbpViewField) {
                listByKey_1 = this._nbpShadowData.filter(function (item) {
                    var _startWithCode = !item.emptyValue ? getPropertyValueByPath(item, _this.nbpViewField)
                        .toLowerCase().trim().startsWith(event.key.toLowerCase()) : false;
                    return _startWithCode;
                });
            }
            else {
                this._nbpShadowData.findIndex(function (item) { return item === _this.value; });
                listByKey_1 = this._nbpShadowData.filter(function (item) {
                    var _startWithCode = item ? item.toLowerCase().trim().startsWith(event.key.toLowerCase()) : '';
                    return _startWithCode;
                });
            }
            currentSelection = (listByKey_1 && listByKey_1.length > 0) ? this._nbpShadowData.findIndex(function (item) { return listByKey_1[0] === item; }) : null;
            if (currentSelection !== null) {
                this._dropdownVisible = true;
                this.scrollToSelectedIndex(currentSelection);
                // this._onPreSelectEvent(this._nbpShadowData[currentSelection]);
                this._preSelectedElementIndex = currentSelection;
            }
        }
    };
    /**
     * Alla pressione di "delete" o "backspace", se la checkbox è vuota ne svuoto anche il value (in modo da triggerare anche la validation).
     * Svuoto anche il _preSelectedElementIndex.
     * @param event
     * @private
     */
    NbpComboComponent.prototype._checkEmpty = function (event) {
        if (!this.visibleTextBox.nativeElement.value) {
            this._preSelectedElementIndex = undefined;
            this.value = undefined;
        }
    };
    NbpComboComponent.prototype._evaluateCurrentSelectionIndex = function () {
        var _this = this;
        var currentSelection;
        if (this.value) {
            if (this.nbpKeyField && this.nbpBindingToKeyField) {
                currentSelection = this._nbpShadowData.findIndex(function (item) { return getPropertyValueByPath(item, _this.nbpKeyField) === _this.value; });
            }
            else {
                currentSelection = this._nbpShadowData.findIndex(function (item) { return item === _this.value; });
            }
            return currentSelection;
        }
        return 0;
    };
    NbpComboComponent.prototype.scrollToSelectedIndex = function (elementIndex) {
        if (this._dropdownVisible) {
            setTimeout(function () {
                document.getElementById('combo-element_' + elementIndex).scrollIntoViewIfNeeded();
            });
        }
    };
    NbpComboComponent.prototype._moveNext = function (event) {
        if (this._nbpDisabled) {
            return;
        }
        event.preventDefault();
        var currentSelection = this._preSelectedElementIndex !== null && this._preSelectedElementIndex !== undefined ?
            this._preSelectedElementIndex : (this.nbpShowEmptyValue) ? 0 : -1;
        var nextSelection = currentSelection + 1;
        if (nextSelection >= this._nbpShadowData.length) {
            nextSelection = currentSelection;
        }
        this.scrollToSelectedIndex(nextSelection);
        // this._onPreSelectEvent(this._nbpShadowData[nextSelection]);
        this._preSelectedElementIndex = nextSelection;
    };
    NbpComboComponent.prototype._movePrev = function (event) {
        if (this._nbpDisabled) {
            return;
        }
        event.preventDefault();
        var currentSelection = this._preSelectedElementIndex !== null && this._preSelectedElementIndex !== undefined ?
            this._preSelectedElementIndex : 0;
        var nextSelection = currentSelection < 0 ? 0 : currentSelection - 1;
        if (nextSelection < 0) {
            nextSelection = currentSelection;
        }
        this.scrollToSelectedIndex(nextSelection);
        // this._onPreSelectEvent(this._nbpShadowData[nextSelection]);
        this._preSelectedElementIndex = nextSelection;
    };
    NbpComboComponent.prototype.confirmSelection = function () {
        this._onSelectEvent(this._nbpShadowData[this._preSelectedElementIndex]);
    };
    NbpComboComponent.prototype.ngOnInit = function () {
        this.ngControl = this.inj.get(NgControl);
        this._prepareDataSource();
        this.initVisibleTextBox();
        if (this.firstFocus) {
            this.elementRef.nativeElement.focus();
            this.elementRef.nativeElement.click();
            // this._focusActive = true;
        }
    };
    NbpComboComponent.prototype.initVisibleTextBox = function () {
        try {
            this._preSelectedElementIndex = this._nbpShadowData.indexOf(this.model);
            this._nbpViewValue = this._getViewModelElement(this._nbpShadowData[this._preSelectedElementIndex]);
        }
        catch (e) {
            console.warn('failed to initialize combo: ' + e);
        }
    };
    NbpComboComponent.prototype.ngOnChanges = function (changes) {
        if (changes.nbpDataSource && !changes.nbpDataSource.firstChange) {
            this._prepareDataSource();
        }
    };
    NbpComboComponent.prototype._prepareDataSource = function () {
        var _this = this;
        if (this.nbpDataSource) {
            this.datasourceSubscription = this.nbpDataSource.getDataStream().subscribe(function (value) {
                if (value.data && value.data.length) {
                    _this._nbpShadowData = value.data.slice(); // 0, 50
                    if (_this.nbpShowEmptyValue) {
                        if (_this.nbpViewField) {
                            var emptyObject = { emptyValue: true };
                            emptyObject = generateObjectFromStringPath(emptyObject, _this.nbpViewField);
                            if (_this.nbpKeyField) {
                                emptyObject = generateObjectFromStringPath(emptyObject, _this.nbpKeyField);
                            }
                            _this._nbpShadowData.unshift(emptyObject);
                        }
                        else {
                            _this._nbpShadowData.unshift(null);
                        }
                    }
                }
                else {
                    _this._nbpShadowData = new Array();
                }
                ;
            }, function (error) {
                console.log(JSON.stringify(error));
            });
            this.nbpDataSource.sendDataEvent();
        }
    };
    NbpComboComponent.prototype._getViewValue = function (item) {
        if (!item) {
            return null;
        }
        return this.formatterService.displayValueTransformation(this._getViewModelElement(item), this.nbpFilter, this.nbpFilterArg);
    };
    NbpComboComponent.prototype._getModelValue = function (item) {
        if (!item || item && item.emptyValue) {
            return;
        }
        return (this._nbpBindingToKeyField && this.nbpKeyField) ? getPropertyValueByPath(item, this.nbpKeyField) : item;
    };
    NbpComboComponent.prototype._getDropDownLabel = function (item) {
        return this.formatterService.displayValueTransformation((this.nbpViewField ? getPropertyValueByPath(item, this.nbpViewField) : item), this.nbpFilter, this.nbpFilterArg);
    };
    NbpComboComponent.prototype._onSelectEvent = function (item) {
        this._nbpViewValue = this._getViewModelElement(item);
        this.value = this._getModelValue(item);
        this.nbpValidation.emit(this.model);
        this._dropdownVisible = !this._dropdownVisible;
        this.element.nativeElement.querySelector('.dropdown-inner-container').focus();
        // TODO validazione anche su set programmatica del value
        event.stopPropagation();
        this.select.emit(item);
    };
    NbpComboComponent.prototype._onPreSelectEvent = function (item) {
        this._nbpViewValue = this._getViewModelElement(item);
        this.value = this._getModelValue(item);
        this.nbpValidation.emit(this.model);
        // TODO validazione anche su set programmatica del value
        event.stopPropagation();
        this.select.emit(item);
    };
    NbpComboComponent.prototype._getViewModelElement = function (item) {
        if (item) {
            if (this._nbpBindingToKeyField) {
                return this.nbpKeyField ? this._getSelectedElement(item) : item;
            }
            else {
                return this.nbpViewField ? getPropertyValueByPath(item, this.nbpViewField) : item;
            }
        }
        return;
    };
    NbpComboComponent.prototype._getSelectedElement = function (item) {
        var _this = this;
        var elems = this._nbpShadowData ? this._nbpShadowData.filter(function (elem) {
            return getPropertyValueByPath(elem, _this.nbpKeyField) === item;
        }) : [];
        if (elems.length > 0) {
            var selectedElem = elems[0];
            return this.nbpViewField ? getPropertyValueByPath(selectedElem, this.nbpViewField) : item;
        }
        else {
            return item;
        }
    };
    NbpComboComponent.prototype._isOptionSelected = function (item) {
        if (!item || !this.value) {
            return false;
        }
        if (this.nbpKeyField && item && getPropertyValueByPath(item, this.nbpKeyField)) {
            var currentValue = this.value;
            if (this._nbpBindingToKeyField) {
                return getPropertyValueByPath(item, this.nbpKeyField) === currentValue;
            }
            else {
                return getPropertyValueByPath(item, this.nbpKeyField) === getPropertyValueByPath(currentValue, this.nbpKeyField);
            }
        }
        if (item) {
            return item === this.value;
        }
        return false;
    };
    NbpComboComponent.prototype._onInputFocus = function (event) {
        if (!this._nbpDisabled
            && this._nbpShadowData
            && this._nbpShadowData.length > 0 && !this._arrowClicked) {
            this._dropdownVisible = !this._dropdownVisible;
            this._dropdownVisible ? this.dropdownOpened.emit() : this.dropdownClosed.emit();
        }
    };
    NbpComboComponent.prototype._onInputClick = function (event) {
        if (!this._nbpDisabled
            && this._nbpShadowData
            && this._nbpShadowData.length > 0) {
            this._dropdownVisible = !this._dropdownVisible;
            this._arrowClicked = !this._arrowClicked;
            if (this._evaluateCurrentSelectionIndex() >= 0 && !this._multi) {
                this.scrollToSelectedIndex(this._evaluateCurrentSelectionIndex());
            }
        }
    };
    NbpComboComponent.prototype._onTouchedEvent = function (event) {
        this.touch();
        this.showErrorLabel = true;
    };
    NbpComboComponent.prototype._closeDropDown = function (target) {
        if (!this._dropdownVisible) {
            return;
        }
        var parentFound = false;
        while (target != null && !parentFound) {
            if (target === this.element.nativeElement) {
                parentFound = true;
            }
            target = target.parentElement;
        }
        if (!parentFound) {
            this._dropdownVisible = false;
            this._arrowClicked = false;
            this.dropdownClosed.emit();
        }
    };
    NbpComboComponent.prototype._focusOutFn = function (event) {
        this._focusActive = false;
        this._onTouchedEvent(event);
        this._onPreSelectEvent(this._nbpShadowData[this._preSelectedElementIndex]);
    };
    NbpComboComponent.prototype.onFocusout = function (target) {
        this._closeDropDown(target);
    };
    NbpComboComponent.prototype.onClick = function (target) {
        this._closeDropDown(target);
    };
    return NbpComboComponent;
}(NbpInputElementBase));
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpComboComponent.prototype, "name", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpComboComponent.prototype, "id", void 0);
__decorate([
    HostBinding('class.nbp-show-validation'),
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboComponent.prototype, "nbpShowValidation", null);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpComboComponent.prototype, "nbpErrorMessage", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpComboComponent.prototype, "showErrorLabel", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboComponent.prototype, "disabled", null);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboComponent.prototype, "required", null);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpComboComponent.prototype, "placeholder", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboComponent.prototype, "nbpShowEmptyValue", null);
__decorate([
    Input(),
    __metadata("design:type", INbpDataSource)
], NbpComboComponent.prototype, "nbpDataSource", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpComboComponent.prototype, "nbpViewField", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], NbpComboComponent.prototype, "nbpKeyField", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboComponent.prototype, "nbpHighlightedDefault", null);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpComboComponent.prototype, "nbpFilter", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], NbpComboComponent.prototype, "nbpFilterArg", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [Object])
], NbpComboComponent.prototype, "nbpBindingToKeyField", null);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpComboComponent.prototype, "select", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NbpComboComponent.prototype, "dropdownClosed", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], NbpComboComponent.prototype, "dropdownOpened", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], NbpComboComponent.prototype, "nbpValidation", void 0);
__decorate([
    ViewChild(NgModel),
    __metadata("design:type", NgModel)
], NbpComboComponent.prototype, "model", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], NbpComboComponent.prototype, "firstFocus", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], NbpComboComponent.prototype, "tabindex", void 0);
__decorate([
    ViewChild('autoFocusCombo'),
    __metadata("design:type", ElementRef)
], NbpComboComponent.prototype, "elementRef", void 0);
__decorate([
    ViewChild('visibleTextBox'),
    __metadata("design:type", ElementRef)
], NbpComboComponent.prototype, "visibleTextBox", void 0);
__decorate([
    HostListener('document: focusout', ['$event.target']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [HTMLElement]),
    __metadata("design:returntype", void 0)
], NbpComboComponent.prototype, "onFocusout", null);
__decorate([
    HostListener('document: click', ['$event.target']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [HTMLElement]),
    __metadata("design:returntype", void 0)
], NbpComboComponent.prototype, "onClick", null);
NbpComboComponent = __decorate([
    Component({
        selector: 'nbp-combo',template: "<div class=\"dropdown-container\"><input type=\"hidden\" [(ngModel)]=\"value\"><div class=\"dropdown-inner-container\" (click)=\"_onInputClick($event)\" (keypress)=\"_searchByKey($event)\" (keydown.arrowdown)=\"_moveNext($event)\" (keydown.arrowup)=\"_movePrev($event)\" (keydown.enter)=\"_manageDropdownState($event)\" (keydown.esc)=\"_manageDropdownState($event)\" (keydown.spacebar)=\"_onSelectByKeyboard($event)\" (keydown.space)=\"_onSelectByKeyboard($event)\" (keydown.tab)=\"_manageDropdownState($event)\" (keydown.shift.tab)=\"_manageDropdownState($event)\" (keyup.shift.tab)=\"_manageDropdownState($event)\" (keyup)=\"_checkEmpty($event)\" (focus)=\"_focusActive = true;\" [ngClass]=\"{'collapse-disabled' : disabled}\" (focusout)=\"_focusOutFn($event)\"><input #visibleTextBox type=\"text\" [id]=\"id\" [name]=\"name\" [placeholder]=\"placeholder\" class=\"dropdown-input-box form-control\" [ngClass]=\"{ 'dropdown-input-box-green': _dropdownVisible || _focusActive, 'highlight-default':nbpHighlightedDefault, 'input-box-disabled': disabled}\" [value]=\"_getViewValue(value)\" [required]=\"required\" autocomplete=\"off\" (focus)=\"_focusActive = true;\" (focusout)=\"_focusOutFn($event)\" #autoFocusCombo [tabindex]=\"tabindex\"> <button class=\"dropdown-button\" [ngClass]=\"{'collapse-disabled' : disabled}\" type=\"button\" tabindex=\"-1\"><span class=\"dropdown-button-border\"></span> <span [ngClass]=\"{'triangolo-bottom' : !_dropdownVisible, 'triangolo-top' : _dropdownVisible}\"></span></button></div><ul *ngIf=\"_dropdownVisible\"><li *ngFor=\"let item of _nbpShadowData; let i = index; let last = last\" (click)=\"_onSelectEvent(item)\" [id]=\"'combo-element_'+i\" [ngClass]=\"{'active': _isOptionSelected(item), 'preselected':_isOptionPreselected(item), 'last': last }\"><nbp-checkbox #checkbox [checked]=\"_isOptionSelected(item)\" (change)=\"_onSelectEvent(item, checkbox.checked)\" *ngIf=\"_showCheckbox(item)\" nbpDisplayInline></nbp-checkbox><span>{{ _getDropDownLabel(item) }}</span></li></ul></div>",
        styles: [":host{position:relative;display:block}:host(.ng-touched.ng-invalid,.nbp-show-validation.ng-invalid) .dropdown-input-box{border:.125em solid #c63200!important;-webkit-box-shadow:inset 0 .0625em .0625em rgba(0,0,0,.075);box-shadow:inset 0 .0625em .0625em rgba(0,0,0,.075)}:host(.ng-touched.ng-invalid,.nbp-show-validation.ng-invalid) .dropdown-input-box:focus{border:.125em solid #c63200!important}:host(.ng-dirty.ng-invalid,.nbp-show-validation.ng-invalid) .dropdown-input-box{border:.125em solid #c63200!important;-webkit-box-shadow:inset 0 .0625em .0625em rgba(0,0,0,.075);box-shadow:inset 0 .0625em .0625em rgba(0,0,0,.075)}:host(.ng-dirty.ng-invalid,.nbp-show-validation.ng-invalid) .dropdown-input-box:focus{border:.125em solid #c63200!important}.highlight-default{border:.125em solid #fce07f!important;outline:0;box-shadow:none!important}.dropdown-container{display:inline-block;position:relative;width:100%}.dropdown-container ul{opacity:1;background:#fff;border-top-width:0;border-right-color:#323232;border-right-style:solid;border-right-width:.0625em;border-bottom-color:#323232;border-bottom-style:solid;border-bottom-width:.0625em;border-left-color:#323232;border-left-style:solid;border-left-width:.0625em;border-radius:0 0 .125em .125em;list-style:none;margin:0 0 .625em 0;padding:0;position:absolute;width:100%;z-index:1000;overflow-y:auto;max-height:12.8125em}.dropdown-container ul>li{cursor:pointer;min-height:3.125em;padding:.75em .625em .5em .625em;border-bottom:.0625em solid #ccc}.dropdown-container ul>li:hover{background:#c4dbbc}.dropdown-container ul>li.preselected{background:#c4dbbc}.dropdown-container ul>li.preselected.active{box-shadow:inset 0 0 1em #258900}.dropdown-container ul>li.active{background:#89bc72}.dropdown-container ul>li.last{border-bottom:none}.dropdown-container ul.ng-hide-add{transition:all ease-out .25s}.dropdown-container ul.ng-hide-remove{transition:all ease-out .25s}.dropdown-container ul.ng-hide{opacity:0;max-height:0}.dropdown-inner-container{position:relative;z-index:999}.dropdown-inner-container.collapse-disabled{pointer-events:none}.dropdown-label{color:#6f6f6f;font-size:1em;margin:0 0 0 .75em!important}.dropdown-input-box{margin:0;border-collapse:separate;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;padding:.625em;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;color:#323232;border:.0625em solid #323232;border-radius:.125em;font-size:1em!important;height:3.125em}.dropdown-input-box:focus{border:.125em solid #89bc72;outline:0;box-shadow:none!important}.dropdown-input-box::-ms-clear{display:none}.dropdown-input-box:disabled{background-color:#fff!important;cursor:default;box-shadow:none}.dropdown-input-box.input-box-disabled{background-color:#fff!important;border:.0625em solid #6f6f6f;opacity:.5!important;cursor:not-allowed}::placeholder{color:#6f6f6f;opacity:1}:-ms-input-placeholder{color:#6f6f6f}::-ms-input-placeholder{color:#6f6f6f}.dropdown-input-box-green{border:.125em solid #89bc72;outline:0;box-shadow:none!important}.dropdown-input-box-disabled{background-color:#fff!important;opacity:.5!important}.dropdown-button{display:table;position:absolute;top:0;right:0;width:2.5em;height:3.125em;text-align:center;color:#ccc;border-color:transparent;background-color:transparent!important}.dropdown-button[role=\"\"]{cursor:default;opacity:.5}.dropdown-button.collapse-disabled .dropdown-button-border{border-left:.0625em solid #6f6f6f!important}.dropdown-button.collapse-disabled .triangolo-bottom{border-style:solid;border-width:.5em .40625em;border-color:#6f6f6f transparent transparent transparent}.dropdown-button.collapse-disabled .triangolo-top{border-style:solid;border-width:0 .40625em .5em .40625em;border-color:transparent transparent #6f6f6f transparent}@media all and (-ms-high-contrast:none),(-ms-high-contrast:active){.dropdown-button span{position:absolute}.dropdown-button.collapse-disabled .triangolo-top{transform:translateY(-50%);top:50%}.dropdown-button.collapse-disabled .triangolo-bottom{transform:translateY(-50%);top:50%}.triangolo-top{transform:translateY(-50%);top:50%}.triangolo-bottom{transform:translateY(-50%);top:50%}}.dropdown-button-border{position:absolute;display:inline-block;height:60%;top:20%;left:0;border-left:.0625em solid #323232}.has-error .dropdown-input-box{border:.125em solid #c63200;-webkit-box-shadow:inset 0 .0625em .0625em rgba(0,0,0,.075);box-shadow:inset 0 .0625em .0625em rgba(0,0,0,.075)}.has-error .dropdown-input-box:focus{border:.125em solid #c63200}.dropdown-error-message{color:#c63200;font-size:1em;margin-top:.625em;margin-bottom:.625em;opacity:1}.dropdown-error-message.ng-hide-add{transition:all ease-out .25s}.dropdown-error-message.ng-hide-remove{transition:all ease-out .25s}.dropdown-error-message.ng-hide{opacity:0;margin-top:-.625em}.triangolo-top{width:0;height:0;border-style:solid;border-width:0 .625em .625em .625em;border-color:transparent transparent #258900 transparent;display:inline-block!important}.triangolo-bottom{width:0;height:0;border-style:solid;border-width:.625em .625em 0 .625em;border-color:#258900 transparent transparent transparent;display:inline-block!important}.yellow-border{border:.125em solid #fce07f}:host-context(.contrasto-on) :host(.ng-invalid.nbp-show-validation) .dropdown-input-box{border-color:#000!important}:host-context(.contrasto-on) :host(.ng-invalid.nbp-show-validation) .dropdown-input-box:focus{border-color:#000!important}:host-context(.contrasto-on) .highlight-default{border-color:#000!important}:host-context(.contrasto-on) .dropdown-container ul{background:#fff;border-right-color:#000;border-bottom-color:#000;border-left-color:#000}:host-context(.contrasto-on) .dropdown-container ul>li{border-bottom-color:#fff}:host-context(.contrasto-on) .dropdown-container ul>li:hover{background:#000;color:#fff}:host-context(.contrasto-on) .dropdown-container ul>li.active{background:#000;color:#fff}:host-context(.contrasto-on) .dropdown-label{color:#000}:host-context(.contrasto-on) .dropdown-input-box{color:#000;border-color:#000}:host-context(.contrasto-on) .dropdown-input-box:focus{border-color:#000}:host-context(.contrasto-on) .dropdown-input-box:disabled{background-color:#fff!important}:host-context(.contrasto-on) .dropdown-input-box.input-box-disabled{background-color:#fff!important;border-color:#000}:host-context(.contrasto-on) .dropdown-input-box-green{border-color:#000}:host-context(.contrasto-on) .dropdown-input-box-disabled{background-color:#fff!important}:host-context(.contrasto-on) .dropdown-button{color:#000}:host-context(.contrasto-on) .dropdown-button.collapse-disabled .dropdown-button-border{border-left-color:#000!important}:host-context(.contrasto-on) .dropdown-button.collapse-disabled .triangolo-bottom{border-top-color:#000}:host-context(.contrasto-on) .dropdown-button.collapse-disabled .triangolo-top{border-bottom-color:#000}:host-context(.contrasto-on) .dropdown-button-border{border-left-color:#000}:host-context(.contrasto-on) .has-error .dropdown-input-box{border-color:#000}:host-context(.contrasto-on) .has-error .dropdown-input-box:focus{border-color:#000}:host-context(.contrasto-on) .dropdown-error-message{color:#000}:host-context(.contrasto-on) .triangolo-top{border-bottom-color:#000}:host-context(.contrasto-on) .triangolo-bottom{border-top-color:#000}:host-context(.contrasto-on) .yellow-border{border-color:#fff} /*# sourceMappingURL=nbp-combo.component.css.map */ "],
        viewProviders: [NbpFormatterService],
        providers: [NBP_COMBO_CONTROL_VALUE_ACCESSOR, NBP_INPUT_ELEMENT_BASE],
        changeDetection: ChangeDetectionStrategy.OnPush
        // animations: [
        //   trigger('dropdownAnimationState', [
        //     state('expanded', style({ opacity: 1, visibility: 'visible', display: 'block' })),
        //     state('collapsed, void', style({ opacity: 0, visibility: 'hidden', display: 'none' })),
        //     transition('expanded => collapsed', animate('100ms ease-in')),
        //     transition('collapsed => expanded', animate('100ms ease-out'))])
        // ]
    }),
    __param(0, Optional()), __param(0, Inject(NG_VALIDATORS)),
    __param(1, Optional()), __param(1, Inject(NG_ASYNC_VALIDATORS)),
    __param(2, Optional()),
    __metadata("design:paramtypes", [Array, Array, NbpFormContainerComponent,
        NbpFormatterService,
        ElementRef, Injector, ChangeDetectorRef])
], NbpComboComponent);
export { NbpComboComponent };
