// cjlib/nbp/nbp.module.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FlexLayoutModule } from "@angular/flex-layout";
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { CurrencyPipe, DatePipe, DecimalPipe, LowerCasePipe, PercentPipe, UpperCasePipe } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgbDatepickerModule } from '@ng-bootstrap/ng-bootstrap';
import { RplDatepickerModule } from './nbp-datepicker-generic/@ng-bootstrap/datepicker/datepicker-reply.module';
export { RplDatepickerModule } from './nbp-datepicker-generic/@ng-bootstrap/datepicker/datepicker-reply.module';
import { NbpDatepickerGenericComponent } from './nbp-datepicker-generic/nbp-datepicker-generic.component';
export { NbpDatepickerGenericComponent } from './nbp-datepicker-generic/nbp-datepicker-generic.component';
import { RplCalendarModule } from './nbp-calendar-generic/@ng-bootstrap/datepicker/datepicker-reply.module';
export { RplCalendarModule } from './nbp-calendar-generic/@ng-bootstrap/datepicker/datepicker-reply.module';
import { NbpCalendarGenericComponent } from './nbp-calendar-generic/nbp-calendar-generic.component';
export { NbpCalendarGenericComponent } from './nbp-calendar-generic/nbp-calendar-generic.component';
export * from './nbp-commons/nbp-commons.enums';
import { NbpButtonComponent } from './nbp-button/nbp-button.component';
export { NbpButtonComponent } from './nbp-button/nbp-button.component';
import { NbpButtonGenerationComponent } from './nbp-button-generation/nbp-button-generation.component';
export { NbpButtonGenerationComponent } from './nbp-button-generation/nbp-button-generation.component';
import { NbpLabelComponent } from './nbp-label/nbp-label.component';
export { NbpLabelComponent } from './nbp-label/nbp-label.component';
import { NbpInputTextComponent } from './nbp-input-text/nbp-input-text.component';
export { NbpInputTextComponent } from './nbp-input-text/nbp-input-text.component';
import { NbpInputContainerComponent } from './nbp-input-container/nbp-input-container.component';
export { NbpInputContainerComponent } from './nbp-input-container/nbp-input-container.component';
import { NbpCheckboxComponent } from './nbp-checkbox/nbp-checkbox.component';
export { NbpCheckboxComponent } from './nbp-checkbox/nbp-checkbox.component';
import { NbpFooterComponent } from './nbp-footer/nbp-footer.component';
export { NbpFooterComponent } from './nbp-footer/nbp-footer.component';
import { NbpHeaderComponent } from './nbp-header/nbp-header.component';
export { NbpHeaderComponent } from './nbp-header/nbp-header.component';
import { NbpHeaderLightComponent } from './nbp-header-light/nbp-header-light.component';
export { NbpHeaderLightComponent } from './nbp-header-light/nbp-header-light.component';
import { NbpNavigationHeaderComponent } from './nbp-navigation-header/nbp-navigation-header.component';
export { NbpNavigationHeaderComponent } from './nbp-navigation-header/nbp-navigation-header.component';
import { NbpTableComponent } from './nbp-table/nbp-table.component';
export { NbpTableComponent } from './nbp-table/nbp-table.component';
import { NbpModalComponent } from './nbp-modal/nbp-modal.component';
export { NbpModalComponent } from './nbp-modal/nbp-modal.component';
import { NbpComboComponent } from './nbp-combo/nbp-combo.component';
export { NbpComboComponent } from './nbp-combo/nbp-combo.component';
// MULTISELECT
import { NbpComboMultiComponent } from './nbp-combo-multi/nbp-combo-multi.component';
export { NbpComboMultiComponent } from './nbp-combo-multi/nbp-combo-multi.component';
//
import { NbpDatepickerHtml5Component } from './nbp-datepicker-html5/nbp-datepicker-html5.component';
export { NbpDatepickerHtml5Component } from './nbp-datepicker-html5/nbp-datepicker-html5.component';
import { NbpSwitchComponent } from './nbp-switch/nbp-switch.component';
export { NbpSwitchComponent } from './nbp-switch/nbp-switch.component';
export { NpbSwitchDispalyType } from './nbp-switch/nbp-switch.enums';
import { NbpCustomerInfoComponent } from './nbp-customer-info/nbp-customer-info.component';
export { NbpCustomerInfoComponent } from './nbp-customer-info/nbp-customer-info.component';
import { NbpAccordionComponent } from './nbp-accordion/nbp-accordion.component';
export { NbpAccordionComponent } from './nbp-accordion/nbp-accordion.component';
import { NbpAccordionButtonComponent } from './nbp-accordion-button/nbp-accordion-button.component';
export { NbpAccordionButtonComponent } from './nbp-accordion-button/nbp-accordion-button.component';
import { NbpAccordionShowmoreComponent } from './nbp-accordion-showmore/nbp-accordion-showmore.component';
export { NbpAccordionShowmoreComponent } from './nbp-accordion-showmore/nbp-accordion-showmore.component';
import { NbpAccordionShowmoreCircleComponent } from './nbp-accordion-showmore-circle/nbp-accordion-showmore-circle.component';
export { NbpAccordionShowmoreCircleComponent } from './nbp-accordion-showmore-circle/nbp-accordion-showmore-circle.component';
import { NbpRadioComponent } from './nbp-radio/nbp-radio.component';
export { NbpRadioComponent } from './nbp-radio/nbp-radio.component';
import { NbpProgressComponent } from './nbp-progress/nbp-progress.component';
export { NbpProgressComponent } from './nbp-progress/nbp-progress.component';
import { NbpProgressStickyContentComponent } from './nbp-progress/nbp-progress-sticky-content.component';
export { NbpProgressStickyContentComponent } from './nbp-progress/nbp-progress-sticky-content.component';
import { NbpFormContainerComponent } from './nbp-form-container/nbp-form-container.component';
export { NbpFormContainerComponent } from './nbp-form-container/nbp-form-container.component';
import { NbpFormAutoScrollDirective } from './nbp-form-container/nbp-form-autoscroll.directive';
export { NbpFormAutoScrollDirective } from './nbp-form-container/nbp-form-autoscroll.directive';
import { NbpPagingComponent } from './nbp-paging/nbp-paging.component';
export { NbpPagingComponent } from './nbp-paging/nbp-paging.component';
import { NbpPaginationComponent } from './nbp-pagination/nbp-pagination.component';
export { NbpPaginationComponent } from './nbp-pagination/nbp-pagination.component';
import { NbpLoadingComponent } from './nbp-loading/nbp-loading.component';
export { NbpLoadingComponent } from './nbp-loading/nbp-loading.component';
import { NbpBreadcrumbComponent } from './nbp-breadcrumb/nbp-breadcrumb.component';
export { NbpBreadcrumbComponent } from './nbp-breadcrumb/nbp-breadcrumb.component';
import { NbpTooltipComponent } from './nbp-tooltip/nbp-tooltip.component';
export { NbpTooltipComponent } from './nbp-tooltip/nbp-tooltip.component';
import { NbpBurgerMenuComponent } from './nbp-burger-menu/nbp-burger-menu.component';
export { NbpBurgerMenuComponent } from './nbp-burger-menu/nbp-burger-menu.component';
import { NbpIconComponent } from './nbp-icon/nbp-icon.component';
export { NbpIconComponent } from './nbp-icon/nbp-icon.component';
import { NbpLoggedUserComponent } from './nbp-logged-user/nbp-logged-user.component';
export { NbpLoggedUserComponent } from './nbp-logged-user/nbp-logged-user.component';
import { NbpPanelComponent } from './nbp-panel/nbp-panel.component';
export { NbpPanelComponent } from './nbp-panel/nbp-panel.component';
import { NbpTableColumnComponent } from './nbp-table-column/nbp-table-column.component';
export { NbpTableColumnComponent } from './nbp-table-column/nbp-table-column.component';
import { NbpCustomerNameComponent } from './nbp-customer-name/nbp-customer-name.component';
export { NbpCustomerNameComponent } from './nbp-customer-name/nbp-customer-name.component';
import { NbpFeedbackComponent } from './nbp-feedback/nbp-feedback.component';
export { NbpFeedbackComponent } from './nbp-feedback/nbp-feedback.component';
import { NbpDataContainerComponent } from './nbp-data-container/nbp-data-container.component';
export { NbpDataContainerComponent } from './nbp-data-container/nbp-data-container.component';
export { NbpButtonType } from './nbp-button/nbp-button.enums';
export { NbpAccordionEvent } from './nbp-accordion/nbp-accordion.interfaces';
export { NbpDataSource } from './nbp-datasource/nbp-datasource.service';
export { INbpDataSource } from './nbp-datasource/nbp-datasource.interfaces';
export { NbpFeedbackType } from './nbp-feedback/nbp-feedback.enums';
export { NbpDatepickerComponent } from './nbp-datepicker/nbp-datepicker.component';
import { NbpDatepickerComponent } from './nbp-datepicker/nbp-datepicker.component';
export { NbpSequenzialeGenericComponent } from './nbp-sequenziale-generic/nbp-sequenziale-generic.component';
import { NbpSequenzialeGenericComponent } from './nbp-sequenziale-generic/nbp-sequenziale-generic.component';
export { NbpSequenzialeSottostepGenericComponent } from './nbp-sequenziale-sottostep-generic/nbp-sequenziale-sottostep-generic.component';
import { NbpSequenzialeSottostepGenericComponent } from './nbp-sequenziale-sottostep-generic/nbp-sequenziale-sottostep-generic.component';
import { NbpDividingLineComponent } from './nbp-dividing-line/nbp-dividing-line.component';
export { NbpDividingLineComponent } from './nbp-dividing-line/nbp-dividing-line.component';
// COMPONENTI POS
import { NbpTitledPanelComponent } from './nbp-titled-panel/nbp-titled-panel.component';
export { NbpTitledPanelComponent } from './nbp-titled-panel/nbp-titled-panel.component';
import { NbpSelectionPanelComponent } from './nbp-selection-panel/nbp-selection-panel.component';
export { NbpSelectionPanelComponent } from './nbp-selection-panel/nbp-selection-panel.component';
import { NbpShadowedPanelComponent } from './nbp-shadowed-panel/nbp-shadowed-panel.component';
export { NbpShadowedPanelComponent } from './nbp-shadowed-panel/nbp-shadowed-panel.component';
import { NbpSpinnerComponent } from './nbp-spinner/nbp-spinner.component';
export { NbpSpinnerComponent } from './nbp-spinner/nbp-spinner.component';
import { NbpTabBarComponent } from './nbp-tab-bar/nbp-tab-bar.component';
export { NbpTabBarComponent } from './nbp-tab-bar/nbp-tab-bar.component';
import { NbpLoadingService } from './nbp-loading/nbp-loading.service';
export { NbpLoadingService } from './nbp-loading/nbp-loading.service';
import { NbpLoadingInterceptor } from './nbp-loading/nbp-loading-interceptor.service';
export { NbpLoadingInterceptor } from './nbp-loading/nbp-loading-interceptor.service';
import { GlobalLoadingIdentificationService } from './nbp-loading/nbp-loading-interceptor.service';
export { GlobalLoadingIdentificationService } from './nbp-loading/nbp-loading-interceptor.service';
import { NbpDebugInfoComponent } from './nbp-debug-info/nbp-debug-info.component';
export { NbpDebugInfoComponent } from './nbp-debug-info/nbp-debug-info.component';
import { NbpErrorAdviceComponent } from './nbp-error-advice/nbp-error-advice.component';
export { NbpErrorAdviceComponent } from './nbp-error-advice/nbp-error-advice.component';
import { NbpSignTypeSelectComponent } from './nbp-sign-type-select/nbp-sign-type-select.component';
export { NbpSignTypeSelectComponent } from './nbp-sign-type-select/nbp-sign-type-select.component';
import { NbpSignInfoHeaderComponent } from './nbp-sign-info-header/nbp-sign-info-header.component';
export { NbpSignInfoHeaderComponent } from './nbp-sign-info-header/nbp-sign-info-header.component';
import { NbpRadioSelectionComponent } from './nbp-radio-selection/nbp-radio-selection.component';
export { NbpRadioSelectionComponent } from './nbp-radio-selection/nbp-radio-selection.component';
import { NbpSignSpilayerComponent } from './nbp-sign-spilayer/nbp-sign-spilayer.component';
export { NbpSignSpilayerComponent } from './nbp-sign-spilayer/nbp-sign-spilayer.component';
import { NbpPaperlessSignComponent } from './nbp-paperless-sign/nbp-paperless-sign.component';
export { NbpPaperlessSignComponent } from './nbp-paperless-sign/nbp-paperless-sign.component';
import { NbpSignComponent } from './nbp-sign/nbp-sign.component';
export { NbpSignComponent } from './nbp-sign/nbp-sign.component';
import { NbpSpilayerUtilityService } from './nbp-sign-spilayer/nbp-spilayer-utility.service';
import { SpilayerMockUtilityService } from './nbp-sign-spilayer/spilayer-mock-utility.service';
import { NbpSignService } from './nbp-sign/nbp-sign.service';
import { NbpSpilayerService } from './nbp-sign/nbp-spilayer.service';
// export { NbpSpilayerUtilityService } from './nbp-sign-spilayer/nbp-spilayer-utility.service';
export { NbpAutocompleteDatasource } from './nbp-datasource/nbp-autocomplete-datasource.service';
import { NbpPendingLoadingStatusService } from './nbp-loading/nbp-pending-loading-status.service';
export { NbpPendingLoadingStatusService } from './nbp-loading/nbp-pending-loading-status.service';
import { NbpStepNavigabiliComponent } from './nbp-step-navigabili/nbp-step-navigabili.component';
export { NbpStepNavigabiliComponent } from './nbp-step-navigabili/nbp-step-navigabili.component';
import { NbpComboFilterComponent } from './nbp-combo-filter/nbp-combo-filter.component';
export { NbpComboFilterComponent } from './nbp-combo-filter/nbp-combo-filter.component';
import { NbpToggleTabsetComponent } from './nbp-tab-toggle/nbp-toggle-tabset.component';
export { NbpToggleTabsetComponent } from './nbp-tab-toggle/nbp-toggle-tabset.component';
import { NbpToggleTabComponent } from './nbp-tab-toggle/nbp-toggle-tab.component';
export { NbpToggleTabComponent } from './nbp-tab-toggle/nbp-toggle-tab.component';
import { NbpSwitchBinaryComponent } from './nbp-switch-binary/nbp-switch-binary.component';
export { NbpSwitchBinaryComponent } from './nbp-switch-binary/nbp-switch-binary.component';
import { NbpAccordionLightComponent } from './nbp-accordion-light/nbp-accordion-light.component';
export { NbpAccordionLightComponent } from './nbp-accordion-light/nbp-accordion-light.component';
import { NbpAlertBoxComponent } from './nbp-alert-box/nbp-alert-box.component';
export { NbpAlertBoxComponent } from './nbp-alert-box/nbp-alert-box.component';
import { NbpSliceAndEllipsisPipe } from './nbp-common-pipe/nbp-commons.pipe';
export { NbpSliceAndEllipsisPipe } from './nbp-common-pipe/nbp-commons.pipe';
import { NbpImportoPipe } from './nbp-input-text/nbp-importo.pipe';
export { NbpImportoPipe } from './nbp-input-text/nbp-importo.pipe';
import { NbpEllipsisDirective } from './nbp-commons/nbp-commons.directive';
export { NbpEllipsisDirective } from './nbp-commons/nbp-commons.directive';
import { NbpPaperSignComponent } from './nbp-paper-sign/nbp-paper-sign.component';
export { NbpPaperSignComponent } from './nbp-paper-sign/nbp-paper-sign.component';
import { ContoPipe } from '../utility/nbp-formatter.service';
import { DownloadService } from './../utility/services/download.service';
export { DownloadService } from './../utility/services/download.service';
import { NbpHeaderGenericComponent } from './nbp-header-generic/nbp-header-generic.component';
export { NbpHeaderGenericComponent } from './nbp-header-generic/nbp-header-generic.component';
import { NbpFloaterComponent } from './nbp-floater/nbp-floater.component';
export { NbpFloaterComponent } from './nbp-floater/nbp-floater.component';
import { SmoothScrollToDirective } from './nbp-floater/smoothScrollTo.directive';
export { SmoothScrollToDirective } from './nbp-floater/smoothScrollTo.directive';
import { NbpFilterComponent } from './nbp-filter/nbp-filter.component';
export { NbpFilterComponent } from './nbp-filter/nbp-filter.component';
import { NbpSpinnerGenericComponent } from './nbp-spinner-generic/nbp-spinner-generic.component';
export { NbpSpinnerGenericComponent } from './nbp-spinner-generic/nbp-spinner-generic.component';
import { NbpStickyTabGenericComponent } from './nbp-sticky-tab-generic/nbp-sticky-tab-generic.component';
export { NbpStickyTabGenericComponent } from './nbp-sticky-tab-generic/nbp-sticky-tab-generic.component';
import { NbpHeaderGenericNoTranslateComponent } from './nbp-header-generic-no-translate/nbp-header-generic-no-translate.component';
export { NbpHeaderGenericNoTranslateComponent } from './nbp-header-generic-no-translate/nbp-header-generic-no-translate.component';
import { NbpBurgerMenuGenericNoTranslateComponent } from './nbp-burger-menu-generic-no-translate/nbp-burger-menu-generic-no-translate.component';
export { NbpBurgerMenuGenericNoTranslateComponent } from './nbp-burger-menu-generic-no-translate/nbp-burger-menu-generic-no-translate.component';
import { NbpBurgerMenuGenericComponent } from './nbp-burger-menu-generic/nbp-burger-menu-generic.component';
export { NbpBurgerMenuGenericComponent } from './nbp-burger-menu-generic/nbp-burger-menu-generic.component';
import { NbpMenuOperationComponent } from './nbp-menu-operation/nbp-menu-operation.component';
export { NbpMenuOperationComponent } from './nbp-menu-operation/nbp-menu-operation.component';
import { NbpLoggedUserGenericComponent } from './nbp-logged-user-generic/nbp-logged-user-generic.component';
import { NbpSwitchGenericComponent } from './nbp-switch-generic/nbp-switch-generic.component';
import { NbpCustomerInfoGenericComponent } from './nbp-customer-info-generic/nbp-customer-info-generic.component';
export { NbpCustomerInfoGenericComponent } from './nbp-customer-info-generic/nbp-customer-info-generic.component';
export { NbpLoggedUserGenericComponent } from './nbp-logged-user-generic/nbp-logged-user-generic.component';
export { NbpSwitchGenericComponent } from './nbp-switch-generic/nbp-switch-generic.component';
import { NbpCustomValidationService } from './nbp-validators/nbp-custom-validation.service';
export { NbpCustomValidationService } from './nbp-validators/nbp-custom-validation.service';
import { NbpBoxWithEllipsisComponent } from './nbp-box-with-ellipsis/nbp-box-with-ellipsis.component';
export { NbpBoxWithEllipsisComponent } from './nbp-box-with-ellipsis/nbp-box-with-ellipsis.component';
import { NbpCurrencyPipe } from './nbp-pipes/nbp-currency.pipe';
import { NbpFormatterDirective } from './nbp-pipes/nbp-formatter.directive';
export { NbpFormatterDirective } from './nbp-pipes/nbp-formatter.directive';
// ----- validatori
import { EqualValidator } from './nbp-validators/equal-validator.directive';
export { EqualValidator } from './nbp-validators/equal-validator.directive';
// import { DateValidator } from './nbp-validators/date-validator.directive';
// export { DateValidator } from './nbp-validators/date-validator.directive';
import { ArrayRequiredValidator } from './nbp-validators/combo-multi-validator.directive';
export { ArrayRequiredValidator } from './nbp-validators/combo-multi-validator.directive';
import { MaxSelectionValidator } from './nbp-validators/max-selection-validator.directive';
export { MaxSelectionValidator } from './nbp-validators/max-selection-validator.directive';
import { NbpCheckboxRequiredValidator } from './nbp-checkbox/nbp-checkbox.component';
export { NbpCheckboxRequiredValidator } from './nbp-checkbox/nbp-checkbox.component';
import { ContainsValidator } from './nbp-validators/contains-validator.directive';
export { ContainsValidator } from './nbp-validators/contains-validator.directive';
import { CustomValidator } from './nbp-validators/nbp-custom-validator.directive';
export { CustomValidator } from './nbp-validators/nbp-custom-validator.directive';
import { NbpModalDialogComponent } from './nbp-modal-dialog/nbp-modal-dialog.component';
export { NbpModalDialogComponent } from './nbp-modal-dialog/nbp-modal-dialog.component';
import { NbpModalDialogService } from './nbp-modal-dialog/nbp-modal-dialog.service';
export { NbpModalDialogService } from './nbp-modal-dialog/nbp-modal-dialog.service';
export { NbpEmbeddingService } from './nbp-embedding/nbp-embedding.service';
import { NbpViewSettingsService } from './nbp-view-settings/nbp-view-settings.service';
export { NbpViewSettingsService } from './nbp-view-settings/nbp-view-settings.service';
import { NbpWrapperComponent } from './nbp-wrapper/nbp-wrapper.component';
export { NbpWrapperComponent } from './nbp-wrapper/nbp-wrapper.component';
import { NbpSliderComponent } from './nbp-slider/nbp-slider.component';
export { NbpSliderComponent } from './nbp-slider/nbp-slider.component';
import { NbpSliderKeyPointsComponent } from './nbp-slider-keypoints/nbp-slider-keypoints.component';
export { NbpSliderKeyPointsComponent } from './nbp-slider-keypoints/nbp-slider-keypoints.component';
import { NouisliderModule } from './nbp-slider/wrapper/nouislider';
export { NouisliderModule } from './nbp-slider/wrapper/nouislider';
import { NbpPaginationLettersComponent } from './nbp-pagination-letters/nbp-pagination-letters.component';
export { NbpPaginationLettersComponent } from './nbp-pagination-letters/nbp-pagination-letters.component';
import { NbpUploadFileComponent } from './nbp-upload-file/nbp-upload-file.component';
export { NbpUploadFileComponent } from './nbp-upload-file/nbp-upload-file.component';
import { FileUploadModule } from 'ng2-file-upload';
export { FileUploadModule } from 'ng2-file-upload';
import { NbpBottomPageButtonComponent } from './nbp-bottom-page-button/nbp-bottom-page-button.component';
export { NbpBottomPageButtonComponent } from './nbp-bottom-page-button/nbp-bottom-page-button.component';
import { NbpSliderRangeComponent } from './nbp-slider-range/nbp-slider-range.component';
import { NbpSliderRangeBarComponent } from './nbp-slider-range/nbp-slider-range-bar/nbp-slider-range-bar.component';
import { NbpSliderRangePickerComponent } from './nbp-slider-range/nbp-slider-range-picker/nbp-slider-range-picker.component';
import { NbpSliderRangePopoverComponent } from './nbp-slider-range/nbp-slider-range-popover/nbp-slider-range-popover.component';
import { NbpSliderRangeBarDirective } from './nbp-slider-range/nbp-slider-range-bar/nbp-slider-range-bar.directive';
import { NbpSliderRangePickerDirective } from './nbp-slider-range/nbp-slider-range-picker/nbp-slider-range-picker.directive';
export { NbpSliderRangeComponent } from './nbp-slider-range/nbp-slider-range.component';
import { NbpSliderStoricoComponent } from './nbp-slider-storico/nbp-slider-storico.component';
import { NbpSliderStoricoBarComponent } from './nbp-slider-storico/nbp-slider-storico-bar/nbp-slider-storico-bar.component';
import { NbpSliderStoricoPickerComponent } from './nbp-slider-storico/nbp-slider-storico-picker/nbp-slider-storico-picker.component';
import { NbpSliderStoricoPopoverComponent } from './nbp-slider-storico/nbp-slider-storico-popover/nbp-slider-storico-popover.component';
import { NbpSliderStoricoBarDirective } from './nbp-slider-storico/nbp-slider-storico-bar/nbp-slider-storico-bar.directive';
import { NbpSliderStoricoPickerDirective } from './nbp-slider-storico/nbp-slider-storico-picker/nbp-slider-storico-picker.directive';
export { NbpSliderStoricoComponent } from './nbp-slider-storico/nbp-slider-storico.component';
import { NbpIndexAlphabeticComponent } from './nbp-index-alphabetic/nbp-index-alphabetic.component';
export { NbpIndexAlphabeticComponent } from './nbp-index-alphabetic/nbp-index-alphabetic.component';
import { NbpOnBoardingComponent } from './nbp-on-boarding/nbp-on-boarding.component';
export { NbpOnBoardingComponent } from './nbp-on-boarding/nbp-on-boarding.component';
import { NbpEtichettaCheckComponent } from './nbp-etichetta-check/nbp-etichetta-check.component';
export { NbpEtichettaCheckComponent } from './nbp-etichetta-check/nbp-etichetta-check.component';
import { NbpSelMercatoComponent } from './nbp-sel-mercato/nbp-sel-mercato.component';
export { NbpSelMercatoComponent } from './nbp-sel-mercato/nbp-sel-mercato.component';
import { NbpEtichettaFiltroComponent } from './nbp-etichetta-filtro/nbp-etichetta-filtro.component';
export { NbpEtichettaFiltroComponent } from './nbp-etichetta-filtro/nbp-etichetta-filtro.component';
import { NbpSearchBoxComponent } from './nbp-search-box/nbp-search-box.component';
export { NbpSearchBoxComponent } from './nbp-search-box/nbp-search-box.component';
import { ClickOutsideModule } from 'ng4-click-outside';
export { ClickOutsideModule } from 'ng4-click-outside';
import { NbpAutoFocusInModalDirective } from './nbp-modal/nbp-auto-focus-in-modal.directive';
export { NbpAutoFocusInModalDirective } from './nbp-modal/nbp-auto-focus-in-modal.directive';
export { NbpCalendarDateModel } from './nbp-calendar-generic/@ng-bootstrap/datepicker/nbp-calendar-date-model';
var importList = [
    //BrowserAnimationsModule, dipendenza importata separatamente per problema relativo
    CommonModule,
    FlexLayoutModule,
    FormsModule,
    NouisliderModule,
    NgbDatepickerModule.forRoot(),
    RplDatepickerModule.forRoot(),
    RplCalendarModule.forRoot(),
    FileUploadModule,
    ClickOutsideModule,
];
var declarationList = [
    NbpButtonComponent,
    NbpButtonGenerationComponent,
    NbpLabelComponent,
    NbpInputContainerComponent,
    NbpInputTextComponent,
    NbpCheckboxComponent,
    NbpFooterComponent,
    NbpHeaderComponent,
    NbpHeaderLightComponent,
    NbpNavigationHeaderComponent,
    NbpComboComponent,
    NbpComboMultiComponent,
    NbpDatepickerHtml5Component,
    NbpTableComponent,
    NbpModalComponent,
    NbpSwitchComponent,
    NbpCustomerInfoComponent,
    NbpAccordionComponent,
    NbpAccordionButtonComponent,
    NbpAccordionShowmoreComponent,
    NbpAccordionShowmoreCircleComponent,
    NbpRadioComponent,
    NbpProgressComponent,
    NbpProgressStickyContentComponent,
    NbpFormContainerComponent,
    NbpPagingComponent,
    NbpPaginationComponent,
    NbpLoadingComponent,
    NbpBreadcrumbComponent,
    NbpTooltipComponent,
    NbpBurgerMenuComponent,
    NbpIconComponent,
    NbpLoggedUserComponent,
    NbpPanelComponent,
    NbpCustomerNameComponent,
    NbpTableColumnComponent,
    NbpFeedbackComponent,
    NbpDataContainerComponent,
    NbpDatepickerComponent,
    NbpDatepickerGenericComponent,
    NbpCalendarGenericComponent,
    NbpSequenzialeGenericComponent,
    NbpSequenzialeSottostepGenericComponent,
    NbpDividingLineComponent,
    NbpSelectionPanelComponent,
    NbpTitledPanelComponent,
    NbpShadowedPanelComponent,
    NbpSpinnerComponent,
    NbpTabBarComponent,
    NbpDebugInfoComponent,
    NbpErrorAdviceComponent,
    NbpSignInfoHeaderComponent,
    NbpSignTypeSelectComponent,
    NbpRadioSelectionComponent,
    NbpSignSpilayerComponent,
    NbpPaperlessSignComponent,
    NbpSignComponent,
    NbpStepNavigabiliComponent,
    NbpComboFilterComponent,
    NbpToggleTabsetComponent,
    NbpToggleTabComponent,
    NbpSwitchBinaryComponent,
    NbpAccordionLightComponent,
    NbpAlertBoxComponent,
    NbpPaperSignComponent,
    NbpSliceAndEllipsisPipe,
    NbpEllipsisDirective,
    ContoPipe,
    NbpImportoPipe,
    NbpFormAutoScrollDirective,
    NbpHeaderGenericComponent,
    NbpFloaterComponent,
    NbpFilterComponent,
    NbpSpinnerGenericComponent,
    NbpSliderComponent,
    NbpSliderKeyPointsComponent,
    NbpPaginationLettersComponent,
    NbpStickyTabGenericComponent,
    NbpHeaderGenericNoTranslateComponent,
    SmoothScrollToDirective,
    NbpBurgerMenuGenericNoTranslateComponent,
    NbpBurgerMenuGenericComponent,
    NbpMenuOperationComponent,
    NbpCustomerInfoGenericComponent,
    NbpLoggedUserGenericComponent,
    NbpSwitchGenericComponent,
    NbpUploadFileComponent,
    NbpSliderRangeComponent,
    NbpSliderRangeBarComponent,
    NbpSliderRangePickerComponent,
    NbpSliderRangePopoverComponent,
    NbpSliderRangeBarDirective,
    NbpSliderRangePickerDirective,
    NbpSliderStoricoComponent,
    NbpSliderStoricoBarComponent,
    NbpSliderStoricoPickerComponent,
    NbpSliderStoricoPopoverComponent,
    NbpSliderStoricoBarDirective,
    NbpSliderStoricoPickerDirective,
    NbpIndexAlphabeticComponent,
    NbpEtichettaCheckComponent,
    NbpSelMercatoComponent,
    NbpEtichettaFiltroComponent,
    NbpSearchBoxComponent,
    // ----- validatori
    EqualValidator,
    ArrayRequiredValidator,
    ContainsValidator,
    CustomValidator,
    NbpCheckboxRequiredValidator,
    MaxSelectionValidator,
    NbpModalDialogComponent,
    NbpWrapperComponent,
    NbpBoxWithEllipsisComponent,
    NbpCurrencyPipe,
    NbpFormatterDirective,
    NbpBottomPageButtonComponent,
    NbpOnBoardingComponent,
    NbpAutoFocusInModalDirective,
];
var providersList = [
    CurrencyPipe,
    NbpImportoPipe,
    DatePipe,
    DecimalPipe,
    LowerCasePipe,
    PercentPipe,
    UpperCasePipe,
    ContoPipe,
    NbpCurrencyPipe,
    NbpLoadingService,
    GlobalLoadingIdentificationService,
    NbpLoadingInterceptor,
    NbpSpilayerUtilityService,
    SpilayerMockUtilityService,
    NbpSignService,
    NbpSpilayerService,
    NbpPendingLoadingStatusService,
    NbpCustomValidationService,
    DownloadService,
    NbpModalDialogService,
    NbpViewSettingsService
];
var exportList = [
    NbpButtonComponent,
    NbpButtonGenerationComponent,
    NbpLabelComponent,
    NbpInputTextComponent,
    NbpInputContainerComponent,
    NbpCheckboxComponent,
    NbpFooterComponent,
    NbpHeaderComponent,
    NbpHeaderLightComponent,
    NbpNavigationHeaderComponent,
    NbpComboComponent,
    NbpComboMultiComponent,
    NbpDatepickerHtml5Component,
    NbpTableComponent,
    NbpModalComponent,
    NbpSwitchComponent,
    NbpCustomerInfoComponent,
    NbpAccordionComponent,
    NbpAccordionButtonComponent,
    NbpAccordionShowmoreComponent,
    NbpAccordionShowmoreCircleComponent,
    NbpRadioComponent,
    NbpProgressComponent,
    NbpProgressStickyContentComponent,
    NbpFormContainerComponent,
    NbpPagingComponent,
    NbpPaginationComponent,
    NbpLoadingComponent,
    NbpBreadcrumbComponent,
    NbpTooltipComponent,
    NbpBurgerMenuComponent,
    NbpIconComponent,
    NbpLoggedUserComponent,
    NbpPanelComponent,
    NbpCustomerNameComponent,
    NbpTableColumnComponent,
    NbpFeedbackComponent,
    NbpDataContainerComponent,
    NbpDatepickerComponent,
    NbpDatepickerGenericComponent,
    NbpCalendarGenericComponent,
    NbpSequenzialeGenericComponent,
    NbpSequenzialeSottostepGenericComponent,
    NbpDividingLineComponent,
    NbpSelectionPanelComponent,
    NbpTitledPanelComponent,
    NbpShadowedPanelComponent,
    NbpSpinnerComponent,
    NbpTabBarComponent,
    NbpDebugInfoComponent,
    NbpErrorAdviceComponent,
    NbpSignInfoHeaderComponent,
    NbpSignTypeSelectComponent,
    NbpRadioSelectionComponent,
    NbpSignSpilayerComponent,
    NbpPaperlessSignComponent,
    NbpSignComponent,
    NbpStepNavigabiliComponent,
    NbpComboFilterComponent,
    NbpToggleTabsetComponent,
    NbpToggleTabComponent,
    NbpSwitchBinaryComponent,
    NbpAccordionLightComponent,
    NbpAlertBoxComponent,
    NbpPaperSignComponent,
    NbpSliceAndEllipsisPipe,
    NbpEllipsisDirective,
    ContoPipe,
    NbpImportoPipe,
    NbpFormAutoScrollDirective,
    NbpHeaderGenericComponent,
    NbpFloaterComponent,
    NbpFilterComponent,
    NbpSpinnerGenericComponent,
    NbpSliderComponent,
    NbpSliderKeyPointsComponent,
    NbpPaginationLettersComponent,
    NbpStickyTabGenericComponent,
    NbpHeaderGenericNoTranslateComponent,
    SmoothScrollToDirective,
    NbpBurgerMenuGenericNoTranslateComponent,
    NbpBurgerMenuGenericComponent,
    NbpMenuOperationComponent,
    NbpCustomerInfoGenericComponent,
    NbpLoggedUserGenericComponent,
    NbpSwitchGenericComponent,
    NbpUploadFileComponent,
    NbpSliderRangeComponent,
    NbpSliderStoricoComponent,
    NbpIndexAlphabeticComponent,
    NbpEtichettaCheckComponent,
    NbpSelMercatoComponent,
    NbpEtichettaFiltroComponent,
    NbpSearchBoxComponent,
    // ----- validatori
    EqualValidator,
    // DateValidator,
    ArrayRequiredValidator,
    ContainsValidator,
    CustomValidator,
    NbpCheckboxRequiredValidator,
    MaxSelectionValidator,
    NbpModalDialogComponent,
    NbpWrapperComponent,
    NbpBoxWithEllipsisComponent,
    NbpCurrencyPipe,
    NbpFormatterDirective,
    NbpBottomPageButtonComponent,
    NbpOnBoardingComponent,
    NbpAutoFocusInModalDirective,
];
/**
 * Modulo da importare direttamente in caso di problemi relativi a doppi import di BrowserAnimationModule
 * (il modulo espone gli stessi componenti di NbpModule ma non importa BrowserAnimationModule).
 * La problematica è stata riscontrata solo in caso di lazy loading e import del modulo a livello dei children.
 * La libreria ha comunque l'esigenza, per alcuni componenti, che BrowserAnimationModule sia importato a livello
 * del root dell'applicazione.
 */
var NbpModule_noBrowserAnimation = (function () {
    function NbpModule_noBrowserAnimation() {
    }
    return NbpModule_noBrowserAnimation;
}());
NbpModule_noBrowserAnimation = __decorate([
    NgModule({
        imports: importList,
        declarations: declarationList,
        providers: providersList,
        exports: exportList,
    })
], NbpModule_noBrowserAnimation);
export { NbpModule_noBrowserAnimation };
var NbpModule = (function () {
    function NbpModule() {
    }
    return NbpModule;
}());
NbpModule = __decorate([
    NgModule({
        imports: [BrowserAnimationsModule, NbpModule_noBrowserAnimation],
        exports: [NbpModule_noBrowserAnimation],
    })
], NbpModule);
export { NbpModule };
